{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": [],
      "gpuType": "T4",
      "authorship_tag": "ABX9TyPee+eFRL/q3t+OKexcC37G",
      "include_colab_link": true
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    },
    "accelerator": "GPU",
    "widgets": {
      "application/vnd.jupyter.widget-state+json": {
        "2f5c908b47c34249a826f542f5adf054": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "PlayModel",
          "model_module_version": "2.0.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "2.0.0",
            "_model_name": "PlayModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "2.0.0",
            "_view_name": "PlayView",
            "description": "",
            "description_allow_html": false,
            "disabled": false,
            "interval": 100,
            "layout": "IPY_MODEL_b7bef3eeb335450292f0b76a43a54dc0",
            "max": 0,
            "min": 0,
            "playing": false,
            "repeat": false,
            "show_repeat": true,
            "step": 1,
            "style": "IPY_MODEL_c6370a6952144c73bf7fdc39cf3d8a3b",
            "tabbable": null,
            "tooltip": null,
            "value": 0
          }
        },
        "c1b930a01abf44a086ff2fad756f0b94": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "IntSliderModel",
          "model_module_version": "2.0.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "2.0.0",
            "_model_name": "IntSliderModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "2.0.0",
            "_view_name": "IntSliderView",
            "behavior": "drag-tap",
            "continuous_update": true,
            "description": "",
            "description_allow_html": false,
            "disabled": false,
            "layout": "IPY_MODEL_f3712a1a444d4cd293b7565cc305c0c1",
            "max": 0,
            "min": 0,
            "orientation": "horizontal",
            "readout": true,
            "readout_format": "d",
            "step": 1,
            "style": "IPY_MODEL_3233db53b097403a95ade1daf78e1fbb",
            "tabbable": null,
            "tooltip": null,
            "value": 0
          }
        },
        "4dcfe5e1f00a469d8696518d42d95c0e": {
          "model_module": "nglview-js-widgets",
          "model_name": "NGLModel",
          "model_module_version": "3.1.5",
          "state": {
            "_camera_orientation": [
              194.91017817324231,
              0,
              0,
              0,
              0,
              194.91017817324231,
              0,
              0,
              0,
              0,
              194.91017817324231,
              0,
              4.291742324829102,
              30.68375587463379,
              -36.923973083496094,
              1
            ],
            "_camera_str": "orthographic",
            "_dom_classes": [],
            "_gui_theme": null,
            "_ibtn_fullscreen": "IPY_MODEL_1516b48783174c179c0072f2e152b2fe",
            "_igui": null,
            "_iplayer": "IPY_MODEL_d432e843235d47d0b4e4c7344a29ef37",
            "_model_module": "nglview-js-widgets",
            "_model_module_version": "3.1.5",
            "_model_name": "NGLModel",
            "_ngl_color_dict": {},
            "_ngl_coordinate_resource": {},
            "_ngl_full_stage_parameters": {
              "impostor": true,
              "quality": "medium",
              "workerDefault": true,
              "sampleLevel": 0,
              "backgroundColor": "white",
              "rotateSpeed": 2,
              "zoomSpeed": 1.2,
              "panSpeed": 1,
              "clipNear": 0,
              "clipFar": 100,
              "clipDist": 10,
              "clipMode": "scene",
              "clipScale": "relative",
              "fogNear": 50,
              "fogFar": 100,
              "cameraFov": 40,
              "cameraEyeSep": 0.3,
              "cameraType": "perspective",
              "lightColor": 14540253,
              "lightIntensity": 1.2,
              "ambientColor": 14540253,
              "ambientIntensity": 0.3,
              "hoverTimeout": 0,
              "tooltip": true,
              "mousePreset": "default"
            },
            "_ngl_msg_archive": [
              {
                "target": "Stage",
                "type": "call_method",
                "methodName": "loadFile",
                "reconstruc_color_scheme": false,
                "args": [
                  {
                    "type": "blob",
                    "data": "CRYST1    1.000    1.000    1.000  90.00  90.00  90.00 P 1                      \nATOM      1  N   MET A   1      46.045 -35.208  95.008  1.00  0.00           N  \nATOM      2  CA  MET A   1      44.930 -34.404  94.463  1.00  0.00           C  \nATOM      3  C   MET A   1      44.976 -34.552  92.940  1.00  0.00           C  \nATOM      4  O   MET A   1      45.987 -34.176  92.379  1.00  0.00           O  \nATOM      5  CB  MET A   1      45.137 -32.925  94.839  1.00  0.00           C  \nATOM      6  CG  MET A   1      44.916 -32.641  96.335  1.00  0.00           C  \nATOM      7  SD  MET A   1      45.626 -31.054  96.845  1.00  0.00           S  \nATOM      8  CE  MET A   1      44.858 -30.830  98.477  1.00  0.00           C  \nATOM      9  N   GLY A   2      44.098 -35.219  92.217  1.00  0.00           N  \nATOM     10  CA  GLY A   2      42.701 -35.577  92.412  1.00  0.00           C  \nATOM     11  C   GLY A   2      41.944 -35.245  91.118  1.00  0.00           C  \nATOM     12  O   GLY A   2      41.835 -34.074  90.806  1.00  0.00           O  \nATOM     13  N   HIS A   3      41.427 -36.281  90.414  1.00  0.00           N  \nATOM     14  CA  HIS A   3      40.360 -36.200  89.379  1.00  0.00           C  \nATOM     15  C   HIS A   3      40.703 -35.590  87.995  1.00  0.00           C  \nATOM     16  O   HIS A   3      41.465 -34.656  87.889  1.00  0.00           O  \nATOM     17  CB  HIS A   3      39.148 -35.519  90.018  1.00  0.00           C  \nATOM     18  CG  HIS A   3      38.638 -36.258  91.245  1.00  0.00           C  \nATOM     19  ND1 HIS A   3      38.346 -37.599  91.311  1.00  0.00           N  \nATOM     20  CD2 HIS A   3      38.390 -35.722  92.475  1.00  0.00           C  \nATOM     21  CE1 HIS A   3      37.939 -37.863  92.569  1.00  0.00           C  \nATOM     22  NE2 HIS A   3      37.955 -36.757  93.301  1.00  0.00           N  \nATOM     23  N   ARG A   4      40.138 -35.978  86.901  1.00  0.00           N  \nATOM     24  CA  ARG A   4      39.296 -37.091  86.404  1.00  0.00           C  \nATOM     25  C   ARG A   4      39.302 -36.985  84.869  1.00  0.00           C  \nATOM     26  O   ARG A   4      39.345 -35.893  84.318  1.00  0.00           O  \nATOM     27  CB  ARG A   4      37.825 -36.971  86.895  1.00  0.00           C  \nATOM     28  CG  ARG A   4      37.416 -38.042  87.920  1.00  0.00           C  \nATOM     29  CD  ARG A   4      35.980 -37.793  88.420  1.00  0.00           C  \nATOM     30  NE  ARG A   4      35.610 -38.694  89.535  1.00  0.00           N  \nATOM     31  CZ  ARG A   4      34.390 -39.135  89.829  1.00  0.00           C  \nATOM     32  NH1 ARG A   4      33.349 -38.835  89.103  1.00  0.00           N  \nATOM     33  NH2 ARG A   4      34.197 -39.892  90.873  1.00  0.00           N  \nATOM     34  N   ARG A   5      39.209 -38.142  84.168  1.00  0.00           N  \nATOM     35  CA  ARG A   5      39.015 -38.265  82.707  1.00  0.00           C  \nATOM     36  C   ARG A   5      37.657 -37.704  82.290  1.00  0.00           C  \nATOM     37  O   ARG A   5      36.675 -38.036  82.938  1.00  0.00           O  \nATOM     38  CB  ARG A   5      39.018 -39.763  82.354  1.00  0.00           C  \nATOM     39  CG  ARG A   5      40.345 -40.509  82.596  1.00  0.00           C  \nATOM     40  CD  ARG A   5      40.129 -42.020  82.369  1.00  0.00           C  \nATOM     41  NE  ARG A   5      41.401 -42.762  82.426  1.00  0.00           N  \nATOM     42  CZ  ARG A   5      41.563 -44.057  82.215  1.00  0.00           C  \nATOM     43  NH1 ARG A   5      40.554 -44.849  81.972  1.00  0.00           N  \nATOM     44  NH2 ARG A   5      42.753 -44.601  82.238  1.00  0.00           N  \nATOM     45  N   PHE A   6      37.592 -37.034  81.147  1.00  0.00           N  \nATOM     46  CA  PHE A   6      36.404 -37.070  80.274  1.00  0.00           C  \nATOM     47  C   PHE A   6      36.835 -36.938  78.805  1.00  0.00           C  \nATOM     48  O   PHE A   6      37.464 -35.963  78.414  1.00  0.00           O  \nATOM     49  CB  PHE A   6      35.351 -36.015  80.666  1.00  0.00           C  \nATOM     50  CG  PHE A   6      34.194 -36.616  81.447  1.00  0.00           C  \nATOM     51  CD1 PHE A   6      33.149 -37.274  80.763  1.00  0.00           C  \nATOM     52  CD2 PHE A   6      34.177 -36.582  82.856  1.00  0.00           C  \nATOM     53  CE1 PHE A   6      32.106 -37.901  81.475  1.00  0.00           C  \nATOM     54  CE2 PHE A   6      33.142 -37.212  83.572  1.00  0.00           C  \nATOM     55  CZ  PHE A   6      32.114 -37.876  82.883  1.00  0.00           C  \nATOM     56  N   GLY A   7      36.529 -37.985  78.004  1.00  0.00           N  \nATOM     57  CA  GLY A   7      36.693 -37.967  76.552  1.00  0.00           C  \nATOM     58  C   GLY A   7      35.499 -37.267  75.899  1.00  0.00           C  \nATOM     59  O   GLY A   7      34.362 -37.475  76.305  1.00  0.00           O  \nATOM     60  N   PHE A   8      35.785 -36.499  74.867  1.00  0.00           N  \nATOM     61  CA  PHE A   8      34.739 -35.964  73.995  1.00  0.00           C  \nATOM     62  C   PHE A   8      35.052 -36.362  72.554  1.00  0.00           C  \nATOM     63  O   PHE A   8      36.084 -36.012  71.999  1.00  0.00           O  \nATOM     64  CB  PHE A   8      34.565 -34.446  74.195  1.00  0.00           C  \nATOM     65  CG  PHE A   8      33.114 -34.035  74.331  1.00  0.00           C  \nATOM     66  CD1 PHE A   8      32.396 -33.567  73.220  1.00  0.00           C  \nATOM     67  CD2 PHE A   8      32.480 -34.137  75.591  1.00  0.00           C  \nATOM     68  CE1 PHE A   8      31.047 -33.204  73.366  1.00  0.00           C  \nATOM     69  CE2 PHE A   8      31.126 -33.778  75.723  1.00  0.00           C  \nATOM     70  CZ  PHE A   8      30.409 -33.302  74.615  1.00  0.00           C  \nATOM     71  N   ARG A   9      34.165 -37.185  71.976  1.00  0.00           N  \nATOM     72  CA  ARG A   9      34.220 -37.612  70.574  1.00  0.00           C  \nATOM     73  C   ARG A   9      33.686 -36.477  69.700  1.00  0.00           C  \nATOM     74  O   ARG A   9      32.488 -36.206  69.748  1.00  0.00           O  \nATOM     75  CB  ARG A   9      33.375 -38.886  70.386  1.00  0.00           C  \nATOM     76  CG  ARG A   9      33.977 -40.156  71.000  1.00  0.00           C  \nATOM     77  CD  ARG A   9      33.048 -41.363  70.733  1.00  0.00           C  \nATOM     78  NE  ARG A   9      33.692 -42.641  71.054  1.00  0.00           N  \nATOM     79  CZ  ARG A   9      33.280 -43.844  70.659  1.00  0.00           C  \nATOM     80  NH1 ARG A   9      32.187 -44.011  69.966  1.00  0.00           N  \nATOM     81  NH2 ARG A   9      33.960 -44.909  70.960  1.00  0.00           N  \nATOM     82  N   LEU A  10      34.551 -35.863  68.908  1.00  0.00           N  \nATOM     83  CA  LEU A  10      34.090 -35.003  67.815  1.00  0.00           C  \nATOM     84  C   LEU A  10      33.566 -35.879  66.664  1.00  0.00           C  \nATOM     85  O   LEU A  10      34.326 -36.616  66.040  1.00  0.00           O  \nATOM     86  CB  LEU A  10      35.213 -34.048  67.346  1.00  0.00           C  \nATOM     87  CG  LEU A  10      34.956 -32.560  67.702  1.00  0.00           C  \nATOM     88  CD1 LEU A  10      35.254 -32.261  69.159  1.00  0.00           C  \nATOM     89  CD2 LEU A  10      35.806 -31.653  66.833  1.00  0.00           C  \nATOM     90  N   ARG A  11      32.290 -35.817  66.395  1.00  0.00           N  \nATOM     91  CA  ARG A  11      31.679 -36.319  65.163  1.00  0.00           C  \nATOM     92  C   ARG A  11      31.849 -35.250  64.081  1.00  0.00           C  \nATOM     93  O   ARG A  11      31.312 -34.157  64.216  1.00  0.00           O  \nATOM     94  CB  ARG A  11      30.189 -36.641  65.390  1.00  0.00           C  \nATOM     95  CG  ARG A  11      29.965 -37.960  66.129  1.00  0.00           C  \nATOM     96  CD  ARG A  11      28.455 -38.230  66.274  1.00  0.00           C  \nATOM     97  NE  ARG A  11      28.173 -39.624  66.685  1.00  0.00           N  \nATOM     98  CZ  ARG A  11      26.982 -40.204  66.709  1.00  0.00           C  \nATOM     99  NH1 ARG A  11      25.889 -39.544  66.424  1.00  0.00           N  \nATOM    100  NH2 ARG A  11      26.849 -41.459  67.017  1.00  0.00           N  \nATOM    101  N   ALA A  12      32.588 -35.572  63.032  1.00  0.00           N  \nATOM    102  CA  ALA A  12      32.638 -34.766  61.825  1.00  0.00           C  \nATOM    103  C   ALA A  12      31.297 -34.879  61.082  1.00  0.00           C  \nATOM    104  O   ALA A  12      30.903 -35.968  60.666  1.00  0.00           O  \nATOM    105  CB  ALA A  12      33.822 -35.235  60.967  1.00  0.00           C  \nATOM    106  N   ALA A  13      30.586 -33.768  60.940  1.00  0.00           N  \nATOM    107  CA  ALA A  13      29.413 -33.659  60.083  1.00  0.00           C  \nATOM    108  C   ALA A  13      29.892 -33.325  58.665  1.00  0.00           C  \nATOM    109  O   ALA A  13      30.413 -32.241  58.420  1.00  0.00           O  \nATOM    110  CB  ALA A  13      28.459 -32.612  60.668  1.00  0.00           C  \nATOM    111  N   ALA A  14      29.749 -34.276  57.750  1.00  0.00           N  \nATOM    112  CA  ALA A  14      29.949 -34.040  56.328  1.00  0.00           C  \nATOM    113  C   ALA A  14      28.762 -33.233  55.791  1.00  0.00           C  \nATOM    114  O   ALA A  14      27.637 -33.722  55.735  1.00  0.00           O  \nATOM    115  CB  ALA A  14      30.118 -35.394  55.622  1.00  0.00           C  \nATOM    116  N   ALA A  15      29.002 -31.974  55.397  1.00  0.00           N  \nATOM    117  CA  ALA A  15      28.034 -31.159  54.683  1.00  0.00           C  \nATOM    118  C   ALA A  15      28.068 -31.536  53.195  1.00  0.00           C  \nATOM    119  O   ALA A  15      29.005 -31.204  52.473  1.00  0.00           O  \nATOM    120  CB  ALA A  15      28.357 -29.676  54.934  1.00  0.00           C  \nATOM    121  N   SER A  16      27.046 -32.258  52.740  1.00  0.00           N  \nATOM    122  CA  SER A  16      26.831 -32.552  51.324  1.00  0.00           C  \nATOM    123  C   SER A  16      26.263 -31.314  50.631  1.00  0.00           C  \nATOM    124  O   SER A  16      25.105 -30.963  50.834  1.00  0.00           O  \nATOM    125  CB  SER A  16      25.878 -33.746  51.173  1.00  0.00           C  \nATOM    126  OG  SER A  16      26.368 -34.863  51.887  1.00  0.00           O  \nATOM    127  N   LEU A  17      27.087 -30.644  49.811  1.00  0.00           N  \nATOM    128  CA  LEU A  17      26.599 -29.630  48.872  1.00  0.00           C  \nATOM    129  C   LEU A  17      25.805 -30.328  47.761  1.00  0.00           C  \nATOM    130  O   LEU A  17      26.372 -30.966  46.871  1.00  0.00           O  \nATOM    131  CB  LEU A  17      27.767 -28.808  48.284  1.00  0.00           C  \nATOM    132  CG  LEU A  17      27.903 -27.385  48.835  1.00  0.00           C  \nATOM    133  CD1 LEU A  17      28.727 -27.353  50.116  1.00  0.00           C  \nATOM    134  CD2 LEU A  17      28.583 -26.467  47.813  1.00  0.00           C  \nATOM    135  N   LEU A  18      24.483 -30.203  47.793  1.00  0.00           N  \nATOM    136  CA  LEU A  18      23.616 -30.505  46.658  1.00  0.00           C  \nATOM    137  C   LEU A  18      23.703 -29.342  45.658  1.00  0.00           C  \nATOM    138  O   LEU A  18      23.089 -28.295  45.839  1.00  0.00           O  \nATOM    139  CB  LEU A  18      22.169 -30.745  47.163  1.00  0.00           C  \nATOM    140  CG  LEU A  18      21.951 -32.167  47.700  1.00  0.00           C  \nATOM    141  CD1 LEU A  18      20.805 -32.195  48.713  1.00  0.00           C  \nATOM    142  CD2 LEU A  18      21.611 -33.139  46.569  1.00  0.00           C  \nATOM    143  N   LEU A  19      24.482 -29.524  44.590  1.00  0.00           N  \nATOM    144  CA  LEU A  19      24.438 -28.683  43.400  1.00  0.00           C  \nATOM    145  C   LEU A  19      23.096 -28.911  42.690  1.00  0.00           C  \nATOM    146  O   LEU A  19      22.910 -29.888  41.973  1.00  0.00           O  \nATOM    147  CB  LEU A  19      25.645 -29.010  42.489  1.00  0.00           C  \nATOM    148  CG  LEU A  19      26.902 -28.205  42.872  1.00  0.00           C  \nATOM    149  CD1 LEU A  19      28.161 -29.067  42.760  1.00  0.00           C  \nATOM    150  CD2 LEU A  19      27.071 -26.995  41.945  1.00  0.00           C  \nATOM    151  N   GLY A  20      22.144 -28.004  42.916  1.00  0.00           N  \nATOM    152  CA  GLY A  20      20.906 -27.938  42.162  1.00  0.00           C  \nATOM    153  C   GLY A  20      21.193 -27.506  40.725  1.00  0.00           C  \nATOM    154  O   GLY A  20      21.427 -26.340  40.461  1.00  0.00           O  \nATOM    155  N   SER A  21      21.176 -28.465  39.803  1.00  0.00           N  \nATOM    156  CA  SER A  21      21.181 -28.194  38.364  1.00  0.00           C  \nATOM    157  C   SER A  21      19.830 -27.591  37.978  1.00  0.00           C  \nATOM    158  O   SER A  21      18.844 -28.296  37.782  1.00  0.00           O  \nATOM    159  CB  SER A  21      21.442 -29.483  37.577  1.00  0.00           C  \nATOM    160  OG  SER A  21      22.634 -30.098  38.028  1.00  0.00           O  \nATOM    161  N   SER A  22      19.771 -26.258  37.868  1.00  0.00           N  \nATOM    162  CA  SER A  22      18.636 -25.556  37.282  1.00  0.00           C  \nATOM    163  C   SER A  22      18.623 -25.816  35.777  1.00  0.00           C  \nATOM    164  O   SER A  22      19.370 -25.203  35.019  1.00  0.00           O  \nATOM    165  CB  SER A  22      18.665 -24.060  37.627  1.00  0.00           C  \nATOM    166  OG  SER A  22      19.928 -23.611  38.046  1.00  0.00           O  \nATOM    167  N   LEU A  23      17.781 -26.759  35.332  1.00  0.00           N  \nATOM    168  CA  LEU A  23      17.400 -26.850  33.925  1.00  0.00           C  \nATOM    169  C   LEU A  23      16.675 -25.549  33.554  1.00  0.00           C  \nATOM    170  O   LEU A  23      15.505 -25.362  33.877  1.00  0.00           O  \nATOM    171  CB  LEU A  23      16.507 -28.077  33.646  1.00  0.00           C  \nATOM    172  CG  LEU A  23      17.270 -29.358  33.276  1.00  0.00           C  \nATOM    173  CD1 LEU A  23      17.347 -30.335  34.461  1.00  0.00           C  \nATOM    174  CD2 LEU A  23      16.585 -30.085  32.113  1.00  0.00           C  \nATOM    175  N   LEU A  24      17.375 -24.654  32.862  1.00  0.00           N  \nATOM    176  CA  LEU A  24      16.751 -23.602  32.082  1.00  0.00           C  \nATOM    177  C   LEU A  24      16.002 -24.277  30.930  1.00  0.00           C  \nATOM    178  O   LEU A  24      16.575 -24.593  29.892  1.00  0.00           O  \nATOM    179  CB  LEU A  24      17.824 -22.604  31.593  1.00  0.00           C  \nATOM    180  CG  LEU A  24      18.167 -21.535  32.641  1.00  0.00           C  \nATOM    181  CD1 LEU A  24      19.618 -21.070  32.487  1.00  0.00           C  \nATOM    182  CD2 LEU A  24      17.264 -20.306  32.493  1.00  0.00           C  \nATOM    183  N   ALA A  25      14.718 -24.529  31.142  1.00  0.00           N  \nATOM    184  CA  ALA A  25      13.806 -24.854  30.058  1.00  0.00           C  \nATOM    185  C   ALA A  25      13.763 -23.644  29.117  1.00  0.00           C  \nATOM    186  O   ALA A  25      13.193 -22.603  29.438  1.00  0.00           O  \nATOM    187  CB  ALA A  25      12.437 -25.208  30.640  1.00  0.00           C  \nATOM    188  N   SER A  26      14.419 -23.758  27.963  1.00  0.00           N  \nATOM    189  CA  SER A  26      14.270 -22.826  26.859  1.00  0.00           C  \nATOM    190  C   SER A  26      12.816 -22.864  26.400  1.00  0.00           C  \nATOM    191  O   SER A  26      12.386 -23.813  25.747  1.00  0.00           O  \nATOM    192  CB  SER A  26      15.231 -23.199  25.724  1.00  0.00           C  \nATOM    193  OG  SER A  26      15.194 -24.586  25.481  1.00  0.00           O  \nATOM    194  N   LEU A  27      12.051 -21.839  26.778  1.00  0.00           N  \nATOM    195  CA  LEU A  27      10.737 -21.604  26.197  1.00  0.00           C  \nATOM    196  C   LEU A  27      10.912 -21.497  24.677  1.00  0.00           C  \nATOM    197  O   LEU A  27      11.803 -20.771  24.223  1.00  0.00           O  \nATOM    198  CB  LEU A  27      10.137 -20.306  26.779  1.00  0.00           C  \nATOM    199  CG  LEU A  27       9.691 -20.421  28.249  1.00  0.00           C  \nATOM    200  CD1 LEU A  27       9.480 -19.023  28.837  1.00  0.00           C  \nATOM    201  CD2 LEU A  27       8.392 -21.203  28.387  1.00  0.00           C  \nATOM    202  N   PRO A  28      10.113 -22.224  23.875  1.00  0.00           N  \nATOM    203  CA  PRO A  28      10.153 -22.044  22.437  1.00  0.00           C  \nATOM    204  C   PRO A  28       9.837 -20.581  22.140  1.00  0.00           C  \nATOM    205  O   PRO A  28       8.860 -20.032  22.649  1.00  0.00           O  \nATOM    206  CB  PRO A  28       9.122 -23.019  21.855  1.00  0.00           C  \nATOM    207  CG  PRO A  28       8.155 -23.245  23.016  1.00  0.00           C  \nATOM    208  CD  PRO A  28       9.017 -23.091  24.264  1.00  0.00           C  \nATOM    209  N   ALA A  29      10.691 -19.937  21.350  1.00  0.00           N  \nATOM    210  CA  ALA A  29      10.437 -18.599  20.846  1.00  0.00           C  \nATOM    211  C   ALA A  29       9.034 -18.590  20.233  1.00  0.00           C  \nATOM    212  O   ALA A  29       8.764 -19.331  19.287  1.00  0.00           O  \nATOM    213  CB  ALA A  29      11.517 -18.226  19.830  1.00  0.00           C  \nATOM    214  N   ALA A  30       8.143 -17.798  20.818  1.00  0.00           N  \nATOM    215  CA  ALA A  30       6.811 -17.611  20.286  1.00  0.00           C  \nATOM    216  C   ALA A  30       6.975 -17.171  18.830  1.00  0.00           C  \nATOM    217  O   ALA A  30       7.562 -16.128  18.550  1.00  0.00           O  \nATOM    218  CB  ALA A  30       6.058 -16.582  21.136  1.00  0.00           C  \nATOM    219  N   ALA A  31       6.501 -18.009  17.907  1.00  0.00           N  \nATOM    220  CA  ALA A  31       6.411 -17.661  16.502  1.00  0.00           C  \nATOM    221  C   ALA A  31       5.720 -16.304  16.423  1.00  0.00           C  \nATOM    222  O   ALA A  31       4.616 -16.139  16.941  1.00  0.00           O  \nATOM    223  CB  ALA A  31       5.635 -18.766  15.773  1.00  0.00           C  \nATOM    224  N   GLN A  32       6.400 -15.313  15.853  1.00  0.00           N  \nATOM    225  CA  GLN A  32       5.790 -14.024  15.575  1.00  0.00           C  \nATOM    226  C   GLN A  32       4.497 -14.293  14.824  1.00  0.00           C  \nATOM    227  O   GLN A  32       4.507 -14.882  13.750  1.00  0.00           O  \nATOM    228  CB  GLN A  32       6.740 -13.132  14.762  1.00  0.00           C  \nATOM    229  CG  GLN A  32       7.881 -12.558  15.623  1.00  0.00           C  \nATOM    230  CD  GLN A  32       8.781 -11.608  14.847  1.00  0.00           C  \nATOM    231  OE1 GLN A  32       8.696 -11.428  13.640  1.00  0.00           O  \nATOM    232  NE2 GLN A  32       9.701 -10.940  15.508  1.00  0.00           N  \nATOM    233  N   GLY A  33       3.384 -13.939  15.466  1.00  0.00           N  \nATOM    234  CA  GLY A  33       2.056 -14.293  15.002  1.00  0.00           C  \nATOM    235  C   GLY A  33       1.882 -13.885  13.548  1.00  0.00           C  \nATOM    236  O   GLY A  33       2.156 -12.751  13.164  1.00  0.00           O  \nATOM    237  N   THR A  34       1.405 -14.819  12.744  1.00  0.00           N  \nATOM    238  CA  THR A  34       0.747 -14.510  11.486  1.00  0.00           C  \nATOM    239  C   THR A  34      -0.216 -13.340  11.718  1.00  0.00           C  \nATOM    240  O   THR A  34      -0.978 -13.395  12.687  1.00  0.00           O  \nATOM    241  CB  THR A  34      -0.053 -15.726  10.985  1.00  0.00           C  \nATOM    242  OG1 THR A  34      -0.613 -16.432  12.064  1.00  0.00           O  \nATOM    243  CG2 THR A  34       0.835 -16.705  10.230  1.00  0.00           C  \nATOM    244  N   PRO A  35      -0.186 -12.270  10.897  1.00  0.00           N  \nATOM    245  CA  PRO A  35      -1.120 -11.168  11.047  1.00  0.00           C  \nATOM    246  C   PRO A  35      -2.539 -11.737  11.082  1.00  0.00           C  \nATOM    247  O   PRO A  35      -2.955 -12.400  10.136  1.00  0.00           O  \nATOM    248  CB  PRO A  35      -0.891 -10.240   9.844  1.00  0.00           C  \nATOM    249  CG  PRO A  35       0.520 -10.603   9.377  1.00  0.00           C  \nATOM    250  CD  PRO A  35       0.668 -12.080   9.753  1.00  0.00           C  \nATOM    251  N   SER A  36      -3.227 -11.527  12.198  1.00  0.00           N  \nATOM    252  CA  SER A  36      -4.588 -11.991  12.397  1.00  0.00           C  \nATOM    253  C   SER A  36      -5.464 -11.559  11.224  1.00  0.00           C  \nATOM    254  O   SER A  36      -5.374 -10.431  10.731  1.00  0.00           O  \nATOM    255  CB  SER A  36      -5.129 -11.472  13.738  1.00  0.00           C  \nATOM    256  OG  SER A  36      -4.944 -10.077  13.846  1.00  0.00           O  \nATOM    257  N   SER A  37      -6.294 -12.473  10.776  1.00  0.00           N  \nATOM    258  CA  SER A  37      -7.246 -12.380   9.666  1.00  0.00           C  \nATOM    259  C   SER A  37      -8.355 -11.351   9.928  1.00  0.00           C  \nATOM    260  O   SER A  37      -9.533 -11.685  10.024  1.00  0.00           O  \nATOM    261  CB  SER A  37      -7.788 -13.793   9.416  1.00  0.00           C  \nATOM    262  OG  SER A  37      -8.103 -14.415  10.647  1.00  0.00           O  \nATOM    263  N   GLY A  38      -7.988 -10.074  10.037  1.00  0.00           N  \nATOM    264  CA  GLY A  38      -8.873  -8.918  10.133  1.00  0.00           C  \nATOM    265  C   GLY A  38      -8.816  -8.012   8.902  1.00  0.00           C  \nATOM    266  O   GLY A  38      -9.103  -6.827   8.987  1.00  0.00           O  \nATOM    267  N   THR A  39      -8.387  -8.523   7.744  1.00  0.00           N  \nATOM    268  CA  THR A  39      -8.163  -7.714   6.536  1.00  0.00           C  \nATOM    269  C   THR A  39      -9.450  -7.271   5.840  1.00  0.00           C  \nATOM    270  O   THR A  39      -9.368  -6.466   4.915  1.00  0.00           O  \nATOM    271  CB  THR A  39      -7.240  -8.436   5.544  1.00  0.00           C  \nATOM    272  OG1 THR A  39      -7.751  -9.710   5.257  1.00  0.00           O  \nATOM    273  CG2 THR A  39      -5.830  -8.614   6.108  1.00  0.00           C  \nATOM    274  N   GLY A  40     -10.615  -7.739   6.279  1.00  0.00           N  \nATOM    275  CA  GLY A  40     -11.900  -7.416   5.669  1.00  0.00           C  \nATOM    276  C   GLY A  40     -11.962  -7.814   4.189  1.00  0.00           C  \nATOM    277  O   GLY A  40     -12.543  -7.082   3.397  1.00  0.00           O  \nATOM    278  N   ALA A  41     -11.316  -8.912   3.808  1.00  0.00           N  \nATOM    279  CA  ALA A  41     -11.449  -9.464   2.468  1.00  0.00           C  \nATOM    280  C   ALA A  41     -12.833 -10.127   2.315  1.00  0.00           C  \nATOM    281  O   ALA A  41     -13.301 -10.762   3.262  1.00  0.00           O  \nATOM    282  CB  ALA A  41     -10.297 -10.448   2.215  1.00  0.00           C  \nATOM    283  N   PRO A  42     -13.502  -9.996   1.157  1.00  0.00           N  \nATOM    284  CA  PRO A  42     -14.780 -10.652   0.911  1.00  0.00           C  \nATOM    285  C   PRO A  42     -14.653 -12.177   0.927  1.00  0.00           C  \nATOM    286  O   PRO A  42     -13.590 -12.744   0.635  1.00  0.00           O  \nATOM    287  CB  PRO A  42     -15.274 -10.132  -0.443  1.00  0.00           C  \nATOM    288  CG  PRO A  42     -14.031  -9.580  -1.131  1.00  0.00           C  \nATOM    289  CD  PRO A  42     -13.081  -9.224   0.004  1.00  0.00           C  \nATOM    290  N   ALA A  43     -15.756 -12.867   1.239  1.00  0.00           N  \nATOM    291  CA  ALA A  43     -15.819 -14.321   1.160  1.00  0.00           C  \nATOM    292  C   ALA A  43     -15.430 -14.805  -0.252  1.00  0.00           C  \nATOM    293  O   ALA A  43     -15.876 -14.246  -1.258  1.00  0.00           O  \nATOM    294  CB  ALA A  43     -17.224 -14.783   1.565  1.00  0.00           C  \nATOM    295  N   GLY A  44     -14.566 -15.811  -0.326  1.00  0.00           N  \nATOM    296  CA  GLY A  44     -14.042 -16.325  -1.593  1.00  0.00           C  \nATOM    297  C   GLY A  44     -12.954 -15.466  -2.258  1.00  0.00           C  \nATOM    298  O   GLY A  44     -12.511 -15.793  -3.359  1.00  0.00           O  \nATOM    299  N   ALA A  45     -12.500 -14.377  -1.620  1.00  0.00           N  \nATOM    300  CA  ALA A  45     -11.300 -13.676  -2.059  1.00  0.00           C  \nATOM    301  C   ALA A  45     -10.047 -14.527  -1.805  1.00  0.00           C  \nATOM    302  O   ALA A  45      -9.962 -15.238  -0.805  1.00  0.00           O  \nATOM    303  CB  ALA A  45     -11.193 -12.323  -1.361  1.00  0.00           C  \nATOM    304  N   SER A  46      -9.056 -14.434  -2.698  1.00  0.00           N  \nATOM    305  CA  SER A  46      -7.763 -15.081  -2.489  1.00  0.00           C  \nATOM    306  C   SER A  46      -7.119 -14.569  -1.196  1.00  0.00           C  \nATOM    307  O   SER A  46      -7.111 -13.356  -0.967  1.00  0.00           O  \nATOM    308  CB  SER A  46      -6.817 -14.827  -3.673  1.00  0.00           C  \nATOM    309  OG  SER A  46      -7.404 -15.319  -4.863  1.00  0.00           O  \nATOM    310  N   PRO A  47      -6.569 -15.446  -0.341  1.00  0.00           N  \nATOM    311  CA  PRO A  47      -5.861 -15.025   0.860  1.00  0.00           C  \nATOM    312  C   PRO A  47      -4.648 -14.165   0.490  1.00  0.00           C  \nATOM    313  O   PRO A  47      -4.100 -14.269  -0.609  1.00  0.00           O  \nATOM    314  CB  PRO A  47      -5.464 -16.315   1.583  1.00  0.00           C  \nATOM    315  CG  PRO A  47      -5.366 -17.336   0.451  1.00  0.00           C  \nATOM    316  CD  PRO A  47      -6.449 -16.881  -0.527  1.00  0.00           C  \nATOM    317  N   ALA A  48      -4.223 -13.309   1.438  1.00  0.00           N  \nATOM    318  CA  ALA A  48      -3.008 -12.537   1.262  1.00  0.00           C  \nATOM    319  C   ALA A  48      -1.814 -13.467   0.998  1.00  0.00           C  \nATOM    320  O   ALA A  48      -1.616 -14.455   1.703  1.00  0.00           O  \nATOM    321  CB  ALA A  48      -2.791 -11.644   2.490  1.00  0.00           C  \nATOM    322  N   VAL A  49      -1.019 -13.148  -0.020  1.00  0.00           N  \nATOM    323  CA  VAL A  49       0.108 -13.972  -0.458  1.00  0.00           C  \nATOM    324  C   VAL A  49       1.439 -13.385   0.006  1.00  0.00           C  \nATOM    325  O   VAL A  49       1.561 -12.185   0.272  1.00  0.00           O  \nATOM    326  CB  VAL A  49       0.109 -14.215  -1.983  1.00  0.00           C  \nATOM    327  CG1 VAL A  49      -1.127 -15.003  -2.423  1.00  0.00           C  \nATOM    328  CG2 VAL A  49       0.185 -12.922  -2.790  1.00  0.00           C  \nATOM    329  N   THR A  50       2.475 -14.230   0.063  1.00  0.00           N  \nATOM    330  CA  THR A  50       3.850 -13.800   0.338  1.00  0.00           C  \nATOM    331  C   THR A  50       4.304 -12.754  -0.679  1.00  0.00           C  \nATOM    332  O   THR A  50       4.168 -12.952  -1.886  1.00  0.00           O  \nATOM    333  CB  THR A  50       4.824 -14.987   0.309  1.00  0.00           C  \nATOM    334  OG1 THR A  50       4.403 -15.962   1.222  1.00  0.00           O  \nATOM    335  CG2 THR A  50       6.252 -14.600   0.696  1.00  0.00           C  \nATOM    336  N   VAL A  51       4.875 -11.657  -0.185  1.00  0.00           N  \nATOM    337  CA  VAL A  51       5.400 -10.559  -1.004  1.00  0.00           C  \nATOM    338  C   VAL A  51       6.917 -10.662  -1.100  1.00  0.00           C  \nATOM    339  O   VAL A  51       7.600 -10.915  -0.107  1.00  0.00           O  \nATOM    340  CB  VAL A  51       4.955  -9.199  -0.455  1.00  0.00           C  \nATOM    341  CG1 VAL A  51       5.545  -8.024  -1.246  1.00  0.00           C  \nATOM    342  CG2 VAL A  51       3.430  -9.079  -0.515  1.00  0.00           C  \nATOM    343  N   ARG A  52       7.448 -10.419  -2.292  1.00  0.00           N  \nATOM    344  CA  ARG A  52       8.868 -10.187  -2.542  1.00  0.00           C  \nATOM    345  C   ARG A  52       9.085  -8.761  -3.042  1.00  0.00           C  \nATOM    346  O   ARG A  52       8.149  -8.129  -3.533  1.00  0.00           O  \nATOM    347  CB  ARG A  52       9.411 -11.225  -3.529  1.00  0.00           C  \nATOM    348  CG  ARG A  52       9.321 -12.653  -2.978  1.00  0.00           C  \nATOM    349  CD  ARG A  52       9.947 -13.627  -3.974  1.00  0.00           C  \nATOM    350  NE  ARG A  52       9.882 -15.023  -3.487  1.00  0.00           N  \nATOM    351  CZ  ARG A  52      10.893 -15.854  -3.288  1.00  0.00           C  \nATOM    352  NH1 ARG A  52      12.134 -15.509  -3.502  1.00  0.00           N  \nATOM    353  NH2 ARG A  52      10.670 -17.060  -2.872  1.00  0.00           N  \nATOM    354  N   ALA A  53      10.314  -8.265  -2.929  1.00  0.00           N  \nATOM    355  CA  ALA A  53      10.667  -6.939  -3.417  1.00  0.00           C  \nATOM    356  C   ALA A  53      12.075  -6.924  -4.016  1.00  0.00           C  \nATOM    357  O   ALA A  53      12.930  -7.725  -3.626  1.00  0.00           O  \nATOM    358  CB  ALA A  53      10.530  -5.924  -2.278  1.00  0.00           C  \nATOM    359  N   PHE A  54      12.318  -6.007  -4.942  1.00  0.00           N  \nATOM    360  CA  PHE A  54      13.642  -5.681  -5.464  1.00  0.00           C  \nATOM    361  C   PHE A  54      13.731  -4.192  -5.821  1.00  0.00           C  \nATOM    362  O   PHE A  54      12.718  -3.556  -6.131  1.00  0.00           O  \nATOM    363  CB  PHE A  54      13.981  -6.572  -6.665  1.00  0.00           C  \nATOM    364  CG  PHE A  54      13.114  -6.354  -7.892  1.00  0.00           C  \nATOM    365  CD1 PHE A  54      11.869  -7.004  -8.003  1.00  0.00           C  \nATOM    366  CD2 PHE A  54      13.555  -5.508  -8.937  1.00  0.00           C  \nATOM    367  CE1 PHE A  54      11.067  -6.816  -9.139  1.00  0.00           C  \nATOM    368  CE2 PHE A  54      12.749  -5.323 -10.068  1.00  0.00           C  \nATOM    369  CZ  PHE A  54      11.508  -5.972 -10.178  1.00  0.00           C  \nATOM    370  N   SER A  55      14.948  -3.636  -5.784  1.00  0.00           N  \nATOM    371  CA  SER A  55      15.196  -2.258  -6.211  1.00  0.00           C  \nATOM    372  C   SER A  55      15.461  -2.200  -7.712  1.00  0.00           C  \nATOM    373  O   SER A  55      16.135  -3.070  -8.267  1.00  0.00           O  \nATOM    374  CB  SER A  55      16.367  -1.650  -5.438  1.00  0.00           C  \nATOM    375  OG  SER A  55      16.073  -1.657  -4.055  1.00  0.00           O  \nATOM    376  N   THR A  56      14.972  -1.162  -8.381  1.00  0.00           N  \nATOM    377  CA  THR A  56      15.290  -0.921  -9.789  1.00  0.00           C  \nATOM    378  C   THR A  56      16.720  -0.388  -9.905  1.00  0.00           C  \nATOM    379  O   THR A  56      17.037   0.681  -9.404  1.00  0.00           O  \nATOM    380  CB  THR A  56      14.305   0.063 -10.440  1.00  0.00           C  \nATOM    381  OG1 THR A  56      14.269   1.272  -9.735  1.00  0.00           O  \nATOM    382  CG2 THR A  56      12.885  -0.501 -10.470  1.00  0.00           C  \nATOM    383  N   VAL A  57      17.601  -1.124 -10.604  1.00  0.00           N  \nATOM    384  CA  VAL A  57      19.039  -0.798 -10.715  1.00  0.00           C  \nATOM    385  C   VAL A  57      19.304   0.372 -11.675  1.00  0.00           C  \nATOM    386  O   VAL A  57      20.287   1.093 -11.518  1.00  0.00           O  \nATOM    387  CB  VAL A  57      19.828  -2.056 -11.141  1.00  0.00           C  \nATOM    388  CG1 VAL A  57      21.342  -1.823 -11.158  1.00  0.00           C  \nATOM    389  CG2 VAL A  57      19.560  -3.244 -10.203  1.00  0.00           C  \nATOM    390  N   ARG A  58      18.438   0.577 -12.673  1.00  0.00           N  \nATOM    391  CA  ARG A  58      18.537   1.669 -13.657  1.00  0.00           C  \nATOM    392  C   ARG A  58      17.156   2.274 -13.893  1.00  0.00           C  \nATOM    393  O   ARG A  58      16.439   1.788 -14.768  1.00  0.00           O  \nATOM    394  CB  ARG A  58      19.168   1.168 -14.970  1.00  0.00           C  \nATOM    395  CG  ARG A  58      20.619   0.696 -14.792  1.00  0.00           C  \nATOM    396  CD  ARG A  58      21.247   0.386 -16.151  1.00  0.00           C  \nATOM    397  NE  ARG A  58      22.612  -0.148 -15.996  1.00  0.00           N  \nATOM    398  CZ  ARG A  58      23.539  -0.252 -16.941  1.00  0.00           C  \nATOM    399  NH1 ARG A  58      23.321   0.149 -18.162  1.00  0.00           N  \nATOM    400  NH2 ARG A  58      24.702  -0.773 -16.674  1.00  0.00           N  \nATOM    401  N   PRO A  59      16.752   3.292 -13.116  1.00  0.00           N  \nATOM    402  CA  PRO A  59      15.446   3.920 -13.284  1.00  0.00           C  \nATOM    403  C   PRO A  59      15.324   4.616 -14.640  1.00  0.00           C  \nATOM    404  O   PRO A  59      16.190   5.389 -15.038  1.00  0.00           O  \nATOM    405  CB  PRO A  59      15.288   4.893 -12.113  1.00  0.00           C  \nATOM    406  CG  PRO A  59      16.283   4.374 -11.082  1.00  0.00           C  \nATOM    407  CD  PRO A  59      17.414   3.820 -11.935  1.00  0.00           C  \nATOM    408  N   GLY A  60      14.239   4.342 -15.368  1.00  0.00           N  \nATOM    409  CA  GLY A  60      13.935   4.922 -16.666  1.00  0.00           C  \nATOM    410  C   GLY A  60      13.045   6.159 -16.557  1.00  0.00           C  \nATOM    411  O   GLY A  60      11.860   6.085 -16.867  1.00  0.00           O  \nATOM    412  N   GLY A  61      13.597   7.296 -16.123  1.00  0.00           N  \nATOM    413  CA  GLY A  61      12.871   8.579 -16.071  1.00  0.00           C  \nATOM    414  C   GLY A  61      12.324   8.985 -14.695  1.00  0.00           C  \nATOM    415  O   GLY A  61      11.453   9.846 -14.633  1.00  0.00           O  \nATOM    416  N   PHE A  62      12.830   8.387 -13.610  1.00  0.00           N  \nATOM    417  CA  PHE A  62      12.592   8.747 -12.204  1.00  0.00           C  \nATOM    418  C   PHE A  62      13.841   8.430 -11.363  1.00  0.00           C  \nATOM    419  O   PHE A  62      14.748   7.753 -11.846  1.00  0.00           O  \nATOM    420  CB  PHE A  62      11.345   8.016 -11.678  1.00  0.00           C  \nATOM    421  CG  PHE A  62      11.401   6.507 -11.759  1.00  0.00           C  \nATOM    422  CD1 PHE A  62      11.024   5.843 -12.946  1.00  0.00           C  \nATOM    423  CD2 PHE A  62      11.819   5.743 -10.644  1.00  0.00           C  \nATOM    424  CE1 PHE A  62      11.071   4.447 -13.018  1.00  0.00           C  \nATOM    425  CE2 PHE A  62      11.862   4.339 -10.719  1.00  0.00           C  \nATOM    426  CZ  PHE A  62      11.490   3.700 -11.908  1.00  0.00           C  \nATOM    427  N   GLY A  63      13.923   8.921 -10.126  1.00  0.00           N  \nATOM    428  CA  GLY A  63      15.152   8.887  -9.322  1.00  0.00           C  \nATOM    429  C   GLY A  63      15.555   7.503  -8.815  1.00  0.00           C  \nATOM    430  O   GLY A  63      16.688   7.071  -9.011  1.00  0.00           O  \nATOM    431  N   ALA A  64      14.638   6.789  -8.168  1.00  0.00           N  \nATOM    432  CA  ALA A  64      14.832   5.411  -7.701  1.00  0.00           C  \nATOM    433  C   ALA A  64      13.481   4.707  -7.559  1.00  0.00           C  \nATOM    434  O   ALA A  64      12.440   5.358  -7.497  1.00  0.00           O  \nATOM    435  CB  ALA A  64      15.583   5.419  -6.362  1.00  0.00           C  \nATOM    436  N   GLY A  65      13.475   3.380  -7.483  1.00  0.00           N  \nATOM    437  CA  GLY A  65      12.238   2.641  -7.288  1.00  0.00           C  \nATOM    438  C   GLY A  65      12.432   1.301  -6.590  1.00  0.00           C  \nATOM    439  O   GLY A  65      13.454   0.638  -6.748  1.00  0.00           O  \nATOM    440  N   THR A  66      11.421   0.906  -5.841  1.00  0.00           N  \nATOM    441  CA  THR A  66      11.294  -0.431  -5.254  1.00  0.00           C  \nATOM    442  C   THR A  66      10.029  -1.089  -5.788  1.00  0.00           C  \nATOM    443  O   THR A  66       8.940  -0.526  -5.676  1.00  0.00           O  \nATOM    444  CB  THR A  66      11.255  -0.375  -3.727  1.00  0.00           C  \nATOM    445  OG1 THR A  66      12.396   0.288  -3.234  1.00  0.00           O  \nATOM    446  CG2 THR A  66      11.243  -1.769  -3.093  1.00  0.00           C  \nATOM    447  N   VAL A  67      10.178  -2.270  -6.365  1.00  0.00           N  \nATOM    448  CA  VAL A  67       9.069  -3.081  -6.882  1.00  0.00           C  \nATOM    449  C   VAL A  67       8.723  -4.153  -5.865  1.00  0.00           C  \nATOM    450  O   VAL A  67       9.579  -4.956  -5.490  1.00  0.00           O  \nATOM    451  CB  VAL A  67       9.418  -3.693  -8.252  1.00  0.00           C  \nATOM    452  CG1 VAL A  67       8.238  -4.510  -8.800  1.00  0.00           C  \nATOM    453  CG2 VAL A  67       9.769  -2.618  -9.276  1.00  0.00           C  \nATOM    454  N   TYR A  68       7.470  -4.183  -5.453  1.00  0.00           N  \nATOM    455  CA  TYR A  68       6.871  -5.218  -4.621  1.00  0.00           C  \nATOM    456  C   TYR A  68       5.957  -6.081  -5.490  1.00  0.00           C  \nATOM    457  O   TYR A  68       5.206  -5.554  -6.307  1.00  0.00           O  \nATOM    458  CB  TYR A  68       6.067  -4.594  -3.479  1.00  0.00           C  \nATOM    459  CG  TYR A  68       6.843  -3.624  -2.614  1.00  0.00           C  \nATOM    460  CD1 TYR A  68       7.423  -4.054  -1.405  1.00  0.00           C  \nATOM    461  CD2 TYR A  68       6.998  -2.280  -3.023  1.00  0.00           C  \nATOM    462  CE1 TYR A  68       8.156  -3.156  -0.609  1.00  0.00           C  \nATOM    463  CE2 TYR A  68       7.734  -1.388  -2.242  1.00  0.00           C  \nATOM    464  CZ  TYR A  68       8.314  -1.818  -1.036  1.00  0.00           C  \nATOM    465  OH  TYR A  68       9.038  -0.948  -0.277  1.00  0.00           O  \nATOM    466  N   TYR A  69       6.001  -7.396  -5.311  1.00  0.00           N  \nATOM    467  CA  TYR A  69       5.198  -8.308  -6.122  1.00  0.00           C  \nATOM    468  C   TYR A  69       4.806  -9.572  -5.347  1.00  0.00           C  \nATOM    469  O   TYR A  69       5.553 -10.021  -4.474  1.00  0.00           O  \nATOM    470  CB  TYR A  69       5.948  -8.641  -7.424  1.00  0.00           C  \nATOM    471  CG  TYR A  69       7.215  -9.447  -7.244  1.00  0.00           C  \nATOM    472  CD1 TYR A  69       8.401  -8.817  -6.818  1.00  0.00           C  \nATOM    473  CD2 TYR A  69       7.209 -10.831  -7.502  1.00  0.00           C  \nATOM    474  CE1 TYR A  69       9.580  -9.574  -6.646  1.00  0.00           C  \nATOM    475  CE2 TYR A  69       8.385 -11.588  -7.338  1.00  0.00           C  \nATOM    476  CZ  TYR A  69       9.571 -10.955  -6.905  1.00  0.00           C  \nATOM    477  OH  TYR A  69      10.703 -11.688  -6.740  1.00  0.00           O  \nATOM    478  N   PRO A  70       3.645 -10.166  -5.659  1.00  0.00           N  \nATOM    479  CA  PRO A  70       3.237 -11.414  -5.064  1.00  0.00           C  \nATOM    480  C   PRO A  70       4.155 -12.557  -5.531  1.00  0.00           C  \nATOM    481  O   PRO A  70       4.433 -12.695  -6.728  1.00  0.00           O  \nATOM    482  CB  PRO A  70       1.789 -11.606  -5.501  1.00  0.00           C  \nATOM    483  CG  PRO A  70       1.704 -10.866  -6.837  1.00  0.00           C  \nATOM    484  CD  PRO A  70       2.666  -9.709  -6.644  1.00  0.00           C  \nATOM    485  N   ALA A  71       4.621 -13.394  -4.601  1.00  0.00           N  \nATOM    486  CA  ALA A  71       5.450 -14.550  -4.934  1.00  0.00           C  \nATOM    487  C   ALA A  71       4.701 -15.555  -5.819  1.00  0.00           C  \nATOM    488  O   ALA A  71       5.284 -16.114  -6.751  1.00  0.00           O  \nATOM    489  CB  ALA A  71       5.946 -15.196  -3.639  1.00  0.00           C  \nATOM    490  N   ALA A  72       3.396 -15.730  -5.556  1.00  0.00           N  \nATOM    491  CA  ALA A  72       2.490 -16.549  -6.348  1.00  0.00           C  \nATOM    492  C   ALA A  72       1.110 -15.878  -6.390  1.00  0.00           C  \nATOM    493  O   ALA A  72       0.484 -15.720  -5.345  1.00  0.00           O  \nATOM    494  CB  ALA A  72       2.420 -17.944  -5.716  1.00  0.00           C  \nATOM    495  N   ALA A  73       0.651 -15.473  -7.572  1.00  0.00           N  \nATOM    496  CA  ALA A  73      -0.652 -14.814  -7.749  1.00  0.00           C  \nATOM    497  C   ALA A  73      -1.660 -15.644  -8.569  1.00  0.00           C  \nATOM    498  O   ALA A  73      -2.779 -15.203  -8.771  1.00  0.00           O  \nATOM    499  CB  ALA A  73      -0.423 -13.427  -8.353  1.00  0.00           C  \nATOM    500  N   GLY A  74      -1.261 -16.810  -9.070  1.00  0.00           N  \nATOM    501  CA  GLY A  74      -2.128 -17.657  -9.909  1.00  0.00           C  \nATOM    502  C   GLY A  74      -2.410 -17.109 -11.317  1.00  0.00           C  \nATOM    503  O   GLY A  74      -3.165 -17.726 -12.059  1.00  0.00           O  \nATOM    504  N   GLY A  75      -1.802 -15.976 -11.696  1.00  0.00           N  \nATOM    505  CA  GLY A  75      -1.964 -15.331 -12.993  1.00  0.00           C  \nATOM    506  C   GLY A  75      -1.309 -13.946 -13.029  1.00  0.00           C  \nATOM    507  O   GLY A  75      -0.749 -13.496 -12.030  1.00  0.00           O  \nATOM    508  N   ALA A  76      -1.377 -13.264 -14.176  1.00  0.00           N  \nATOM    509  CA  ALA A  76      -0.904 -11.895 -14.309  1.00  0.00           C  \nATOM    510  C   ALA A  76      -1.794 -10.919 -13.521  1.00  0.00           C  \nATOM    511  O   ALA A  76      -3.022 -11.024 -13.530  1.00  0.00           O  \nATOM    512  CB  ALA A  76      -0.807 -11.529 -15.794  1.00  0.00           C  \nATOM    513  N   VAL A  77      -1.165  -9.945 -12.852  1.00  0.00           N  \nATOM    514  CA  VAL A  77      -1.838  -8.963 -11.989  1.00  0.00           C  \nATOM    515  C   VAL A  77      -1.614  -7.538 -12.493  1.00  0.00           C  \nATOM    516  O   VAL A  77      -0.647  -7.265 -13.205  1.00  0.00           O  \nATOM    517  CB  VAL A  77      -1.409  -9.109 -10.523  1.00  0.00           C  \nATOM    518  CG1 VAL A  77      -1.742 -10.502  -9.985  1.00  0.00           C  \nATOM    519  CG2 VAL A  77       0.078  -8.821 -10.278  1.00  0.00           C  \nATOM    520  N   GLY A  78      -2.499  -6.612 -12.113  1.00  0.00           N  \nATOM    521  CA  GLY A  78      -2.329  -5.195 -12.433  1.00  0.00           C  \nATOM    522  C   GLY A  78      -1.120  -4.567 -11.731  1.00  0.00           C  \nATOM    523  O   GLY A  78      -0.638  -5.087 -10.720  1.00  0.00           O  \nATOM    524  N   ALA A  79      -0.635  -3.429 -12.253  1.00  0.00           N  \nATOM    525  CA  ALA A  79       0.485  -2.682 -11.678  1.00  0.00           C  \nATOM    526  C   ALA A  79       0.063  -1.303 -11.151  1.00  0.00           C  \nATOM    527  O   ALA A  79      -0.728  -0.602 -11.776  1.00  0.00           O  \nATOM    528  CB  ALA A  79       1.615  -2.580 -12.700  1.00  0.00           C  \nATOM    529  N   ILE A  80       0.630  -0.889 -10.002  1.00  0.00           N  \nATOM    530  CA  ILE A  80       0.340   0.386  -9.343  1.00  0.00           C  \nATOM    531  C   ILE A  80       1.657   1.141  -9.121  1.00  0.00           C  \nATOM    532  O   ILE A  80       2.579   0.599  -8.514  1.00  0.00           O  \nATOM    533  CB  ILE A  80      -0.420   0.166  -8.016  1.00  0.00           C  \nATOM    534  CG1 ILE A  80      -1.746  -0.601  -8.233  1.00  0.00           C  \nATOM    535  CG2 ILE A  80      -0.707   1.521  -7.332  1.00  0.00           C  \nATOM    536  CD1 ILE A  80      -2.382  -1.121  -6.939  1.00  0.00           C  \nATOM    537  N   ALA A  81       1.735   2.406  -9.553  1.00  0.00           N  \nATOM    538  CA  ALA A  81       2.847   3.288  -9.220  1.00  0.00           C  \nATOM    539  C   ALA A  81       2.455   4.295  -8.134  1.00  0.00           C  \nATOM    540  O   ALA A  81       1.390   4.910  -8.200  1.00  0.00           O  \nATOM    541  CB  ALA A  81       3.369   3.990 -10.476  1.00  0.00           C  \nATOM    542  N   ILE A  82       3.335   4.486  -7.138  1.00  0.00           N  \nATOM    543  CA  ILE A  82       3.091   5.338  -5.968  1.00  0.00           C  \nATOM    544  C   ILE A  82       4.217   6.359  -5.821  1.00  0.00           C  \nATOM    545  O   ILE A  82       5.396   6.001  -5.795  1.00  0.00           O  \nATOM    546  CB  ILE A  82       2.909   4.506  -4.682  1.00  0.00           C  \nATOM    547  CG1 ILE A  82       2.025   3.257  -4.922  1.00  0.00           C  \nATOM    548  CG2 ILE A  82       2.324   5.415  -3.583  1.00  0.00           C  \nATOM    549  CD1 ILE A  82       1.661   2.476  -3.661  1.00  0.00           C  \nATOM    550  N   VAL A  83       3.846   7.648  -5.703  1.00  0.00           N  \nATOM    551  CA  VAL A  83       4.779   8.786  -5.675  1.00  0.00           C  \nATOM    552  C   VAL A  83       4.664   9.541  -4.344  1.00  0.00           C  \nATOM    553  O   VAL A  83       3.551   9.822  -3.902  1.00  0.00           O  \nATOM    554  CB  VAL A  83       4.524   9.729  -6.863  1.00  0.00           C  \nATOM    555  CG1 VAL A  83       5.621  10.796  -6.981  1.00  0.00           C  \nATOM    556  CG2 VAL A  83       4.486   8.976  -8.190  1.00  0.00           C  \nATOM    557  N   PRO A  84       5.784   9.911  -3.677  1.00  0.00           N  \nATOM    558  CA  PRO A  84       5.754  10.670  -2.431  1.00  0.00           C  \nATOM    559  C   PRO A  84       5.419  12.155  -2.656  1.00  0.00           C  \nATOM    560  O   PRO A  84       5.307  12.632  -3.787  1.00  0.00           O  \nATOM    561  CB  PRO A  84       7.145  10.465  -1.829  1.00  0.00           C  \nATOM    562  CG  PRO A  84       8.043  10.348  -3.053  1.00  0.00           C  \nATOM    563  CD  PRO A  84       7.160   9.604  -4.050  1.00  0.00           C  \nATOM    564  N   GLY A  85       5.284  12.910  -1.566  1.00  0.00           N  \nATOM    565  CA  GLY A  85       5.114  14.361  -1.573  1.00  0.00           C  \nATOM    566  C   GLY A  85       6.428  15.148  -1.710  1.00  0.00           C  \nATOM    567  O   GLY A  85       7.516  14.577  -1.822  1.00  0.00           O  \nATOM    568  N   TYR A  86       6.316  16.478  -1.690  1.00  0.00           N  \nATOM    569  CA  TYR A  86       7.443  17.411  -1.685  1.00  0.00           C  \nATOM    570  C   TYR A  86       8.377  17.123  -0.501  1.00  0.00           C  \nATOM    571  O   TYR A  86       7.903  16.796   0.579  1.00  0.00           O  \nATOM    572  CB  TYR A  86       6.891  18.837  -1.628  1.00  0.00           C  \nATOM    573  CG  TYR A  86       7.911  19.927  -1.870  1.00  0.00           C  \nATOM    574  CD1 TYR A  86       8.463  20.645  -0.791  1.00  0.00           C  \nATOM    575  CD2 TYR A  86       8.284  20.258  -3.192  1.00  0.00           C  \nATOM    576  CE1 TYR A  86       9.373  21.686  -1.025  1.00  0.00           C  \nATOM    577  CE2 TYR A  86       9.197  21.301  -3.426  1.00  0.00           C  \nATOM    578  CZ  TYR A  86       9.738  22.020  -2.347  1.00  0.00           C  \nATOM    579  OH  TYR A  86      10.614  23.035  -2.576  1.00  0.00           O  \nATOM    580  N   THR A  87       9.690  17.192  -0.723  1.00  0.00           N  \nATOM    581  CA  THR A  87      10.785  16.830   0.209  1.00  0.00           C  \nATOM    582  C   THR A  87      10.828  15.356   0.662  1.00  0.00           C  \nATOM    583  O   THR A  87      11.791  14.952   1.311  1.00  0.00           O  \nATOM    584  CB  THR A  87      10.882  17.761   1.429  1.00  0.00           C  \nATOM    585  OG1 THR A  87       9.798  17.583   2.297  1.00  0.00           O  \nATOM    586  CG2 THR A  87      10.956  19.244   1.048  1.00  0.00           C  \nATOM    587  N   ALA A  88       9.841  14.547   0.302  1.00  0.00           N  \nATOM    588  CA  ALA A  88       9.689  13.186   0.791  1.00  0.00           C  \nATOM    589  C   ALA A  88      10.255  12.135  -0.164  1.00  0.00           C  \nATOM    590  O   ALA A  88      10.496  12.369  -1.344  1.00  0.00           O  \nATOM    591  CB  ALA A  88       8.211  12.963   1.125  1.00  0.00           C  \nATOM    592  N   ARG A  89      10.456  10.924   0.386  1.00  0.00           N  \nATOM    593  CA  ARG A  89      10.977   9.742  -0.305  1.00  0.00           C  \nATOM    594  C   ARG A  89      10.039   8.548  -0.150  1.00  0.00           C  \nATOM    595  O   ARG A  89       9.044   8.618   0.576  1.00  0.00           O  \nATOM    596  CB  ARG A  89      12.383   9.430   0.233  1.00  0.00           C  \nATOM    597  CG  ARG A  89      13.400  10.509  -0.140  1.00  0.00           C  \nATOM    598  CD  ARG A  89      14.817  10.074   0.271  1.00  0.00           C  \nATOM    599  NE  ARG A  89      15.828  11.064  -0.122  1.00  0.00           N  \nATOM    600  CZ  ARG A  89      16.366  11.206  -1.322  1.00  0.00           C  \nATOM    601  NH1 ARG A  89      16.038  10.439  -2.335  1.00  0.00           N  \nATOM    602  NH2 ARG A  89      17.256  12.129  -1.531  1.00  0.00           N  \nATOM    603  N   GLN A  90      10.409   7.420  -0.763  1.00  0.00           N  \nATOM    604  CA  GLN A  90       9.684   6.148  -0.682  1.00  0.00           C  \nATOM    605  C   GLN A  90       9.301   5.746   0.747  1.00  0.00           C  \nATOM    606  O   GLN A  90       8.247   5.139   0.952  1.00  0.00           O  \nATOM    607  CB  GLN A  90      10.564   5.027  -1.277  1.00  0.00           C  \nATOM    608  CG  GLN A  90      10.694   5.103  -2.793  1.00  0.00           C  \nATOM    609  CD  GLN A  90      11.600   3.991  -3.308  1.00  0.00           C  \nATOM    610  OE1 GLN A  90      11.202   2.841  -3.427  1.00  0.00           O  \nATOM    611  NE2 GLN A  90      12.848   4.273  -3.593  1.00  0.00           N  \nATOM    612  N   SER A  91      10.116   6.103   1.746  1.00  0.00           N  \nATOM    613  CA  SER A  91       9.846   5.794   3.154  1.00  0.00           C  \nATOM    614  C   SER A  91       8.473   6.284   3.627  1.00  0.00           C  \nATOM    615  O   SER A  91       7.847   5.597   4.430  1.00  0.00           O  \nATOM    616  CB  SER A  91      10.941   6.380   4.049  1.00  0.00           C  \nATOM    617  OG  SER A  91      11.042   7.780   3.853  1.00  0.00           O  \nATOM    618  N   SER A  92       7.964   7.408   3.098  1.00  0.00           N  \nATOM    619  CA  SER A  92       6.661   7.970   3.451  1.00  0.00           C  \nATOM    620  C   SER A  92       5.482   7.115   2.959  1.00  0.00           C  \nATOM    621  O   SER A  92       4.452   7.057   3.622  1.00  0.00           O  \nATOM    622  CB  SER A  92       6.538   9.401   2.901  1.00  0.00           C  \nATOM    623  OG  SER A  92       6.584   9.414   1.491  1.00  0.00           O  \nATOM    624  N   ILE A  93       5.636   6.393   1.847  1.00  0.00           N  \nATOM    625  CA  ILE A  93       4.566   5.659   1.155  1.00  0.00           C  \nATOM    626  C   ILE A  93       4.796   4.142   1.089  1.00  0.00           C  \nATOM    627  O   ILE A  93       3.925   3.397   0.642  1.00  0.00           O  \nATOM    628  CB  ILE A  93       4.361   6.242  -0.257  1.00  0.00           C  \nATOM    629  CG1 ILE A  93       5.680   6.226  -1.060  1.00  0.00           C  \nATOM    630  CG2 ILE A  93       3.749   7.658  -0.172  1.00  0.00           C  \nATOM    631  CD1 ILE A  93       5.556   6.591  -2.540  1.00  0.00           C  \nATOM    632  N   LYS A  94       5.966   3.653   1.540  1.00  0.00           N  \nATOM    633  CA  LYS A  94       6.405   2.258   1.338  1.00  0.00           C  \nATOM    634  C   LYS A  94       5.466   1.196   1.908  1.00  0.00           C  \nATOM    635  O   LYS A  94       5.470   0.082   1.413  1.00  0.00           O  \nATOM    636  CB  LYS A  94       7.831   2.061   1.879  1.00  0.00           C  \nATOM    637  CG  LYS A  94       7.923   2.101   3.418  1.00  0.00           C  \nATOM    638  CD  LYS A  94       9.355   1.903   3.915  1.00  0.00           C  \nATOM    639  CE  LYS A  94       9.360   1.909   5.451  1.00  0.00           C  \nATOM    640  NZ  LYS A  94      10.711   1.652   6.001  1.00  0.00           N  \nATOM    641  N   TRP A  95       4.658   1.528   2.929  1.00  0.00           N  \nATOM    642  CA  TRP A  95       3.745   0.561   3.545  1.00  0.00           C  \nATOM    643  C   TRP A  95       2.685   0.028   2.572  1.00  0.00           C  \nATOM    644  O   TRP A  95       2.251  -1.120   2.695  1.00  0.00           O  \nATOM    645  CB  TRP A  95       3.070   1.194   4.766  1.00  0.00           C  \nATOM    646  CG  TRP A  95       1.912   2.099   4.470  1.00  0.00           C  \nATOM    647  CD1 TRP A  95       1.987   3.405   4.136  1.00  0.00           C  \nATOM    648  CD2 TRP A  95       0.491   1.772   4.504  1.00  0.00           C  \nATOM    649  NE1 TRP A  95       0.712   3.918   3.965  1.00  0.00           N  \nATOM    650  CE2 TRP A  95      -0.244   2.954   4.197  1.00  0.00           C  \nATOM    651  CE3 TRP A  95      -0.240   0.607   4.782  1.00  0.00           C  \nATOM    652  CZ2 TRP A  95      -1.646   2.984   4.201  1.00  0.00           C  \nATOM    653  CZ3 TRP A  95      -1.648   0.620   4.784  1.00  0.00           C  \nATOM    654  CH2 TRP A  95      -2.355   1.807   4.496  1.00  0.00           C  \nATOM    655  N   TRP A  96       2.326   0.827   1.561  1.00  0.00           N  \nATOM    656  CA  TRP A  96       1.423   0.417   0.493  1.00  0.00           C  \nATOM    657  C   TRP A  96       1.995  -0.701  -0.376  1.00  0.00           C  \nATOM    658  O   TRP A  96       1.225  -1.520  -0.867  1.00  0.00           O  \nATOM    659  CB  TRP A  96       1.094   1.630  -0.382  1.00  0.00           C  \nATOM    660  CG  TRP A  96       0.091   2.567   0.203  1.00  0.00           C  \nATOM    661  CD1 TRP A  96       0.286   3.877   0.510  1.00  0.00           C  \nATOM    662  CD2 TRP A  96      -1.308   2.299   0.492  1.00  0.00           C  \nATOM    663  NE1 TRP A  96      -0.890   4.429   0.954  1.00  0.00           N  \nATOM    664  CE2 TRP A  96      -1.911   3.505   0.959  1.00  0.00           C  \nATOM    665  CE3 TRP A  96      -2.134   1.157   0.392  1.00  0.00           C  \nATOM    666  CZ2 TRP A  96      -3.279   3.579   1.283  1.00  0.00           C  \nATOM    667  CZ3 TRP A  96      -3.501   1.222   0.725  1.00  0.00           C  \nATOM    668  CH2 TRP A  96      -4.069   2.429   1.167  1.00  0.00           C  \nATOM    669  N   GLY A  97       3.320  -0.752  -0.528  1.00  0.00           N  \nATOM    670  CA  GLY A  97       4.008  -1.770  -1.318  1.00  0.00           C  \nATOM    671  C   GLY A  97       3.575  -3.190  -0.930  1.00  0.00           C  \nATOM    672  O   GLY A  97       2.840  -3.823  -1.684  1.00  0.00           O  \nATOM    673  N   PRO A  98       3.956  -3.676   0.260  1.00  0.00           N  \nATOM    674  CA  PRO A  98       3.552  -5.002   0.711  1.00  0.00           C  \nATOM    675  C   PRO A  98       2.040  -5.125   0.942  1.00  0.00           C  \nATOM    676  O   PRO A  98       1.479  -6.196   0.713  1.00  0.00           O  \nATOM    677  CB  PRO A  98       4.356  -5.246   1.994  1.00  0.00           C  \nATOM    678  CG  PRO A  98       4.680  -3.844   2.505  1.00  0.00           C  \nATOM    679  CD  PRO A  98       4.849  -3.049   1.216  1.00  0.00           C  \nATOM    680  N   ARG A  99       1.344  -4.040   1.357  1.00  0.00           N  \nATOM    681  CA  ARG A  99      -0.103  -4.080   1.601  1.00  0.00           C  \nATOM    682  C   ARG A  99      -0.891  -4.405   0.332  1.00  0.00           C  \nATOM    683  O   ARG A  99      -1.775  -5.253   0.386  1.00  0.00           O  \nATOM    684  CB  ARG A  99      -0.566  -2.757   2.231  1.00  0.00           C  \nATOM    685  CG  ARG A  99      -2.087  -2.688   2.521  1.00  0.00           C  \nATOM    686  CD  ARG A  99      -2.561  -3.792   3.471  1.00  0.00           C  \nATOM    687  NE  ARG A  99      -3.996  -3.668   3.778  1.00  0.00           N  \nATOM    688  CZ  ARG A  99      -4.636  -4.275   4.761  1.00  0.00           C  \nATOM    689  NH1 ARG A  99      -4.047  -5.093   5.585  1.00  0.00           N  \nATOM    690  NH2 ARG A  99      -5.913  -4.064   4.945  1.00  0.00           N  \nATOM    691  N   LEU A 100      -0.591  -3.758  -0.787  1.00  0.00           N  \nATOM    692  CA  LEU A 100      -1.260  -3.995  -2.062  1.00  0.00           C  \nATOM    693  C   LEU A 100      -0.749  -5.269  -2.750  1.00  0.00           C  \nATOM    694  O   LEU A 100      -1.546  -6.066  -3.238  1.00  0.00           O  \nATOM    695  CB  LEU A 100      -1.074  -2.761  -2.961  1.00  0.00           C  \nATOM    696  CG  LEU A 100      -1.841  -1.506  -2.484  1.00  0.00           C  \nATOM    697  CD1 LEU A 100      -1.274  -0.263  -3.175  1.00  0.00           C  \nATOM    698  CD2 LEU A 100      -3.328  -1.600  -2.819  1.00  0.00           C  \nATOM    699  N   ALA A 101       0.572  -5.500  -2.722  1.00  0.00           N  \nATOM    700  CA  ALA A 101       1.166  -6.662  -3.376  1.00  0.00           C  \nATOM    701  C   ALA A 101       0.692  -7.994  -2.762  1.00  0.00           C  \nATOM    702  O   ALA A 101       0.446  -8.949  -3.502  1.00  0.00           O  \nATOM    703  CB  ALA A 101       2.691  -6.532  -3.337  1.00  0.00           C  \nATOM    704  N   SER A 102       0.465  -8.041  -1.448  1.00  0.00           N  \nATOM    705  CA  SER A 102      -0.108  -9.224  -0.803  1.00  0.00           C  \nATOM    706  C   SER A 102      -1.546  -9.530  -1.244  1.00  0.00           C  \nATOM    707  O   SER A 102      -2.002 -10.657  -1.053  1.00  0.00           O  \nATOM    708  CB  SER A 102      -0.047  -9.102   0.726  1.00  0.00           C  \nATOM    709  OG  SER A 102      -0.856  -8.036   1.196  1.00  0.00           O  \nATOM    710  N   HIS A 103      -2.244  -8.575  -1.866  1.00  0.00           N  \nATOM    711  CA  HIS A 103      -3.589  -8.746  -2.418  1.00  0.00           C  \nATOM    712  C   HIS A 103      -3.592  -8.888  -3.949  1.00  0.00           C  \nATOM    713  O   HIS A 103      -4.643  -8.773  -4.578  1.00  0.00           O  \nATOM    714  CB  HIS A 103      -4.504  -7.622  -1.919  1.00  0.00           C  \nATOM    715  CG  HIS A 103      -4.717  -7.633  -0.426  1.00  0.00           C  \nATOM    716  ND1 HIS A 103      -3.840  -7.158   0.516  1.00  0.00           N  \nATOM    717  CD2 HIS A 103      -5.797  -8.141   0.257  1.00  0.00           C  \nATOM    718  CE1 HIS A 103      -4.372  -7.380   1.731  1.00  0.00           C  \nATOM    719  NE2 HIS A 103      -5.570  -7.968   1.628  1.00  0.00           N  \nATOM    720  N   GLY A 104      -2.438  -9.164  -4.556  1.00  0.00           N  \nATOM    721  CA  GLY A 104      -2.354  -9.488  -5.974  1.00  0.00           C  \nATOM    722  C   GLY A 104      -2.167  -8.283  -6.891  1.00  0.00           C  \nATOM    723  O   GLY A 104      -2.851  -8.173  -7.902  1.00  0.00           O  \nATOM    724  N   PHE A 105      -1.231  -7.390  -6.563  1.00  0.00           N  \nATOM    725  CA  PHE A 105      -0.766  -6.322  -7.445  1.00  0.00           C  \nATOM    726  C   PHE A 105       0.759  -6.300  -7.522  1.00  0.00           C  \nATOM    727  O   PHE A 105       1.440  -6.611  -6.550  1.00  0.00           O  \nATOM    728  CB  PHE A 105      -1.297  -4.959  -6.956  1.00  0.00           C  \nATOM    729  CG  PHE A 105      -2.812  -4.860  -6.963  1.00  0.00           C  \nATOM    730  CD1 PHE A 105      -3.499  -4.701  -8.181  1.00  0.00           C  \nATOM    731  CD2 PHE A 105      -3.531  -4.967  -5.758  1.00  0.00           C  \nATOM    732  CE1 PHE A 105      -4.899  -4.651  -8.193  1.00  0.00           C  \nATOM    733  CE2 PHE A 105      -4.940  -4.911  -5.773  1.00  0.00           C  \nATOM    734  CZ  PHE A 105      -5.623  -4.759  -6.992  1.00  0.00           C  \nATOM    735  N   VAL A 106       1.323  -5.885  -8.656  1.00  0.00           N  \nATOM    736  CA  VAL A 106       2.706  -5.412  -8.705  1.00  0.00           C  \nATOM    737  C   VAL A 106       2.704  -3.934  -8.346  1.00  0.00           C  \nATOM    738  O   VAL A 106       1.953  -3.158  -8.923  1.00  0.00           O  \nATOM    739  CB  VAL A 106       3.364  -5.687 -10.067  1.00  0.00           C  \nATOM    740  CG1 VAL A 106       4.731  -5.009 -10.203  1.00  0.00           C  \nATOM    741  CG2 VAL A 106       3.584  -7.199 -10.232  1.00  0.00           C  \nATOM    742  N   VAL A 107       3.512  -3.526  -7.376  1.00  0.00           N  \nATOM    743  CA  VAL A 107       3.532  -2.151  -6.865  1.00  0.00           C  \nATOM    744  C   VAL A 107       4.935  -1.592  -7.003  1.00  0.00           C  \nATOM    745  O   VAL A 107       5.878  -2.209  -6.522  1.00  0.00           O  \nATOM    746  CB  VAL A 107       3.060  -2.069  -5.406  1.00  0.00           C  \nATOM    747  CG1 VAL A 107       2.939  -0.617  -4.939  1.00  0.00           C  \nATOM    748  CG2 VAL A 107       1.692  -2.741  -5.214  1.00  0.00           C  \nATOM    749  N   VAL A 108       5.076  -0.416  -7.597  1.00  0.00           N  \nATOM    750  CA  VAL A 108       6.343   0.313  -7.621  1.00  0.00           C  \nATOM    751  C   VAL A 108       6.202   1.594  -6.805  1.00  0.00           C  \nATOM    752  O   VAL A 108       5.398   2.464  -7.123  1.00  0.00           O  \nATOM    753  CB  VAL A 108       6.855   0.530  -9.065  1.00  0.00           C  \nATOM    754  CG1 VAL A 108       5.897   1.307  -9.966  1.00  0.00           C  \nATOM    755  CG2 VAL A 108       8.229   1.215  -9.072  1.00  0.00           C  \nATOM    756  N   THR A 109       6.986   1.726  -5.741  1.00  0.00           N  \nATOM    757  CA  THR A 109       7.184   3.005  -5.047  1.00  0.00           C  \nATOM    758  C   THR A 109       8.404   3.690  -5.619  1.00  0.00           C  \nATOM    759  O   THR A 109       9.411   3.019  -5.863  1.00  0.00           O  \nATOM    760  CB  THR A 109       7.334   2.839  -3.526  1.00  0.00           C  \nATOM    761  OG1 THR A 109       8.417   1.998  -3.201  1.00  0.00           O  \nATOM    762  CG2 THR A 109       6.088   2.244  -2.879  1.00  0.00           C  \nATOM    763  N   ILE A 110       8.368   5.002  -5.816  1.00  0.00           N  \nATOM    764  CA  ILE A 110       9.495   5.751  -6.392  1.00  0.00           C  \nATOM    765  C   ILE A 110      10.010   6.863  -5.475  1.00  0.00           C  \nATOM    766  O   ILE A 110       9.264   7.443  -4.688  1.00  0.00           O  \nATOM    767  CB  ILE A 110       9.188   6.268  -7.816  1.00  0.00           C  \nATOM    768  CG1 ILE A 110       8.174   7.428  -7.848  1.00  0.00           C  \nATOM    769  CG2 ILE A 110       8.727   5.102  -8.717  1.00  0.00           C  \nATOM    770  CD1 ILE A 110       8.052   8.094  -9.230  1.00  0.00           C  \nATOM    771  N   ASP A 111      11.298   7.180  -5.618  1.00  0.00           N  \nATOM    772  CA  ASP A 111      11.845   8.503  -5.365  1.00  0.00           C  \nATOM    773  C   ASP A 111      11.827   9.279  -6.687  1.00  0.00           C  \nATOM    774  O   ASP A 111      12.124   8.720  -7.747  1.00  0.00           O  \nATOM    775  CB  ASP A 111      13.275   8.418  -4.810  1.00  0.00           C  \nATOM    776  CG  ASP A 111      13.369   7.896  -3.374  1.00  0.00           C  \nATOM    777  OD1 ASP A 111      12.417   8.091  -2.584  1.00  0.00           O  \nATOM    778  OD2 ASP A 111      14.444   7.360  -3.001  1.00  0.00           O  \nATOM    779  N   THR A 112      11.478  10.563  -6.626  1.00  0.00           N  \nATOM    780  CA  THR A 112      11.545  11.461  -7.780  1.00  0.00           C  \nATOM    781  C   THR A 112      12.985  11.887  -8.068  1.00  0.00           C  \nATOM    782  O   THR A 112      13.856  11.751  -7.210  1.00  0.00           O  \nATOM    783  CB  THR A 112      10.621  12.671  -7.585  1.00  0.00           C  \nATOM    784  OG1 THR A 112      10.829  13.274  -6.328  1.00  0.00           O  \nATOM    785  CG2 THR A 112       9.151  12.245  -7.630  1.00  0.00           C  \nATOM    786  N   ASN A 113      13.246  12.409  -9.277  1.00  0.00           N  \nATOM    787  CA  ASN A 113      14.578  12.905  -9.667  1.00  0.00           C  \nATOM    788  C   ASN A 113      15.084  13.983  -8.702  1.00  0.00           C  \nATOM    789  O   ASN A 113      16.236  13.946  -8.271  1.00  0.00           O  \nATOM    790  CB  ASN A 113      14.505  13.466 -11.098  1.00  0.00           C  \nATOM    791  CG  ASN A 113      14.334  12.384 -12.148  1.00  0.00           C  \nATOM    792  OD1 ASN A 113      14.842  11.286 -12.044  1.00  0.00           O  \nATOM    793  ND2 ASN A 113      13.624  12.665 -13.212  1.00  0.00           N  \nATOM    794  N   SER A 114      14.203  14.905  -8.305  1.00  0.00           N  \nATOM    795  CA  SER A 114      14.405  15.841  -7.209  1.00  0.00           C  \nATOM    796  C   SER A 114      13.273  15.706  -6.190  1.00  0.00           C  \nATOM    797  O   SER A 114      12.103  15.572  -6.552  1.00  0.00           O  \nATOM    798  CB  SER A 114      14.466  17.280  -7.743  1.00  0.00           C  \nATOM    799  OG  SER A 114      14.596  18.184  -6.667  1.00  0.00           O  \nATOM    800  N   THR A 115      13.589  15.791  -4.904  1.00  0.00           N  \nATOM    801  CA  THR A 115      12.550  15.850  -3.865  1.00  0.00           C  \nATOM    802  C   THR A 115      11.704  17.124  -3.956  1.00  0.00           C  \nATOM    803  O   THR A 115      10.613  17.169  -3.382  1.00  0.00           O  \nATOM    804  CB  THR A 115      13.144  15.725  -2.460  1.00  0.00           C  \nATOM    805  OG1 THR A 115      14.158  16.680  -2.252  1.00  0.00           O  \nATOM    806  CG2 THR A 115      13.774  14.349  -2.222  1.00  0.00           C  \nATOM    807  N   LEU A 116      12.164  18.126  -4.701  1.00  0.00           N  \nATOM    808  CA  LEU A 116      11.503  19.419  -4.926  1.00  0.00           C  \nATOM    809  C   LEU A 116      10.635  19.449  -6.200  1.00  0.00           C  \nATOM    810  O   LEU A 116      10.068  20.492  -6.532  1.00  0.00           O  \nATOM    811  CB  LEU A 116      12.565  20.533  -4.940  1.00  0.00           C  \nATOM    812  CG  LEU A 116      13.576  20.531  -3.772  1.00  0.00           C  \nATOM    813  CD1 LEU A 116      14.481  21.751  -3.878  1.00  0.00           C  \nATOM    814  CD2 LEU A 116      12.910  20.537  -2.398  1.00  0.00           C  \nATOM    815  N   ASP A 117      10.528  18.333  -6.925  1.00  0.00           N  \nATOM    816  CA  ASP A 117       9.707  18.251  -8.132  1.00  0.00           C  \nATOM    817  C   ASP A 117       8.236  18.604  -7.857  1.00  0.00           C  \nATOM    818  O   ASP A 117       7.666  18.251  -6.819  1.00  0.00           O  \nATOM    819  CB  ASP A 117       9.796  16.855  -8.763  1.00  0.00           C  \nATOM    820  CG  ASP A 117      11.090  16.592  -9.527  1.00  0.00           C  \nATOM    821  OD1 ASP A 117      11.674  17.558 -10.057  1.00  0.00           O  \nATOM    822  OD2 ASP A 117      11.442  15.400  -9.664  1.00  0.00           O  \nATOM    823  N   GLN A 118       7.609  19.280  -8.817  1.00  0.00           N  \nATOM    824  CA  GLN A 118       6.206  19.699  -8.803  1.00  0.00           C  \nATOM    825  C   GLN A 118       5.273  18.585  -9.326  1.00  0.00           C  \nATOM    826  O   GLN A 118       5.750  17.621  -9.932  1.00  0.00           O  \nATOM    827  CB  GLN A 118       6.083  20.994  -9.620  1.00  0.00           C  \nATOM    828  CG  GLN A 118       6.892  22.170  -9.022  1.00  0.00           C  \nATOM    829  CD  GLN A 118       6.487  22.476  -7.574  1.00  0.00           C  \nATOM    830  OE1 GLN A 118       5.319  22.631  -7.259  1.00  0.00           O  \nATOM    831  NE2 GLN A 118       7.428  22.535  -6.663  1.00  0.00           N  \nATOM    832  N   PRO A 119       3.934  18.701  -9.138  1.00  0.00           N  \nATOM    833  CA  PRO A 119       2.997  17.631  -9.493  1.00  0.00           C  \nATOM    834  C   PRO A 119       3.078  17.146 -10.945  1.00  0.00           C  \nATOM    835  O   PRO A 119       3.030  15.940 -11.196  1.00  0.00           O  \nATOM    836  CB  PRO A 119       1.607  18.186  -9.162  1.00  0.00           C  \nATOM    837  CG  PRO A 119       1.880  19.131  -7.995  1.00  0.00           C  \nATOM    838  CD  PRO A 119       3.236  19.735  -8.384  1.00  0.00           C  \nATOM    839  N   SER A 120       3.259  18.037 -11.918  1.00  0.00           N  \nATOM    840  CA  SER A 120       3.393  17.663 -13.332  1.00  0.00           C  \nATOM    841  C   SER A 120       4.633  16.799 -13.612  1.00  0.00           C  \nATOM    842  O   SER A 120       4.552  15.797 -14.329  1.00  0.00           O  \nATOM    843  CB  SER A 120       3.437  18.936 -14.194  1.00  0.00           C  \nATOM    844  OG  SER A 120       4.495  19.774 -13.764  1.00  0.00           O  \nATOM    845  N   SER A 121       5.779  17.128 -12.991  1.00  0.00           N  \nATOM    846  CA  SER A 121       6.991  16.305 -13.044  1.00  0.00           C  \nATOM    847  C   SER A 121       6.771  14.952 -12.369  1.00  0.00           C  \nATOM    848  O   SER A 121       7.087  13.912 -12.940  1.00  0.00           O  \nATOM    849  CB  SER A 121       8.175  17.044 -12.397  1.00  0.00           C  \nATOM    850  OG  SER A 121       9.332  16.229 -12.460  1.00  0.00           O  \nATOM    851  N   ARG A 122       6.126  14.947 -11.195  1.00  0.00           N  \nATOM    852  CA  ARG A 122       5.774  13.718 -10.465  1.00  0.00           C  \nATOM    853  C   ARG A 122       4.871  12.806 -11.293  1.00  0.00           C  \nATOM    854  O   ARG A 122       5.134  11.603 -11.335  1.00  0.00           O  \nATOM    855  CB  ARG A 122       5.124  14.060  -9.122  1.00  0.00           C  \nATOM    856  CG  ARG A 122       6.139  14.697  -8.150  1.00  0.00           C  \nATOM    857  CD  ARG A 122       5.463  15.130  -6.840  1.00  0.00           C  \nATOM    858  NE  ARG A 122       6.420  15.886  -6.011  1.00  0.00           N  \nATOM    859  CZ  ARG A 122       7.392  15.393  -5.260  1.00  0.00           C  \nATOM    860  NH1 ARG A 122       7.519  14.121  -5.023  1.00  0.00           N  \nATOM    861  NH2 ARG A 122       8.273  16.192  -4.754  1.00  0.00           N  \nATOM    862  N   SER A 123       3.885  13.348 -11.991  1.00  0.00           N  \nATOM    863  CA  SER A 123       3.022  12.611 -12.921  1.00  0.00           C  \nATOM    864  C   SER A 123       3.831  11.945 -14.038  1.00  0.00           C  \nATOM    865  O   SER A 123       3.709  10.745 -14.292  1.00  0.00           O  \nATOM    866  CB  SER A 123       1.982  13.563 -13.515  1.00  0.00           C  \nATOM    867  OG  SER A 123       1.202  12.905 -14.492  1.00  0.00           O  \nATOM    868  N   SER A 124       4.734  12.702 -14.675  1.00  0.00           N  \nATOM    869  CA  SER A 124       5.600  12.175 -15.740  1.00  0.00           C  \nATOM    870  C   SER A 124       6.472  11.014 -15.241  1.00  0.00           C  \nATOM    871  O   SER A 124       6.577   9.978 -15.904  1.00  0.00           O  \nATOM    872  CB  SER A 124       6.482  13.290 -16.301  1.00  0.00           C  \nATOM    873  OG  SER A 124       5.677  14.324 -16.828  1.00  0.00           O  \nATOM    874  N   GLN A 125       7.044  11.151 -14.051  1.00  0.00           N  \nATOM    875  CA  GLN A 125       7.868  10.123 -13.410  1.00  0.00           C  \nATOM    876  C   GLN A 125       7.046   8.908 -12.963  1.00  0.00           C  \nATOM    877  O   GLN A 125       7.503   7.772 -13.104  1.00  0.00           O  \nATOM    878  CB  GLN A 125       8.608  10.758 -12.226  1.00  0.00           C  \nATOM    879  CG  GLN A 125       9.647  11.799 -12.688  1.00  0.00           C  \nATOM    880  CD  GLN A 125      10.379  12.413 -11.514  1.00  0.00           C  \nATOM    881  OE1 GLN A 125      11.102  11.740 -10.802  1.00  0.00           O  \nATOM    882  NE2 GLN A 125      10.247  13.694 -11.277  1.00  0.00           N  \nATOM    883  N   GLN A 126       5.822   9.118 -12.503  1.00  0.00           N  \nATOM    884  CA  GLN A 126       4.881   8.049 -12.173  1.00  0.00           C  \nATOM    885  C   GLN A 126       4.547   7.184 -13.389  1.00  0.00           C  \nATOM    886  O   GLN A 126       4.620   5.950 -13.320  1.00  0.00           O  \nATOM    887  CB  GLN A 126       3.620   8.681 -11.576  1.00  0.00           C  \nATOM    888  CG  GLN A 126       2.590   7.634 -11.133  1.00  0.00           C  \nATOM    889  CD  GLN A 126       1.463   8.281 -10.358  1.00  0.00           C  \nATOM    890  OE1 GLN A 126       0.735   9.127 -10.852  1.00  0.00           O  \nATOM    891  NE2 GLN A 126       1.289   7.937  -9.094  1.00  0.00           N  \nATOM    892  N   MET A 127       4.261   7.813 -14.517  1.00  0.00           N  \nATOM    893  CA  MET A 127       4.006   7.100 -15.770  1.00  0.00           C  \nATOM    894  C   MET A 127       5.286   6.435 -16.313  1.00  0.00           C  \nATOM    895  O   MET A 127       5.209   5.346 -16.883  1.00  0.00           O  \nATOM    896  CB  MET A 127       3.393   8.056 -16.799  1.00  0.00           C  \nATOM    897  CG  MET A 127       1.999   8.564 -16.382  1.00  0.00           C  \nATOM    898  SD  MET A 127       0.787   7.287 -15.906  1.00  0.00           S  \nATOM    899  CE  MET A 127       0.636   6.368 -17.457  1.00  0.00           C  \nATOM    900  N   ALA A 128       6.463   7.028 -16.092  1.00  0.00           N  \nATOM    901  CA  ALA A 128       7.738   6.390 -16.415  1.00  0.00           C  \nATOM    902  C   ALA A 128       7.972   5.114 -15.592  1.00  0.00           C  \nATOM    903  O   ALA A 128       8.397   4.099 -16.146  1.00  0.00           O  \nATOM    904  CB  ALA A 128       8.878   7.405 -16.234  1.00  0.00           C  \nATOM    905  N   ALA A 129       7.605   5.122 -14.312  1.00  0.00           N  \nATOM    906  CA  ALA A 129       7.683   3.941 -13.458  1.00  0.00           C  \nATOM    907  C   ALA A 129       6.765   2.809 -13.942  1.00  0.00           C  \nATOM    908  O   ALA A 129       7.203   1.663 -14.021  1.00  0.00           O  \nATOM    909  CB  ALA A 129       7.381   4.365 -12.017  1.00  0.00           C  \nATOM    910  N   LEU A 130       5.526   3.118 -14.362  1.00  0.00           N  \nATOM    911  CA  LEU A 130       4.631   2.132 -14.963  1.00  0.00           C  \nATOM    912  C   LEU A 130       5.180   1.569 -16.277  1.00  0.00           C  \nATOM    913  O   LEU A 130       5.152   0.355 -16.481  1.00  0.00           O  \nATOM    914  CB  LEU A 130       3.234   2.743 -15.176  1.00  0.00           C  \nATOM    915  CG  LEU A 130       2.433   2.990 -13.883  1.00  0.00           C  \nATOM    916  CD1 LEU A 130       1.112   3.691 -14.231  1.00  0.00           C  \nATOM    917  CD2 LEU A 130       2.103   1.694 -13.144  1.00  0.00           C  \nATOM    918  N   ARG A 131       5.756   2.406 -17.127  1.00  0.00           N  \nATOM    919  CA  ARG A 131       6.433   1.948 -18.362  1.00  0.00           C  \nATOM    920  C   ARG A 131       7.608   1.023 -18.062  1.00  0.00           C  \nATOM    921  O   ARG A 131       7.795   0.031 -18.751  1.00  0.00           O  \nATOM    922  CB  ARG A 131       6.912   3.137 -19.205  1.00  0.00           C  \nATOM    923  CG  ARG A 131       5.762   3.866 -19.917  1.00  0.00           C  \nATOM    924  CD  ARG A 131       6.281   4.923 -20.896  1.00  0.00           C  \nATOM    925  NE  ARG A 131       7.079   5.993 -20.257  1.00  0.00           N  \nATOM    926  CZ  ARG A 131       6.623   7.125 -19.735  1.00  0.00           C  \nATOM    927  NH1 ARG A 131       5.354   7.397 -19.672  1.00  0.00           N  \nATOM    928  NH2 ARG A 131       7.452   8.010 -19.259  1.00  0.00           N  \nATOM    929  N   GLN A 132       8.388   1.317 -17.016  1.00  0.00           N  \nATOM    930  CA  GLN A 132       9.478   0.432 -16.617  1.00  0.00           C  \nATOM    931  C   GLN A 132       8.966  -0.902 -16.067  1.00  0.00           C  \nATOM    932  O   GLN A 132       9.534  -1.936 -16.411  1.00  0.00           O  \nATOM    933  CB  GLN A 132      10.408   1.138 -15.620  1.00  0.00           C  \nATOM    934  CG  GLN A 132      11.638   0.255 -15.359  1.00  0.00           C  \nATOM    935  CD  GLN A 132      12.744   0.988 -14.612  1.00  0.00           C  \nATOM    936  OE1 GLN A 132      12.593   2.086 -14.117  1.00  0.00           O  \nATOM    937  NE2 GLN A 132      13.909   0.405 -14.510  1.00  0.00           N  \nATOM    938  N   VAL A 133       7.889  -0.919 -15.298  1.00  0.00           N  \nATOM    939  CA  VAL A 133       7.247  -2.173 -14.865  1.00  0.00           C  \nATOM    940  C   VAL A 133       6.783  -2.995 -16.072  1.00  0.00           C  \nATOM    941  O   VAL A 133       7.041  -4.197 -16.111  1.00  0.00           O  \nATOM    942  CB  VAL A 133       6.091  -1.887 -13.887  1.00  0.00           C  \nATOM    943  CG1 VAL A 133       5.234  -3.126 -13.610  1.00  0.00           C  \nATOM    944  CG2 VAL A 133       6.640  -1.415 -12.536  1.00  0.00           C  \nATOM    945  N   ALA A 134       6.194  -2.368 -17.089  1.00  0.00           N  \nATOM    946  CA  ALA A 134       5.823  -3.049 -18.330  1.00  0.00           C  \nATOM    947  C   ALA A 134       7.049  -3.603 -19.079  1.00  0.00           C  \nATOM    948  O   ALA A 134       7.024  -4.744 -19.545  1.00  0.00           O  \nATOM    949  CB  ALA A 134       5.016  -2.070 -19.197  1.00  0.00           C  \nATOM    950  N   SER A 135       8.154  -2.857 -19.143  1.00  0.00           N  \nATOM    951  CA  SER A 135       9.416  -3.326 -19.728  1.00  0.00           C  \nATOM    952  C   SER A 135      10.001  -4.519 -18.957  1.00  0.00           C  \nATOM    953  O   SER A 135      10.413  -5.516 -19.562  1.00  0.00           O  \nATOM    954  CB  SER A 135      10.424  -2.175 -19.774  1.00  0.00           C  \nATOM    955  OG  SER A 135      11.644  -2.597 -20.343  1.00  0.00           O  \nATOM    956  N   LEU A 136      10.003  -4.463 -17.630  1.00  0.00           N  \nATOM    957  CA  LEU A 136      10.469  -5.547 -16.769  1.00  0.00           C  \nATOM    958  C   LEU A 136       9.593  -6.800 -16.894  1.00  0.00           C  \nATOM    959  O   LEU A 136      10.111  -7.907 -16.777  1.00  0.00           O  \nATOM    960  CB  LEU A 136      10.509  -5.056 -15.310  1.00  0.00           C  \nATOM    961  CG  LEU A 136      11.633  -4.058 -14.991  1.00  0.00           C  \nATOM    962  CD1 LEU A 136      11.415  -3.482 -13.590  1.00  0.00           C  \nATOM    963  CD2 LEU A 136      13.014  -4.711 -15.025  1.00  0.00           C  \nATOM    964  N   ASN A 137       8.304  -6.649 -17.171  1.00  0.00           N  \nATOM    965  CA  ASN A 137       7.413  -7.768 -17.472  1.00  0.00           C  \nATOM    966  C   ASN A 137       7.817  -8.528 -18.738  1.00  0.00           C  \nATOM    967  O   ASN A 137       7.616  -9.741 -18.809  1.00  0.00           O  \nATOM    968  CB  ASN A 137       5.980  -7.232 -17.596  1.00  0.00           C  \nATOM    969  CG  ASN A 137       4.986  -8.356 -17.819  1.00  0.00           C  \nATOM    970  OD1 ASN A 137       4.831  -9.217 -16.970  1.00  0.00           O  \nATOM    971  ND2 ASN A 137       4.311  -8.379 -18.934  1.00  0.00           N  \nATOM    972  N   GLY A 138       8.418  -7.845 -19.726  1.00  0.00           N  \nATOM    973  CA  GLY A 138       8.953  -8.447 -20.948  1.00  0.00           C  \nATOM    974  C   GLY A 138      10.418  -8.897 -20.845  1.00  0.00           C  \nATOM    975  O   GLY A 138      10.972  -9.382 -21.830  1.00  0.00           O  \nATOM    976  N   THR A 139      11.079  -8.731 -19.704  1.00  0.00           N  \nATOM    977  CA  THR A 139      12.515  -9.015 -19.522  1.00  0.00           C  \nATOM    978  C   THR A 139      12.708 -10.343 -18.790  1.00  0.00           C  \nATOM    979  O   THR A 139      12.407 -10.434 -17.602  1.00  0.00           O  \nATOM    980  CB  THR A 139      13.192  -7.869 -18.758  1.00  0.00           C  \nATOM    981  OG1 THR A 139      13.040  -6.650 -19.447  1.00  0.00           O  \nATOM    982  CG2 THR A 139      14.702  -8.086 -18.605  1.00  0.00           C  \nATOM    983  N   SER A 140      13.256 -11.365 -19.468  1.00  0.00           N  \nATOM    984  CA  SER A 140      13.350 -12.733 -18.931  1.00  0.00           C  \nATOM    985  C   SER A 140      14.143 -12.874 -17.622  1.00  0.00           C  \nATOM    986  O   SER A 140      13.851 -13.757 -16.818  1.00  0.00           O  \nATOM    987  CB  SER A 140      13.952 -13.666 -19.983  1.00  0.00           C  \nATOM    988  OG  SER A 140      15.226 -13.190 -20.382  1.00  0.00           O  \nATOM    989  N   SER A 141      15.127 -11.999 -17.370  1.00  0.00           N  \nATOM    990  CA  SER A 141      15.891 -11.956 -16.115  1.00  0.00           C  \nATOM    991  C   SER A 141      15.164 -11.232 -14.968  1.00  0.00           C  \nATOM    992  O   SER A 141      15.650 -11.233 -13.839  1.00  0.00           O  \nATOM    993  CB  SER A 141      17.263 -11.326 -16.364  1.00  0.00           C  \nATOM    994  OG  SER A 141      17.108 -10.009 -16.860  1.00  0.00           O  \nATOM    995  N   SER A 142      14.034 -10.589 -15.238  1.00  0.00           N  \nATOM    996  CA  SER A 142      13.290  -9.847 -14.229  1.00  0.00           C  \nATOM    997  C   SER A 142      12.507 -10.775 -13.294  1.00  0.00           C  \nATOM    998  O   SER A 142      11.818 -11.682 -13.764  1.00  0.00           O  \nATOM    999  CB  SER A 142      12.322  -8.867 -14.894  1.00  0.00           C  \nATOM   1000  OG  SER A 142      11.549  -8.195 -13.916  1.00  0.00           O  \nATOM   1001  N   PRO A 143      12.476 -10.504 -11.978  1.00  0.00           N  \nATOM   1002  CA  PRO A 143      11.626 -11.245 -11.041  1.00  0.00           C  \nATOM   1003  C   PRO A 143      10.121 -11.163 -11.337  1.00  0.00           C  \nATOM   1004  O   PRO A 143       9.355 -11.993 -10.836  1.00  0.00           O  \nATOM   1005  CB  PRO A 143      11.936 -10.646  -9.670  1.00  0.00           C  \nATOM   1006  CG  PRO A 143      13.335 -10.059  -9.822  1.00  0.00           C  \nATOM   1007  CD  PRO A 143      13.339  -9.568 -11.261  1.00  0.00           C  \nATOM   1008  N   ILE A 144       9.686 -10.168 -12.130  1.00  0.00           N  \nATOM   1009  CA  ILE A 144       8.292  -9.966 -12.537  1.00  0.00           C  \nATOM   1010  C   ILE A 144       8.025 -10.325 -14.008  1.00  0.00           C  \nATOM   1011  O   ILE A 144       6.968  -9.983 -14.526  1.00  0.00           O  \nATOM   1012  CB  ILE A 144       7.772  -8.554 -12.177  1.00  0.00           C  \nATOM   1013  CG1 ILE A 144       8.557  -7.438 -12.886  1.00  0.00           C  \nATOM   1014  CG2 ILE A 144       7.786  -8.364 -10.651  1.00  0.00           C  \nATOM   1015  CD1 ILE A 144       7.865  -6.063 -12.823  1.00  0.00           C  \nATOM   1016  N   TYR A 145       8.947 -11.055 -14.667  1.00  0.00           N  \nATOM   1017  CA  TYR A 145       8.747 -11.532 -16.037  1.00  0.00           C  \nATOM   1018  C   TYR A 145       7.426 -12.310 -16.174  1.00  0.00           C  \nATOM   1019  O   TYR A 145       7.180 -13.259 -15.434  1.00  0.00           O  \nATOM   1020  CB  TYR A 145       9.925 -12.414 -16.456  1.00  0.00           C  \nATOM   1021  CG  TYR A 145       9.791 -12.979 -17.858  1.00  0.00           C  \nATOM   1022  CD1 TYR A 145       9.640 -14.368 -18.068  1.00  0.00           C  \nATOM   1023  CD2 TYR A 145       9.792 -12.108 -18.970  1.00  0.00           C  \nATOM   1024  CE1 TYR A 145       9.512 -14.881 -19.375  1.00  0.00           C  \nATOM   1025  CE2 TYR A 145       9.657 -12.613 -20.264  1.00  0.00           C  \nATOM   1026  CZ  TYR A 145       9.521 -13.999 -20.478  1.00  0.00           C  \nATOM   1027  OH  TYR A 145       9.398 -14.486 -21.739  1.00  0.00           O  \nATOM   1028  N   GLY A 146       6.566 -11.869 -17.091  1.00  0.00           N  \nATOM   1029  CA  GLY A 146       5.261 -12.488 -17.355  1.00  0.00           C  \nATOM   1030  C   GLY A 146       4.231 -12.376 -16.231  1.00  0.00           C  \nATOM   1031  O   GLY A 146       3.189 -13.021 -16.302  1.00  0.00           O  \nATOM   1032  N   LYS A 147       4.492 -11.569 -15.178  1.00  0.00           N  \nATOM   1033  CA  LYS A 147       3.594 -11.462 -14.009  1.00  0.00           C  \nATOM   1034  C   LYS A 147       2.627 -10.283 -14.063  1.00  0.00           C  \nATOM   1035  O   LYS A 147       1.780 -10.168 -13.170  1.00  0.00           O  \nATOM   1036  CB  LYS A 147       4.399 -11.420 -12.701  1.00  0.00           C  \nATOM   1037  CG  LYS A 147       5.234 -12.688 -12.505  1.00  0.00           C  \nATOM   1038  CD  LYS A 147       5.881 -12.693 -11.124  1.00  0.00           C  \nATOM   1039  CE  LYS A 147       6.720 -13.967 -10.995  1.00  0.00           C  \nATOM   1040  NZ  LYS A 147       7.417 -14.030  -9.701  1.00  0.00           N  \nATOM   1041  N   VAL A 148       2.765  -9.391 -15.041  1.00  0.00           N  \nATOM   1042  CA  VAL A 148       2.005  -8.144 -15.099  1.00  0.00           C  \nATOM   1043  C   VAL A 148       1.059  -8.137 -16.288  1.00  0.00           C  \nATOM   1044  O   VAL A 148       1.485  -8.284 -17.431  1.00  0.00           O  \nATOM   1045  CB  VAL A 148       2.932  -6.911 -15.139  1.00  0.00           C  \nATOM   1046  CG1 VAL A 148       2.138  -5.606 -15.046  1.00  0.00           C  \nATOM   1047  CG2 VAL A 148       3.944  -6.920 -13.991  1.00  0.00           C  \nATOM   1048  N   ASP A 149      -0.222  -7.897 -16.023  1.00  0.00           N  \nATOM   1049  CA  ASP A 149      -1.172  -7.469 -17.033  1.00  0.00           C  \nATOM   1050  C   ASP A 149      -1.000  -5.963 -17.259  1.00  0.00           C  \nATOM   1051  O   ASP A 149      -1.495  -5.126 -16.501  1.00  0.00           O  \nATOM   1052  CB  ASP A 149      -2.582  -7.858 -16.603  1.00  0.00           C  \nATOM   1053  CG  ASP A 149      -3.660  -7.426 -17.597  1.00  0.00           C  \nATOM   1054  OD1 ASP A 149      -3.324  -6.760 -18.609  1.00  0.00           O  \nATOM   1055  OD2 ASP A 149      -4.829  -7.729 -17.291  1.00  0.00           O  \nATOM   1056  N   THR A 150      -0.260  -5.618 -18.313  1.00  0.00           N  \nATOM   1057  CA  THR A 150       0.101  -4.233 -18.643  1.00  0.00           C  \nATOM   1058  C   THR A 150      -1.079  -3.386 -19.128  1.00  0.00           C  \nATOM   1059  O   THR A 150      -0.942  -2.167 -19.248  1.00  0.00           O  \nATOM   1060  CB  THR A 150       1.233  -4.197 -19.682  1.00  0.00           C  \nATOM   1061  OG1 THR A 150       0.856  -4.883 -20.850  1.00  0.00           O  \nATOM   1062  CG2 THR A 150       2.508  -4.866 -19.162  1.00  0.00           C  \nATOM   1063  N   ALA A 151      -2.248  -3.978 -19.359  1.00  0.00           N  \nATOM   1064  CA  ALA A 151      -3.479  -3.248 -19.663  1.00  0.00           C  \nATOM   1065  C   ALA A 151      -4.230  -2.787 -18.394  1.00  0.00           C  \nATOM   1066  O   ALA A 151      -5.180  -2.004 -18.496  1.00  0.00           O  \nATOM   1067  CB  ALA A 151      -4.364  -4.135 -20.546  1.00  0.00           C  \nATOM   1068  N   ARG A 152      -3.809  -3.237 -17.200  1.00  0.00           N  \nATOM   1069  CA  ARG A 152      -4.434  -2.913 -15.913  1.00  0.00           C  \nATOM   1070  C   ARG A 152      -3.458  -2.161 -15.016  1.00  0.00           C  \nATOM   1071  O   ARG A 152      -2.701  -2.758 -14.257  1.00  0.00           O  \nATOM   1072  CB  ARG A 152      -4.986  -4.176 -15.245  1.00  0.00           C  \nATOM   1073  CG  ARG A 152      -6.131  -4.811 -16.046  1.00  0.00           C  \nATOM   1074  CD  ARG A 152      -6.819  -5.927 -15.269  1.00  0.00           C  \nATOM   1075  NE  ARG A 152      -5.917  -7.067 -15.012  1.00  0.00           N  \nATOM   1076  CZ  ARG A 152      -5.618  -7.634 -13.857  1.00  0.00           C  \nATOM   1077  NH1 ARG A 152      -6.089  -7.219 -12.720  1.00  0.00           N  \nATOM   1078  NH2 ARG A 152      -4.827  -8.651 -13.841  1.00  0.00           N  \nATOM   1079  N   MET A 153      -3.470  -0.844 -15.104  1.00  0.00           N  \nATOM   1080  CA  MET A 153      -2.581   0.028 -14.337  1.00  0.00           C  \nATOM   1081  C   MET A 153      -3.363   1.009 -13.453  1.00  0.00           C  \nATOM   1082  O   MET A 153      -4.438   1.472 -13.823  1.00  0.00           O  \nATOM   1083  CB  MET A 153      -1.626   0.763 -15.279  1.00  0.00           C  \nATOM   1084  CG  MET A 153      -0.665  -0.211 -15.979  1.00  0.00           C  \nATOM   1085  SD  MET A 153       0.591   0.630 -16.973  1.00  0.00           S  \nATOM   1086  CE  MET A 153       1.587  -0.806 -17.450  1.00  0.00           C  \nATOM   1087  N   GLY A 154      -2.795   1.327 -12.291  1.00  0.00           N  \nATOM   1088  CA  GLY A 154      -3.325   2.282 -11.328  1.00  0.00           C  \nATOM   1089  C   GLY A 154      -2.265   3.219 -10.779  1.00  0.00           C  \nATOM   1090  O   GLY A 154      -1.057   2.969 -10.885  1.00  0.00           O  \nATOM   1091  N   VAL A 155      -2.717   4.327 -10.187  1.00  0.00           N  \nATOM   1092  CA  VAL A 155      -1.841   5.385  -9.663  1.00  0.00           C  \nATOM   1093  C   VAL A 155      -2.235   5.789  -8.246  1.00  0.00           C  \nATOM   1094  O   VAL A 155      -3.408   5.919  -7.915  1.00  0.00           O  \nATOM   1095  CB  VAL A 155      -1.792   6.604 -10.606  1.00  0.00           C  \nATOM   1096  CG1 VAL A 155      -1.068   6.251 -11.907  1.00  0.00           C  \nATOM   1097  CG2 VAL A 155      -3.171   7.182 -10.956  1.00  0.00           C  \nATOM   1098  N   MET A 156      -1.226   6.007  -7.409  1.00  0.00           N  \nATOM   1099  CA  MET A 156      -1.380   6.529  -6.047  1.00  0.00           C  \nATOM   1100  C   MET A 156      -0.265   7.520  -5.736  1.00  0.00           C  \nATOM   1101  O   MET A 156       0.804   7.486  -6.339  1.00  0.00           O  \nATOM   1102  CB  MET A 156      -1.372   5.395  -5.012  1.00  0.00           C  \nATOM   1103  CG  MET A 156      -2.435   4.326  -5.246  1.00  0.00           C  \nATOM   1104  SD  MET A 156      -2.472   3.008  -3.993  1.00  0.00           S  \nATOM   1105  CE  MET A 156      -2.999   3.926  -2.522  1.00  0.00           C  \nATOM   1106  N   GLY A 157      -0.484   8.392  -4.760  1.00  0.00           N  \nATOM   1107  CA  GLY A 157       0.581   9.273  -4.310  1.00  0.00           C  \nATOM   1108  C   GLY A 157       0.130  10.173  -3.168  1.00  0.00           C  \nATOM   1109  O   GLY A 157      -1.064  10.317  -2.906  1.00  0.00           O  \nATOM   1110  N   TRP A 158       1.100  10.771  -2.474  1.00  0.00           N  \nATOM   1111  CA  TRP A 158       0.862  11.644  -1.338  1.00  0.00           C  \nATOM   1112  C   TRP A 158       1.126  13.112  -1.691  1.00  0.00           C  \nATOM   1113  O   TRP A 158       2.160  13.424  -2.283  1.00  0.00           O  \nATOM   1114  CB  TRP A 158       1.721  11.189  -0.156  1.00  0.00           C  \nATOM   1115  CG  TRP A 158       1.723  12.092   1.034  1.00  0.00           C  \nATOM   1116  CD1 TRP A 158       0.641  12.653   1.614  1.00  0.00           C  \nATOM   1117  CD2 TRP A 158       2.873  12.574   1.783  1.00  0.00           C  \nATOM   1118  NE1 TRP A 158       1.040  13.453   2.673  1.00  0.00           N  \nATOM   1119  CE2 TRP A 158       2.402  13.430   2.822  1.00  0.00           C  \nATOM   1120  CE3 TRP A 158       4.263  12.369   1.695  1.00  0.00           C  \nATOM   1121  CZ2 TRP A 158       3.270  14.051   3.738  1.00  0.00           C  \nATOM   1122  CZ3 TRP A 158       5.137  12.984   2.603  1.00  0.00           C  \nATOM   1123  CH2 TRP A 158       4.649  13.817   3.619  1.00  0.00           C  \nATOM   1124  N   SER A 159       0.221  14.010  -1.278  1.00  0.00           N  \nATOM   1125  CA  SER A 159       0.376  15.461  -1.432  1.00  0.00           C  \nATOM   1126  C   SER A 159       0.616  15.833  -2.906  1.00  0.00           C  \nATOM   1127  O   SER A 159      -0.203  15.459  -3.758  1.00  0.00           O  \nATOM   1128  CB  SER A 159       1.466  15.930  -0.460  1.00  0.00           C  \nATOM   1129  OG  SER A 159       1.573  17.334  -0.409  1.00  0.00           O  \nATOM   1130  N   MET A 160       1.728  16.477  -3.263  1.00  0.00           N  \nATOM   1131  CA  MET A 160       2.093  16.728  -4.667  1.00  0.00           C  \nATOM   1132  C   MET A 160       2.177  15.450  -5.519  1.00  0.00           C  \nATOM   1133  O   MET A 160       1.899  15.484  -6.712  1.00  0.00           O  \nATOM   1134  CB  MET A 160       3.435  17.461  -4.759  1.00  0.00           C  \nATOM   1135  CG  MET A 160       3.390  18.891  -4.218  1.00  0.00           C  \nATOM   1136  SD  MET A 160       4.926  19.794  -4.570  1.00  0.00           S  \nATOM   1137  CE  MET A 160       4.492  21.397  -3.840  1.00  0.00           C  \nATOM   1138  N   GLY A 161       2.529  14.306  -4.926  1.00  0.00           N  \nATOM   1139  CA  GLY A 161       2.438  13.005  -5.601  1.00  0.00           C  \nATOM   1140  C   GLY A 161       0.988  12.563  -5.828  1.00  0.00           C  \nATOM   1141  O   GLY A 161       0.697  11.869  -6.801  1.00  0.00           O  \nATOM   1142  N   GLY A 162       0.057  13.003  -4.963  1.00  0.00           N  \nATOM   1143  CA  GLY A 162      -1.386  12.828  -5.126  1.00  0.00           C  \nATOM   1144  C   GLY A 162      -1.947  13.689  -6.256  1.00  0.00           C  \nATOM   1145  O   GLY A 162      -2.638  13.159  -7.128  1.00  0.00           O  \nATOM   1146  N   GLY A 163      -1.561  14.974  -6.311  1.00  0.00           N  \nATOM   1147  CA  GLY A 163      -1.840  15.852  -7.455  1.00  0.00           C  \nATOM   1148  C   GLY A 163      -1.284  15.276  -8.760  1.00  0.00           C  \nATOM   1149  O   GLY A 163      -2.005  15.143  -9.747  1.00  0.00           O  \nATOM   1150  N   GLY A 164      -0.036  14.787  -8.736  1.00  0.00           N  \nATOM   1151  CA  GLY A 164       0.563  14.078  -9.864  1.00  0.00           C  \nATOM   1152  C   GLY A 164      -0.228  12.838 -10.311  1.00  0.00           C  \nATOM   1153  O   GLY A 164      -0.359  12.592 -11.514  1.00  0.00           O  \nATOM   1154  N   SER A 165      -0.813  12.095  -9.371  1.00  0.00           N  \nATOM   1155  CA  SER A 165      -1.665  10.932  -9.681  1.00  0.00           C  \nATOM   1156  C   SER A 165      -2.970  11.339 -10.382  1.00  0.00           C  \nATOM   1157  O   SER A 165      -3.417  10.648 -11.294  1.00  0.00           O  \nATOM   1158  CB  SER A 165      -1.989  10.133  -8.418  1.00  0.00           C  \nATOM   1159  OG  SER A 165      -0.799   9.707  -7.783  1.00  0.00           O  \nATOM   1160  N   LEU A 166      -3.553  12.477  -9.993  1.00  0.00           N  \nATOM   1161  CA  LEU A 166      -4.727  13.044 -10.653  1.00  0.00           C  \nATOM   1162  C   LEU A 166      -4.398  13.565 -12.058  1.00  0.00           C  \nATOM   1163  O   LEU A 166      -5.145  13.299 -12.994  1.00  0.00           O  \nATOM   1164  CB  LEU A 166      -5.302  14.167  -9.775  1.00  0.00           C  \nATOM   1165  CG  LEU A 166      -5.901  13.662  -8.452  1.00  0.00           C  \nATOM   1166  CD1 LEU A 166      -6.000  14.825  -7.468  1.00  0.00           C  \nATOM   1167  CD2 LEU A 166      -7.296  13.078  -8.665  1.00  0.00           C  \nATOM   1168  N   ILE A 167      -3.246  14.222 -12.233  1.00  0.00           N  \nATOM   1169  CA  ILE A 167      -2.740  14.620 -13.561  1.00  0.00           C  \nATOM   1170  C   ILE A 167      -2.521  13.373 -14.434  1.00  0.00           C  \nATOM   1171  O   ILE A 167      -2.932  13.345 -15.592  1.00  0.00           O  \nATOM   1172  CB  ILE A 167      -1.445  15.457 -13.430  1.00  0.00           C  \nATOM   1173  CG1 ILE A 167      -1.720  16.799 -12.712  1.00  0.00           C  \nATOM   1174  CG2 ILE A 167      -0.833  15.727 -14.818  1.00  0.00           C  \nATOM   1175  CD1 ILE A 167      -0.444  17.535 -12.278  1.00  0.00           C  \nATOM   1176  N   SER A 168      -1.922  12.308 -13.870  1.00  0.00           N  \nATOM   1177  CA  SER A 168      -1.749  11.041 -14.574  1.00  0.00           C  \nATOM   1178  C   SER A 168      -3.093  10.460 -15.033  1.00  0.00           C  \nATOM   1179  O   SER A 168      -3.199  10.031 -16.175  1.00  0.00           O  \nATOM   1180  CB  SER A 168      -1.027  10.004 -13.705  1.00  0.00           C  \nATOM   1181  OG  SER A 168       0.301  10.405 -13.471  1.00  0.00           O  \nATOM   1182  N   ALA A 169      -4.114  10.489 -14.171  1.00  0.00           N  \nATOM   1183  CA  ALA A 169      -5.455  10.003 -14.497  1.00  0.00           C  \nATOM   1184  C   ALA A 169      -6.152  10.867 -15.565  1.00  0.00           C  \nATOM   1185  O   ALA A 169      -6.774  10.324 -16.479  1.00  0.00           O  \nATOM   1186  CB  ALA A 169      -6.268   9.935 -13.202  1.00  0.00           C  \nATOM   1187  N   ALA A 170      -6.016  12.192 -15.489  1.00  0.00           N  \nATOM   1188  CA  ALA A 170      -6.558  13.115 -16.483  1.00  0.00           C  \nATOM   1189  C   ALA A 170      -5.938  12.894 -17.873  1.00  0.00           C  \nATOM   1190  O   ALA A 170      -6.649  12.840 -18.873  1.00  0.00           O  \nATOM   1191  CB  ALA A 170      -6.321  14.546 -15.988  1.00  0.00           C  \nATOM   1192  N   ASN A 171      -4.618  12.697 -17.927  1.00  0.00           N  \nATOM   1193  CA  ASN A 171      -3.872  12.535 -19.178  1.00  0.00           C  \nATOM   1194  C   ASN A 171      -3.900  11.107 -19.736  1.00  0.00           C  \nATOM   1195  O   ASN A 171      -3.580  10.903 -20.902  1.00  0.00           O  \nATOM   1196  CB  ASN A 171      -2.431  13.009 -18.943  1.00  0.00           C  \nATOM   1197  CG  ASN A 171      -2.326  14.508 -18.699  1.00  0.00           C  \nATOM   1198  OD1 ASN A 171      -3.211  15.292 -18.993  1.00  0.00           O  \nATOM   1199  ND2 ASN A 171      -1.205  14.968 -18.196  1.00  0.00           N  \nATOM   1200  N   ASN A 172      -4.260  10.103 -18.921  1.00  0.00           N  \nATOM   1201  CA  ASN A 172      -4.253   8.692 -19.309  1.00  0.00           C  \nATOM   1202  C   ASN A 172      -5.561   8.008 -18.879  1.00  0.00           C  \nATOM   1203  O   ASN A 172      -5.570   7.230 -17.917  1.00  0.00           O  \nATOM   1204  CB  ASN A 172      -3.005   7.977 -18.752  1.00  0.00           C  \nATOM   1205  CG  ASN A 172      -1.697   8.661 -19.091  1.00  0.00           C  \nATOM   1206  OD1 ASN A 172      -1.024   8.339 -20.052  1.00  0.00           O  \nATOM   1207  ND2 ASN A 172      -1.280   9.615 -18.286  1.00  0.00           N  \nATOM   1208  N   PRO A 173      -6.688   8.251 -19.591  1.00  0.00           N  \nATOM   1209  CA  PRO A 173      -7.996   7.711 -19.218  1.00  0.00           C  \nATOM   1210  C   PRO A 173      -8.107   6.177 -19.304  1.00  0.00           C  \nATOM   1211  O   PRO A 173      -9.089   5.587 -18.852  1.00  0.00           O  \nATOM   1212  CB  PRO A 173      -8.988   8.411 -20.164  1.00  0.00           C  \nATOM   1213  CG  PRO A 173      -8.138   8.730 -21.390  1.00  0.00           C  \nATOM   1214  CD  PRO A 173      -6.788   9.077 -20.783  1.00  0.00           C  \nATOM   1215  N   SER A 174      -7.089   5.496 -19.855  1.00  0.00           N  \nATOM   1216  CA  SER A 174      -6.977   4.033 -19.850  1.00  0.00           C  \nATOM   1217  C   SER A 174      -6.564   3.448 -18.491  1.00  0.00           C  \nATOM   1218  O   SER A 174      -6.742   2.244 -18.289  1.00  0.00           O  \nATOM   1219  CB  SER A 174      -5.988   3.585 -20.933  1.00  0.00           C  \nATOM   1220  OG  SER A 174      -4.709   4.141 -20.683  1.00  0.00           O  \nATOM   1221  N   LEU A 175      -6.040   4.275 -17.564  1.00  0.00           N  \nATOM   1222  CA  LEU A 175      -5.794   3.876 -16.180  1.00  0.00           C  \nATOM   1223  C   LEU A 175      -7.095   3.370 -15.548  1.00  0.00           C  \nATOM   1224  O   LEU A 175      -8.176   3.907 -15.787  1.00  0.00           O  \nATOM   1225  CB  LEU A 175      -5.212   5.046 -15.366  1.00  0.00           C  \nATOM   1226  CG  LEU A 175      -3.764   5.416 -15.718  1.00  0.00           C  \nATOM   1227  CD1 LEU A 175      -3.365   6.697 -14.987  1.00  0.00           C  \nATOM   1228  CD2 LEU A 175      -2.777   4.316 -15.305  1.00  0.00           C  \nATOM   1229  N   LYS A 176      -6.988   2.325 -14.723  1.00  0.00           N  \nATOM   1230  CA  LYS A 176      -8.167   1.645 -14.170  1.00  0.00           C  \nATOM   1231  C   LYS A 176      -8.606   2.219 -12.832  1.00  0.00           C  \nATOM   1232  O   LYS A 176      -9.799   2.178 -12.530  1.00  0.00           O  \nATOM   1233  CB  LYS A 176      -7.916   0.131 -14.071  1.00  0.00           C  \nATOM   1234  CG  LYS A 176      -7.508  -0.531 -15.401  1.00  0.00           C  \nATOM   1235  CD  LYS A 176      -8.488  -0.227 -16.549  1.00  0.00           C  \nATOM   1236  CE  LYS A 176      -8.033  -0.948 -17.816  1.00  0.00           C  \nATOM   1237  NZ  LYS A 176      -8.766  -0.470 -19.007  1.00  0.00           N  \nATOM   1238  N   ALA A 177      -7.684   2.757 -12.050  1.00  0.00           N  \nATOM   1239  CA  ALA A 177      -7.997   3.364 -10.765  1.00  0.00           C  \nATOM   1240  C   ALA A 177      -6.943   4.394 -10.340  1.00  0.00           C  \nATOM   1241  O   ALA A 177      -5.758   4.249 -10.642  1.00  0.00           O  \nATOM   1242  CB  ALA A 177      -8.129   2.257  -9.710  1.00  0.00           C  \nATOM   1243  N   ALA A 178      -7.377   5.394  -9.574  1.00  0.00           N  \nATOM   1244  CA  ALA A 178      -6.528   6.322  -8.849  1.00  0.00           C  \nATOM   1245  C   ALA A 178      -6.916   6.342  -7.363  1.00  0.00           C  \nATOM   1246  O   ALA A 178      -8.097   6.286  -7.020  1.00  0.00           O  \nATOM   1247  CB  ALA A 178      -6.630   7.710  -9.499  1.00  0.00           C  \nATOM   1248  N   ALA A 179      -5.926   6.437  -6.454  1.00  0.00           N  \nATOM   1249  CA  ALA A 179      -6.182   6.634  -5.025  1.00  0.00           C  \nATOM   1250  C   ALA A 179      -5.146   7.592  -4.398  1.00  0.00           C  \nATOM   1251  O   ALA A 179      -4.264   7.158  -3.654  1.00  0.00           O  \nATOM   1252  CB  ALA A 179      -6.299   5.271  -4.324  1.00  0.00           C  \nATOM   1253  N   PRO A 180      -5.207   8.908  -4.717  1.00  0.00           N  \nATOM   1254  CA  PRO A 180      -4.373   9.907  -4.068  1.00  0.00           C  \nATOM   1255  C   PRO A 180      -4.692  10.044  -2.573  1.00  0.00           C  \nATOM   1256  O   PRO A 180      -5.834   9.885  -2.142  1.00  0.00           O  \nATOM   1257  CB  PRO A 180      -4.638  11.212  -4.823  1.00  0.00           C  \nATOM   1258  CG  PRO A 180      -6.067  11.034  -5.335  1.00  0.00           C  \nATOM   1259  CD  PRO A 180      -6.112   9.548  -5.665  1.00  0.00           C  \nATOM   1260  N   GLN A 181      -3.680  10.422  -1.780  1.00  0.00           N  \nATOM   1261  CA  GLN A 181      -3.793  10.707  -0.351  1.00  0.00           C  \nATOM   1262  C   GLN A 181      -3.303  12.127  -0.030  1.00  0.00           C  \nATOM   1263  O   GLN A 181      -2.206  12.510  -0.443  1.00  0.00           O  \nATOM   1264  CB  GLN A 181      -3.083   9.624   0.478  1.00  0.00           C  \nATOM   1265  CG  GLN A 181      -1.601   9.425   0.129  1.00  0.00           C  \nATOM   1266  CD  GLN A 181      -0.916   8.357   0.986  1.00  0.00           C  \nATOM   1267  OE1 GLN A 181      -1.187   7.172   0.893  1.00  0.00           O  \nATOM   1268  NE2 GLN A 181       0.004   8.739   1.839  1.00  0.00           N  \nATOM   1269  N   ALA A 182      -4.130  12.898   0.703  1.00  0.00           N  \nATOM   1270  CA  ALA A 182      -3.915  14.317   1.000  1.00  0.00           C  \nATOM   1271  C   ALA A 182      -3.453  15.116  -0.244  1.00  0.00           C  \nATOM   1272  O   ALA A 182      -2.402  15.754  -0.199  1.00  0.00           O  \nATOM   1273  CB  ALA A 182      -2.987  14.431   2.217  1.00  0.00           C  \nATOM   1274  N   PRO A 183      -4.165  15.016  -1.394  1.00  0.00           N  \nATOM   1275  CA  PRO A 183      -3.667  15.531  -2.661  1.00  0.00           C  \nATOM   1276  C   PRO A 183      -3.515  17.053  -2.642  1.00  0.00           C  \nATOM   1277  O   PRO A 183      -4.373  17.761  -2.117  1.00  0.00           O  \nATOM   1278  CB  PRO A 183      -4.678  15.092  -3.718  1.00  0.00           C  \nATOM   1279  CG  PRO A 183      -5.978  14.998  -2.926  1.00  0.00           C  \nATOM   1280  CD  PRO A 183      -5.499  14.475  -1.573  1.00  0.00           C  \nATOM   1281  N   TRP A 184      -2.466  17.513  -3.302  1.00  0.00           N  \nATOM   1282  CA  TRP A 184      -2.252  18.922  -3.585  1.00  0.00           C  \nATOM   1283  C   TRP A 184      -1.809  19.110  -5.025  1.00  0.00           C  \nATOM   1284  O   TRP A 184      -0.913  18.406  -5.507  1.00  0.00           O  \nATOM   1285  CB  TRP A 184      -1.221  19.514  -2.604  1.00  0.00           C  \nATOM   1286  CG  TRP A 184      -1.012  20.988  -2.755  1.00  0.00           C  \nATOM   1287  CD1 TRP A 184      -1.643  21.949  -2.043  1.00  0.00           C  \nATOM   1288  CD2 TRP A 184      -0.147  21.702  -3.700  1.00  0.00           C  \nATOM   1289  NE1 TRP A 184      -1.237  23.192  -2.470  1.00  0.00           N  \nATOM   1290  CE2 TRP A 184      -0.320  23.105  -3.491  1.00  0.00           C  \nATOM   1291  CE3 TRP A 184       0.750  21.315  -4.709  1.00  0.00           C  \nATOM   1292  CZ2 TRP A 184       0.362  24.072  -4.252  1.00  0.00           C  \nATOM   1293  CZ3 TRP A 184       1.440  22.280  -5.482  1.00  0.00           C  \nATOM   1294  CH2 TRP A 184       1.248  23.645  -5.256  1.00  0.00           C  \nATOM   1295  N   ASP A 185      -2.395  20.088  -5.692  1.00  0.00           N  \nATOM   1296  CA  ASP A 185      -1.962  20.639  -6.975  1.00  0.00           C  \nATOM   1297  C   ASP A 185      -2.303  22.135  -7.028  1.00  0.00           C  \nATOM   1298  O   ASP A 185      -3.126  22.623  -6.255  1.00  0.00           O  \nATOM   1299  CB  ASP A 185      -2.605  19.853  -8.129  1.00  0.00           C  \nATOM   1300  CG  ASP A 185      -2.042  20.311  -9.472  1.00  0.00           C  \nATOM   1301  OD1 ASP A 185      -0.800  20.255  -9.634  1.00  0.00           O  \nATOM   1302  OD2 ASP A 185      -2.823  20.813 -10.313  1.00  0.00           O  \nATOM   1303  N   SER A 186      -1.651  22.891  -7.927  1.00  0.00           N  \nATOM   1304  CA  SER A 186      -1.933  24.318  -8.126  1.00  0.00           C  \nATOM   1305  C   SER A 186      -3.347  24.578  -8.672  1.00  0.00           C  \nATOM   1306  O   SER A 186      -3.889  25.659  -8.473  1.00  0.00           O  \nATOM   1307  CB  SER A 186      -0.888  24.925  -9.063  1.00  0.00           C  \nATOM   1308  OG  SER A 186      -0.846  24.234 -10.298  1.00  0.00           O  \nATOM   1309  N   SER A 187      -3.949  23.592  -9.334  1.00  0.00           N  \nATOM   1310  CA  SER A 187      -5.336  23.599  -9.781  1.00  0.00           C  \nATOM   1311  C   SER A 187      -6.141  22.539  -9.030  1.00  0.00           C  \nATOM   1312  O   SER A 187      -5.701  21.395  -8.889  1.00  0.00           O  \nATOM   1313  CB  SER A 187      -5.408  23.332 -11.286  1.00  0.00           C  \nATOM   1314  OG  SER A 187      -6.756  23.357 -11.726  1.00  0.00           O  \nATOM   1315  N   THR A 188      -7.354  22.892  -8.617  1.00  0.00           N  \nATOM   1316  CA  THR A 188      -8.302  21.921  -8.053  1.00  0.00           C  \nATOM   1317  C   THR A 188      -9.286  21.381  -9.092  1.00  0.00           C  \nATOM   1318  O   THR A 188     -10.099  20.515  -8.789  1.00  0.00           O  \nATOM   1319  CB  THR A 188      -9.055  22.518  -6.857  1.00  0.00           C  \nATOM   1320  OG1 THR A 188      -9.712  23.695  -7.244  1.00  0.00           O  \nATOM   1321  CG2 THR A 188      -8.101  22.876  -5.714  1.00  0.00           C  \nATOM   1322  N   ASN A 189      -9.238  21.877 -10.342  1.00  0.00           N  \nATOM   1323  CA  ASN A 189     -10.152  21.452 -11.394  1.00  0.00           C  \nATOM   1324  C   ASN A 189      -9.638  20.193 -12.108  1.00  0.00           C  \nATOM   1325  O   ASN A 189      -8.819  20.261 -13.023  1.00  0.00           O  \nATOM   1326  CB  ASN A 189     -10.423  22.616 -12.357  1.00  0.00           C  \nATOM   1327  CG  ASN A 189     -11.466  22.267 -13.398  1.00  0.00           C  \nATOM   1328  OD1 ASN A 189     -11.974  21.156 -13.505  1.00  0.00           O  \nATOM   1329  ND2 ASN A 189     -11.841  23.209 -14.225  1.00  0.00           N  \nATOM   1330  N   PHE A 190     -10.200  19.049 -11.737  1.00  0.00           N  \nATOM   1331  CA  PHE A 190      -9.968  17.738 -12.358  1.00  0.00           C  \nATOM   1332  C   PHE A 190     -11.225  17.188 -13.049  1.00  0.00           C  \nATOM   1333  O   PHE A 190     -11.410  15.979 -13.160  1.00  0.00           O  \nATOM   1334  CB  PHE A 190      -9.341  16.794 -11.329  1.00  0.00           C  \nATOM   1335  CG  PHE A 190      -7.906  17.141 -11.013  1.00  0.00           C  \nATOM   1336  CD1 PHE A 190      -6.884  16.693 -11.871  1.00  0.00           C  \nATOM   1337  CD2 PHE A 190      -7.589  17.937  -9.903  1.00  0.00           C  \nATOM   1338  CE1 PHE A 190      -5.544  17.024 -11.594  1.00  0.00           C  \nATOM   1339  CE2 PHE A 190      -6.252  18.272  -9.633  1.00  0.00           C  \nATOM   1340  CZ  PHE A 190      -5.230  17.816 -10.478  1.00  0.00           C  \nATOM   1341  N   SER A 191     -12.104  18.081 -13.574  1.00  0.00           N  \nATOM   1342  CA  SER A 191     -13.375  17.715 -14.226  1.00  0.00           C  \nATOM   1343  C   SER A 191     -13.230  16.786 -15.444  1.00  0.00           C  \nATOM   1344  O   SER A 191     -14.182  16.104 -15.816  1.00  0.00           O  \nATOM   1345  CB  SER A 191     -14.120  18.990 -14.649  1.00  0.00           C  \nATOM   1346  OG  SER A 191     -13.317  19.792 -15.482  1.00  0.00           O  \nATOM   1347  N   SER A 192     -12.043  16.724 -16.055  1.00  0.00           N  \nATOM   1348  CA  SER A 192     -11.745  15.825 -17.175  1.00  0.00           C  \nATOM   1349  C   SER A 192     -11.516  14.360 -16.756  1.00  0.00           C  \nATOM   1350  O   SER A 192     -11.575  13.477 -17.614  1.00  0.00           O  \nATOM   1351  CB  SER A 192     -10.534  16.341 -17.950  1.00  0.00           C  \nATOM   1352  OG  SER A 192      -9.394  16.392 -17.109  1.00  0.00           O  \nATOM   1353  N   VAL A 193     -11.276  14.082 -15.466  1.00  0.00           N  \nATOM   1354  CA  VAL A 193     -10.964  12.731 -14.970  1.00  0.00           C  \nATOM   1355  C   VAL A 193     -12.181  11.822 -15.117  1.00  0.00           C  \nATOM   1356  O   VAL A 193     -13.256  12.071 -14.571  1.00  0.00           O  \nATOM   1357  CB  VAL A 193     -10.442  12.757 -13.522  1.00  0.00           C  \nATOM   1358  CG1 VAL A 193     -10.231  11.348 -12.955  1.00  0.00           C  \nATOM   1359  CG2 VAL A 193      -9.082  13.469 -13.449  1.00  0.00           C  \nATOM   1360  N   THR A 194     -11.986  10.711 -15.831  1.00  0.00           N  \nATOM   1361  CA  THR A 194     -12.959   9.625 -15.985  1.00  0.00           C  \nATOM   1362  C   THR A 194     -12.553   8.355 -15.237  1.00  0.00           C  \nATOM   1363  O   THR A 194     -13.345   7.430 -15.100  1.00  0.00           O  \nATOM   1364  CB  THR A 194     -13.202   9.295 -17.467  1.00  0.00           C  \nATOM   1365  OG1 THR A 194     -11.988   8.960 -18.095  1.00  0.00           O  \nATOM   1366  CG2 THR A 194     -13.793  10.480 -18.237  1.00  0.00           C  \nATOM   1367  N   VAL A 195     -11.323   8.297 -14.739  1.00  0.00           N  \nATOM   1368  CA  VAL A 195     -10.769   7.169 -13.983  1.00  0.00           C  \nATOM   1369  C   VAL A 195     -11.434   7.089 -12.600  1.00  0.00           C  \nATOM   1370  O   VAL A 195     -11.502   8.113 -11.919  1.00  0.00           O  \nATOM   1371  CB  VAL A 195      -9.238   7.314 -13.854  1.00  0.00           C  \nATOM   1372  CG1 VAL A 195      -8.607   6.182 -13.045  1.00  0.00           C  \nATOM   1373  CG2 VAL A 195      -8.572   7.331 -15.235  1.00  0.00           C  \nATOM   1374  N   PRO A 196     -11.900   5.913 -12.150  1.00  0.00           N  \nATOM   1375  CA  PRO A 196     -12.362   5.711 -10.782  1.00  0.00           C  \nATOM   1376  C   PRO A 196     -11.353   6.217  -9.745  1.00  0.00           C  \nATOM   1377  O   PRO A 196     -10.212   5.739  -9.693  1.00  0.00           O  \nATOM   1378  CB  PRO A 196     -12.597   4.198 -10.664  1.00  0.00           C  \nATOM   1379  CG  PRO A 196     -12.997   3.812 -12.084  1.00  0.00           C  \nATOM   1380  CD  PRO A 196     -12.093   4.710 -12.930  1.00  0.00           C  \nATOM   1381  N   THR A 197     -11.758   7.191  -8.916  1.00  0.00           N  \nATOM   1382  CA  THR A 197     -10.837   7.921  -8.041  1.00  0.00           C  \nATOM   1383  C   THR A 197     -11.306   7.930  -6.587  1.00  0.00           C  \nATOM   1384  O   THR A 197     -12.351   8.490  -6.265  1.00  0.00           O  \nATOM   1385  CB  THR A 197     -10.621   9.359  -8.548  1.00  0.00           C  \nATOM   1386  OG1 THR A 197     -10.091   9.322  -9.859  1.00  0.00           O  \nATOM   1387  CG2 THR A 197      -9.604  10.125  -7.699  1.00  0.00           C  \nATOM   1388  N   LEU A 198     -10.499   7.347  -5.685  1.00  0.00           N  \nATOM   1389  CA  LEU A 198     -10.655   7.440  -4.231  1.00  0.00           C  \nATOM   1390  C   LEU A 198      -9.678   8.486  -3.689  1.00  0.00           C  \nATOM   1391  O   LEU A 198      -8.469   8.299  -3.794  1.00  0.00           O  \nATOM   1392  CB  LEU A 198     -10.428   6.044  -3.615  1.00  0.00           C  \nATOM   1393  CG  LEU A 198     -10.205   6.014  -2.079  1.00  0.00           C  \nATOM   1394  CD1 LEU A 198     -11.399   6.540  -1.293  1.00  0.00           C  \nATOM   1395  CD2 LEU A 198      -9.957   4.564  -1.646  1.00  0.00           C  \nATOM   1396  N   ILE A 199     -10.192   9.538  -3.063  1.00  0.00           N  \nATOM   1397  CA  ILE A 199      -9.376  10.530  -2.359  1.00  0.00           C  \nATOM   1398  C   ILE A 199      -9.376  10.207  -0.868  1.00  0.00           C  \nATOM   1399  O   ILE A 199     -10.427  10.215  -0.234  1.00  0.00           O  \nATOM   1400  CB  ILE A 199      -9.868  11.964  -2.660  1.00  0.00           C  \nATOM   1401  CG1 ILE A 199      -9.702  12.259  -4.168  1.00  0.00           C  \nATOM   1402  CG2 ILE A 199      -9.097  12.983  -1.803  1.00  0.00           C  \nATOM   1403  CD1 ILE A 199     -10.191  13.646  -4.598  1.00  0.00           C  \nATOM   1404  N   PHE A 200      -8.202   9.936  -0.300  1.00  0.00           N  \nATOM   1405  CA  PHE A 200      -7.989   9.985   1.140  1.00  0.00           C  \nATOM   1406  C   PHE A 200      -7.654  11.423   1.534  1.00  0.00           C  \nATOM   1407  O   PHE A 200      -6.705  11.995   1.005  1.00  0.00           O  \nATOM   1408  CB  PHE A 200      -6.865   9.032   1.555  1.00  0.00           C  \nATOM   1409  CG  PHE A 200      -7.147   7.569   1.299  1.00  0.00           C  \nATOM   1410  CD1 PHE A 200      -8.026   6.869   2.145  1.00  0.00           C  \nATOM   1411  CD2 PHE A 200      -6.508   6.900   0.235  1.00  0.00           C  \nATOM   1412  CE1 PHE A 200      -8.254   5.495   1.937  1.00  0.00           C  \nATOM   1413  CE2 PHE A 200      -6.747   5.526   0.027  1.00  0.00           C  \nATOM   1414  CZ  PHE A 200      -7.615   4.827   0.880  1.00  0.00           C  \nATOM   1415  N   ALA A 201      -8.413  12.002   2.477  1.00  0.00           N  \nATOM   1416  CA  ALA A 201      -8.161  13.315   3.060  1.00  0.00           C  \nATOM   1417  C   ALA A 201      -7.822  13.184   4.549  1.00  0.00           C  \nATOM   1418  O   ALA A 201      -8.086  12.158   5.179  1.00  0.00           O  \nATOM   1419  CB  ALA A 201      -9.371  14.210   2.789  1.00  0.00           C  \nATOM   1420  N   CYS A 202      -7.221  14.230   5.123  1.00  0.00           N  \nATOM   1421  CA  CYS A 202      -6.838  14.292   6.528  1.00  0.00           C  \nATOM   1422  C   CYS A 202      -7.468  15.559   7.133  1.00  0.00           C  \nATOM   1423  O   CYS A 202      -7.109  16.653   6.716  1.00  0.00           O  \nATOM   1424  CB  CYS A 202      -5.306  14.312   6.620  1.00  0.00           C  \nATOM   1425  SG  CYS A 202      -4.405  12.942   5.818  1.00  0.00           S  \nATOM   1426  N   GLU A 203      -8.380  15.425   8.083  1.00  0.00           N  \nATOM   1427  CA  GLU A 203      -9.248  16.519   8.585  1.00  0.00           C  \nATOM   1428  C   GLU A 203      -8.459  17.772   9.011  1.00  0.00           C  \nATOM   1429  O   GLU A 203      -8.772  18.883   8.585  1.00  0.00           O  \nATOM   1430  CB  GLU A 203     -10.075  15.969   9.759  1.00  0.00           C  \nATOM   1431  CG  GLU A 203     -11.084  16.981  10.345  1.00  0.00           C  \nATOM   1432  CD  GLU A 203     -11.821  16.407  11.574  1.00  0.00           C  \nATOM   1433  OE1 GLU A 203     -12.967  16.824  11.843  1.00  0.00           O  \nATOM   1434  OE2 GLU A 203     -11.221  15.550  12.273  1.00  0.00           O  \nATOM   1435  N   ASN A 204      -7.397  17.586   9.793  1.00  0.00           N  \nATOM   1436  CA  ASN A 204      -6.563  18.680  10.312  1.00  0.00           C  \nATOM   1437  C   ASN A 204      -5.314  18.927   9.446  1.00  0.00           C  \nATOM   1438  O   ASN A 204      -4.242  19.265   9.963  1.00  0.00           O  \nATOM   1439  CB  ASN A 204      -6.249  18.416  11.795  1.00  0.00           C  \nATOM   1440  CG  ASN A 204      -7.482  18.414  12.685  1.00  0.00           C  \nATOM   1441  OD1 ASN A 204      -8.500  19.011  12.411  1.00  0.00           O  \nATOM   1442  ND2 ASN A 204      -7.411  17.765  13.826  1.00  0.00           N  \nATOM   1443  N   ASP A 205      -5.408  18.711   8.143  1.00  0.00           N  \nATOM   1444  CA  ASP A 205      -4.316  18.991   7.213  1.00  0.00           C  \nATOM   1445  C   ASP A 205      -4.051  20.498   7.134  1.00  0.00           C  \nATOM   1446  O   ASP A 205      -4.874  21.267   6.653  1.00  0.00           O  \nATOM   1447  CB  ASP A 205      -4.620  18.383   5.841  1.00  0.00           C  \nATOM   1448  CG  ASP A 205      -3.396  18.323   4.927  1.00  0.00           C  \nATOM   1449  OD1 ASP A 205      -2.476  19.162   5.075  1.00  0.00           O  \nATOM   1450  OD2 ASP A 205      -3.319  17.390   4.100  1.00  0.00           O  \nATOM   1451  N   SER A 206      -2.874  20.921   7.603  1.00  0.00           N  \nATOM   1452  CA  SER A 206      -2.432  22.325   7.561  1.00  0.00           C  \nATOM   1453  C   SER A 206      -1.489  22.641   6.403  1.00  0.00           C  \nATOM   1454  O   SER A 206      -1.093  23.796   6.233  1.00  0.00           O  \nATOM   1455  CB  SER A 206      -1.789  22.699   8.901  1.00  0.00           C  \nATOM   1456  OG  SER A 206      -0.651  21.895   9.160  1.00  0.00           O  \nATOM   1457  N   ILE A 207      -1.094  21.633   5.621  1.00  0.00           N  \nATOM   1458  CA  ILE A 207      -0.180  21.778   4.476  1.00  0.00           C  \nATOM   1459  C   ILE A 207      -0.978  21.863   3.177  1.00  0.00           C  \nATOM   1460  O   ILE A 207      -0.727  22.739   2.351  1.00  0.00           O  \nATOM   1461  CB  ILE A 207       0.828  20.603   4.444  1.00  0.00           C  \nATOM   1462  CG1 ILE A 207       1.646  20.470   5.752  1.00  0.00           C  \nATOM   1463  CG2 ILE A 207       1.779  20.727   3.233  1.00  0.00           C  \nATOM   1464  CD1 ILE A 207       2.474  21.697   6.148  1.00  0.00           C  \nATOM   1465  N   ALA A 208      -1.954  20.962   3.010  1.00  0.00           N  \nATOM   1466  CA  ALA A 208      -2.927  20.968   1.936  1.00  0.00           C  \nATOM   1467  C   ALA A 208      -4.337  20.928   2.552  1.00  0.00           C  \nATOM   1468  O   ALA A 208      -4.974  19.869   2.537  1.00  0.00           O  \nATOM   1469  CB  ALA A 208      -2.624  19.782   1.002  1.00  0.00           C  \nATOM   1470  N   PRO A 209      -4.822  22.047   3.129  1.00  0.00           N  \nATOM   1471  CA  PRO A 209      -6.103  22.070   3.833  1.00  0.00           C  \nATOM   1472  C   PRO A 209      -7.236  21.492   2.990  1.00  0.00           C  \nATOM   1473  O   PRO A 209      -7.383  21.826   1.806  1.00  0.00           O  \nATOM   1474  CB  PRO A 209      -6.349  23.531   4.210  1.00  0.00           C  \nATOM   1475  CG  PRO A 209      -4.927  24.103   4.305  1.00  0.00           C  \nATOM   1476  CD  PRO A 209      -4.181  23.346   3.213  1.00  0.00           C  \nATOM   1477  N   VAL A 210      -8.048  20.628   3.602  1.00  0.00           N  \nATOM   1478  CA  VAL A 210      -9.070  19.847   2.891  1.00  0.00           C  \nATOM   1479  C   VAL A 210     -10.040  20.742   2.115  1.00  0.00           C  \nATOM   1480  O   VAL A 210     -10.356  20.452   0.964  1.00  0.00           O  \nATOM   1481  CB  VAL A 210      -9.823  18.955   3.893  1.00  0.00           C  \nATOM   1482  CG1 VAL A 210     -11.039  18.272   3.258  1.00  0.00           C  \nATOM   1483  CG2 VAL A 210      -8.892  17.861   4.430  1.00  0.00           C  \nATOM   1484  N   ASN A 211     -10.453  21.875   2.724  1.00  0.00           N  \nATOM   1485  CA  ASN A 211     -11.406  22.832   2.152  1.00  0.00           C  \nATOM   1486  C   ASN A 211     -10.871  23.598   0.927  1.00  0.00           C  \nATOM   1487  O   ASN A 211     -11.669  24.143   0.173  1.00  0.00           O  \nATOM   1488  CB  ASN A 211     -11.865  23.795   3.255  1.00  0.00           C  \nATOM   1489  CG  ASN A 211     -10.762  24.708   3.794  1.00  0.00           C  \nATOM   1490  OD1 ASN A 211      -9.588  24.553   3.532  1.00  0.00           O  \nATOM   1491  ND2 ASN A 211     -11.116  25.694   4.584  1.00  0.00           N  \nATOM   1492  N   SER A 212      -9.559  23.635   0.717  1.00  0.00           N  \nATOM   1493  CA  SER A 212      -8.929  24.318  -0.412  1.00  0.00           C  \nATOM   1494  C   SER A 212      -8.199  23.380  -1.376  1.00  0.00           C  \nATOM   1495  O   SER A 212      -7.773  23.814  -2.443  1.00  0.00           O  \nATOM   1496  CB  SER A 212      -7.990  25.424   0.091  1.00  0.00           C  \nATOM   1497  OG  SER A 212      -6.971  24.890   0.921  1.00  0.00           O  \nATOM   1498  N   SER A 213      -8.078  22.106  -1.032  1.00  0.00           N  \nATOM   1499  CA  SER A 213      -7.357  21.105  -1.822  1.00  0.00           C  \nATOM   1500  C   SER A 213      -8.224  19.874  -2.100  1.00  0.00           C  \nATOM   1501  O   SER A 213      -8.894  19.811  -3.129  1.00  0.00           O  \nATOM   1502  CB  SER A 213      -6.023  20.746  -1.155  1.00  0.00           C  \nATOM   1503  OG  SER A 213      -5.222  21.901  -0.996  1.00  0.00           O  \nATOM   1504  N   ALA A 214      -8.258  18.892  -1.199  1.00  0.00           N  \nATOM   1505  CA  ALA A 214      -8.851  17.580  -1.447  1.00  0.00           C  \nATOM   1506  C   ALA A 214     -10.354  17.639  -1.772  1.00  0.00           C  \nATOM   1507  O   ALA A 214     -10.809  16.945  -2.687  1.00  0.00           O  \nATOM   1508  CB  ALA A 214      -8.579  16.696  -0.224  1.00  0.00           C  \nATOM   1509  N   LEU A 215     -11.117  18.482  -1.053  1.00  0.00           N  \nATOM   1510  CA  LEU A 215     -12.561  18.580  -1.260  1.00  0.00           C  \nATOM   1511  C   LEU A 215     -12.935  19.266  -2.590  1.00  0.00           C  \nATOM   1512  O   LEU A 215     -13.670  18.647  -3.359  1.00  0.00           O  \nATOM   1513  CB  LEU A 215     -13.220  19.196  -0.009  1.00  0.00           C  \nATOM   1514  CG  LEU A 215     -14.747  19.328  -0.115  1.00  0.00           C  \nATOM   1515  CD1 LEU A 215     -15.404  17.951  -0.304  1.00  0.00           C  \nATOM   1516  CD2 LEU A 215     -15.298  19.963   1.159  1.00  0.00           C  \nATOM   1517  N   PRO A 216     -12.389  20.445  -2.958  1.00  0.00           N  \nATOM   1518  CA  PRO A 216     -12.604  21.023  -4.288  1.00  0.00           C  \nATOM   1519  C   PRO A 216     -12.172  20.102  -5.433  1.00  0.00           C  \nATOM   1520  O   PRO A 216     -12.858  20.024  -6.458  1.00  0.00           O  \nATOM   1521  CB  PRO A 216     -11.803  22.328  -4.301  1.00  0.00           C  \nATOM   1522  CG  PRO A 216     -11.773  22.735  -2.832  1.00  0.00           C  \nATOM   1523  CD  PRO A 216     -11.685  21.396  -2.110  1.00  0.00           C  \nATOM   1524  N   ILE A 217     -11.073  19.355  -5.258  1.00  0.00           N  \nATOM   1525  CA  ILE A 217     -10.650  18.332  -6.228  1.00  0.00           C  \nATOM   1526  C   ILE A 217     -11.739  17.267  -6.386  1.00  0.00           C  \nATOM   1527  O   ILE A 217     -12.138  16.960  -7.510  1.00  0.00           O  \nATOM   1528  CB  ILE A 217      -9.294  17.716  -5.816  1.00  0.00           C  \nATOM   1529  CG1 ILE A 217      -8.151  18.733  -6.037  1.00  0.00           C  \nATOM   1530  CG2 ILE A 217      -9.007  16.425  -6.614  1.00  0.00           C  \nATOM   1531  CD1 ILE A 217      -6.815  18.333  -5.371  1.00  0.00           C  \nATOM   1532  N   TYR A 218     -12.243  16.718  -5.279  1.00  0.00           N  \nATOM   1533  CA  TYR A 218     -13.323  15.732  -5.310  1.00  0.00           C  \nATOM   1534  C   TYR A 218     -14.584  16.291  -5.981  1.00  0.00           C  \nATOM   1535  O   TYR A 218     -15.164  15.636  -6.852  1.00  0.00           O  \nATOM   1536  CB  TYR A 218     -13.631  15.257  -3.882  1.00  0.00           C  \nATOM   1537  CG  TYR A 218     -14.849  14.361  -3.803  1.00  0.00           C  \nATOM   1538  CD1 TYR A 218     -16.127  14.909  -3.506  1.00  0.00           C  \nATOM   1539  CD2 TYR A 218     -14.737  12.981  -4.066  1.00  0.00           C  \nATOM   1540  CE1 TYR A 218     -17.259  14.088  -3.467  1.00  0.00           C  \nATOM   1541  CE2 TYR A 218     -15.872  12.152  -4.025  1.00  0.00           C  \nATOM   1542  CZ  TYR A 218     -17.134  12.705  -3.724  1.00  0.00           C  \nATOM   1543  OH  TYR A 218     -18.232  11.914  -3.690  1.00  0.00           O  \nATOM   1544  N   ASP A 219     -14.989  17.507  -5.610  1.00  0.00           N  \nATOM   1545  CA  ASP A 219     -16.212  18.132  -6.112  1.00  0.00           C  \nATOM   1546  C   ASP A 219     -16.123  18.447  -7.616  1.00  0.00           C  \nATOM   1547  O   ASP A 219     -17.124  18.322  -8.323  1.00  0.00           O  \nATOM   1548  CB  ASP A 219     -16.538  19.377  -5.259  1.00  0.00           C  \nATOM   1549  CG  ASP A 219     -17.043  19.024  -3.843  1.00  0.00           C  \nATOM   1550  OD1 ASP A 219     -17.514  17.878  -3.643  1.00  0.00           O  \nATOM   1551  OD2 ASP A 219     -17.010  19.907  -2.954  1.00  0.00           O  \nATOM   1552  N   SER A 220     -14.928  18.732  -8.129  1.00  0.00           N  \nATOM   1553  CA  SER A 220     -14.707  19.013  -9.562  1.00  0.00           C  \nATOM   1554  C   SER A 220     -14.980  17.821 -10.484  1.00  0.00           C  \nATOM   1555  O   SER A 220     -15.338  18.007 -11.653  1.00  0.00           O  \nATOM   1556  CB  SER A 220     -13.274  19.515  -9.774  1.00  0.00           C  \nATOM   1557  OG  SER A 220     -12.320  18.475  -9.802  1.00  0.00           O  \nATOM   1558  N   MET A 221     -14.824  16.587  -9.992  1.00  0.00           N  \nATOM   1559  CA  MET A 221     -14.974  15.374 -10.799  1.00  0.00           C  \nATOM   1560  C   MET A 221     -16.444  14.956 -10.889  1.00  0.00           C  \nATOM   1561  O   MET A 221     -17.036  14.541  -9.894  1.00  0.00           O  \nATOM   1562  CB  MET A 221     -14.120  14.244 -10.223  1.00  0.00           C  \nATOM   1563  CG  MET A 221     -12.615  14.508 -10.313  1.00  0.00           C  \nATOM   1564  SD  MET A 221     -11.605  13.117  -9.712  1.00  0.00           S  \nATOM   1565  CE  MET A 221     -11.924  13.211  -7.930  1.00  0.00           C  \nATOM   1566  N   SER A 222     -17.038  14.980 -12.079  1.00  0.00           N  \nATOM   1567  CA  SER A 222     -18.446  14.602 -12.301  1.00  0.00           C  \nATOM   1568  C   SER A 222     -18.651  13.413 -13.244  1.00  0.00           C  \nATOM   1569  O   SER A 222     -19.779  12.975 -13.447  1.00  0.00           O  \nATOM   1570  CB  SER A 222     -19.227  15.823 -12.807  1.00  0.00           C  \nATOM   1571  OG  SER A 222     -18.662  16.290 -14.016  1.00  0.00           O  \nATOM   1572  N   ARG A 223     -17.568  12.863 -13.812  1.00  0.00           N  \nATOM   1573  CA  ARG A 223     -17.615  11.888 -14.912  1.00  0.00           C  \nATOM   1574  C   ARG A 223     -17.111  10.496 -14.546  1.00  0.00           C  \nATOM   1575  O   ARG A 223     -16.944   9.661 -15.439  1.00  0.00           O  \nATOM   1576  CB  ARG A 223     -16.845  12.458 -16.115  1.00  0.00           C  \nATOM   1577  CG  ARG A 223     -17.414  13.793 -16.622  1.00  0.00           C  \nATOM   1578  CD  ARG A 223     -16.587  14.339 -17.789  1.00  0.00           C  \nATOM   1579  NE  ARG A 223     -16.627  13.447 -18.968  1.00  0.00           N  \nATOM   1580  CZ  ARG A 223     -15.923  13.583 -20.077  1.00  0.00           C  \nATOM   1581  NH1 ARG A 223     -15.089  14.574 -20.240  1.00  0.00           N  \nATOM   1582  NH2 ARG A 223     -16.041  12.727 -21.057  1.00  0.00           N  \nATOM   1583  N   ASN A 224     -16.851  10.242 -13.281  1.00  0.00           N  \nATOM   1584  CA  ASN A 224     -16.213   9.017 -12.809  1.00  0.00           C  \nATOM   1585  C   ASN A 224     -16.897   8.451 -11.561  1.00  0.00           C  \nATOM   1586  O   ASN A 224     -17.591   9.165 -10.832  1.00  0.00           O  \nATOM   1587  CB  ASN A 224     -14.722   9.293 -12.562  1.00  0.00           C  \nATOM   1588  CG  ASN A 224     -14.414  10.290 -11.448  1.00  0.00           C  \nATOM   1589  OD1 ASN A 224     -15.165  11.201 -11.155  1.00  0.00           O  \nATOM   1590  ND2 ASN A 224     -13.274  10.144 -10.820  1.00  0.00           N  \nATOM   1591  N   ALA A 225     -16.654   7.179 -11.294  1.00  0.00           N  \nATOM   1592  CA  ALA A 225     -16.828   6.643  -9.953  1.00  0.00           C  \nATOM   1593  C   ALA A 225     -15.825   7.348  -9.025  1.00  0.00           C  \nATOM   1594  O   ALA A 225     -14.632   7.397  -9.328  1.00  0.00           O  \nATOM   1595  CB  ALA A 225     -16.626   5.131  -9.965  1.00  0.00           C  \nATOM   1596  N   LYS A 226     -16.314   7.922  -7.918  1.00  0.00           N  \nATOM   1597  CA  LYS A 226     -15.457   8.654  -6.992  1.00  0.00           C  \nATOM   1598  C   LYS A 226     -15.837   8.432  -5.539  1.00  0.00           C  \nATOM   1599  O   LYS A 226     -16.997   8.187  -5.216  1.00  0.00           O  \nATOM   1600  CB  LYS A 226     -15.407  10.148  -7.357  1.00  0.00           C  \nATOM   1601  CG  LYS A 226     -16.739  10.891  -7.160  1.00  0.00           C  \nATOM   1602  CD  LYS A 226     -16.570  12.389  -7.470  1.00  0.00           C  \nATOM   1603  CE  LYS A 226     -17.864  13.167  -7.189  1.00  0.00           C  \nATOM   1604  NZ  LYS A 226     -17.671  14.620  -7.421  1.00  0.00           N  \nATOM   1605  N   GLN A 227     -14.847   8.553  -4.665  1.00  0.00           N  \nATOM   1606  CA  GLN A 227     -15.032   8.510  -3.224  1.00  0.00           C  \nATOM   1607  C   GLN A 227     -14.088   9.487  -2.530  1.00  0.00           C  \nATOM   1608  O   GLN A 227     -12.947   9.657  -2.945  1.00  0.00           O  \nATOM   1609  CB  GLN A 227     -14.847   7.070  -2.735  1.00  0.00           C  \nATOM   1610  CG  GLN A 227     -15.180   6.880  -1.246  1.00  0.00           C  \nATOM   1611  CD  GLN A 227     -15.287   5.405  -0.901  1.00  0.00           C  \nATOM   1612  OE1 GLN A 227     -14.456   4.826  -0.227  1.00  0.00           O  \nATOM   1613  NE2 GLN A 227     -16.317   4.728  -1.362  1.00  0.00           N  \nATOM   1614  N   PHE A 228     -14.584  10.096  -1.456  1.00  0.00           N  \nATOM   1615  CA  PHE A 228     -13.851  10.936  -0.530  1.00  0.00           C  \nATOM   1616  C   PHE A 228     -13.925  10.311   0.858  1.00  0.00           C  \nATOM   1617  O   PHE A 228     -15.009   9.998   1.340  1.00  0.00           O  \nATOM   1618  CB  PHE A 228     -14.468  12.338  -0.566  1.00  0.00           C  \nATOM   1619  CG  PHE A 228     -13.773  13.339   0.305  1.00  0.00           C  \nATOM   1620  CD1 PHE A 228     -14.210  13.560   1.633  1.00  0.00           C  \nATOM   1621  CD2 PHE A 228     -12.690  14.076  -0.213  1.00  0.00           C  \nATOM   1622  CE1 PHE A 228     -13.583  14.520   2.416  1.00  0.00           C  \nATOM   1623  CE2 PHE A 228     -12.074  15.035   0.579  1.00  0.00           C  \nATOM   1624  CZ  PHE A 228     -12.515  15.268   1.891  1.00  0.00           C  \nATOM   1625  N   LEU A 229     -12.773  10.077   1.475  1.00  0.00           N  \nATOM   1626  CA  LEU A 229     -12.657   9.499   2.803  1.00  0.00           C  \nATOM   1627  C   LEU A 229     -11.706  10.348   3.645  1.00  0.00           C  \nATOM   1628  O   LEU A 229     -10.488  10.267   3.468  1.00  0.00           O  \nATOM   1629  CB  LEU A 229     -12.166   8.045   2.684  1.00  0.00           C  \nATOM   1630  CG  LEU A 229     -12.162   7.303   4.029  1.00  0.00           C  \nATOM   1631  CD1 LEU A 229     -13.577   6.873   4.418  1.00  0.00           C  \nATOM   1632  CD2 LEU A 229     -11.316   6.039   3.945  1.00  0.00           C  \nATOM   1633  N   GLU A 230     -12.263  11.131   4.558  1.00  0.00           N  \nATOM   1634  CA  GLU A 230     -11.497  11.983   5.459  1.00  0.00           C  \nATOM   1635  C   GLU A 230     -11.218  11.270   6.785  1.00  0.00           C  \nATOM   1636  O   GLU A 230     -12.113  10.733   7.437  1.00  0.00           O  \nATOM   1637  CB  GLU A 230     -12.240  13.298   5.667  1.00  0.00           C  \nATOM   1638  CG  GLU A 230     -11.376  14.333   6.378  1.00  0.00           C  \nATOM   1639  CD  GLU A 230     -12.160  15.628   6.550  1.00  0.00           C  \nATOM   1640  OE1 GLU A 230     -12.953  15.684   7.516  1.00  0.00           O  \nATOM   1641  OE2 GLU A 230     -11.994  16.511   5.699  1.00  0.00           O  \nATOM   1642  N   ILE A 231      -9.954  11.269   7.193  1.00  0.00           N  \nATOM   1643  CA  ILE A 231      -9.505  10.675   8.453  1.00  0.00           C  \nATOM   1644  C   ILE A 231      -9.671  11.685   9.589  1.00  0.00           C  \nATOM   1645  O   ILE A 231      -9.027  12.739   9.580  1.00  0.00           O  \nATOM   1646  CB  ILE A 231      -8.058  10.164   8.340  1.00  0.00           C  \nATOM   1647  CG1 ILE A 231      -7.892   9.127   7.196  1.00  0.00           C  \nATOM   1648  CG2 ILE A 231      -7.568   9.576   9.665  1.00  0.00           C  \nATOM   1649  CD1 ILE A 231      -8.814   7.905   7.276  1.00  0.00           C  \nATOM   1650  N   ASN A 232     -10.483  11.325  10.576  1.00  0.00           N  \nATOM   1651  CA  ASN A 232     -10.786  12.131  11.750  1.00  0.00           C  \nATOM   1652  C   ASN A 232      -9.507  12.501  12.521  1.00  0.00           C  \nATOM   1653  O   ASN A 232      -8.702  11.629  12.868  1.00  0.00           O  \nATOM   1654  CB  ASN A 232     -11.787  11.353  12.621  1.00  0.00           C  \nATOM   1655  CG  ASN A 232     -12.309  12.128  13.815  1.00  0.00           C  \nATOM   1656  OD1 ASN A 232     -11.828  13.167  14.196  1.00  0.00           O  \nATOM   1657  ND2 ASN A 232     -13.309  11.627  14.492  1.00  0.00           N  \nATOM   1658  N   GLY A 233      -9.334  13.809  12.775  1.00  0.00           N  \nATOM   1659  CA  GLY A 233      -8.181  14.379  13.478  1.00  0.00           C  \nATOM   1660  C   GLY A 233      -6.838  14.228  12.759  1.00  0.00           C  \nATOM   1661  O   GLY A 233      -5.791  14.485  13.352  1.00  0.00           O  \nATOM   1662  N   GLY A 234      -6.834  13.758  11.511  1.00  0.00           N  \nATOM   1663  CA  GLY A 234      -5.620  13.437  10.777  1.00  0.00           C  \nATOM   1664  C   GLY A 234      -4.796  14.670  10.399  1.00  0.00           C  \nATOM   1665  O   GLY A 234      -5.315  15.622   9.820  1.00  0.00           O  \nATOM   1666  N   SER A 235      -3.489  14.647  10.656  1.00  0.00           N  \nATOM   1667  CA  SER A 235      -2.531  15.591  10.071  1.00  0.00           C  \nATOM   1668  C   SER A 235      -2.214  15.232   8.615  1.00  0.00           C  \nATOM   1669  O   SER A 235      -2.536  14.136   8.159  1.00  0.00           O  \nATOM   1670  CB  SER A 235      -1.243  15.616  10.901  1.00  0.00           C  \nATOM   1671  OG  SER A 235      -0.543  14.385  10.781  1.00  0.00           O  \nATOM   1672  N   HIS A 236      -1.481  16.078   7.898  1.00  0.00           N  \nATOM   1673  CA  HIS A 236      -1.112  15.872   6.482  1.00  0.00           C  \nATOM   1674  C   HIS A 236      -0.527  14.479   6.150  1.00  0.00           C  \nATOM   1675  O   HIS A 236      -0.635  13.985   5.024  1.00  0.00           O  \nATOM   1676  CB  HIS A 236      -0.100  16.953   6.089  1.00  0.00           C  \nATOM   1677  CG  HIS A 236       0.183  16.991   4.605  1.00  0.00           C  \nATOM   1678  ND1 HIS A 236      -0.766  17.136   3.612  1.00  0.00           N  \nATOM   1679  CD2 HIS A 236       1.401  16.905   3.971  1.00  0.00           C  \nATOM   1680  CE1 HIS A 236      -0.146  17.141   2.437  1.00  0.00           C  \nATOM   1681  NE2 HIS A 236       1.178  17.002   2.597  1.00  0.00           N  \nATOM   1682  N   SER A 237       0.087  13.803   7.118  1.00  0.00           N  \nATOM   1683  CA  SER A 237       0.683  12.466   6.970  1.00  0.00           C  \nATOM   1684  C   SER A 237      -0.234  11.314   7.390  1.00  0.00           C  \nATOM   1685  O   SER A 237       0.244  10.187   7.552  1.00  0.00           O  \nATOM   1686  CB  SER A 237       2.034  12.438   7.694  1.00  0.00           C  \nATOM   1687  OG  SER A 237       1.875  12.632   9.086  1.00  0.00           O  \nATOM   1688  N   CYS A 238      -1.557  11.552   7.574  1.00  0.00           N  \nATOM   1689  CA  CYS A 238      -2.494  10.593   8.156  1.00  0.00           C  \nATOM   1690  C   CYS A 238      -2.588   9.235   7.428  1.00  0.00           C  \nATOM   1691  O   CYS A 238      -3.008   8.249   8.024  1.00  0.00           O  \nATOM   1692  CB  CYS A 238      -3.879  11.245   8.263  1.00  0.00           C  \nATOM   1693  SG  CYS A 238      -4.835  11.296   6.717  1.00  0.00           S  \nATOM   1694  N   ALA A 239      -2.179   9.172   6.154  1.00  0.00           N  \nATOM   1695  CA  ALA A 239      -2.178   7.957   5.341  1.00  0.00           C  \nATOM   1696  C   ALA A 239      -0.762   7.422   5.016  1.00  0.00           C  \nATOM   1697  O   ALA A 239      -0.621   6.442   4.277  1.00  0.00           O  \nATOM   1698  CB  ALA A 239      -3.023   8.234   4.087  1.00  0.00           C  \nATOM   1699  N   ASN A 240       0.289   8.028   5.575  1.00  0.00           N  \nATOM   1700  CA  ASN A 240       1.679   7.644   5.314  1.00  0.00           C  \nATOM   1701  C   ASN A 240       2.115   6.386   6.083  1.00  0.00           C  \nATOM   1702  O   ASN A 240       1.408   5.860   6.940  1.00  0.00           O  \nATOM   1703  CB  ASN A 240       2.592   8.850   5.603  1.00  0.00           C  \nATOM   1704  CG  ASN A 240       2.516   9.928   4.546  1.00  0.00           C  \nATOM   1705  OD1 ASN A 240       1.743   9.877   3.598  1.00  0.00           O  \nATOM   1706  ND2 ASN A 240       3.345  10.936   4.655  1.00  0.00           N  \nATOM   1707  N   SER A 241       3.315   5.892   5.799  1.00  0.00           N  \nATOM   1708  CA  SER A 241       3.929   4.804   6.562  1.00  0.00           C  \nATOM   1709  C   SER A 241       3.955   5.125   8.059  1.00  0.00           C  \nATOM   1710  O   SER A 241       4.370   6.216   8.444  1.00  0.00           O  \nATOM   1711  CB  SER A 241       5.370   4.549   6.102  1.00  0.00           C  \nATOM   1712  OG  SER A 241       5.417   4.260   4.716  1.00  0.00           O  \nATOM   1713  N   GLY A 242       3.547   4.177   8.889  1.00  0.00           N  \nATOM   1714  CA  GLY A 242       3.472   4.361  10.344  1.00  0.00           C  \nATOM   1715  C   GLY A 242       2.166   4.987  10.855  1.00  0.00           C  \nATOM   1716  O   GLY A 242       2.037   5.172  12.062  1.00  0.00           O  \nATOM   1717  N   ASN A 243       1.184   5.277   9.983  1.00  0.00           N  \nATOM   1718  CA  ASN A 243      -0.114   5.787  10.423  1.00  0.00           C  \nATOM   1719  C   ASN A 243      -0.864   4.780  11.324  1.00  0.00           C  \nATOM   1720  O   ASN A 243      -0.721   3.560  11.187  1.00  0.00           O  \nATOM   1721  CB  ASN A 243      -0.969   6.226   9.222  1.00  0.00           C  \nATOM   1722  CG  ASN A 243      -1.622   5.067   8.486  1.00  0.00           C  \nATOM   1723  OD1 ASN A 243      -2.688   4.592   8.851  1.00  0.00           O  \nATOM   1724  ND2 ASN A 243      -1.017   4.557   7.446  1.00  0.00           N  \nATOM   1725  N   SER A 244      -1.726   5.289  12.217  1.00  0.00           N  \nATOM   1726  CA  SER A 244      -2.525   4.484  13.156  1.00  0.00           C  \nATOM   1727  C   SER A 244      -3.814   3.902  12.549  1.00  0.00           C  \nATOM   1728  O   SER A 244      -4.524   3.141  13.211  1.00  0.00           O  \nATOM   1729  CB  SER A 244      -2.857   5.336  14.385  1.00  0.00           C  \nATOM   1730  OG  SER A 244      -3.496   6.526  13.983  1.00  0.00           O  \nATOM   1731  N   ASN A 245      -4.131   4.218  11.287  1.00  0.00           N  \nATOM   1732  CA  ASN A 245      -5.365   3.846  10.601  1.00  0.00           C  \nATOM   1733  C   ASN A 245      -5.158   2.828   9.469  1.00  0.00           C  \nATOM   1734  O   ASN A 245      -6.030   2.647   8.620  1.00  0.00           O  \nATOM   1735  CB  ASN A 245      -6.084   5.129  10.147  1.00  0.00           C  \nATOM   1736  CG  ASN A 245      -6.549   5.958  11.327  1.00  0.00           C  \nATOM   1737  OD1 ASN A 245      -7.323   5.486  12.156  1.00  0.00           O  \nATOM   1738  ND2 ASN A 245      -6.089   7.179  11.464  1.00  0.00           N  \nATOM   1739  N   GLN A 246      -4.028   2.105   9.456  1.00  0.00           N  \nATOM   1740  CA  GLN A 246      -3.657   1.172   8.377  1.00  0.00           C  \nATOM   1741  C   GLN A 246      -4.719   0.094   8.087  1.00  0.00           C  \nATOM   1742  O   GLN A 246      -4.841  -0.348   6.943  1.00  0.00           O  \nATOM   1743  CB  GLN A 246      -2.330   0.473   8.707  1.00  0.00           C  \nATOM   1744  CG  GLN A 246      -1.134   1.416   8.567  1.00  0.00           C  \nATOM   1745  CD  GLN A 246       0.202   0.728   8.805  1.00  0.00           C  \nATOM   1746  OE1 GLN A 246       0.304  -0.464   9.028  1.00  0.00           O  \nATOM   1747  NE2 GLN A 246       1.292   1.453   8.745  1.00  0.00           N  \nATOM   1748  N   ALA A 247      -5.477  -0.327   9.102  1.00  0.00           N  \nATOM   1749  CA  ALA A 247      -6.554  -1.295   8.914  1.00  0.00           C  \nATOM   1750  C   ALA A 247      -7.663  -0.725   8.003  1.00  0.00           C  \nATOM   1751  O   ALA A 247      -8.007  -1.349   7.004  1.00  0.00           O  \nATOM   1752  CB  ALA A 247      -7.089  -1.714  10.287  1.00  0.00           C  \nATOM   1753  N   LEU A 248      -8.180   0.481   8.319  1.00  0.00           N  \nATOM   1754  CA  LEU A 248      -9.220   1.148   7.546  1.00  0.00           C  \nATOM   1755  C   LEU A 248      -8.714   1.615   6.175  1.00  0.00           C  \nATOM   1756  O   LEU A 248      -9.283   1.243   5.147  1.00  0.00           O  \nATOM   1757  CB  LEU A 248      -9.771   2.333   8.377  1.00  0.00           C  \nATOM   1758  CG  LEU A 248     -10.780   3.222   7.626  1.00  0.00           C  \nATOM   1759  CD1 LEU A 248     -12.049   2.463   7.265  1.00  0.00           C  \nATOM   1760  CD2 LEU A 248     -11.142   4.417   8.493  1.00  0.00           C  \nATOM   1761  N   ILE A 249      -7.633   2.391   6.167  1.00  0.00           N  \nATOM   1762  CA  ILE A 249      -7.066   2.975   4.944  1.00  0.00           C  \nATOM   1763  C   ILE A 249      -6.642   1.852   3.990  1.00  0.00           C  \nATOM   1764  O   ILE A 249      -6.987   1.863   2.809  1.00  0.00           O  \nATOM   1765  CB  ILE A 249      -5.900   3.927   5.304  1.00  0.00           C  \nATOM   1766  CG1 ILE A 249      -6.406   5.116   6.153  1.00  0.00           C  \nATOM   1767  CG2 ILE A 249      -5.208   4.445   4.031  1.00  0.00           C  \nATOM   1768  CD1 ILE A 249      -5.282   6.011   6.707  1.00  0.00           C  \nATOM   1769  N   GLY A 250      -5.942   0.839   4.504  1.00  0.00           N  \nATOM   1770  CA  GLY A 250      -5.526  -0.306   3.711  1.00  0.00           C  \nATOM   1771  C   GLY A 250      -6.698  -1.133   3.184  1.00  0.00           C  \nATOM   1772  O   GLY A 250      -6.659  -1.556   2.038  1.00  0.00           O  \nATOM   1773  N   LYS A 251      -7.773  -1.337   3.985  1.00  0.00           N  \nATOM   1774  CA  LYS A 251      -9.008  -2.003   3.529  1.00  0.00           C  \nATOM   1775  C   LYS A 251      -9.613  -1.248   2.340  1.00  0.00           C  \nATOM   1776  O   LYS A 251      -9.863  -1.859   1.311  1.00  0.00           O  \nATOM   1777  CB  LYS A 251     -10.016  -2.139   4.692  1.00  0.00           C  \nATOM   1778  CG  LYS A 251     -11.356  -2.788   4.262  1.00  0.00           C  \nATOM   1779  CD  LYS A 251     -12.435  -2.783   5.374  1.00  0.00           C  \nATOM   1780  CE  LYS A 251     -13.779  -3.256   4.786  1.00  0.00           C  \nATOM   1781  NZ  LYS A 251     -14.962  -3.031   5.651  1.00  0.00           N  \nATOM   1782  N   LYS A 252      -9.802   0.064   2.476  1.00  0.00           N  \nATOM   1783  CA  LYS A 252     -10.406   0.896   1.431  1.00  0.00           C  \nATOM   1784  C   LYS A 252      -9.549   0.982   0.166  1.00  0.00           C  \nATOM   1785  O   LYS A 252     -10.075   0.773  -0.923  1.00  0.00           O  \nATOM   1786  CB  LYS A 252     -10.716   2.295   1.996  1.00  0.00           C  \nATOM   1787  CG  LYS A 252     -11.990   2.325   2.866  1.00  0.00           C  \nATOM   1788  CD  LYS A 252     -13.234   2.053   1.999  1.00  0.00           C  \nATOM   1789  CE  LYS A 252     -14.542   2.320   2.718  1.00  0.00           C  \nATOM   1790  NZ  LYS A 252     -15.679   2.180   1.766  1.00  0.00           N  \nATOM   1791  N   GLY A 253      -8.237   1.201   0.313  1.00  0.00           N  \nATOM   1792  CA  GLY A 253      -7.331   1.257  -0.837  1.00  0.00           C  \nATOM   1793  C   GLY A 253      -7.251  -0.065  -1.601  1.00  0.00           C  \nATOM   1794  O   GLY A 253      -7.369  -0.080  -2.821  1.00  0.00           O  \nATOM   1795  N   VAL A 254      -7.137  -1.196  -0.889  1.00  0.00           N  \nATOM   1796  CA  VAL A 254      -7.169  -2.529  -1.524  1.00  0.00           C  \nATOM   1797  C   VAL A 254      -8.512  -2.760  -2.212  1.00  0.00           C  \nATOM   1798  O   VAL A 254      -8.523  -3.120  -3.383  1.00  0.00           O  \nATOM   1799  CB  VAL A 254      -6.862  -3.650  -0.517  1.00  0.00           C  \nATOM   1800  CG1 VAL A 254      -7.147  -5.052  -1.085  1.00  0.00           C  \nATOM   1801  CG2 VAL A 254      -5.386  -3.623  -0.094  1.00  0.00           C  \nATOM   1802  N   ALA A 255      -9.634  -2.522  -1.520  1.00  0.00           N  \nATOM   1803  CA  ALA A 255     -10.958  -2.753  -2.084  1.00  0.00           C  \nATOM   1804  C   ALA A 255     -11.214  -1.916  -3.348  1.00  0.00           C  \nATOM   1805  O   ALA A 255     -11.726  -2.432  -4.328  1.00  0.00           O  \nATOM   1806  CB  ALA A 255     -12.008  -2.468  -1.005  1.00  0.00           C  \nATOM   1807  N   TRP A 256     -10.777  -0.653  -3.350  1.00  0.00           N  \nATOM   1808  CA  TRP A 256     -10.883   0.232  -4.507  1.00  0.00           C  \nATOM   1809  C   TRP A 256     -10.104  -0.304  -5.705  1.00  0.00           C  \nATOM   1810  O   TRP A 256     -10.650  -0.439  -6.800  1.00  0.00           O  \nATOM   1811  CB  TRP A 256     -10.396   1.633  -4.117  1.00  0.00           C  \nATOM   1812  CG  TRP A 256     -10.760   2.673  -5.120  1.00  0.00           C  \nATOM   1813  CD1 TRP A 256      -9.963   3.170  -6.098  1.00  0.00           C  \nATOM   1814  CD2 TRP A 256     -12.041   3.348  -5.269  1.00  0.00           C  \nATOM   1815  NE1 TRP A 256     -10.665   4.107  -6.836  1.00  0.00           N  \nATOM   1816  CE2 TRP A 256     -11.954   4.246  -6.369  1.00  0.00           C  \nATOM   1817  CE3 TRP A 256     -13.272   3.278  -4.584  1.00  0.00           C  \nATOM   1818  CZ2 TRP A 256     -13.038   5.040  -6.767  1.00  0.00           C  \nATOM   1819  CZ3 TRP A 256     -14.365   4.071  -4.977  1.00  0.00           C  \nATOM   1820  CH2 TRP A 256     -14.254   4.952  -6.063  1.00  0.00           C  \nATOM   1821  N   MET A 257      -8.847  -0.677  -5.489  1.00  0.00           N  \nATOM   1822  CA  MET A 257      -8.006  -1.229  -6.551  1.00  0.00           C  \nATOM   1823  C   MET A 257      -8.544  -2.582  -7.060  1.00  0.00           C  \nATOM   1824  O   MET A 257      -8.596  -2.794  -8.263  1.00  0.00           O  \nATOM   1825  CB  MET A 257      -6.555  -1.358  -6.068  1.00  0.00           C  \nATOM   1826  CG  MET A 257      -5.892   0.003  -5.800  1.00  0.00           C  \nATOM   1827  SD  MET A 257      -5.813   1.099  -7.240  1.00  0.00           S  \nATOM   1828  CE  MET A 257      -5.051   2.557  -6.507  1.00  0.00           C  \nATOM   1829  N   LYS A 258      -9.005  -3.469  -6.172  1.00  0.00           N  \nATOM   1830  CA  LYS A 258      -9.627  -4.735  -6.583  1.00  0.00           C  \nATOM   1831  C   LYS A 258     -10.887  -4.502  -7.412  1.00  0.00           C  \nATOM   1832  O   LYS A 258     -11.027  -5.078  -8.484  1.00  0.00           O  \nATOM   1833  CB  LYS A 258      -9.941  -5.609  -5.359  1.00  0.00           C  \nATOM   1834  CG  LYS A 258      -8.709  -6.165  -4.629  1.00  0.00           C  \nATOM   1835  CD  LYS A 258      -7.808  -7.070  -5.476  1.00  0.00           C  \nATOM   1836  CE  LYS A 258      -8.534  -8.336  -5.941  1.00  0.00           C  \nATOM   1837  NZ  LYS A 258      -7.679  -9.120  -6.855  1.00  0.00           N  \nATOM   1838  N   ARG A 259     -11.786  -3.618  -6.957  1.00  0.00           N  \nATOM   1839  CA  ARG A 259     -13.039  -3.333  -7.665  1.00  0.00           C  \nATOM   1840  C   ARG A 259     -12.808  -2.783  -9.066  1.00  0.00           C  \nATOM   1841  O   ARG A 259     -13.466  -3.236  -9.996  1.00  0.00           O  \nATOM   1842  CB  ARG A 259     -13.908  -2.392  -6.817  1.00  0.00           C  \nATOM   1843  CG  ARG A 259     -15.211  -1.951  -7.514  1.00  0.00           C  \nATOM   1844  CD  ARG A 259     -16.120  -3.134  -7.870  1.00  0.00           C  \nATOM   1845  NE  ARG A 259     -17.243  -2.703  -8.715  1.00  0.00           N  \nATOM   1846  CZ  ARG A 259     -17.256  -2.621 -10.031  1.00  0.00           C  \nATOM   1847  NH1 ARG A 259     -16.219  -2.895 -10.776  1.00  0.00           N  \nATOM   1848  NH2 ARG A 259     -18.345  -2.259 -10.633  1.00  0.00           N  \nATOM   1849  N   PHE A 260     -11.894  -1.823  -9.206  1.00  0.00           N  \nATOM   1850  CA  PHE A 260     -11.764  -1.077 -10.456  1.00  0.00           C  \nATOM   1851  C   PHE A 260     -10.610  -1.536 -11.356  1.00  0.00           C  \nATOM   1852  O   PHE A 260     -10.748  -1.486 -12.574  1.00  0.00           O  \nATOM   1853  CB  PHE A 260     -11.713   0.419 -10.143  1.00  0.00           C  \nATOM   1854  CG  PHE A 260     -13.008   0.948  -9.551  1.00  0.00           C  \nATOM   1855  CD1 PHE A 260     -14.163   1.027 -10.353  1.00  0.00           C  \nATOM   1856  CD2 PHE A 260     -13.071   1.343  -8.204  1.00  0.00           C  \nATOM   1857  CE1 PHE A 260     -15.376   1.502  -9.815  1.00  0.00           C  \nATOM   1858  CE2 PHE A 260     -14.287   1.811  -7.666  1.00  0.00           C  \nATOM   1859  CZ  PHE A 260     -15.432   1.888  -8.465  1.00  0.00           C  \nATOM   1860  N   MET A 261      -9.496  -2.029 -10.811  1.00  0.00           N  \nATOM   1861  CA  MET A 261      -8.411  -2.570 -11.638  1.00  0.00           C  \nATOM   1862  C   MET A 261      -8.683  -4.004 -12.078  1.00  0.00           C  \nATOM   1863  O   MET A 261      -8.358  -4.359 -13.211  1.00  0.00           O  \nATOM   1864  CB  MET A 261      -7.053  -2.500 -10.930  1.00  0.00           C  \nATOM   1865  CG  MET A 261      -6.603  -1.071 -10.640  1.00  0.00           C  \nATOM   1866  SD  MET A 261      -4.963  -0.957  -9.867  1.00  0.00           S  \nATOM   1867  CE  MET A 261      -3.913  -1.727 -11.119  1.00  0.00           C  \nATOM   1868  N   ASP A 262      -9.291  -4.813 -11.212  1.00  0.00           N  \nATOM   1869  CA  ASP A 262      -9.583  -6.217 -11.506  1.00  0.00           C  \nATOM   1870  C   ASP A 262     -11.032  -6.443 -11.955  1.00  0.00           C  \nATOM   1871  O   ASP A 262     -11.392  -7.566 -12.295  1.00  0.00           O  \nATOM   1872  CB  ASP A 262      -9.219  -7.101 -10.296  1.00  0.00           C  \nATOM   1873  CG  ASP A 262      -7.725  -7.134  -9.967  1.00  0.00           C  \nATOM   1874  OD1 ASP A 262      -6.890  -6.720 -10.790  1.00  0.00           O  \nATOM   1875  OD2 ASP A 262      -7.389  -7.653  -8.878  1.00  0.00           O  \nATOM   1876  N   ASN A 263     -11.863  -5.392 -11.961  1.00  0.00           N  \nATOM   1877  CA  ASN A 263     -13.312  -5.487 -12.139  1.00  0.00           C  \nATOM   1878  C   ASN A 263     -13.969  -6.504 -11.180  1.00  0.00           C  \nATOM   1879  O   ASN A 263     -14.975  -7.136 -11.500  1.00  0.00           O  \nATOM   1880  CB  ASN A 263     -13.638  -5.698 -13.632  1.00  0.00           C  \nATOM   1881  CG  ASN A 263     -15.095  -5.421 -13.952  1.00  0.00           C  \nATOM   1882  OD1 ASN A 263     -15.719  -4.521 -13.407  1.00  0.00           O  \nATOM   1883  ND2 ASN A 263     -15.670  -6.152 -14.871  1.00  0.00           N  \nATOM   1884  N   ASP A 264     -13.384  -6.673  -9.988  1.00  0.00           N  \nATOM   1885  CA  ASP A 264     -13.787  -7.705  -9.033  1.00  0.00           C  \nATOM   1886  C   ASP A 264     -14.968  -7.223  -8.184  1.00  0.00           C  \nATOM   1887  O   ASP A 264     -14.814  -6.571  -7.148  1.00  0.00           O  \nATOM   1888  CB  ASP A 264     -12.572  -8.126  -8.192  1.00  0.00           C  \nATOM   1889  CG  ASP A 264     -12.855  -9.343  -7.309  1.00  0.00           C  \nATOM   1890  OD1 ASP A 264     -14.020  -9.788  -7.178  1.00  0.00           O  \nATOM   1891  OD2 ASP A 264     -11.874  -9.900  -6.771  1.00  0.00           O  \nATOM   1892  N   THR A 265     -16.187  -7.581  -8.619  1.00  0.00           N  \nATOM   1893  CA  THR A 265     -17.437  -7.173  -7.962  1.00  0.00           C  \nATOM   1894  C   THR A 265     -17.610  -7.759  -6.554  1.00  0.00           C  \nATOM   1895  O   THR A 265     -18.417  -7.240  -5.788  1.00  0.00           O  \nATOM   1896  CB  THR A 265     -18.661  -7.524  -8.813  1.00  0.00           C  \nATOM   1897  OG1 THR A 265     -18.674  -8.901  -9.091  1.00  0.00           O  \nATOM   1898  CG2 THR A 265     -18.661  -6.780 -10.144  1.00  0.00           C  \nATOM   1899  N   ARG A 266     -16.815  -8.770  -6.152  1.00  0.00           N  \nATOM   1900  CA  ARG A 266     -16.788  -9.237  -4.760  1.00  0.00           C  \nATOM   1901  C   ARG A 266     -16.365  -8.122  -3.796  1.00  0.00           C  \nATOM   1902  O   ARG A 266     -16.706  -8.165  -2.622  1.00  0.00           O  \nATOM   1903  CB  ARG A 266     -15.843 -10.434  -4.613  1.00  0.00           C  \nATOM   1904  CG  ARG A 266     -16.252 -11.638  -5.472  1.00  0.00           C  \nATOM   1905  CD  ARG A 266     -15.218 -12.765  -5.386  1.00  0.00           C  \nATOM   1906  NE  ARG A 266     -13.926 -12.357  -5.951  1.00  0.00           N  \nATOM   1907  CZ  ARG A 266     -12.820 -13.086  -6.013  1.00  0.00           C  \nATOM   1908  NH1 ARG A 266     -12.780 -14.310  -5.561  1.00  0.00           N  \nATOM   1909  NH2 ARG A 266     -11.747 -12.588  -6.515  1.00  0.00           N  \nATOM   1910  N   TYR A 267     -15.645  -7.121  -4.299  1.00  0.00           N  \nATOM   1911  CA  TYR A 267     -15.186  -5.968  -3.519  1.00  0.00           C  \nATOM   1912  C   TYR A 267     -16.120  -4.751  -3.607  1.00  0.00           C  \nATOM   1913  O   TYR A 267     -15.846  -3.770  -2.918  1.00  0.00           O  \nATOM   1914  CB  TYR A 267     -13.758  -5.602  -3.934  1.00  0.00           C  \nATOM   1915  CG  TYR A 267     -12.700  -6.583  -3.453  1.00  0.00           C  \nATOM   1916  CD1 TYR A 267     -11.974  -6.316  -2.271  1.00  0.00           C  \nATOM   1917  CD2 TYR A 267     -12.438  -7.758  -4.176  1.00  0.00           C  \nATOM   1918  CE1 TYR A 267     -10.987  -7.210  -1.829  1.00  0.00           C  \nATOM   1919  CE2 TYR A 267     -11.454  -8.659  -3.734  1.00  0.00           C  \nATOM   1920  CZ  TYR A 267     -10.719  -8.388  -2.564  1.00  0.00           C  \nATOM   1921  OH  TYR A 267      -9.761  -9.259  -2.144  1.00  0.00           O  \nATOM   1922  N   SER A 268     -17.224  -4.772  -4.374  1.00  0.00           N  \nATOM   1923  CA  SER A 268     -18.137  -3.628  -4.548  1.00  0.00           C  \nATOM   1924  C   SER A 268     -18.604  -3.037  -3.220  1.00  0.00           C  \nATOM   1925  O   SER A 268     -18.415  -1.853  -2.987  1.00  0.00           O  \nATOM   1926  CB  SER A 268     -19.371  -4.012  -5.378  1.00  0.00           C  \nATOM   1927  OG  SER A 268     -19.004  -4.288  -6.703  1.00  0.00           O  \nATOM   1928  N   THR A 269     -19.113  -3.865  -2.301  1.00  0.00           N  \nATOM   1929  CA  THR A 269     -19.537  -3.396  -0.976  1.00  0.00           C  \nATOM   1930  C   THR A 269     -18.389  -2.718  -0.231  1.00  0.00           C  \nATOM   1931  O   THR A 269     -18.524  -1.583   0.206  1.00  0.00           O  \nATOM   1932  CB  THR A 269     -20.100  -4.550  -0.136  1.00  0.00           C  \nATOM   1933  OG1 THR A 269     -21.121  -5.195  -0.854  1.00  0.00           O  \nATOM   1934  CG2 THR A 269     -20.690  -4.086   1.189  1.00  0.00           C  \nATOM   1935  N   PHE A 270     -17.226  -3.352  -0.141  1.00  0.00           N  \nATOM   1936  CA  PHE A 270     -16.075  -2.800   0.583  1.00  0.00           C  \nATOM   1937  C   PHE A 270     -15.522  -1.517  -0.055  1.00  0.00           C  \nATOM   1938  O   PHE A 270     -15.058  -0.619   0.655  1.00  0.00           O  \nATOM   1939  CB  PHE A 270     -14.961  -3.848   0.674  1.00  0.00           C  \nATOM   1940  CG  PHE A 270     -15.370  -5.110   1.403  1.00  0.00           C  \nATOM   1941  CD1 PHE A 270     -15.315  -5.167   2.808  1.00  0.00           C  \nATOM   1942  CD2 PHE A 270     -15.824  -6.227   0.683  1.00  0.00           C  \nATOM   1943  CE1 PHE A 270     -15.715  -6.330   3.490  1.00  0.00           C  \nATOM   1944  CE2 PHE A 270     -16.233  -7.388   1.368  1.00  0.00           C  \nATOM   1945  CZ  PHE A 270     -16.177  -7.436   2.766  1.00  0.00           C  \nATOM   1946  N   ALA A 271     -15.569  -1.422  -1.379  1.00  0.00           N  \nATOM   1947  CA  ALA A 271     -15.157  -0.238  -2.119  1.00  0.00           C  \nATOM   1948  C   ALA A 271     -16.143   0.912  -1.893  1.00  0.00           C  \nATOM   1949  O   ALA A 271     -15.708   1.998  -1.541  1.00  0.00           O  \nATOM   1950  CB  ALA A 271     -15.010  -0.602  -3.602  1.00  0.00           C  \nATOM   1951  N   CYS A 272     -17.452   0.675  -2.015  1.00  0.00           N  \nATOM   1952  CA  CYS A 272     -18.468   1.727  -2.103  1.00  0.00           C  \nATOM   1953  C   CYS A 272     -19.137   2.077  -0.761  1.00  0.00           C  \nATOM   1954  O   CYS A 272     -19.677   3.175  -0.632  1.00  0.00           O  \nATOM   1955  CB  CYS A 272     -19.503   1.315  -3.147  1.00  0.00           C  \nATOM   1956  SG  CYS A 272     -18.799   1.008  -4.794  1.00  0.00           S  \nATOM   1957  N   GLU A 273     -19.101   1.201   0.256  1.00  0.00           N  \nATOM   1958  CA  GLU A 273     -19.759   1.431   1.548  1.00  0.00           C  \nATOM   1959  C   GLU A 273     -19.172   2.622   2.319  1.00  0.00           C  \nATOM   1960  O   GLU A 273     -17.951   2.841   2.328  1.00  0.00           O  \nATOM   1961  CB  GLU A 273     -19.744   0.158   2.423  1.00  0.00           C  \nATOM   1962  CG  GLU A 273     -18.362  -0.189   3.024  1.00  0.00           C  \nATOM   1963  CD  GLU A 273     -18.265  -1.543   3.761  1.00  0.00           C  \nATOM   1964  OE1 GLU A 273     -17.136  -1.874   4.233  1.00  0.00           O  \nATOM   1965  OE2 GLU A 273     -19.276  -2.268   3.833  1.00  0.00           O  \nATOM   1966  N   ASN A 274     -20.050   3.349   3.040  1.00  0.00           N  \nATOM   1967  CA  ASN A 274     -19.601   4.201   4.132  1.00  0.00           C  \nATOM   1968  C   ASN A 274     -19.054   3.303   5.256  1.00  0.00           C  \nATOM   1969  O   ASN A 274     -19.773   2.409   5.707  1.00  0.00           O  \nATOM   1970  CB  ASN A 274     -20.761   5.103   4.600  1.00  0.00           C  \nATOM   1971  CG  ASN A 274     -20.335   6.115   5.650  1.00  0.00           C  \nATOM   1972  OD1 ASN A 274     -19.368   5.945   6.381  1.00  0.00           O  \nATOM   1973  ND2 ASN A 274     -21.029   7.213   5.777  1.00  0.00           N  \nATOM   1974  N   PRO A 275     -17.814   3.518   5.744  1.00  0.00           N  \nATOM   1975  CA  PRO A 275     -17.246   2.699   6.813  1.00  0.00           C  \nATOM   1976  C   PRO A 275     -18.003   2.777   8.147  1.00  0.00           C  \nATOM   1977  O   PRO A 275     -17.745   1.931   9.002  1.00  0.00           O  \nATOM   1978  CB  PRO A 275     -15.811   3.197   6.995  1.00  0.00           C  \nATOM   1979  CG  PRO A 275     -15.481   3.855   5.671  1.00  0.00           C  \nATOM   1980  CD  PRO A 275     -16.813   4.433   5.222  1.00  0.00           C  \nATOM   1981  N   ASN A 276     -18.862   3.788   8.365  1.00  0.00           N  \nATOM   1982  CA  ASN A 276     -19.571   4.077   9.620  1.00  0.00           C  \nATOM   1983  C   ASN A 276     -18.641   3.988  10.849  1.00  0.00           C  \nATOM   1984  O   ASN A 276     -18.919   3.306  11.829  1.00  0.00           O  \nATOM   1985  CB  ASN A 276     -20.833   3.201   9.713  1.00  0.00           C  \nATOM   1986  CG  ASN A 276     -21.845   3.561   8.636  1.00  0.00           C  \nATOM   1987  OD1 ASN A 276     -22.224   4.701   8.471  1.00  0.00           O  \nATOM   1988  ND2 ASN A 276     -22.315   2.597   7.885  1.00  0.00           N  \nATOM   1989  N   SER A 277     -17.480   4.640  10.759  1.00  0.00           N  \nATOM   1990  CA  SER A 277     -16.394   4.525  11.727  1.00  0.00           C  \nATOM   1991  C   SER A 277     -16.087   5.878  12.351  1.00  0.00           C  \nATOM   1992  O   SER A 277     -15.970   6.872  11.639  1.00  0.00           O  \nATOM   1993  CB  SER A 277     -15.146   3.975  11.026  1.00  0.00           C  \nATOM   1994  OG  SER A 277     -14.074   3.812  11.940  1.00  0.00           O  \nATOM   1995  N   THR A 278     -15.824   5.915  13.659  1.00  0.00           N  \nATOM   1996  CA  THR A 278     -15.349   7.118  14.377  1.00  0.00           C  \nATOM   1997  C   THR A 278     -13.992   7.623  13.879  1.00  0.00           C  \nATOM   1998  O   THR A 278     -13.578   8.727  14.227  1.00  0.00           O  \nATOM   1999  CB  THR A 278     -15.226   6.841  15.880  1.00  0.00           C  \nATOM   2000  OG1 THR A 278     -14.368   5.739  16.103  1.00  0.00           O  \nATOM   2001  CG2 THR A 278     -16.577   6.491  16.515  1.00  0.00           C  \nATOM   2002  N   ARG A 279     -13.286   6.827  13.072  1.00  0.00           N  \nATOM   2003  CA  ARG A 279     -12.025   7.214  12.421  1.00  0.00           C  \nATOM   2004  C   ARG A 279     -12.226   8.035  11.142  1.00  0.00           C  \nATOM   2005  O   ARG A 279     -11.238   8.428  10.532  1.00  0.00           O  \nATOM   2006  CB  ARG A 279     -11.198   5.957  12.114  1.00  0.00           C  \nATOM   2007  CG  ARG A 279     -10.803   5.190  13.378  1.00  0.00           C  \nATOM   2008  CD  ARG A 279      -9.883   4.022  13.008  1.00  0.00           C  \nATOM   2009  NE  ARG A 279      -9.345   3.356  14.210  1.00  0.00           N  \nATOM   2010  CZ  ARG A 279      -9.946   2.411  14.928  1.00  0.00           C  \nATOM   2011  NH1 ARG A 279     -11.113   1.933  14.586  1.00  0.00           N  \nATOM   2012  NH2 ARG A 279      -9.368   1.949  15.995  1.00  0.00           N  \nATOM   2013  N   VAL A 280     -13.483   8.239  10.717  1.00  0.00           N  \nATOM   2014  CA  VAL A 280     -13.863   8.979   9.512  1.00  0.00           C  \nATOM   2015  C   VAL A 280     -14.686  10.184   9.943  1.00  0.00           C  \nATOM   2016  O   VAL A 280     -15.698  10.005  10.613  1.00  0.00           O  \nATOM   2017  CB  VAL A 280     -14.646   8.075   8.541  1.00  0.00           C  \nATOM   2018  CG1 VAL A 280     -15.094   8.825   7.282  1.00  0.00           C  \nATOM   2019  CG2 VAL A 280     -13.782   6.888   8.093  1.00  0.00           C  \nATOM   2020  N   SER A 281     -14.241  11.386   9.585  1.00  0.00           N  \nATOM   2021  CA  SER A 281     -14.912  12.660   9.860  1.00  0.00           C  \nATOM   2022  C   SER A 281     -15.797  13.111   8.698  1.00  0.00           C  \nATOM   2023  O   SER A 281     -16.856  13.675   8.936  1.00  0.00           O  \nATOM   2024  CB  SER A 281     -13.873  13.735  10.182  1.00  0.00           C  \nATOM   2025  OG  SER A 281     -12.831  13.699   9.245  1.00  0.00           O  \nATOM   2026  N   ASP A 282     -15.434  12.801   7.454  1.00  0.00           N  \nATOM   2027  CA  ASP A 282     -16.275  13.024   6.271  1.00  0.00           C  \nATOM   2028  C   ASP A 282     -16.158  11.845   5.287  1.00  0.00           C  \nATOM   2029  O   ASP A 282     -15.083  11.282   5.049  1.00  0.00           O  \nATOM   2030  CB  ASP A 282     -15.943  14.382   5.607  1.00  0.00           C  \nATOM   2031  CG  ASP A 282     -16.968  14.853   4.551  1.00  0.00           C  \nATOM   2032  OD1 ASP A 282     -17.940  14.120   4.261  1.00  0.00           O  \nATOM   2033  OD2 ASP A 282     -16.800  15.954   3.973  1.00  0.00           O  \nATOM   2034  N   PHE A 283     -17.303  11.447   4.732  1.00  0.00           N  \nATOM   2035  CA  PHE A 283     -17.397  10.405   3.726  1.00  0.00           C  \nATOM   2036  C   PHE A 283     -18.361  10.818   2.618  1.00  0.00           C  \nATOM   2037  O   PHE A 283     -19.541  11.089   2.867  1.00  0.00           O  \nATOM   2038  CB  PHE A 283     -17.845   9.082   4.347  1.00  0.00           C  \nATOM   2039  CG  PHE A 283     -18.021   7.981   3.318  1.00  0.00           C  \nATOM   2040  CD1 PHE A 283     -19.282   7.756   2.717  1.00  0.00           C  \nATOM   2041  CD2 PHE A 283     -16.922   7.203   2.915  1.00  0.00           C  \nATOM   2042  CE1 PHE A 283     -19.433   6.769   1.729  1.00  0.00           C  \nATOM   2043  CE2 PHE A 283     -17.070   6.213   1.932  1.00  0.00           C  \nATOM   2044  CZ  PHE A 283     -18.322   5.987   1.339  1.00  0.00           C  \nATOM   2045  N   ARG A 284     -17.904  10.768   1.388  1.00  0.00           N  \nATOM   2046  CA  ARG A 284     -18.708  11.031   0.195  1.00  0.00           C  \nATOM   2047  C   ARG A 284     -18.410   9.981  -0.864  1.00  0.00           C  \nATOM   2048  O   ARG A 284     -17.288   9.493  -0.978  1.00  0.00           O  \nATOM   2049  CB  ARG A 284     -18.456  12.453  -0.337  1.00  0.00           C  \nATOM   2050  CG  ARG A 284     -18.504  13.522   0.771  1.00  0.00           C  \nATOM   2051  CD  ARG A 284     -18.322  14.927   0.214  1.00  0.00           C  \nATOM   2052  NE  ARG A 284     -18.176  15.878   1.333  1.00  0.00           N  \nATOM   2053  CZ  ARG A 284     -18.323  17.181   1.308  1.00  0.00           C  \nATOM   2054  NH1 ARG A 284     -18.626  17.846   0.216  1.00  0.00           N  \nATOM   2055  NH2 ARG A 284     -18.136  17.843   2.410  1.00  0.00           N  \nATOM   2056  N   THR A 285     -19.405   9.626  -1.660  1.00  0.00           N  \nATOM   2057  CA  THR A 285     -19.198   8.741  -2.811  1.00  0.00           C  \nATOM   2058  C   THR A 285     -20.235   9.020  -3.894  1.00  0.00           C  \nATOM   2059  O   THR A 285     -21.364   9.412  -3.596  1.00  0.00           O  \nATOM   2060  CB  THR A 285     -19.183   7.266  -2.387  1.00  0.00           C  \nATOM   2061  OG1 THR A 285     -18.699   6.462  -3.435  1.00  0.00           O  \nATOM   2062  CG2 THR A 285     -20.537   6.718  -1.953  1.00  0.00           C  \nATOM   2063  N   ALA A 286     -19.855   8.850  -5.154  1.00  0.00           N  \nATOM   2064  CA  ALA A 286     -20.754   8.970  -6.289  1.00  0.00           C  \nATOM   2065  C   ALA A 286     -20.379   7.966  -7.388  1.00  0.00           C  \nATOM   2066  O   ALA A 286     -19.200   7.668  -7.593  1.00  0.00           O  \nATOM   2067  CB  ALA A 286     -20.730  10.413  -6.808  1.00  0.00           C  \nATOM   2068  N   ASN A 287     -21.390   7.466  -8.105  1.00  0.00           N  \nATOM   2069  CA  ASN A 287     -21.232   6.530  -9.223  1.00  0.00           C  \nATOM   2070  C   ASN A 287     -20.425   5.268  -8.861  1.00  0.00           C  \nATOM   2071  O   ASN A 287     -19.720   4.712  -9.705  1.00  0.00           O  \nATOM   2072  CB  ASN A 287     -20.672   7.280 -10.448  1.00  0.00           C  \nATOM   2073  CG  ASN A 287     -21.452   8.542 -10.784  1.00  0.00           C  \nATOM   2074  OD1 ASN A 287     -22.669   8.587 -10.718  1.00  0.00           O  \nATOM   2075  ND2 ASN A 287     -20.769   9.614 -11.123  1.00  0.00           N  \nATOM   2076  N   CYS A 288     -20.481   4.830  -7.600  1.00  0.00           N  \nATOM   2077  CA  CYS A 288     -19.787   3.641  -7.125  1.00  0.00           C  \nATOM   2078  C   CYS A 288     -20.742   2.448  -7.101  1.00  0.00           C  \nATOM   2079  O   CYS A 288     -21.832   2.540  -6.542  1.00  0.00           O  \nATOM   2080  CB  CYS A 288     -19.178   3.928  -5.746  1.00  0.00           C  \nATOM   2081  SG  CYS A 288     -17.873   2.774  -5.245  1.00  0.00           S  \nATOM   2082  N   SER A 289     -20.359   1.367  -7.737  1.00  0.00           N  \nATOM   2083  CA  SER A 289     -21.146   0.129  -7.847  1.00  0.00           C  \nATOM   2084  C   SER A 289     -20.249  -1.103  -7.878  1.00  0.00           C  \nATOM   2085  O   SER A 289     -19.148  -1.036  -8.457  1.00  0.00           O  \nATOM   2086  CB  SER A 289     -22.022   0.164  -9.103  1.00  0.00           C  \nATOM   2087  OG  SER A 289     -21.191   0.250 -10.232  1.00  0.00           O  \nATOM   2088  OXT SER A 289     -20.699  -2.136  -7.347  1.00  0.00           O  \nEND                                                                             \n",
                    "binary": false
                  }
                ],
                "kwargs": {
                  "defaultRepresentation": true,
                  "ext": "pdb"
                }
              },
              {
                "target": "Stage",
                "type": "call_method",
                "methodName": "loadFile",
                "reconstruc_color_scheme": false,
                "args": [
                  {
                    "type": "blob",
                    "data": "HEADER                                            01-AUG-25                     \nTITLE     ALPHAFOLD MONOMER V2.0 PREDICTION FOR POLY(ETHYLENE TEREPHTHALATE)    \nTITLE    2 HYDROLASE (A0A0K8P6T7)                                               \nREMARK   1                                                                      \nREMARK   1 REFERENCE 1                                                          \nREMARK   1  AUTH   JOHN JUMPER, RICHARD EVANS, ALEXANDER PRITZEL, TIM GREEN,    \nREMARK   1  AUTH 2 MICHAEL FIGURNOV, OLAF RONNEBERGER, KATHRYN TUNYASUVUNAKOOL, \nREMARK   1  AUTH 3 RUSS BATES, AUGUSTIN ZIDEK, ANNA POTAPENKO, ALEX BRIDGLAND,  \nREMARK   1  AUTH 4 CLEMENS MEYER, SIMON A A KOHL, ANDREW J BALLARD,             \nREMARK   1  AUTH 5 ANDREW COWIE, BERNARDINO ROMERA-PAREDES, STANISLAV NIKOLOV,  \nREMARK   1  AUTH 6 RISHUB JAIN, JONAS ADLER, TREVOR BACK, STIG PETERSEN,        \nREMARK   1  AUTH 7 DAVID REIMAN, ELLEN CLANCY, MICHAL ZIELINSKI,                \nREMARK   1  AUTH 8 MARTIN STEINEGGER, MICHALINA PACHOLSKA, TAMAS BERGHAMMER,    \nREMARK   1  AUTH 9 DAVID SILVER, ORIOL VINYALS, ANDREW W SENIOR,                \nREMARK   1  AUTH10 KORAY KAVUKCUOGLU, PUSHMEET KOHLI, DEMIS HASSABIS            \nREMARK   1  TITL   HIGHLY ACCURATE PROTEIN STRUCTURE PREDICTION WITH ALPHAFOLD  \nREMARK   1  REF    NATURE                        V. 596   583 2021              \nREMARK   1  REFN                   ISSN 0028-0836                               \nREMARK   1  PMID   34265844                                                     \nREMARK   1  DOI    10.1038/s41586-021-03819-2                                   \nREMARK   1                                                                      \nREMARK   1 DISCLAIMERS                                                          \nREMARK   1 ALPHAFOLD DATA, COPYRIGHT (2021) DEEPMIND TECHNOLOGIES LIMITED. THE  \nREMARK   1 INFORMATION PROVIDED IS THEORETICAL MODELLING ONLY AND CAUTION SHOULD\nREMARK   1 BE EXERCISED IN ITS USE. IT IS PROVIDED \"AS-IS\" WITHOUT ANY WARRANTY \nREMARK   1 OF ANY KIND, WHETHER EXPRESSED OR IMPLIED. NO WARRANTY IS GIVEN THAT \nREMARK   1 USE OF THE INFORMATION SHALL NOT INFRINGE THE RIGHTS OF ANY THIRD    \nREMARK   1 PARTY. THE INFORMATION IS NOT INTENDED TO BE A SUBSTITUTE FOR        \nREMARK   1 PROFESSIONAL MEDICAL ADVICE, DIAGNOSIS, OR TREATMENT, AND DOES NOT   \nREMARK   1 CONSTITUTE MEDICAL OR OTHER PROFESSIONAL ADVICE. IT IS AVAILABLE FOR \nREMARK   1 ACADEMIC AND COMMERCIAL PURPOSES, UNDER CC-BY 4.0 LICENCE.           \nDBREF       A    1   290  UNP    A0A0K8P6 7 PETH_PISS1     0      29            \nSEQRES   1 A  290  MET ASN PHE PRO ARG ALA SER ARG LEU MET GLN ALA ALA          \nSEQRES   2 A  290  VAL LEU GLY GLY LEU MET ALA VAL SER ALA ALA ALA THR          \nSEQRES   3 A  290  ALA GLN THR ASN PRO TYR ALA ARG GLY PRO ASN PRO THR          \nSEQRES   4 A  290  ALA ALA SER LEU GLU ALA SER ALA GLY PRO PHE THR VAL          \nSEQRES   5 A  290  ARG SER PHE THR VAL SER ARG PRO SER GLY TYR GLY ALA          \nSEQRES   6 A  290  GLY THR VAL TYR TYR PRO THR ASN ALA GLY GLY THR VAL          \nSEQRES   7 A  290  GLY ALA ILE ALA ILE VAL PRO GLY TYR THR ALA ARG GLN          \nSEQRES   8 A  290  SER SER ILE LYS TRP TRP GLY PRO ARG LEU ALA SER HIS          \nSEQRES   9 A  290  GLY PHE VAL VAL ILE THR ILE ASP THR ASN SER THR LEU          \nSEQRES  10 A  290  ASP GLN PRO SER SER ARG SER SER GLN GLN MET ALA ALA          \nSEQRES  11 A  290  LEU ARG GLN VAL ALA SER LEU ASN GLY THR SER SER SER          \nSEQRES  12 A  290  PRO ILE TYR GLY LYS VAL ASP THR ALA ARG MET GLY VAL          \nSEQRES  13 A  290  MET GLY TRP SER MET GLY GLY GLY GLY SER LEU ILE SER          \nSEQRES  14 A  290  ALA ALA ASN ASN PRO SER LEU LYS ALA ALA ALA PRO GLN          \nSEQRES  15 A  290  ALA PRO TRP ASP SER SER THR ASN PHE SER SER VAL THR          \nSEQRES  16 A  290  VAL PRO THR LEU ILE PHE ALA CYS GLU ASN ASP SER ILE          \nSEQRES  17 A  290  ALA PRO VAL ASN SER SER ALA LEU PRO ILE TYR ASP SER          \nSEQRES  18 A  290  MET SER ARG ASN ALA LYS GLN PHE LEU GLU ILE ASN GLY          \nSEQRES  19 A  290  GLY SER HIS SER CYS ALA ASN SER GLY ASN SER ASN GLN          \nSEQRES  20 A  290  ALA LEU ILE GLY LYS LYS GLY VAL ALA TRP MET LYS ARG          \nSEQRES  21 A  290  PHE MET ASP ASN ASP THR ARG TYR SER THR PHE ALA CYS          \nSEQRES  22 A  290  GLU ASN PRO ASN SER THR ARG VAL SER ASP PHE ARG THR          \nSEQRES  23 A  290  ALA ASN CYS SER                                              \nCRYST1    1.000    1.000    1.000  90.00  90.00  90.00 P 1           1          \nATOM      1  N   MET A   1     -54.367 -86.493  -5.312  1.00 54.09           N  \nATOM      2  CA  MET A   1     -53.253 -86.007  -6.153  1.00 54.09           C  \nATOM      3  C   MET A   1     -53.349 -84.495  -6.227  1.00 54.09           C  \nATOM      4  CB  MET A   1     -53.252 -86.663  -7.540  1.00 54.09           C  \nATOM      5  O   MET A   1     -54.418 -83.976  -6.517  1.00 54.09           O  \nATOM      6  CG  MET A   1     -52.645 -88.069  -7.459  1.00 54.09           C  \nATOM      7  SD  MET A   1     -52.837 -89.038  -8.972  1.00 54.09           S  \nATOM      8  CE  MET A   1     -51.982 -90.567  -8.495  1.00 54.09           C  \nATOM      9  N   ASN A   2     -52.262 -83.844  -5.818  1.00 42.47           N  \nATOM     10  CA  ASN A   2     -52.083 -82.405  -5.613  1.00 42.47           C  \nATOM     11  C   ASN A   2     -52.277 -81.587  -6.889  1.00 42.47           C  \nATOM     12  CB  ASN A   2     -50.618 -82.211  -5.159  1.00 42.47           C  \nATOM     13  O   ASN A   2     -51.894 -82.078  -7.937  1.00 42.47           O  \nATOM     14  CG  ASN A   2     -50.348 -82.720  -3.761  1.00 42.47           C  \nATOM     15  ND2 ASN A   2     -49.159 -83.198  -3.481  1.00 42.47           N  \nATOM     16  OD1 ASN A   2     -51.215 -82.705  -2.911  1.00 42.47           O  \nATOM     17  N   PHE A   3     -52.758 -80.343  -6.758  1.00 47.62           N  \nATOM     18  CA  PHE A   3     -52.101 -79.114  -7.249  1.00 47.62           C  \nATOM     19  C   PHE A   3     -52.892 -77.878  -6.766  1.00 47.62           C  \nATOM     20  CB  PHE A   3     -52.002 -79.052  -8.793  1.00 47.62           C  \nATOM     21  O   PHE A   3     -54.009 -77.665  -7.239  1.00 47.62           O  \nATOM     22  CG  PHE A   3     -50.801 -79.763  -9.398  1.00 47.62           C  \nATOM     23  CD1 PHE A   3     -49.495 -79.309  -9.129  1.00 47.62           C  \nATOM     24  CD2 PHE A   3     -50.981 -80.892 -10.220  1.00 47.62           C  \nATOM     25  CE1 PHE A   3     -48.385 -80.014  -9.628  1.00 47.62           C  \nATOM     26  CE2 PHE A   3     -49.874 -81.620 -10.684  1.00 47.62           C  \nATOM     27  CZ  PHE A   3     -48.574 -81.182 -10.388  1.00 47.62           C  \nATOM     28  N   PRO A   4     -52.344 -77.019  -5.885  1.00 56.66           N  \nATOM     29  CA  PRO A   4     -52.840 -75.664  -5.715  1.00 56.66           C  \nATOM     30  C   PRO A   4     -51.988 -74.646  -6.483  1.00 56.66           C  \nATOM     31  CB  PRO A   4     -52.880 -75.413  -4.213  1.00 56.66           C  \nATOM     32  O   PRO A   4     -50.783 -74.799  -6.688  1.00 56.66           O  \nATOM     33  CG  PRO A   4     -51.822 -76.364  -3.647  1.00 56.66           C  \nATOM     34  CD  PRO A   4     -51.426 -77.309  -4.794  1.00 56.66           C  \nATOM     35  N   ARG A   5     -52.693 -73.607  -6.929  1.00 54.34           N  \nATOM     36  CA  ARG A   5     -52.255 -72.507  -7.783  1.00 54.34           C  \nATOM     37  C   ARG A   5     -51.277 -71.559  -7.091  1.00 54.34           C  \nATOM     38  CB  ARG A   5     -53.500 -71.718  -8.224  1.00 54.34           C  \nATOM     39  O   ARG A   5     -51.387 -71.264  -5.907  1.00 54.34           O  \nATOM     40  CG  ARG A   5     -54.322 -72.430  -9.306  1.00 54.34           C  \nATOM     41  CD  ARG A   5     -55.613 -71.643  -9.567  1.00 54.34           C  \nATOM     42  NE  ARG A   5     -56.279 -72.067 -10.813  1.00 54.34           N  \nATOM     43  NH1 ARG A   5     -58.330 -71.106 -10.408  1.00 54.34           N  \nATOM     44  NH2 ARG A   5     -57.953 -72.116 -12.359  1.00 54.34           N  \nATOM     45  CZ  ARG A   5     -57.512 -71.763 -11.184  1.00 54.34           C  \nATOM     46  N   ALA A   6     -50.388 -71.025  -7.920  1.00 47.75           N  \nATOM     47  CA  ALA A   6     -49.497 -69.911  -7.654  1.00 47.75           C  \nATOM     48  C   ALA A   6     -50.242 -68.628  -7.246  1.00 47.75           C  \nATOM     49  CB  ALA A   6     -48.720 -69.668  -8.957  1.00 47.75           C  \nATOM     50  O   ALA A   6     -51.287 -68.301  -7.805  1.00 47.75           O  \nATOM     51  N   SER A   7     -49.640 -67.841  -6.356  1.00 56.06           N  \nATOM     52  CA  SER A   7     -49.825 -66.387  -6.292  1.00 56.06           C  \nATOM     53  C   SER A   7     -48.544 -65.750  -5.760  1.00 56.06           C  \nATOM     54  CB  SER A   7     -51.048 -66.001  -5.463  1.00 56.06           C  \nATOM     55  O   SER A   7     -48.028 -66.119  -4.707  1.00 56.06           O  \nATOM     56  OG  SER A   7     -52.206 -66.298  -6.218  1.00 56.06           O  \nATOM     57  N   ARG A   8     -47.983 -64.863  -6.585  1.00 51.91           N  \nATOM     58  CA  ARG A   8     -46.686 -64.205  -6.434  1.00 51.91           C  \nATOM     59  C   ARG A   8     -46.801 -62.938  -5.578  1.00 51.91           C  \nATOM     60  CB  ARG A   8     -46.157 -63.823  -7.833  1.00 51.91           C  \nATOM     61  O   ARG A   8     -47.704 -62.142  -5.787  1.00 51.91           O  \nATOM     62  CG  ARG A   8     -45.573 -64.984  -8.652  1.00 51.91           C  \nATOM     63  CD  ARG A   8     -45.147 -64.465 -10.036  1.00 51.91           C  \nATOM     64  NE  ARG A   8     -44.343 -65.449 -10.790  1.00 51.91           N  \nATOM     65  NH1 ARG A   8     -44.139 -64.235 -12.734  1.00 51.91           N  \nATOM     66  NH2 ARG A   8     -43.102 -66.198 -12.554  1.00 51.91           N  \nATOM     67  CZ  ARG A   8     -43.869 -65.291 -12.017  1.00 51.91           C  \nATOM     68  N   LEU A   9     -45.818 -62.786  -4.689  1.00 45.91           N  \nATOM     69  CA  LEU A   9     -45.007 -61.593  -4.391  1.00 45.91           C  \nATOM     70  C   LEU A   9     -45.635 -60.197  -4.546  1.00 45.91           C  \nATOM     71  CB  LEU A   9     -43.731 -61.669  -5.257  1.00 45.91           C  \nATOM     72  O   LEU A   9     -45.846 -59.715  -5.654  1.00 45.91           O  \nATOM     73  CG  LEU A   9     -42.708 -62.707  -4.759  1.00 45.91           C  \nATOM     74  CD1 LEU A   9     -42.065 -63.445  -5.933  1.00 45.91           C  \nATOM     75  CD2 LEU A   9     -41.616 -62.023  -3.938  1.00 45.91           C  \nATOM     76  N   MET A  10     -45.670 -59.471  -3.427  1.00 46.41           N  \nATOM     77  CA  MET A  10     -45.492 -58.018  -3.404  1.00 46.41           C  \nATOM     78  C   MET A  10     -44.701 -57.643  -2.138  1.00 46.41           C  \nATOM     79  CB  MET A  10     -46.859 -57.309  -3.530  1.00 46.41           C  \nATOM     80  O   MET A  10     -45.258 -57.500  -1.056  1.00 46.41           O  \nATOM     81  CG  MET A  10     -46.833 -56.213  -4.602  1.00 46.41           C  \nATOM     82  SD  MET A  10     -48.460 -55.469  -4.897  1.00 46.41           S  \nATOM     83  CE  MET A  10     -48.012 -54.157  -6.062  1.00 46.41           C  \nATOM     84  N   GLN A  11     -43.374 -57.551  -2.258  1.00 49.91           N  \nATOM     85  CA  GLN A  11     -42.522 -56.879  -1.275  1.00 49.91           C  \nATOM     86  C   GLN A  11     -42.086 -55.552  -1.892  1.00 49.91           C  \nATOM     87  CB  GLN A  11     -41.321 -57.747  -0.859  1.00 49.91           C  \nATOM     88  O   GLN A  11     -41.332 -55.524  -2.863  1.00 49.91           O  \nATOM     89  CG  GLN A  11     -41.693 -58.780   0.219  1.00 49.91           C  \nATOM     90  CD  GLN A  11     -40.498 -59.605   0.697  1.00 49.91           C  \nATOM     91  NE2 GLN A  11     -40.646 -60.374   1.751  1.00 49.91           N  \nATOM     92  OE1 GLN A  11     -39.417 -59.592   0.135  1.00 49.91           O  \nATOM     93  N   ALA A  12     -42.614 -54.456  -1.354  1.00 51.16           N  \nATOM     94  CA  ALA A  12     -42.207 -53.107  -1.705  1.00 51.16           C  \nATOM     95  C   ALA A  12     -40.880 -52.791  -0.997  1.00 51.16           C  \nATOM     96  CB  ALA A  12     -43.339 -52.145  -1.322  1.00 51.16           C  \nATOM     97  O   ALA A  12     -40.848 -52.576   0.213  1.00 51.16           O  \nATOM     98  N   ALA A  13     -39.779 -52.787  -1.750  1.00 56.09           N  \nATOM     99  CA  ALA A  13     -38.499 -52.272  -1.287  1.00 56.09           C  \nATOM    100  C   ALA A  13     -38.484 -50.745  -1.458  1.00 56.09           C  \nATOM    101  CB  ALA A  13     -37.368 -52.974  -2.049  1.00 56.09           C  \nATOM    102  O   ALA A  13     -38.476 -50.231  -2.575  1.00 56.09           O  \nATOM    103  N   VAL A  14     -38.499 -50.021  -0.339  1.00 54.41           N  \nATOM    104  CA  VAL A  14     -38.322 -48.566  -0.292  1.00 54.41           C  \nATOM    105  C   VAL A  14     -36.832 -48.261  -0.461  1.00 54.41           C  \nATOM    106  CB  VAL A  14     -38.893 -47.994   1.027  1.00 54.41           C  \nATOM    107  O   VAL A  14     -36.058 -48.377   0.485  1.00 54.41           O  \nATOM    108  CG1 VAL A  14     -38.664 -46.482   1.160  1.00 54.41           C  \nATOM    109  CG2 VAL A  14     -40.406 -48.243   1.120  1.00 54.41           C  \nATOM    110  N   LEU A  15     -36.416 -47.877  -1.670  1.00 52.31           N  \nATOM    111  CA  LEU A  15     -35.124 -47.231  -1.911  1.00 52.31           C  \nATOM    112  C   LEU A  15     -35.295 -45.711  -1.778  1.00 52.31           C  \nATOM    113  CB  LEU A  15     -34.543 -47.641  -3.283  1.00 52.31           C  \nATOM    114  O   LEU A  15     -35.726 -45.036  -2.710  1.00 52.31           O  \nATOM    115  CG  LEU A  15     -33.607 -48.869  -3.223  1.00 52.31           C  \nATOM    116  CD1 LEU A  15     -34.199 -50.067  -3.963  1.00 52.31           C  \nATOM    117  CD2 LEU A  15     -32.248 -48.541  -3.847  1.00 52.31           C  \nATOM    118  N   GLY A  16     -34.962 -45.175  -0.603  1.00 54.00           N  \nATOM    119  CA  GLY A  16     -34.786 -43.739  -0.394  1.00 54.00           C  \nATOM    120  C   GLY A  16     -33.458 -43.284  -1.000  1.00 54.00           C  \nATOM    121  O   GLY A  16     -32.394 -43.599  -0.471  1.00 54.00           O  \nATOM    122  N   GLY A  17     -33.517 -42.581  -2.131  1.00 52.59           N  \nATOM    123  CA  GLY A  17     -32.345 -42.017  -2.795  1.00 52.59           C  \nATOM    124  C   GLY A  17     -31.701 -40.902  -1.970  1.00 52.59           C  \nATOM    125  O   GLY A  17     -32.365 -39.936  -1.593  1.00 52.59           O  \nATOM    126  N   LEU A  18     -30.393 -41.012  -1.722  1.00 55.97           N  \nATOM    127  CA  LEU A  18     -29.577 -39.896  -1.250  1.00 55.97           C  \nATOM    128  C   LEU A  18     -29.510 -38.827  -2.348  1.00 55.97           C  \nATOM    129  CB  LEU A  18     -28.153 -40.362  -0.878  1.00 55.97           C  \nATOM    130  O   LEU A  18     -28.858 -39.016  -3.374  1.00 55.97           O  \nATOM    131  CG  LEU A  18     -27.989 -40.869   0.570  1.00 55.97           C  \nATOM    132  CD1 LEU A  18     -27.929 -42.394   0.637  1.00 55.97           C  \nATOM    133  CD2 LEU A  18     -26.697 -40.313   1.174  1.00 55.97           C  \nATOM    134  N   MET A  19     -30.143 -37.678  -2.114  1.00 56.47           N  \nATOM    135  CA  MET A  19     -29.868 -36.460  -2.873  1.00 56.47           C  \nATOM    136  C   MET A  19     -28.484 -35.938  -2.479  1.00 56.47           C  \nATOM    137  CB  MET A  19     -30.971 -35.412  -2.643  1.00 56.47           C  \nATOM    138  O   MET A  19     -28.328 -35.206  -1.503  1.00 56.47           O  \nATOM    139  CG  MET A  19     -32.256 -35.763  -3.407  1.00 56.47           C  \nATOM    140  SD  MET A  19     -32.783 -34.497  -4.600  1.00 56.47           S  \nATOM    141  CE  MET A  19     -34.007 -33.598  -3.611  1.00 56.47           C  \nATOM    142  N   ALA A  20     -27.462 -36.328  -3.241  1.00 56.34           N  \nATOM    143  CA  ALA A  20     -26.179 -35.644  -3.237  1.00 56.34           C  \nATOM    144  C   ALA A  20     -26.380 -34.253  -3.858  1.00 56.34           C  \nATOM    145  CB  ALA A  20     -25.146 -36.501  -3.983  1.00 56.34           C  \nATOM    146  O   ALA A  20     -26.337 -34.089  -5.079  1.00 56.34           O  \nATOM    147  N   VAL A  21     -26.634 -33.250  -3.015  1.00 55.50           N  \nATOM    148  CA  VAL A  21     -26.586 -31.841  -3.413  1.00 55.50           C  \nATOM    149  C   VAL A  21     -25.127 -31.520  -3.720  1.00 55.50           C  \nATOM    150  CB  VAL A  21     -27.185 -30.905  -2.343  1.00 55.50           C  \nATOM    151  O   VAL A  21     -24.336 -31.174  -2.846  1.00 55.50           O  \nATOM    152  CG1 VAL A  21     -27.211 -29.451  -2.838  1.00 55.50           C  \nATOM    153  CG2 VAL A  21     -28.625 -31.306  -1.998  1.00 55.50           C  \nATOM    154  N   SER A  22     -24.749 -31.706  -4.979  1.00 52.66           N  \nATOM    155  CA  SER A  22     -23.507 -31.184  -5.520  1.00 52.66           C  \nATOM    156  C   SER A  22     -23.637 -29.665  -5.502  1.00 52.66           C  \nATOM    157  CB  SER A  22     -23.223 -31.750  -6.924  1.00 52.66           C  \nATOM    158  O   SER A  22     -24.335 -29.074  -6.321  1.00 52.66           O  \nATOM    159  OG  SER A  22     -24.395 -32.175  -7.603  1.00 52.66           O  \nATOM    160  N   ALA A  23     -23.001 -29.021  -4.523  1.00 52.00           N  \nATOM    161  CA  ALA A  23     -22.761 -27.590  -4.573  1.00 52.00           C  \nATOM    162  C   ALA A  23     -21.920 -27.335  -5.828  1.00 52.00           C  \nATOM    163  CB  ALA A  23     -22.065 -27.142  -3.281  1.00 52.00           C  \nATOM    164  O   ALA A  23     -20.712 -27.569  -5.836  1.00 52.00           O  \nATOM    165  N   ALA A  24     -22.574 -26.938  -6.920  1.00 49.31           N  \nATOM    166  CA  ALA A  24     -21.892 -26.469  -8.108  1.00 49.31           C  \nATOM    167  C   ALA A  24     -21.019 -25.291  -7.671  1.00 49.31           C  \nATOM    168  CB  ALA A  24     -22.934 -26.092  -9.170  1.00 49.31           C  \nATOM    169  O   ALA A  24     -21.529 -24.213  -7.362  1.00 49.31           O  \nATOM    170  N   ALA A  25     -19.708 -25.516  -7.577  1.00 54.16           N  \nATOM    171  CA  ALA A  25     -18.748 -24.442  -7.426  1.00 54.16           C  \nATOM    172  C   ALA A  25     -18.963 -23.525  -8.630  1.00 54.16           C  \nATOM    173  CB  ALA A  25     -17.331 -25.030  -7.363  1.00 54.16           C  \nATOM    174  O   ALA A  25     -18.673 -23.907  -9.765  1.00 54.16           O  \nATOM    175  N   THR A  26     -19.556 -22.355  -8.403  1.00 54.66           N  \nATOM    176  CA  THR A  26     -19.723 -21.360  -9.452  1.00 54.66           C  \nATOM    177  C   THR A  26     -18.325 -20.996  -9.929  1.00 54.66           C  \nATOM    178  CB  THR A  26     -20.515 -20.136  -8.965  1.00 54.66           C  \nATOM    179  O   THR A  26     -17.527 -20.407  -9.197  1.00 54.66           O  \nATOM    180  CG2 THR A  26     -22.004 -20.461  -8.836  1.00 54.66           C  \nATOM    181  OG1 THR A  26     -20.095 -19.706  -7.690  1.00 54.66           O  \nATOM    182  N   ALA A  27     -17.982 -21.444 -11.137  1.00 64.25           N  \nATOM    183  CA  ALA A  27     -16.705 -21.124 -11.744  1.00 64.25           C  \nATOM    184  C   ALA A  27     -16.566 -19.596 -11.779  1.00 64.25           C  \nATOM    185  CB  ALA A  27     -16.631 -21.765 -13.136  1.00 64.25           C  \nATOM    186  O   ALA A  27     -17.479 -18.893 -12.217  1.00 64.25           O  \nATOM    187  N   GLN A  28     -15.444 -19.081 -11.276  1.00 72.38           N  \nATOM    188  CA  GLN A  28     -15.148 -17.653 -11.309  1.00 72.38           C  \nATOM    189  C   GLN A  28     -15.065 -17.201 -12.775  1.00 72.38           C  \nATOM    190  CB  GLN A  28     -13.843 -17.407 -10.530  1.00 72.38           C  \nATOM    191  O   GLN A  28     -14.144 -17.588 -13.489  1.00 72.38           O  \nATOM    192  CG  GLN A  28     -13.313 -15.969 -10.621  1.00 72.38           C  \nATOM    193  CD  GLN A  28     -14.308 -14.935 -10.109  1.00 72.38           C  \nATOM    194  NE2 GLN A  28     -14.193 -14.491  -8.876  1.00 72.38           N  \nATOM    195  OE1 GLN A  28     -15.211 -14.512 -10.808  1.00 72.38           O  \nATOM    196  N   THR A  29     -16.030 -16.396 -13.224  1.00 91.88           N  \nATOM    197  CA  THR A  29     -16.135 -15.944 -14.621  1.00 91.88           C  \nATOM    198  C   THR A  29     -15.259 -14.730 -14.926  1.00 91.88           C  \nATOM    199  CB  THR A  29     -17.591 -15.623 -14.989  1.00 91.88           C  \nATOM    200  O   THR A  29     -14.985 -14.458 -16.093  1.00 91.88           O  \nATOM    201  CG2 THR A  29     -18.500 -16.850 -14.922  1.00 91.88           C  \nATOM    202  OG1 THR A  29     -18.116 -14.669 -14.094  1.00 91.88           O  \nATOM    203  N   ASN A  30     -14.788 -14.003 -13.906  1.00 96.19           N  \nATOM    204  CA  ASN A  30     -13.897 -12.860 -14.091  1.00 96.19           C  \nATOM    205  C   ASN A  30     -12.424 -13.319 -14.240  1.00 96.19           C  \nATOM    206  CB  ASN A  30     -14.100 -11.868 -12.932  1.00 96.19           C  \nATOM    207  O   ASN A  30     -11.829 -13.762 -13.251  1.00 96.19           O  \nATOM    208  CG  ASN A  30     -13.174 -10.665 -13.009  1.00 96.19           C  \nATOM    209  ND2 ASN A  30     -13.271  -9.770 -12.064  1.00 96.19           N  \nATOM    210  OD1 ASN A  30     -12.339 -10.521 -13.889  1.00 96.19           O  \nATOM    211  N   PRO A  31     -11.780 -13.137 -15.414  1.00 97.19           N  \nATOM    212  CA  PRO A  31     -10.406 -13.585 -15.661  1.00 97.19           C  \nATOM    213  C   PRO A  31      -9.343 -12.809 -14.865  1.00 97.19           C  \nATOM    214  CB  PRO A  31     -10.198 -13.403 -17.168  1.00 97.19           C  \nATOM    215  O   PRO A  31      -8.188 -13.239 -14.799  1.00 97.19           O  \nATOM    216  CG  PRO A  31     -11.097 -12.216 -17.501  1.00 97.19           C  \nATOM    217  CD  PRO A  31     -12.302 -12.460 -16.598  1.00 97.19           C  \nATOM    218  N   TYR A  32      -9.700 -11.681 -14.247  1.00 98.25           N  \nATOM    219  CA  TYR A  32      -8.804 -10.854 -13.436  1.00 98.25           C  \nATOM    220  C   TYR A  32      -8.977 -11.079 -11.934  1.00 98.25           C  \nATOM    221  CB  TYR A  32      -9.028  -9.382 -13.771  1.00 98.25           C  \nATOM    222  O   TYR A  32      -8.149 -10.608 -11.158  1.00 98.25           O  \nATOM    223  CG  TYR A  32      -9.064  -9.054 -15.247  1.00 98.25           C  \nATOM    224  CD1 TYR A  32      -8.058  -9.537 -16.107  1.00 98.25           C  \nATOM    225  CD2 TYR A  32     -10.123  -8.281 -15.759  1.00 98.25           C  \nATOM    226  CE1 TYR A  32      -8.106  -9.245 -17.481  1.00 98.25           C  \nATOM    227  CE2 TYR A  32     -10.175  -7.990 -17.135  1.00 98.25           C  \nATOM    228  OH  TYR A  32      -9.217  -8.175 -19.318  1.00 98.25           O  \nATOM    229  CZ  TYR A  32      -9.168  -8.472 -17.996  1.00 98.25           C  \nATOM    230  N   ALA A  33     -10.001 -11.824 -11.511  1.00 98.06           N  \nATOM    231  CA  ALA A  33     -10.198 -12.141 -10.108  1.00 98.06           C  \nATOM    232  C   ALA A  33      -9.021 -12.976  -9.578  1.00 98.06           C  \nATOM    233  CB  ALA A  33     -11.540 -12.850  -9.938  1.00 98.06           C  \nATOM    234  O   ALA A  33      -8.628 -13.996 -10.153  1.00 98.06           O  \nATOM    235  N   ARG A  34      -8.428 -12.525  -8.471  1.00 98.31           N  \nATOM    236  CA  ARG A  34      -7.314 -13.198  -7.783  1.00 98.31           C  \nATOM    237  C   ARG A  34      -7.619 -13.288  -6.294  1.00 98.31           C  \nATOM    238  CB  ARG A  34      -5.983 -12.453  -8.011  1.00 98.31           C  \nATOM    239  O   ARG A  34      -8.132 -12.326  -5.719  1.00 98.31           O  \nATOM    240  CG  ARG A  34      -5.564 -12.273  -9.475  1.00 98.31           C  \nATOM    241  CD  ARG A  34      -5.115 -13.559 -10.170  1.00 98.31           C  \nATOM    242  NE  ARG A  34      -4.818 -13.294 -11.587  1.00 98.31           N  \nATOM    243  NH1 ARG A  34      -6.907 -13.755 -12.433  1.00 98.31           N  \nATOM    244  NH2 ARG A  34      -5.306 -13.057 -13.790  1.00 98.31           N  \nATOM    245  CZ  ARG A  34      -5.673 -13.379 -12.585  1.00 98.31           C  \nATOM    246  N   GLY A  35      -7.250 -14.411  -5.681  1.00 97.75           N  \nATOM    247  CA  GLY A  35      -7.364 -14.642  -4.240  1.00 97.75           C  \nATOM    248  C   GLY A  35      -8.765 -15.072  -3.773  1.00 97.75           C  \nATOM    249  O   GLY A  35      -9.710 -15.088  -4.574  1.00 97.75           O  \nATOM    250  N   PRO A  36      -8.902 -15.431  -2.482  1.00 97.38           N  \nATOM    251  CA  PRO A  36     -10.162 -15.853  -1.874  1.00 97.38           C  \nATOM    252  C   PRO A  36     -11.196 -14.720  -1.856  1.00 97.38           C  \nATOM    253  CB  PRO A  36      -9.792 -16.302  -0.454  1.00 97.38           C  \nATOM    254  O   PRO A  36     -10.878 -13.557  -2.115  1.00 97.38           O  \nATOM    255  CG  PRO A  36      -8.593 -15.419  -0.114  1.00 97.38           C  \nATOM    256  CD  PRO A  36      -7.872 -15.326  -1.456  1.00 97.38           C  \nATOM    257  N   ASN A  37     -12.448 -15.063  -1.546  1.00 98.06           N  \nATOM    258  CA  ASN A  37     -13.487 -14.057  -1.342  1.00 98.06           C  \nATOM    259  C   ASN A  37     -13.126 -13.172  -0.134  1.00 98.06           C  \nATOM    260  CB  ASN A  37     -14.861 -14.731  -1.186  1.00 98.06           C  \nATOM    261  O   ASN A  37     -12.775 -13.710   0.920  1.00 98.06           O  \nATOM    262  CG  ASN A  37     -15.370 -15.340  -2.481  1.00 98.06           C  \nATOM    263  ND2 ASN A  37     -16.495 -16.012  -2.438  1.00 98.06           N  \nATOM    264  OD1 ASN A  37     -14.771 -15.214  -3.543  1.00 98.06           O  \nATOM    265  N   PRO A  38     -13.178 -11.836  -0.277  1.00 98.44           N  \nATOM    266  CA  PRO A  38     -12.757 -10.925   0.775  1.00 98.44           C  \nATOM    267  C   PRO A  38     -13.755 -10.889   1.935  1.00 98.44           C  \nATOM    268  CB  PRO A  38     -12.628  -9.558   0.099  1.00 98.44           C  \nATOM    269  O   PRO A  38     -14.969 -10.925   1.749  1.00 98.44           O  \nATOM    270  CG  PRO A  38     -13.660  -9.628  -1.027  1.00 98.44           C  \nATOM    271  CD  PRO A  38     -13.604 -11.091  -1.455  1.00 98.44           C  \nATOM    272  N   THR A  39     -13.220 -10.736   3.141  1.00 98.56           N  \nATOM    273  CA  THR A  39     -13.956 -10.364   4.357  1.00 98.56           C  \nATOM    274  C   THR A  39     -13.382  -9.072   4.933  1.00 98.56           C  \nATOM    275  CB  THR A  39     -13.903 -11.475   5.420  1.00 98.56           C  \nATOM    276  O   THR A  39     -12.215  -8.759   4.681  1.00 98.56           O  \nATOM    277  CG2 THR A  39     -14.317 -12.842   4.882  1.00 98.56           C  \nATOM    278  OG1 THR A  39     -12.604 -11.607   5.955  1.00 98.56           O  \nATOM    279  N   ALA A  40     -14.149  -8.348   5.758  1.00 98.25           N  \nATOM    280  CA  ALA A  40     -13.646  -7.153   6.445  1.00 98.25           C  \nATOM    281  C   ALA A  40     -12.334  -7.444   7.201  1.00 98.25           C  \nATOM    282  CB  ALA A  40     -14.732  -6.619   7.385  1.00 98.25           C  \nATOM    283  O   ALA A  40     -11.342  -6.758   6.979  1.00 98.25           O  \nATOM    284  N   ALA A  41     -12.291  -8.539   7.969  1.00 98.50           N  \nATOM    285  CA  ALA A  41     -11.099  -8.971   8.700  1.00 98.50           C  \nATOM    286  C   ALA A  41      -9.896  -9.236   7.776  1.00 98.50           C  \nATOM    287  CB  ALA A  41     -11.464 -10.220   9.512  1.00 98.50           C  \nATOM    288  O   ALA A  41      -8.788  -8.789   8.063  1.00 98.50           O  \nATOM    289  N   SER A  42     -10.104  -9.910   6.636  1.00 98.50           N  \nATOM    290  CA  SER A  42      -9.020 -10.172   5.676  1.00 98.50           C  \nATOM    291  C   SER A  42      -8.482  -8.901   5.015  1.00 98.50           C  \nATOM    292  CB  SER A  42      -9.477 -11.151   4.591  1.00 98.50           C  \nATOM    293  O   SER A  42      -7.311  -8.847   4.656  1.00 98.50           O  \nATOM    294  OG  SER A  42     -10.369 -10.532   3.681  1.00 98.50           O  \nATOM    295  N   LEU A  43      -9.331  -7.881   4.842  1.00 98.69           N  \nATOM    296  CA  LEU A  43      -8.941  -6.611   4.237  1.00 98.69           C  \nATOM    297  C   LEU A  43      -8.364  -5.632   5.250  1.00 98.69           C  \nATOM    298  CB  LEU A  43     -10.136  -5.970   3.519  1.00 98.69           C  \nATOM    299  O   LEU A  43      -7.638  -4.736   4.841  1.00 98.69           O  \nATOM    300  CG  LEU A  43     -10.722  -6.764   2.344  1.00 98.69           C  \nATOM    301  CD1 LEU A  43     -11.847  -5.949   1.706  1.00 98.69           C  \nATOM    302  CD2 LEU A  43      -9.682  -7.080   1.274  1.00 98.69           C  \nATOM    303  N   GLU A  44      -8.678  -5.776   6.534  1.00 98.62           N  \nATOM    304  CA  GLU A  44      -8.134  -4.969   7.631  1.00 98.62           C  \nATOM    305  C   GLU A  44      -6.805  -5.516   8.164  1.00 98.62           C  \nATOM    306  CB  GLU A  44      -9.189  -4.868   8.742  1.00 98.62           C  \nATOM    307  O   GLU A  44      -6.038  -4.765   8.769  1.00 98.62           O  \nATOM    308  CG  GLU A  44     -10.288  -3.885   8.311  1.00 98.62           C  \nATOM    309  CD  GLU A  44     -11.539  -3.889   9.193  1.00 98.62           C  \nATOM    310  OE1 GLU A  44     -12.517  -3.222   8.757  1.00 98.62           O  \nATOM    311  OE2 GLU A  44     -11.523  -4.519  10.270  1.00 98.62           O  \nATOM    312  N   ALA A  45      -6.498  -6.786   7.886  1.00 98.50           N  \nATOM    313  CA  ALA A  45      -5.216  -7.399   8.198  1.00 98.50           C  \nATOM    314  C   ALA A  45      -4.033  -6.663   7.540  1.00 98.50           C  \nATOM    315  CB  ALA A  45      -5.259  -8.876   7.782  1.00 98.50           C  \nATOM    316  O   ALA A  45      -4.134  -6.076   6.458  1.00 98.50           O  \nATOM    317  N   SER A  46      -2.867  -6.734   8.188  1.00 98.38           N  \nATOM    318  CA  SER A  46      -1.632  -6.103   7.699  1.00 98.38           C  \nATOM    319  C   SER A  46      -1.121  -6.700   6.383  1.00 98.38           C  \nATOM    320  CB  SER A  46      -0.530  -6.227   8.755  1.00 98.38           C  \nATOM    321  O   SER A  46      -0.426  -6.012   5.634  1.00 98.38           O  \nATOM    322  OG  SER A  46      -0.956  -5.649   9.972  1.00 98.38           O  \nATOM    323  N   ALA A  47      -1.451  -7.963   6.102  1.00 98.62           N  \nATOM    324  CA  ALA A  47      -1.058  -8.678   4.896  1.00 98.62           C  \nATOM    325  C   ALA A  47      -2.241  -9.468   4.325  1.00 98.62           C  \nATOM    326  CB  ALA A  47       0.126  -9.600   5.214  1.00 98.62           C  \nATOM    327  O   ALA A  47      -3.014 -10.071   5.068  1.00 98.62           O  \nATOM    328  N   GLY A  48      -2.341  -9.465   3.000  1.00 98.62           N  \nATOM    329  CA  GLY A  48      -3.277 -10.279   2.241  1.00 98.62           C  \nATOM    330  C   GLY A  48      -2.803 -11.724   2.024  1.00 98.62           C  \nATOM    331  O   GLY A  48      -1.812 -12.165   2.608  1.00 98.62           O  \nATOM    332  N   PRO A  49      -3.502 -12.475   1.158  1.00 98.62           N  \nATOM    333  CA  PRO A  49      -3.286 -13.910   0.958  1.00 98.62           C  \nATOM    334  C   PRO A  49      -1.983 -14.280   0.227  1.00 98.62           C  \nATOM    335  CB  PRO A  49      -4.506 -14.361   0.143  1.00 98.62           C  \nATOM    336  O   PRO A  49      -1.582 -15.442   0.266  1.00 98.62           O  \nATOM    337  CG  PRO A  49      -4.851 -13.108  -0.664  1.00 98.62           C  \nATOM    338  CD  PRO A  49      -4.621 -12.006   0.355  1.00 98.62           C  \nATOM    339  N   PHE A  50      -1.338 -13.346  -0.480  1.00 98.75           N  \nATOM    340  CA  PHE A  50      -0.168 -13.646  -1.311  1.00 98.75           C  \nATOM    341  C   PHE A  50       1.149 -13.298  -0.623  1.00 98.75           C  \nATOM    342  CB  PHE A  50      -0.274 -12.952  -2.670  1.00 98.75           C  \nATOM    343  O   PHE A  50       1.292 -12.253   0.013  1.00 98.75           O  \nATOM    344  CG  PHE A  50      -1.539 -13.279  -3.437  1.00 98.75           C  \nATOM    345  CD1 PHE A  50      -1.867 -14.611  -3.745  1.00 98.75           C  \nATOM    346  CD2 PHE A  50      -2.397 -12.243  -3.833  1.00 98.75           C  \nATOM    347  CE1 PHE A  50      -3.037 -14.897  -4.466  1.00 98.75           C  \nATOM    348  CE2 PHE A  50      -3.559 -12.520  -4.573  1.00 98.75           C  \nATOM    349  CZ  PHE A  50      -3.864 -13.852  -4.901  1.00 98.75           C  \nATOM    350  N   THR A  51       2.157 -14.146  -0.825  1.00 98.75           N  \nATOM    351  CA  THR A  51       3.523 -13.887  -0.364  1.00 98.75           C  \nATOM    352  C   THR A  51       4.156 -12.797  -1.224  1.00 98.75           C  \nATOM    353  CB  THR A  51       4.369 -15.163  -0.422  1.00 98.75           C  \nATOM    354  O   THR A  51       4.135 -12.877  -2.452  1.00 98.75           O  \nATOM    355  CG2 THR A  51       5.789 -14.963   0.112  1.00 98.75           C  \nATOM    356  OG1 THR A  51       3.771 -16.188   0.336  1.00 98.75           O  \nATOM    357  N   VAL A  52       4.751 -11.785  -0.588  1.00 98.88           N  \nATOM    358  CA  VAL A  52       5.367 -10.644  -1.278  1.00 98.88           C  \nATOM    359  C   VAL A  52       6.885 -10.765  -1.273  1.00 98.88           C  \nATOM    360  CB  VAL A  52       4.912  -9.304  -0.670  1.00 98.88           C  \nATOM    361  O   VAL A  52       7.495 -11.044  -0.243  1.00 98.88           O  \nATOM    362  CG1 VAL A  52       5.536  -8.100  -1.392  1.00 98.88           C  \nATOM    363  CG2 VAL A  52       3.393  -9.164  -0.770  1.00 98.88           C  \nATOM    364  N   ARG A  53       7.510 -10.482  -2.418  1.00 98.88           N  \nATOM    365  CA  ARG A  53       8.943 -10.165  -2.508  1.00 98.88           C  \nATOM    366  C   ARG A  53       9.132  -8.772  -3.094  1.00 98.88           C  \nATOM    367  CB  ARG A  53       9.689 -11.245  -3.305  1.00 98.88           C  \nATOM    368  O   ARG A  53       8.184  -8.158  -3.582  1.00 98.88           O  \nATOM    369  CG  ARG A  53       9.726 -12.583  -2.556  1.00 98.88           C  \nATOM    370  CD  ARG A  53      10.703 -13.567  -3.216  1.00 98.88           C  \nATOM    371  NE  ARG A  53      12.080 -13.375  -2.714  1.00 98.88           N  \nATOM    372  NH1 ARG A  53      13.261 -14.050  -4.572  1.00 98.88           N  \nATOM    373  NH2 ARG A  53      14.353 -13.502  -2.710  1.00 98.88           N  \nATOM    374  CZ  ARG A  53      13.218 -13.640  -3.335  1.00 98.88           C  \nATOM    375  N   SER A  54      10.359  -8.270  -3.022  1.00 98.88           N  \nATOM    376  CA  SER A  54      10.714  -6.987  -3.621  1.00 98.88           C  \nATOM    377  C   SER A  54      12.106  -6.994  -4.240  1.00 98.88           C  \nATOM    378  CB  SER A  54      10.565  -5.842  -2.611  1.00 98.88           C  \nATOM    379  O   SER A  54      12.923  -7.863  -3.929  1.00 98.88           O  \nATOM    380  OG  SER A  54      11.473  -5.991  -1.535  1.00 98.88           O  \nATOM    381  N   PHE A  55      12.352  -6.032  -5.124  1.00 98.81           N  \nATOM    382  CA  PHE A  55      13.675  -5.703  -5.643  1.00 98.81           C  \nATOM    383  C   PHE A  55      13.759  -4.205  -5.962  1.00 98.81           C  \nATOM    384  CB  PHE A  55      14.001  -6.557  -6.880  1.00 98.81           C  \nATOM    385  O   PHE A  55      12.751  -3.564  -6.266  1.00 98.81           O  \nATOM    386  CG  PHE A  55      13.133  -6.282  -8.094  1.00 98.81           C  \nATOM    387  CD1 PHE A  55      11.880  -6.904  -8.222  1.00 98.81           C  \nATOM    388  CD2 PHE A  55      13.572  -5.395  -9.096  1.00 98.81           C  \nATOM    389  CE1 PHE A  55      11.073  -6.648  -9.341  1.00 98.81           C  \nATOM    390  CE2 PHE A  55      12.761  -5.133 -10.215  1.00 98.81           C  \nATOM    391  CZ  PHE A  55      11.513  -5.764 -10.340  1.00 98.81           C  \nATOM    392  N   THR A  56      14.969  -3.653  -5.912  1.00 98.69           N  \nATOM    393  CA  THR A  56      15.246  -2.273  -6.335  1.00 98.69           C  \nATOM    394  C   THR A  56      15.477  -2.228  -7.841  1.00 98.69           C  \nATOM    395  CB  THR A  56      16.459  -1.700  -5.589  1.00 98.69           C  \nATOM    396  O   THR A  56      16.181  -3.074  -8.395  1.00 98.69           O  \nATOM    397  CG2 THR A  56      16.705  -0.222  -5.887  1.00 98.69           C  \nATOM    398  OG1 THR A  56      16.237  -1.805  -4.203  1.00 98.69           O  \nATOM    399  N   VAL A  57      14.915  -1.228  -8.515  1.00 98.31           N  \nATOM    400  CA  VAL A  57      15.179  -0.974  -9.935  1.00 98.31           C  \nATOM    401  C   VAL A  57      16.597  -0.424 -10.085  1.00 98.31           C  \nATOM    402  CB  VAL A  57      14.140  -0.012 -10.535  1.00 98.31           C  \nATOM    403  O   VAL A  57      16.896   0.657  -9.590  1.00 98.31           O  \nATOM    404  CG1 VAL A  57      14.421   0.222 -12.021  1.00 98.31           C  \nATOM    405  CG2 VAL A  57      12.726  -0.593 -10.396  1.00 98.31           C  \nATOM    406  N   SER A  58      17.466  -1.162 -10.777  1.00 95.81           N  \nATOM    407  CA  SER A  58      18.889  -0.818 -10.915  1.00 95.81           C  \nATOM    408  C   SER A  58      19.134   0.461 -11.719  1.00 95.81           C  \nATOM    409  CB  SER A  58      19.634  -1.983 -11.571  1.00 95.81           C  \nATOM    410  O   SER A  58      20.011   1.247 -11.373  1.00 95.81           O  \nATOM    411  OG  SER A  58      19.046  -2.312 -12.822  1.00 95.81           O  \nATOM    412  N   ARG A  59      18.371   0.671 -12.798  1.00 97.12           N  \nATOM    413  CA  ARG A  59      18.450   1.851 -13.673  1.00 97.12           C  \nATOM    414  C   ARG A  59      17.045   2.400 -13.929  1.00 97.12           C  \nATOM    415  CB  ARG A  59      19.181   1.502 -14.981  1.00 97.12           C  \nATOM    416  O   ARG A  59      16.421   2.005 -14.915  1.00 97.12           O  \nATOM    417  CG  ARG A  59      20.663   1.173 -14.743  1.00 97.12           C  \nATOM    418  CD  ARG A  59      21.388   0.785 -16.037  1.00 97.12           C  \nATOM    419  NE  ARG A  59      21.527   1.922 -16.968  1.00 97.12           N  \nATOM    420  NH1 ARG A  59      23.110   0.983 -18.346  1.00 97.12           N  \nATOM    421  NH2 ARG A  59      22.391   3.056 -18.747  1.00 97.12           N  \nATOM    422  CZ  ARG A  59      22.337   1.981 -18.012  1.00 97.12           C  \nATOM    423  N   PRO A  60      16.516   3.241 -13.028  1.00 98.12           N  \nATOM    424  CA  PRO A  60      15.253   3.940 -13.233  1.00 98.12           C  \nATOM    425  C   PRO A  60      15.194   4.661 -14.586  1.00 98.12           C  \nATOM    426  CB  PRO A  60      15.183   4.933 -12.078  1.00 98.12           C  \nATOM    427  O   PRO A  60      16.138   5.339 -14.987  1.00 98.12           O  \nATOM    428  CG  PRO A  60      15.926   4.224 -10.956  1.00 98.12           C  \nATOM    429  CD  PRO A  60      17.069   3.570 -11.720  1.00 98.12           C  \nATOM    430  N   SER A  61      14.085   4.505 -15.302  1.00 97.69           N  \nATOM    431  CA  SER A  61      13.845   5.109 -16.612  1.00 97.69           C  \nATOM    432  C   SER A  61      12.929   6.316 -16.445  1.00 97.69           C  \nATOM    433  CB  SER A  61      13.266   4.070 -17.574  1.00 97.69           C  \nATOM    434  O   SER A  61      11.707   6.187 -16.461  1.00 97.69           O  \nATOM    435  OG  SER A  61      13.192   4.613 -18.880  1.00 97.69           O  \nATOM    436  N   GLY A  62      13.527   7.493 -16.243  1.00 98.31           N  \nATOM    437  CA  GLY A  62      12.806   8.765 -16.140  1.00 98.31           C  \nATOM    438  C   GLY A  62      12.271   9.117 -14.748  1.00 98.31           C  \nATOM    439  O   GLY A  62      11.369   9.939 -14.666  1.00 98.31           O  \nATOM    440  N   TYR A  63      12.798   8.522 -13.676  1.00 98.81           N  \nATOM    441  CA  TYR A  63      12.512   8.857 -12.271  1.00 98.81           C  \nATOM    442  C   TYR A  63      13.747   8.548 -11.399  1.00 98.81           C  \nATOM    443  CB  TYR A  63      11.259   8.108 -11.798  1.00 98.81           C  \nATOM    444  O   TYR A  63      14.697   7.946 -11.894  1.00 98.81           O  \nATOM    445  CG  TYR A  63      11.360   6.599 -11.849  1.00 98.81           C  \nATOM    446  CD1 TYR A  63      11.063   5.911 -13.040  1.00 98.81           C  \nATOM    447  CD2 TYR A  63      11.746   5.885 -10.701  1.00 98.81           C  \nATOM    448  CE1 TYR A  63      11.156   4.510 -13.085  1.00 98.81           C  \nATOM    449  CE2 TYR A  63      11.832   4.483 -10.741  1.00 98.81           C  \nATOM    450  OH  TYR A  63      11.606   2.442 -11.944  1.00 98.81           O  \nATOM    451  CZ  TYR A  63      11.538   3.794 -11.933  1.00 98.81           C  \nATOM    452  N   GLY A  64      13.763   8.956 -10.124  1.00 98.56           N  \nATOM    453  CA  GLY A  64      14.982   8.961  -9.301  1.00 98.56           C  \nATOM    454  C   GLY A  64      15.417   7.584  -8.786  1.00 98.56           C  \nATOM    455  O   GLY A  64      16.559   7.173  -8.973  1.00 98.56           O  \nATOM    456  N   ALA A  65      14.510   6.854  -8.137  1.00 98.81           N  \nATOM    457  CA  ALA A  65      14.737   5.487  -7.652  1.00 98.81           C  \nATOM    458  C   ALA A  65      13.409   4.735  -7.562  1.00 98.81           C  \nATOM    459  CB  ALA A  65      15.448   5.513  -6.293  1.00 98.81           C  \nATOM    460  O   ALA A  65      12.352   5.359  -7.541  1.00 98.81           O  \nATOM    461  N   GLY A  66      13.426   3.403  -7.487  1.00 98.81           N  \nATOM    462  CA  GLY A  66      12.185   2.670  -7.245  1.00 98.81           C  \nATOM    463  C   GLY A  66      12.367   1.282  -6.665  1.00 98.81           C  \nATOM    464  O   GLY A  66      13.329   0.583  -6.981  1.00 98.81           O  \nATOM    465  N   THR A  67      11.399   0.885  -5.845  1.00 98.94           N  \nATOM    466  CA  THR A  67      11.277  -0.457  -5.274  1.00 98.94           C  \nATOM    467  C   THR A  67      10.019  -1.105  -5.827  1.00 98.94           C  \nATOM    468  CB  THR A  67      11.217  -0.428  -3.741  1.00 98.94           C  \nATOM    469  O   THR A  67       8.927  -0.552  -5.704  1.00 98.94           O  \nATOM    470  CG2 THR A  67      11.334  -1.832  -3.147  1.00 98.94           C  \nATOM    471  OG1 THR A  67      12.273   0.348  -3.227  1.00 98.94           O  \nATOM    472  N   VAL A  68      10.170  -2.277  -6.440  1.00 98.94           N  \nATOM    473  CA  VAL A  68       9.054  -3.066  -6.967  1.00 98.94           C  \nATOM    474  C   VAL A  68       8.705  -4.151  -5.960  1.00 98.94           C  \nATOM    475  CB  VAL A  68       9.395  -3.668  -8.340  1.00 98.94           C  \nATOM    476  O   VAL A  68       9.549  -4.980  -5.631  1.00 98.94           O  \nATOM    477  CG1 VAL A  68       8.243  -4.526  -8.881  1.00 98.94           C  \nATOM    478  CG2 VAL A  68       9.676  -2.572  -9.376  1.00 98.94           C  \nATOM    479  N   TYR A  69       7.457  -4.170  -5.507  1.00 98.94           N  \nATOM    480  CA  TYR A  69       6.846  -5.211  -4.686  1.00 98.94           C  \nATOM    481  C   TYR A  69       5.932  -6.064  -5.561  1.00 98.94           C  \nATOM    482  CB  TYR A  69       6.043  -4.584  -3.540  1.00 98.94           C  \nATOM    483  O   TYR A  69       5.200  -5.531  -6.396  1.00 98.94           O  \nATOM    484  CG  TYR A  69       6.852  -3.686  -2.631  1.00 98.94           C  \nATOM    485  CD1 TYR A  69       7.407  -4.206  -1.448  1.00 98.94           C  \nATOM    486  CD2 TYR A  69       7.065  -2.336  -2.979  1.00 98.94           C  \nATOM    487  CE1 TYR A  69       8.188  -3.382  -0.615  1.00 98.94           C  \nATOM    488  CE2 TYR A  69       7.829  -1.506  -2.139  1.00 98.94           C  \nATOM    489  OH  TYR A  69       9.149  -1.234  -0.156  1.00 98.94           O  \nATOM    490  CZ  TYR A  69       8.395  -2.028  -0.960  1.00 98.94           C  \nATOM    491  N   TYR A  70       5.955  -7.383  -5.388  1.00 98.94           N  \nATOM    492  CA  TYR A  70       5.150  -8.278  -6.215  1.00 98.94           C  \nATOM    493  C   TYR A  70       4.736  -9.554  -5.468  1.00 98.94           C  \nATOM    494  CB  TYR A  70       5.919  -8.612  -7.503  1.00 98.94           C  \nATOM    495  O   TYR A  70       5.509 -10.056  -4.643  1.00 98.94           O  \nATOM    496  CG  TYR A  70       7.181  -9.435  -7.308  1.00 98.94           C  \nATOM    497  CD1 TYR A  70       8.383  -8.823  -6.896  1.00 98.94           C  \nATOM    498  CD2 TYR A  70       7.138 -10.826  -7.524  1.00 98.94           C  \nATOM    499  CE1 TYR A  70       9.532  -9.609  -6.672  1.00 98.94           C  \nATOM    500  CE2 TYR A  70       8.289 -11.608  -7.330  1.00 98.94           C  \nATOM    501  OH  TYR A  70      10.577 -11.772  -6.660  1.00 98.94           O  \nATOM    502  CZ  TYR A  70       9.483 -10.999  -6.894  1.00 98.94           C  \nATOM    503  N   PRO A  71       3.544 -10.109  -5.764  1.00 98.81           N  \nATOM    504  CA  PRO A  71       3.127 -11.413  -5.268  1.00 98.81           C  \nATOM    505  C   PRO A  71       3.904 -12.528  -5.977  1.00 98.81           C  \nATOM    506  CB  PRO A  71       1.628 -11.503  -5.557  1.00 98.81           C  \nATOM    507  O   PRO A  71       4.043 -12.530  -7.203  1.00 98.81           O  \nATOM    508  CG  PRO A  71       1.477 -10.650  -6.818  1.00 98.81           C  \nATOM    509  CD  PRO A  71       2.480  -9.531  -6.576  1.00 98.81           C  \nATOM    510  N   THR A  72       4.382 -13.518  -5.228  1.00 98.69           N  \nATOM    511  CA  THR A  72       5.052 -14.695  -5.806  1.00 98.69           C  \nATOM    512  C   THR A  72       4.068 -15.770  -6.266  1.00 98.69           C  \nATOM    513  CB  THR A  72       6.060 -15.314  -4.834  1.00 98.69           C  \nATOM    514  O   THR A  72       4.420 -16.576  -7.121  1.00 98.69           O  \nATOM    515  CG2 THR A  72       7.148 -14.324  -4.421  1.00 98.69           C  \nATOM    516  OG1 THR A  72       5.406 -15.736  -3.668  1.00 98.69           O  \nATOM    517  N   ASN A  73       2.844 -15.771  -5.731  1.00 98.31           N  \nATOM    518  CA  ASN A  73       1.864 -16.852  -5.870  1.00 98.31           C  \nATOM    519  C   ASN A  73       0.435 -16.356  -6.183  1.00 98.31           C  \nATOM    520  CB  ASN A  73       1.941 -17.748  -4.615  1.00 98.31           C  \nATOM    521  O   ASN A  73      -0.538 -16.924  -5.699  1.00 98.31           O  \nATOM    522  CG  ASN A  73       1.581 -17.049  -3.310  1.00 98.31           C  \nATOM    523  ND2 ASN A  73       1.340 -17.804  -2.265  1.00 98.31           N  \nATOM    524  OD1 ASN A  73       1.548 -15.831  -3.195  1.00 98.31           O  \nATOM    525  N   ALA A  74       0.291 -15.315  -7.013  1.00 97.94           N  \nATOM    526  CA  ALA A  74      -1.023 -14.754  -7.371  1.00 97.94           C  \nATOM    527  C   ALA A  74      -1.927 -15.693  -8.201  1.00 97.94           C  \nATOM    528  CB  ALA A  74      -0.816 -13.419  -8.094  1.00 97.94           C  \nATOM    529  O   ALA A  74      -3.130 -15.460  -8.286  1.00 97.94           O  \nATOM    530  N   GLY A  75      -1.360 -16.726  -8.837  1.00 96.56           N  \nATOM    531  CA  GLY A  75      -2.086 -17.677  -9.695  1.00 96.56           C  \nATOM    532  C   GLY A  75      -2.336 -17.204 -11.135  1.00 96.56           C  \nATOM    533  O   GLY A  75      -2.851 -17.963 -11.944  1.00 96.56           O  \nATOM    534  N   GLY A  76      -1.950 -15.975 -11.481  1.00 97.56           N  \nATOM    535  CA  GLY A  76      -2.044 -15.411 -12.829  1.00 97.56           C  \nATOM    536  C   GLY A  76      -1.477 -13.993 -12.870  1.00 97.56           C  \nATOM    537  O   GLY A  76      -1.037 -13.474 -11.841  1.00 97.56           O  \nATOM    538  N   THR A  77      -1.512 -13.344 -14.036  1.00 98.75           N  \nATOM    539  CA  THR A  77      -1.065 -11.947 -14.173  1.00 98.75           C  \nATOM    540  C   THR A  77      -1.923 -10.997 -13.335  1.00 98.75           C  \nATOM    541  CB  THR A  77      -1.058 -11.475 -15.633  1.00 98.75           C  \nATOM    542  O   THR A  77      -3.125 -11.224 -13.144  1.00 98.75           O  \nATOM    543  CG2 THR A  77      -0.015 -12.200 -16.476  1.00 98.75           C  \nATOM    544  OG1 THR A  77      -2.322 -11.702 -16.210  1.00 98.75           O  \nATOM    545  N   VAL A  78      -1.305  -9.931 -12.828  1.00 98.88           N  \nATOM    546  CA  VAL A  78      -1.925  -8.912 -11.965  1.00 98.88           C  \nATOM    547  C   VAL A  78      -1.708  -7.503 -12.515  1.00 98.88           C  \nATOM    548  CB  VAL A  78      -1.414  -9.001 -10.518  1.00 98.88           C  \nATOM    549  O   VAL A  78      -0.809  -7.289 -13.323  1.00 98.88           O  \nATOM    550  CG1 VAL A  78      -1.828 -10.323  -9.866  1.00 98.88           C  \nATOM    551  CG2 VAL A  78       0.098  -8.827 -10.376  1.00 98.88           C  \nATOM    552  N   GLY A  79      -2.516  -6.537 -12.078  1.00 98.88           N  \nATOM    553  CA  GLY A  79      -2.317  -5.139 -12.457  1.00 98.88           C  \nATOM    554  C   GLY A  79      -1.092  -4.509 -11.783  1.00 98.88           C  \nATOM    555  O   GLY A  79      -0.566  -5.047 -10.803  1.00 98.88           O  \nATOM    556  N   ALA A  80      -0.654  -3.358 -12.297  1.00 98.94           N  \nATOM    557  CA  ALA A  80       0.482  -2.600 -11.767  1.00 98.94           C  \nATOM    558  C   ALA A  80       0.074  -1.236 -11.189  1.00 98.94           C  \nATOM    559  CB  ALA A  80       1.565  -2.460 -12.839  1.00 98.94           C  \nATOM    560  O   ALA A  80      -0.697  -0.509 -11.809  1.00 98.94           O  \nATOM    561  N   ILE A  81       0.619  -0.856 -10.032  1.00 98.94           N  \nATOM    562  CA  ILE A  81       0.329   0.418  -9.356  1.00 98.94           C  \nATOM    563  C   ILE A  81       1.629   1.190  -9.128  1.00 98.94           C  \nATOM    564  CB  ILE A  81      -0.438   0.191  -8.031  1.00 98.94           C  \nATOM    565  O   ILE A  81       2.568   0.640  -8.554  1.00 98.94           O  \nATOM    566  CG1 ILE A  81      -1.756  -0.579  -8.276  1.00 98.94           C  \nATOM    567  CG2 ILE A  81      -0.737   1.541  -7.346  1.00 98.94           C  \nATOM    568  CD1 ILE A  81      -2.445  -1.062  -6.995  1.00 98.94           C  \nATOM    569  N   ALA A  82       1.682   2.462  -9.526  1.00 98.94           N  \nATOM    570  CA  ALA A  82       2.812   3.348  -9.228  1.00 98.94           C  \nATOM    571  C   ALA A  82       2.461   4.379  -8.150  1.00 98.94           C  \nATOM    572  CB  ALA A  82       3.321   4.000 -10.513  1.00 98.94           C  \nATOM    573  O   ALA A  82       1.444   5.070  -8.250  1.00 98.94           O  \nATOM    574  N   ILE A  83       3.324   4.495  -7.135  1.00 98.94           N  \nATOM    575  CA  ILE A  83       3.093   5.308  -5.937  1.00 98.94           C  \nATOM    576  C   ILE A  83       4.187   6.363  -5.768  1.00 98.94           C  \nATOM    577  CB  ILE A  83       2.924   4.445  -4.668  1.00 98.94           C  \nATOM    578  O   ILE A  83       5.368   6.022  -5.668  1.00 98.94           O  \nATOM    579  CG1 ILE A  83       2.016   3.222  -4.943  1.00 98.94           C  \nATOM    580  CG2 ILE A  83       2.383   5.338  -3.531  1.00 98.94           C  \nATOM    581  CD1 ILE A  83       1.594   2.431  -3.705  1.00 98.94           C  \nATOM    582  N   VAL A  84       3.787   7.639  -5.702  1.00 98.94           N  \nATOM    583  CA  VAL A  84       4.711   8.786  -5.642  1.00 98.94           C  \nATOM    584  C   VAL A  84       4.631   9.521  -4.294  1.00 98.94           C  \nATOM    585  CB  VAL A  84       4.468   9.775  -6.799  1.00 98.94           C  \nATOM    586  O   VAL A  84       3.527   9.818  -3.828  1.00 98.94           O  \nATOM    587  CG1 VAL A  84       5.607  10.797  -6.916  1.00 98.94           C  \nATOM    588  CG2 VAL A  84       4.377   9.053  -8.144  1.00 98.94           C  \nATOM    589  N   PRO A  85       5.766   9.850  -3.652  1.00 98.88           N  \nATOM    590  CA  PRO A  85       5.795  10.636  -2.421  1.00 98.88           C  \nATOM    591  C   PRO A  85       5.446  12.117  -2.658  1.00 98.88           C  \nATOM    592  CB  PRO A  85       7.214  10.437  -1.873  1.00 98.88           C  \nATOM    593  O   PRO A  85       5.306  12.575  -3.795  1.00 98.88           O  \nATOM    594  CG  PRO A  85       8.059  10.243  -3.127  1.00 98.88           C  \nATOM    595  CD  PRO A  85       7.115   9.441  -4.014  1.00 98.88           C  \nATOM    596  N   GLY A  86       5.331  12.874  -1.564  1.00 98.69           N  \nATOM    597  CA  GLY A  86       5.102  14.321  -1.569  1.00 98.69           C  \nATOM    598  C   GLY A  86       6.368  15.162  -1.755  1.00 98.69           C  \nATOM    599  O   GLY A  86       7.469  14.636  -1.950  1.00 98.69           O  \nATOM    600  N   TYR A  87       6.210  16.485  -1.679  1.00 98.44           N  \nATOM    601  CA  TYR A  87       7.320  17.445  -1.681  1.00 98.44           C  \nATOM    602  C   TYR A  87       8.289  17.158  -0.528  1.00 98.44           C  \nATOM    603  CB  TYR A  87       6.758  18.866  -1.585  1.00 98.44           C  \nATOM    604  O   TYR A  87       7.854  16.819   0.571  1.00 98.44           O  \nATOM    605  CG  TYR A  87       7.753  19.959  -1.926  1.00 98.44           C  \nATOM    606  CD1 TYR A  87       8.248  20.802  -0.913  1.00 98.44           C  \nATOM    607  CD2 TYR A  87       8.150  20.168  -3.262  1.00 98.44           C  \nATOM    608  CE1 TYR A  87       9.171  21.822  -1.219  1.00 98.44           C  \nATOM    609  CE2 TYR A  87       9.034  21.219  -3.576  1.00 98.44           C  \nATOM    610  OH  TYR A  87      10.424  23.033  -2.878  1.00 98.44           O  \nATOM    611  CZ  TYR A  87       9.562  22.033  -2.558  1.00 98.44           C  \nATOM    612  N   THR A  88       9.598  17.255  -0.773  1.00 98.31           N  \nATOM    613  CA  THR A  88      10.715  16.880   0.129  1.00 98.31           C  \nATOM    614  C   THR A  88      10.819  15.396   0.516  1.00 98.31           C  \nATOM    615  CB  THR A  88      10.781  17.745   1.397  1.00 98.31           C  \nATOM    616  O   THR A  88      11.851  14.962   1.033  1.00 98.31           O  \nATOM    617  CG2 THR A  88      10.654  19.236   1.106  1.00 98.31           C  \nATOM    618  OG1 THR A  88       9.769  17.389   2.307  1.00 98.31           O  \nATOM    619  N   ALA A  89       9.784  14.600   0.250  1.00 98.38           N  \nATOM    620  CA  ALA A  89       9.665  13.234   0.731  1.00 98.38           C  \nATOM    621  C   ALA A  89      10.303  12.199  -0.208  1.00 98.38           C  \nATOM    622  CB  ALA A  89       8.185  12.969   1.018  1.00 98.38           C  \nATOM    623  O   ALA A  89      10.721  12.491  -1.328  1.00 98.38           O  \nATOM    624  N   ARG A  90      10.374  10.962   0.290  1.00 98.75           N  \nATOM    625  CA  ARG A  90      10.948   9.785  -0.373  1.00 98.75           C  \nATOM    626  C   ARG A  90      10.013   8.583  -0.249  1.00 98.75           C  \nATOM    627  CB  ARG A  90      12.330   9.484   0.237  1.00 98.75           C  \nATOM    628  O   ARG A  90       9.025   8.648   0.480  1.00 98.75           O  \nATOM    629  CG  ARG A  90      13.386  10.497  -0.222  1.00 98.75           C  \nATOM    630  CD  ARG A  90      14.779  10.032   0.216  1.00 98.75           C  \nATOM    631  NE  ARG A  90      15.830  10.977  -0.205  1.00 98.75           N  \nATOM    632  NH1 ARG A  90      15.599  12.516   1.489  1.00 98.75           N  \nATOM    633  NH2 ARG A  90      17.165  12.815  -0.070  1.00 98.75           N  \nATOM    634  CZ  ARG A  90      16.191  12.093   0.406  1.00 98.75           C  \nATOM    635  N   GLN A  91      10.361   7.459  -0.872  1.00 98.75           N  \nATOM    636  CA  GLN A  91       9.633   6.185  -0.799  1.00 98.75           C  \nATOM    637  C   GLN A  91       9.227   5.801   0.632  1.00 98.75           C  \nATOM    638  CB  GLN A  91      10.509   5.053  -1.362  1.00 98.75           C  \nATOM    639  O   GLN A  91       8.151   5.245   0.828  1.00 98.75           O  \nATOM    640  CG  GLN A  91      10.682   5.135  -2.880  1.00 98.75           C  \nATOM    641  CD  GLN A  91      11.617   4.052  -3.400  1.00 98.75           C  \nATOM    642  NE2 GLN A  91      12.872   4.346  -3.651  1.00 98.75           N  \nATOM    643  OE1 GLN A  91      11.236   2.899  -3.544  1.00 98.75           O  \nATOM    644  N   SER A  92      10.036   6.123   1.648  1.00 98.50           N  \nATOM    645  CA  SER A  92       9.735   5.825   3.056  1.00 98.50           C  \nATOM    646  C   SER A  92       8.356   6.317   3.509  1.00 98.50           C  \nATOM    647  CB  SER A  92      10.815   6.415   3.971  1.00 98.50           C  \nATOM    648  O   SER A  92       7.693   5.603   4.265  1.00 98.50           O  \nATOM    649  OG  SER A  92      10.938   7.808   3.764  1.00 98.50           O  \nATOM    650  N   SER A  93       7.874   7.460   3.009  1.00 98.62           N  \nATOM    651  CA  SER A  93       6.565   8.007   3.386  1.00 98.62           C  \nATOM    652  C   SER A  93       5.393   7.181   2.851  1.00 98.62           C  \nATOM    653  CB  SER A  93       6.438   9.466   2.928  1.00 98.62           C  \nATOM    654  O   SER A  93       4.327   7.187   3.453  1.00 98.62           O  \nATOM    655  OG  SER A  93       6.338   9.564   1.522  1.00 98.62           O  \nATOM    656  N   ILE A  94       5.583   6.442   1.754  1.00 98.81           N  \nATOM    657  CA  ILE A  94       4.525   5.685   1.067  1.00 98.81           C  \nATOM    658  C   ILE A  94       4.796   4.171   0.992  1.00 98.81           C  \nATOM    659  CB  ILE A  94       4.185   6.337  -0.299  1.00 98.81           C  \nATOM    660  O   ILE A  94       3.939   3.419   0.534  1.00 98.81           O  \nATOM    661  CG1 ILE A  94       5.374   6.614  -1.248  1.00 98.81           C  \nATOM    662  CG2 ILE A  94       3.462   7.677  -0.073  1.00 98.81           C  \nATOM    663  CD1 ILE A  94       5.881   5.379  -1.981  1.00 98.81           C  \nATOM    664  N   LYS A  95       5.949   3.697   1.492  1.00 98.62           N  \nATOM    665  CA  LYS A  95       6.409   2.302   1.362  1.00 98.62           C  \nATOM    666  C   LYS A  95       5.441   1.254   1.905  1.00 98.62           C  \nATOM    667  CB  LYS A  95       7.812   2.104   1.974  1.00 98.62           C  \nATOM    668  O   LYS A  95       5.442   0.132   1.409  1.00 98.62           O  \nATOM    669  CG  LYS A  95       7.841   2.078   3.514  1.00 98.62           C  \nATOM    670  CD  LYS A  95       9.223   1.716   4.069  1.00 98.62           C  \nATOM    671  CE  LYS A  95       9.100   1.410   5.569  1.00 98.62           C  \nATOM    672  NZ  LYS A  95      10.404   1.032   6.172  1.00 98.62           N  \nATOM    673  N   TRP A  96       4.639   1.581   2.922  1.00 98.75           N  \nATOM    674  CA  TRP A  96       3.731   0.611   3.546  1.00 98.75           C  \nATOM    675  C   TRP A  96       2.674   0.079   2.565  1.00 98.75           C  \nATOM    676  CB  TRP A  96       3.083   1.235   4.785  1.00 98.75           C  \nATOM    677  O   TRP A  96       2.257  -1.075   2.672  1.00 98.75           O  \nATOM    678  CG  TRP A  96       1.906   2.120   4.514  1.00 98.75           C  \nATOM    679  CD1 TRP A  96       1.935   3.396   4.064  1.00 98.75           C  \nATOM    680  CD2 TRP A  96       0.494   1.769   4.620  1.00 98.75           C  \nATOM    681  CE2 TRP A  96      -0.280   2.909   4.261  1.00 98.75           C  \nATOM    682  CE3 TRP A  96      -0.210   0.598   4.977  1.00 98.75           C  \nATOM    683  NE1 TRP A  96       0.645   3.875   3.952  1.00 98.75           N  \nATOM    684  CH2 TRP A  96      -2.352   1.723   4.617  1.00 98.75           C  \nATOM    685  CZ2 TRP A  96      -1.678   2.903   4.271  1.00 98.75           C  \nATOM    686  CZ3 TRP A  96      -1.619   0.577   4.969  1.00 98.75           C  \nATOM    687  N   TRP A  97       2.317   0.877   1.553  1.00 98.88           N  \nATOM    688  CA  TRP A  97       1.430   0.454   0.476  1.00 98.88           C  \nATOM    689  C   TRP A  97       2.016  -0.655  -0.397  1.00 98.88           C  \nATOM    690  CB  TRP A  97       1.085   1.658  -0.396  1.00 98.88           C  \nATOM    691  O   TRP A  97       1.251  -1.466  -0.911  1.00 98.88           O  \nATOM    692  CG  TRP A  97       0.086   2.585   0.202  1.00 98.88           C  \nATOM    693  CD1 TRP A  97       0.269   3.893   0.483  1.00 98.88           C  \nATOM    694  CD2 TRP A  97      -1.303   2.288   0.525  1.00 98.88           C  \nATOM    695  CE2 TRP A  97      -1.923   3.491   0.966  1.00 98.88           C  \nATOM    696  CE3 TRP A  97      -2.110   1.132   0.430  1.00 98.88           C  \nATOM    697  NE1 TRP A  97      -0.919   4.431   0.940  1.00 98.88           N  \nATOM    698  CH2 TRP A  97      -4.067   2.394   1.149  1.00 98.88           C  \nATOM    699  CZ2 TRP A  97      -3.288   3.556   1.262  1.00 98.88           C  \nATOM    700  CZ3 TRP A  97      -3.481   1.181   0.743  1.00 98.88           C  \nATOM    701  N   GLY A  98       3.345  -0.726  -0.538  1.00 98.88           N  \nATOM    702  CA  GLY A  98       4.028  -1.728  -1.359  1.00 98.88           C  \nATOM    703  C   GLY A  98       3.642  -3.157  -0.964  1.00 98.88           C  \nATOM    704  O   GLY A  98       2.952  -3.831  -1.730  1.00 98.88           O  \nATOM    705  N   PRO A  99       4.003  -3.621   0.248  1.00 98.81           N  \nATOM    706  CA  PRO A  99       3.597  -4.937   0.736  1.00 98.81           C  \nATOM    707  C   PRO A  99       2.081  -5.081   0.893  1.00 98.81           C  \nATOM    708  CB  PRO A  99       4.318  -5.124   2.076  1.00 98.81           C  \nATOM    709  O   PRO A  99       1.533  -6.147   0.594  1.00 98.81           O  \nATOM    710  CG  PRO A  99       5.505  -4.165   1.988  1.00 98.81           C  \nATOM    711  CD  PRO A  99       4.942  -3.006   1.172  1.00 98.81           C  \nATOM    712  N   ARG A 100       1.385  -4.017   1.325  1.00 98.88           N  \nATOM    713  CA  ARG A 100      -0.062  -4.076   1.556  1.00 98.88           C  \nATOM    714  C   ARG A 100      -0.832  -4.381   0.274  1.00 98.88           C  \nATOM    715  CB  ARG A 100      -0.558  -2.786   2.228  1.00 98.88           C  \nATOM    716  O   ARG A 100      -1.628  -5.304   0.266  1.00 98.88           O  \nATOM    717  CG  ARG A 100      -2.075  -2.785   2.491  1.00 98.88           C  \nATOM    718  CD  ARG A 100      -2.545  -4.029   3.270  1.00 98.88           C  \nATOM    719  NE  ARG A 100      -3.939  -3.910   3.732  1.00 98.88           N  \nATOM    720  NH1 ARG A 100      -3.526  -2.471   5.471  1.00 98.88           N  \nATOM    721  NH2 ARG A 100      -5.584  -3.086   5.090  1.00 98.88           N  \nATOM    722  CZ  ARG A 100      -4.342  -3.182   4.755  1.00 98.88           C  \nATOM    723  N   LEU A 101      -0.587  -3.666  -0.819  1.00 98.94           N  \nATOM    724  CA  LEU A 101      -1.273  -3.920  -2.090  1.00 98.94           C  \nATOM    725  C   LEU A 101      -0.740  -5.186  -2.769  1.00 98.94           C  \nATOM    726  CB  LEU A 101      -1.129  -2.684  -2.986  1.00 98.94           C  \nATOM    727  O   LEU A 101      -1.531  -5.994  -3.254  1.00 98.94           O  \nATOM    728  CG  LEU A 101      -1.985  -1.492  -2.539  1.00 98.94           C  \nATOM    729  CD1 LEU A 101      -1.432  -0.212  -3.152  1.00 98.94           C  \nATOM    730  CD2 LEU A 101      -3.446  -1.649  -2.959  1.00 98.94           C  \nATOM    731  N   ALA A 102       0.578  -5.418  -2.751  1.00 98.94           N  \nATOM    732  CA  ALA A 102       1.156  -6.590  -3.407  1.00 98.94           C  \nATOM    733  C   ALA A 102       0.647  -7.913  -2.815  1.00 98.94           C  \nATOM    734  CB  ALA A 102       2.685  -6.497  -3.388  1.00 98.94           C  \nATOM    735  O   ALA A 102       0.367  -8.848  -3.568  1.00 98.94           O  \nATOM    736  N   SER A 103       0.432  -7.973  -1.496  1.00 98.88           N  \nATOM    737  CA  SER A 103      -0.130  -9.155  -0.823  1.00 98.88           C  \nATOM    738  C   SER A 103      -1.596  -9.437  -1.181  1.00 98.88           C  \nATOM    739  CB  SER A 103       0.049  -9.070   0.695  1.00 98.88           C  \nATOM    740  O   SER A 103      -2.072 -10.542  -0.939  1.00 98.88           O  \nATOM    741  OG  SER A 103      -0.599  -7.931   1.224  1.00 98.88           O  \nATOM    742  N   HIS A 104      -2.301  -8.498  -1.826  1.00 98.88           N  \nATOM    743  CA  HIS A 104      -3.639  -8.702  -2.401  1.00 98.88           C  \nATOM    744  C   HIS A 104      -3.634  -8.889  -3.926  1.00 98.88           C  \nATOM    745  CB  HIS A 104      -4.582  -7.583  -1.948  1.00 98.88           C  \nATOM    746  O   HIS A 104      -4.692  -9.060  -4.533  1.00 98.88           O  \nATOM    747  CG  HIS A 104      -4.993  -7.739  -0.507  1.00 98.88           C  \nATOM    748  CD2 HIS A 104      -4.329  -7.253   0.583  1.00 98.88           C  \nATOM    749  ND1 HIS A 104      -6.113  -8.392  -0.038  1.00 98.88           N  \nATOM    750  CE1 HIS A 104      -6.121  -8.288   1.305  1.00 98.88           C  \nATOM    751  NE2 HIS A 104      -5.060  -7.586   1.721  1.00 98.88           N  \nATOM    752  N   GLY A 105      -2.461  -8.961  -4.555  1.00 98.81           N  \nATOM    753  CA  GLY A 105      -2.335  -9.367  -5.951  1.00 98.81           C  \nATOM    754  C   GLY A 105      -2.195  -8.183  -6.894  1.00 98.81           C  \nATOM    755  O   GLY A 105      -2.973  -8.063  -7.831  1.00 98.81           O  \nATOM    756  N   PHE A 106      -1.190  -7.345  -6.650  1.00 98.94           N  \nATOM    757  CA  PHE A 106      -0.759  -6.268  -7.543  1.00 98.94           C  \nATOM    758  C   PHE A 106       0.767  -6.244  -7.628  1.00 98.94           C  \nATOM    759  CB  PHE A 106      -1.290  -4.917  -7.038  1.00 98.94           C  \nATOM    760  O   PHE A 106       1.450  -6.550  -6.652  1.00 98.94           O  \nATOM    761  CG  PHE A 106      -2.801  -4.813  -7.060  1.00 98.94           C  \nATOM    762  CD1 PHE A 106      -3.471  -4.700  -8.293  1.00 98.94           C  \nATOM    763  CD2 PHE A 106      -3.544  -4.872  -5.865  1.00 98.94           C  \nATOM    764  CE1 PHE A 106      -4.874  -4.676  -8.330  1.00 98.94           C  \nATOM    765  CE2 PHE A 106      -4.948  -4.833  -5.902  1.00 98.94           C  \nATOM    766  CZ  PHE A 106      -5.612  -4.746  -7.138  1.00 98.94           C  \nATOM    767  N   VAL A 107       1.325  -5.841  -8.765  1.00 98.94           N  \nATOM    768  CA  VAL A 107       2.713  -5.363  -8.795  1.00 98.94           C  \nATOM    769  C   VAL A 107       2.699  -3.890  -8.408  1.00 98.94           C  \nATOM    770  CB  VAL A 107       3.385  -5.634 -10.151  1.00 98.94           C  \nATOM    771  O   VAL A 107       1.952  -3.105  -8.978  1.00 98.94           O  \nATOM    772  CG1 VAL A 107       4.768  -4.979 -10.239  1.00 98.94           C  \nATOM    773  CG2 VAL A 107       3.584  -7.148 -10.316  1.00 98.94           C  \nATOM    774  N   VAL A 108       3.498  -3.495  -7.426  1.00 98.94           N  \nATOM    775  CA  VAL A 108       3.488  -2.130  -6.889  1.00 98.94           C  \nATOM    776  C   VAL A 108       4.884  -1.554  -6.993  1.00 98.94           C  \nATOM    777  CB  VAL A 108       2.999  -2.090  -5.434  1.00 98.94           C  \nATOM    778  O   VAL A 108       5.826  -2.153  -6.485  1.00 98.94           O  \nATOM    779  CG1 VAL A 108       2.834  -0.642  -4.951  1.00 98.94           C  \nATOM    780  CG2 VAL A 108       1.666  -2.815  -5.242  1.00 98.94           C  \nATOM    781  N   ILE A 109       5.027  -0.390  -7.616  1.00 98.94           N  \nATOM    782  CA  ILE A 109       6.281   0.354  -7.638  1.00 98.94           C  \nATOM    783  C   ILE A 109       6.148   1.622  -6.799  1.00 98.94           C  \nATOM    784  CB  ILE A 109       6.792   0.569  -9.077  1.00 98.94           C  \nATOM    785  O   ILE A 109       5.334   2.496  -7.085  1.00 98.94           O  \nATOM    786  CG1 ILE A 109       8.199   1.193  -9.013  1.00 98.94           C  \nATOM    787  CG2 ILE A 109       5.842   1.395  -9.963  1.00 98.94           C  \nATOM    788  CD1 ILE A 109       8.954   1.161 -10.343  1.00 98.94           C  \nATOM    789  N   THR A 110       6.955   1.727  -5.748  1.00 98.94           N  \nATOM    790  CA  THR A 110       7.151   2.979  -5.011  1.00 98.94           C  \nATOM    791  C   THR A 110       8.352   3.690  -5.602  1.00 98.94           C  \nATOM    792  CB  THR A 110       7.369   2.729  -3.513  1.00 98.94           C  \nATOM    793  O   THR A 110       9.376   3.037  -5.800  1.00 98.94           O  \nATOM    794  CG2 THR A 110       6.143   2.058  -2.889  1.00 98.94           C  \nATOM    795  OG1 THR A 110       8.467   1.888  -3.254  1.00 98.94           O  \nATOM    796  N   ILE A 111       8.258   4.991  -5.875  1.00 98.94           N  \nATOM    797  CA  ILE A 111       9.378   5.737  -6.469  1.00 98.94           C  \nATOM    798  C   ILE A 111       9.907   6.843  -5.562  1.00 98.94           C  \nATOM    799  CB  ILE A 111       9.066   6.251  -7.889  1.00 98.94           C  \nATOM    800  O   ILE A 111       9.165   7.415  -4.771  1.00 98.94           O  \nATOM    801  CG1 ILE A 111       8.002   7.370  -7.918  1.00 98.94           C  \nATOM    802  CG2 ILE A 111       8.682   5.071  -8.796  1.00 98.94           C  \nATOM    803  CD1 ILE A 111       7.875   8.038  -9.295  1.00 98.94           C  \nATOM    804  N   ASP A 112      11.190   7.155  -5.701  1.00 98.88           N  \nATOM    805  CA  ASP A 112      11.713   8.486  -5.412  1.00 98.88           C  \nATOM    806  C   ASP A 112      11.696   9.290  -6.709  1.00 98.88           C  \nATOM    807  CB  ASP A 112      13.131   8.437  -4.831  1.00 98.88           C  \nATOM    808  O   ASP A 112      11.977   8.760  -7.790  1.00 98.88           O  \nATOM    809  CG  ASP A 112      13.155   7.817  -3.438  1.00 98.88           C  \nATOM    810  OD1 ASP A 112      12.357   8.260  -2.587  1.00 98.88           O  \nATOM    811  OD2 ASP A 112      13.930   6.867  -3.198  1.00 98.88           O  \nATOM    812  N   THR A 113      11.359  10.568  -6.602  1.00 98.81           N  \nATOM    813  CA  THR A 113      11.412  11.490  -7.734  1.00 98.81           C  \nATOM    814  C   THR A 113      12.857  11.903  -8.030  1.00 98.81           C  \nATOM    815  CB  THR A 113      10.493  12.695  -7.488  1.00 98.81           C  \nATOM    816  O   THR A 113      13.725  11.778  -7.166  1.00 98.81           O  \nATOM    817  CG2 THR A 113       9.029  12.263  -7.465  1.00 98.81           C  \nATOM    818  OG1 THR A 113      10.734  13.287  -6.232  1.00 98.81           O  \nATOM    819  N   ASN A 114      13.136  12.382  -9.246  1.00 98.62           N  \nATOM    820  CA  ASN A 114      14.458  12.861  -9.670  1.00 98.62           C  \nATOM    821  C   ASN A 114      14.991  13.941  -8.721  1.00 98.62           C  \nATOM    822  CB  ASN A 114      14.354  13.441 -11.093  1.00 98.62           C  \nATOM    823  O   ASN A 114      16.157  13.921  -8.335  1.00 98.62           O  \nATOM    824  CG  ASN A 114      14.100  12.390 -12.152  1.00 98.62           C  \nATOM    825  ND2 ASN A 114      13.313  12.702 -13.154  1.00 98.62           N  \nATOM    826  OD1 ASN A 114      14.598  11.282 -12.101  1.00 98.62           O  \nATOM    827  N   SER A 115      14.111  14.852  -8.313  1.00 98.62           N  \nATOM    828  CA  SER A 115      14.333  15.793  -7.228  1.00 98.62           C  \nATOM    829  C   SER A 115      13.221  15.649  -6.203  1.00 98.62           C  \nATOM    830  CB  SER A 115      14.381  17.215  -7.780  1.00 98.62           C  \nATOM    831  O   SER A 115      12.038  15.569  -6.545  1.00 98.62           O  \nATOM    832  OG  SER A 115      14.515  18.140  -6.715  1.00 98.62           O  \nATOM    833  N   THR A 116      13.559  15.697  -4.916  1.00 98.56           N  \nATOM    834  CA  THR A 116      12.539  15.762  -3.858  1.00 98.56           C  \nATOM    835  C   THR A 116      11.685  17.033  -3.964  1.00 98.56           C  \nATOM    836  CB  THR A 116      13.170  15.661  -2.467  1.00 98.56           C  \nATOM    837  O   THR A 116      10.597  17.079  -3.390  1.00 98.56           O  \nATOM    838  CG2 THR A 116      13.927  14.352  -2.248  1.00 98.56           C  \nATOM    839  OG1 THR A 116      14.076  16.720  -2.268  1.00 98.56           O  \nATOM    840  N   LEU A 117      12.132  18.027  -4.744  1.00 98.62           N  \nATOM    841  CA  LEU A 117      11.456  19.301  -4.996  1.00 98.62           C  \nATOM    842  C   LEU A 117      10.623  19.320  -6.291  1.00 98.62           C  \nATOM    843  CB  LEU A 117      12.486  20.448  -4.989  1.00 98.62           C  \nATOM    844  O   LEU A 117      10.092  20.362  -6.666  1.00 98.62           O  \nATOM    845  CG  LEU A 117      13.502  20.440  -3.833  1.00 98.62           C  \nATOM    846  CD1 LEU A 117      14.407  21.665  -3.950  1.00 98.62           C  \nATOM    847  CD2 LEU A 117      12.852  20.447  -2.447  1.00 98.62           C  \nATOM    848  N   ASP A 118      10.502  18.189  -6.993  1.00 98.75           N  \nATOM    849  CA  ASP A 118       9.678  18.106  -8.203  1.00 98.75           C  \nATOM    850  C   ASP A 118       8.212  18.491  -7.924  1.00 98.75           C  \nATOM    851  CB  ASP A 118       9.756  16.690  -8.797  1.00 98.75           C  \nATOM    852  O   ASP A 118       7.637  18.101  -6.903  1.00 98.75           O  \nATOM    853  CG  ASP A 118      11.001  16.470  -9.658  1.00 98.75           C  \nATOM    854  OD1 ASP A 118      11.267  17.346 -10.507  1.00 98.75           O  \nATOM    855  OD2 ASP A 118      11.592  15.369  -9.582  1.00 98.75           O  \nATOM    856  N   GLN A 119       7.602  19.224  -8.858  1.00 98.81           N  \nATOM    857  CA  GLN A 119       6.202  19.674  -8.814  1.00 98.81           C  \nATOM    858  C   GLN A 119       5.230  18.595  -9.344  1.00 98.81           C  \nATOM    859  CB  GLN A 119       6.089  20.985  -9.620  1.00 98.81           C  \nATOM    860  O   GLN A 119       5.682  17.667 -10.028  1.00 98.81           O  \nATOM    861  CG  GLN A 119       6.860  22.155  -8.984  1.00 98.81           C  \nATOM    862  CD  GLN A 119       6.340  22.553  -7.603  1.00 98.81           C  \nATOM    863  NE2 GLN A 119       7.120  23.263  -6.819  1.00 98.81           N  \nATOM    864  OE1 GLN A 119       5.233  22.233  -7.204  1.00 98.81           O  \nATOM    865  N   PRO A 120       3.906  18.687  -9.090  1.00 98.75           N  \nATOM    866  CA  PRO A 120       2.924  17.665  -9.483  1.00 98.75           C  \nATOM    867  C   PRO A 120       3.002  17.188 -10.943  1.00 98.75           C  \nATOM    868  CB  PRO A 120       1.564  18.294  -9.189  1.00 98.75           C  \nATOM    869  O   PRO A 120       2.967  15.982 -11.191  1.00 98.75           O  \nATOM    870  CG  PRO A 120       1.844  19.188  -7.986  1.00 98.75           C  \nATOM    871  CD  PRO A 120       3.244  19.718  -8.288  1.00 98.75           C  \nATOM    872  N   SER A 121       3.194  18.081 -11.921  1.00 98.75           N  \nATOM    873  CA  SER A 121       3.317  17.691 -13.339  1.00 98.75           C  \nATOM    874  C   SER A 121       4.578  16.858 -13.628  1.00 98.75           C  \nATOM    875  CB  SER A 121       3.316  18.930 -14.243  1.00 98.75           C  \nATOM    876  O   SER A 121       4.521  15.871 -14.367  1.00 98.75           O  \nATOM    877  OG  SER A 121       2.303  19.838 -13.867  1.00 98.75           O  \nATOM    878  N   SER A 122       5.714  17.203 -13.000  1.00 98.81           N  \nATOM    879  CA  SER A 122       6.945  16.396 -13.061  1.00 98.81           C  \nATOM    880  C   SER A 122       6.714  15.034 -12.400  1.00 98.81           C  \nATOM    881  CB  SER A 122       8.121  17.141 -12.403  1.00 98.81           C  \nATOM    882  O   SER A 122       7.013  13.995 -12.987  1.00 98.81           O  \nATOM    883  OG  SER A 122       9.294  16.348 -12.411  1.00 98.81           O  \nATOM    884  N   ARG A 123       6.071  15.013 -11.226  1.00 98.88           N  \nATOM    885  CA  ARG A 123       5.702  13.780 -10.509  1.00 98.88           C  \nATOM    886  C   ARG A 123       4.798  12.864 -11.331  1.00 98.88           C  \nATOM    887  CB  ARG A 123       5.033  14.138  -9.177  1.00 98.88           C  \nATOM    888  O   ARG A 123       5.039  11.661 -11.351  1.00 98.88           O  \nATOM    889  CG  ARG A 123       6.042  14.766  -8.218  1.00 98.88           C  \nATOM    890  CD  ARG A 123       5.412  15.065  -6.862  1.00 98.88           C  \nATOM    891  NE  ARG A 123       6.358  15.825  -6.041  1.00 98.88           N  \nATOM    892  NH1 ARG A 123       7.332  14.144  -4.796  1.00 98.88           N  \nATOM    893  NH2 ARG A 123       8.210  16.190  -4.842  1.00 98.88           N  \nATOM    894  CZ  ARG A 123       7.288  15.366  -5.230  1.00 98.88           C  \nATOM    895  N   SER A 124       3.812  13.409 -12.043  1.00 98.94           N  \nATOM    896  CA  SER A 124       2.952  12.651 -12.963  1.00 98.94           C  \nATOM    897  C   SER A 124       3.760  12.003 -14.092  1.00 98.94           C  \nATOM    898  CB  SER A 124       1.894  13.578 -13.566  1.00 98.94           C  \nATOM    899  O   SER A 124       3.641  10.802 -14.345  1.00 98.94           O  \nATOM    900  OG  SER A 124       1.087  12.863 -14.478  1.00 98.94           O  \nATOM    901  N   SER A 125       4.660  12.767 -14.718  1.00 98.88           N  \nATOM    902  CA  SER A 125       5.527  12.264 -15.793  1.00 98.88           C  \nATOM    903  C   SER A 125       6.395  11.094 -15.316  1.00 98.88           C  \nATOM    904  CB  SER A 125       6.414  13.389 -16.334  1.00 98.88           C  \nATOM    905  O   SER A 125       6.487  10.066 -15.991  1.00 98.88           O  \nATOM    906  OG  SER A 125       5.611  14.440 -16.831  1.00 98.88           O  \nATOM    907  N   GLN A 126       6.965  11.227 -14.117  1.00 98.88           N  \nATOM    908  CA  GLN A 126       7.778  10.204 -13.458  1.00 98.88           C  \nATOM    909  C   GLN A 126       6.953   8.975 -13.044  1.00 98.88           C  \nATOM    910  CB  GLN A 126       8.463  10.836 -12.238  1.00 98.88           C  \nATOM    911  O   GLN A 126       7.401   7.844 -13.229  1.00 98.88           O  \nATOM    912  CG  GLN A 126       9.484  11.905 -12.654  1.00 98.88           C  \nATOM    913  CD  GLN A 126      10.172  12.516 -11.458  1.00 98.88           C  \nATOM    914  NE2 GLN A 126      10.005  13.788 -11.196  1.00 98.88           N  \nATOM    915  OE1 GLN A 126      10.884  11.838 -10.743  1.00 98.88           O  \nATOM    916  N   GLN A 127       5.723   9.166 -12.558  1.00 98.94           N  \nATOM    917  CA  GLN A 127       4.797   8.078 -12.222  1.00 98.94           C  \nATOM    918  C   GLN A 127       4.469   7.214 -13.442  1.00 98.94           C  \nATOM    919  CB  GLN A 127       3.500   8.677 -11.671  1.00 98.94           C  \nATOM    920  O   GLN A 127       4.530   5.984 -13.383  1.00 98.94           O  \nATOM    921  CG  GLN A 127       2.592   7.618 -11.036  1.00 98.94           C  \nATOM    922  CD  GLN A 127       1.367   8.275 -10.426  1.00 98.94           C  \nATOM    923  NE2 GLN A 127       0.971   7.891  -9.235  1.00 98.94           N  \nATOM    924  OE1 GLN A 127       0.761   9.155 -11.014  1.00 98.94           O  \nATOM    925  N   MET A 128       4.161   7.854 -14.571  1.00 98.94           N  \nATOM    926  CA  MET A 128       3.877   7.144 -15.815  1.00 98.94           C  \nATOM    927  C   MET A 128       5.134   6.487 -16.394  1.00 98.94           C  \nATOM    928  CB  MET A 128       3.220   8.097 -16.824  1.00 98.94           C  \nATOM    929  O   MET A 128       5.049   5.403 -16.972  1.00 98.94           O  \nATOM    930  CG  MET A 128       1.851   8.612 -16.355  1.00 98.94           C  \nATOM    931  SD  MET A 128       0.651   7.348 -15.838  1.00 98.94           S  \nATOM    932  CE  MET A 128       0.499   6.358 -17.349  1.00 98.94           C  \nATOM    933  N   ALA A 129       6.312   7.092 -16.211  1.00 98.88           N  \nATOM    934  CA  ALA A 129       7.586   6.478 -16.579  1.00 98.88           C  \nATOM    935  C   ALA A 129       7.868   5.203 -15.763  1.00 98.88           C  \nATOM    936  CB  ALA A 129       8.692   7.530 -16.450  1.00 98.88           C  \nATOM    937  O   ALA A 129       8.250   4.180 -16.333  1.00 98.88           O  \nATOM    938  N   ALA A 130       7.565   5.220 -14.464  1.00 98.88           N  \nATOM    939  CA  ALA A 130       7.651   4.056 -13.589  1.00 98.88           C  \nATOM    940  C   ALA A 130       6.719   2.914 -14.036  1.00 98.88           C  \nATOM    941  CB  ALA A 130       7.364   4.525 -12.161  1.00 98.88           C  \nATOM    942  O   ALA A 130       7.155   1.766 -14.128  1.00 98.88           O  \nATOM    943  N   LEU A 131       5.466   3.216 -14.400  1.00 98.88           N  \nATOM    944  CA  LEU A 131       4.542   2.221 -14.966  1.00 98.88           C  \nATOM    945  C   LEU A 131       5.055   1.628 -16.289  1.00 98.88           C  \nATOM    946  CB  LEU A 131       3.152   2.847 -15.171  1.00 98.88           C  \nATOM    947  O   LEU A 131       5.014   0.409 -16.473  1.00 98.88           O  \nATOM    948  CG  LEU A 131       2.328   3.049 -13.889  1.00 98.88           C  \nATOM    949  CD1 LEU A 131       1.027   3.773 -14.224  1.00 98.88           C  \nATOM    950  CD2 LEU A 131       1.981   1.725 -13.196  1.00 98.88           C  \nATOM    951  N   ARG A 132       5.601   2.459 -17.192  1.00 98.75           N  \nATOM    952  CA  ARG A 132       6.236   1.979 -18.434  1.00 98.75           C  \nATOM    953  C   ARG A 132       7.431   1.070 -18.149  1.00 98.75           C  \nATOM    954  CB  ARG A 132       6.676   3.155 -19.320  1.00 98.75           C  \nATOM    955  O   ARG A 132       7.592   0.052 -18.819  1.00 98.75           O  \nATOM    956  CG  ARG A 132       5.496   3.866 -19.997  1.00 98.75           C  \nATOM    957  CD  ARG A 132       5.975   4.904 -21.025  1.00 98.75           C  \nATOM    958  NE  ARG A 132       6.772   5.994 -20.423  1.00 98.75           N  \nATOM    959  NH1 ARG A 132       5.043   7.406 -19.862  1.00 98.75           N  \nATOM    960  NH2 ARG A 132       7.143   8.040 -19.488  1.00 98.75           N  \nATOM    961  CZ  ARG A 132       6.316   7.134 -19.929  1.00 98.75           C  \nATOM    962  N   GLN A 133       8.241   1.386 -17.138  1.00 98.75           N  \nATOM    963  CA  GLN A 133       9.355   0.524 -16.755  1.00 98.75           C  \nATOM    964  C   GLN A 133       8.882  -0.802 -16.151  1.00 98.75           C  \nATOM    965  CB  GLN A 133      10.321   1.266 -15.829  1.00 98.75           C  \nATOM    966  O   GLN A 133       9.448  -1.838 -16.483  1.00 98.75           O  \nATOM    967  CG  GLN A 133      11.568   0.399 -15.592  1.00 98.75           C  \nATOM    968  CD  GLN A 133      12.755   1.193 -15.084  1.00 98.75           C  \nATOM    969  NE2 GLN A 133      13.955   0.693 -15.256  1.00 98.75           N  \nATOM    970  OE1 GLN A 133      12.637   2.286 -14.577  1.00 98.75           O  \nATOM    971  N   VAL A 134       7.816  -0.815 -15.342  1.00 98.81           N  \nATOM    972  CA  VAL A 134       7.195  -2.068 -14.874  1.00 98.81           C  \nATOM    973  C   VAL A 134       6.717  -2.920 -16.056  1.00 98.81           C  \nATOM    974  CB  VAL A 134       6.051  -1.787 -13.877  1.00 98.81           C  \nATOM    975  O   VAL A 134       6.959  -4.127 -16.070  1.00 98.81           O  \nATOM    976  CG1 VAL A 134       5.241  -3.048 -13.547  1.00 98.81           C  \nATOM    977  CG2 VAL A 134       6.611  -1.256 -12.551  1.00 98.81           C  \nATOM    978  N   ALA A 135       6.117  -2.310 -17.083  1.00 98.75           N  \nATOM    979  CA  ALA A 135       5.730  -3.019 -18.305  1.00 98.75           C  \nATOM    980  C   ALA A 135       6.941  -3.583 -19.075  1.00 98.75           C  \nATOM    981  CB  ALA A 135       4.882  -2.085 -19.174  1.00 98.75           C  \nATOM    982  O   ALA A 135       6.902  -4.727 -19.523  1.00 98.75           O  \nATOM    983  N   SER A 136       8.045  -2.837 -19.174  1.00 98.56           N  \nATOM    984  CA  SER A 136       9.296  -3.337 -19.767  1.00 98.56           C  \nATOM    985  C   SER A 136       9.891  -4.506 -18.967  1.00 98.56           C  \nATOM    986  CB  SER A 136      10.304  -2.191 -19.872  1.00 98.56           C  \nATOM    987  O   SER A 136      10.290  -5.522 -19.540  1.00 98.56           O  \nATOM    988  OG  SER A 136      11.573  -2.666 -20.270  1.00 98.56           O  \nATOM    989  N   LEU A 137       9.898  -4.401 -17.638  1.00 98.62           N  \nATOM    990  CA  LEU A 137      10.381  -5.446 -16.736  1.00 98.62           C  \nATOM    991  C   LEU A 137       9.515  -6.709 -16.796  1.00 98.62           C  \nATOM    992  CB  LEU A 137      10.415  -4.884 -15.307  1.00 98.62           C  \nATOM    993  O   LEU A 137      10.038  -7.802 -16.598  1.00 98.62           O  \nATOM    994  CG  LEU A 137      11.548  -3.885 -15.021  1.00 98.62           C  \nATOM    995  CD1 LEU A 137      11.332  -3.269 -13.637  1.00 98.62           C  \nATOM    996  CD2 LEU A 137      12.924  -4.556 -15.035  1.00 98.62           C  \nATOM    997  N   ASN A 138       8.222  -6.588 -17.104  1.00 98.69           N  \nATOM    998  CA  ASN A 138       7.347  -7.735 -17.352  1.00 98.69           C  \nATOM    999  C   ASN A 138       7.779  -8.556 -18.582  1.00 98.69           C  \nATOM   1000  CB  ASN A 138       5.914  -7.219 -17.506  1.00 98.69           C  \nATOM   1001  O   ASN A 138       7.603  -9.771 -18.584  1.00 98.69           O  \nATOM   1002  CG  ASN A 138       4.928  -8.348 -17.719  1.00 98.69           C  \nATOM   1003  ND2 ASN A 138       4.220  -8.361 -18.821  1.00 98.69           N  \nATOM   1004  OD1 ASN A 138       4.772  -9.217 -16.878  1.00 98.69           O  \nATOM   1005  N   GLY A 139       8.367  -7.913 -19.596  1.00 98.38           N  \nATOM   1006  CA  GLY A 139       8.912  -8.561 -20.796  1.00 98.38           C  \nATOM   1007  C   GLY A 139      10.391  -8.957 -20.700  1.00 98.38           C  \nATOM   1008  O   GLY A 139      10.940  -9.495 -21.657  1.00 98.38           O  \nATOM   1009  N   THR A 140      11.057  -8.700 -19.570  1.00 98.44           N  \nATOM   1010  CA  THR A 140      12.491  -8.976 -19.386  1.00 98.44           C  \nATOM   1011  C   THR A 140      12.692 -10.295 -18.644  1.00 98.44           C  \nATOM   1012  CB  THR A 140      13.148  -7.821 -18.616  1.00 98.44           C  \nATOM   1013  O   THR A 140      12.377 -10.376 -17.463  1.00 98.44           O  \nATOM   1014  CG2 THR A 140      14.651  -8.025 -18.429  1.00 98.44           C  \nATOM   1015  OG1 THR A 140      12.988  -6.607 -19.314  1.00 98.44           O  \nATOM   1016  N   SER A 141      13.250 -11.320 -19.293  1.00 97.94           N  \nATOM   1017  CA  SER A 141      13.362 -12.686 -18.739  1.00 97.94           C  \nATOM   1018  C   SER A 141      14.168 -12.801 -17.437  1.00 97.94           C  \nATOM   1019  CB  SER A 141      13.965 -13.617 -19.792  1.00 97.94           C  \nATOM   1020  O   SER A 141      13.902 -13.689 -16.631  1.00 97.94           O  \nATOM   1021  OG  SER A 141      15.221 -13.119 -20.220  1.00 97.94           O  \nATOM   1022  N   SER A 142      15.119 -11.896 -17.195  1.00 97.94           N  \nATOM   1023  CA  SER A 142      15.902 -11.836 -15.949  1.00 97.94           C  \nATOM   1024  C   SER A 142      15.188 -11.113 -14.798  1.00 97.94           C  \nATOM   1025  CB  SER A 142      17.252 -11.170 -16.229  1.00 97.94           C  \nATOM   1026  O   SER A 142      15.664 -11.129 -13.663  1.00 97.94           O  \nATOM   1027  OG  SER A 142      17.067  -9.862 -16.743  1.00 97.94           O  \nATOM   1028  N   SER A 143      14.055 -10.462 -15.067  1.00 98.38           N  \nATOM   1029  CA  SER A 143      13.306  -9.699 -14.073  1.00 98.38           C  \nATOM   1030  C   SER A 143      12.508 -10.620 -13.143  1.00 98.38           C  \nATOM   1031  CB  SER A 143      12.347  -8.750 -14.780  1.00 98.38           C  \nATOM   1032  O   SER A 143      11.791 -11.502 -13.622  1.00 98.38           O  \nATOM   1033  OG  SER A 143      11.570  -8.020 -13.855  1.00 98.38           O  \nATOM   1034  N   PRO A 144      12.491 -10.372 -11.818  1.00 98.62           N  \nATOM   1035  CA  PRO A 144      11.645 -11.134 -10.895  1.00 98.62           C  \nATOM   1036  C   PRO A 144      10.137 -11.077 -11.214  1.00 98.62           C  \nATOM   1037  CB  PRO A 144      11.952 -10.554  -9.512  1.00 98.62           C  \nATOM   1038  O   PRO A 144       9.382 -11.968 -10.809  1.00 98.62           O  \nATOM   1039  CG  PRO A 144      13.359  -9.980  -9.650  1.00 98.62           C  \nATOM   1040  CD  PRO A 144      13.375  -9.472 -11.087  1.00 98.62           C  \nATOM   1041  N   ILE A 145       9.696 -10.056 -11.959  1.00 98.81           N  \nATOM   1042  CA  ILE A 145       8.304  -9.869 -12.401  1.00 98.81           C  \nATOM   1043  C   ILE A 145       8.050 -10.274 -13.859  1.00 98.81           C  \nATOM   1044  CB  ILE A 145       7.779  -8.450 -12.090  1.00 98.81           C  \nATOM   1045  O   ILE A 145       6.984  -9.957 -14.384  1.00 98.81           O  \nATOM   1046  CG1 ILE A 145       8.640  -7.360 -12.756  1.00 98.81           C  \nATOM   1047  CG2 ILE A 145       7.676  -8.283 -10.565  1.00 98.81           C  \nATOM   1048  CD1 ILE A 145       7.925  -6.008 -12.842  1.00 98.81           C  \nATOM   1049  N   TYR A 146       8.979 -10.993 -14.501  1.00 98.69           N  \nATOM   1050  CA  TYR A 146       8.766 -11.540 -15.843  1.00 98.69           C  \nATOM   1051  C   TYR A 146       7.445 -12.323 -15.917  1.00 98.69           C  \nATOM   1052  CB  TYR A 146       9.940 -12.443 -16.236  1.00 98.69           C  \nATOM   1053  O   TYR A 146       7.204 -13.217 -15.097  1.00 98.69           O  \nATOM   1054  CG  TYR A 146       9.819 -13.005 -17.641  1.00 98.69           C  \nATOM   1055  CD1 TYR A 146       9.682 -14.391 -17.845  1.00 98.69           C  \nATOM   1056  CD2 TYR A 146       9.826 -12.134 -18.748  1.00 98.69           C  \nATOM   1057  CE1 TYR A 146       9.556 -14.908 -19.150  1.00 98.69           C  \nATOM   1058  CE2 TYR A 146       9.713 -12.645 -20.054  1.00 98.69           C  \nATOM   1059  OH  TYR A 146       9.457 -14.519 -21.520  1.00 98.69           O  \nATOM   1060  CZ  TYR A 146       9.572 -14.032 -20.258  1.00 98.69           C  \nATOM   1061  N   GLY A 147       6.577 -11.942 -16.858  1.00 98.56           N  \nATOM   1062  CA  GLY A 147       5.268 -12.559 -17.089  1.00 98.56           C  \nATOM   1063  C   GLY A 147       4.245 -12.425 -15.951  1.00 98.56           C  \nATOM   1064  O   GLY A 147       3.268 -13.168 -15.939  1.00 98.56           O  \nATOM   1065  N   LYS A 148       4.445 -11.528 -14.971  1.00 98.81           N  \nATOM   1066  CA  LYS A 148       3.524 -11.359 -13.825  1.00 98.81           C  \nATOM   1067  C   LYS A 148       2.526 -10.215 -13.984  1.00 98.81           C  \nATOM   1068  CB  LYS A 148       4.301 -11.206 -12.508  1.00 98.81           C  \nATOM   1069  O   LYS A 148       1.551 -10.167 -13.238  1.00 98.81           O  \nATOM   1070  CG  LYS A 148       5.080 -12.480 -12.175  1.00 98.81           C  \nATOM   1071  CD  LYS A 148       5.610 -12.475 -10.734  1.00 98.81           C  \nATOM   1072  CE  LYS A 148       6.314 -13.799 -10.422  1.00 98.81           C  \nATOM   1073  NZ  LYS A 148       7.483 -14.021 -11.314  1.00 98.81           N  \nATOM   1074  N   VAL A 149       2.750  -9.298 -14.919  1.00 98.88           N  \nATOM   1075  CA  VAL A 149       1.969  -8.064 -15.057  1.00 98.88           C  \nATOM   1076  C   VAL A 149       1.029  -8.152 -16.252  1.00 98.88           C  \nATOM   1077  CB  VAL A 149       2.867  -6.817 -15.163  1.00 98.88           C  \nATOM   1078  O   VAL A 149       1.448  -8.410 -17.378  1.00 98.88           O  \nATOM   1079  CG1 VAL A 149       2.041  -5.530 -15.082  1.00 98.88           C  \nATOM   1080  CG2 VAL A 149       3.910  -6.760 -14.041  1.00 98.88           C  \nATOM   1081  N   ASP A 150      -0.244  -7.863 -16.013  1.00 98.75           N  \nATOM   1082  CA  ASP A 150      -1.179  -7.463 -17.054  1.00 98.75           C  \nATOM   1083  C   ASP A 150      -1.003  -5.961 -17.302  1.00 98.75           C  \nATOM   1084  CB  ASP A 150      -2.603  -7.822 -16.633  1.00 98.75           C  \nATOM   1085  O   ASP A 150      -1.486  -5.113 -16.547  1.00 98.75           O  \nATOM   1086  CG  ASP A 150      -3.624  -7.549 -17.739  1.00 98.75           C  \nATOM   1087  OD1 ASP A 150      -3.323  -6.766 -18.667  1.00 98.75           O  \nATOM   1088  OD2 ASP A 150      -4.722  -8.129 -17.595  1.00 98.75           O  \nATOM   1089  N   THR A 151      -0.245  -5.629 -18.343  1.00 98.62           N  \nATOM   1090  CA  THR A 151       0.153  -4.243 -18.633  1.00 98.62           C  \nATOM   1091  C   THR A 151      -1.009  -3.365 -19.107  1.00 98.62           C  \nATOM   1092  CB  THR A 151       1.322  -4.185 -19.626  1.00 98.62           C  \nATOM   1093  O   THR A 151      -0.867  -2.145 -19.126  1.00 98.62           O  \nATOM   1094  CG2 THR A 151       2.549  -4.950 -19.123  1.00 98.62           C  \nATOM   1095  OG1 THR A 151       0.959  -4.747 -20.864  1.00 98.62           O  \nATOM   1096  N   ALA A 152      -2.179  -3.943 -19.405  1.00 98.69           N  \nATOM   1097  CA  ALA A 152      -3.395  -3.194 -19.721  1.00 98.69           C  \nATOM   1098  C   ALA A 152      -4.168  -2.733 -18.466  1.00 98.69           C  \nATOM   1099  CB  ALA A 152      -4.259  -4.063 -20.642  1.00 98.69           C  \nATOM   1100  O   ALA A 152      -5.166  -2.013 -18.580  1.00 98.69           O  \nATOM   1101  N   ARG A 153      -3.737  -3.141 -17.261  1.00 98.75           N  \nATOM   1102  CA  ARG A 153      -4.418  -2.849 -15.989  1.00 98.75           C  \nATOM   1103  C   ARG A 153      -3.482  -2.123 -15.032  1.00 98.75           C  \nATOM   1104  CB  ARG A 153      -4.998  -4.142 -15.398  1.00 98.75           C  \nATOM   1105  O   ARG A 153      -2.793  -2.727 -14.214  1.00 98.75           O  \nATOM   1106  CG  ARG A 153      -6.005  -4.792 -16.365  1.00 98.75           C  \nATOM   1107  CD  ARG A 153      -6.720  -6.016 -15.794  1.00 98.75           C  \nATOM   1108  NE  ARG A 153      -5.776  -7.029 -15.307  1.00 98.75           N  \nATOM   1109  NH1 ARG A 153      -5.852  -6.483 -13.094  1.00 98.75           N  \nATOM   1110  NH2 ARG A 153      -4.714  -8.299 -13.767  1.00 98.75           N  \nATOM   1111  CZ  ARG A 153      -5.439  -7.259 -14.056  1.00 98.75           C  \nATOM   1112  N   MET A 154      -3.467  -0.798 -15.143  1.00 98.88           N  \nATOM   1113  CA  MET A 154      -2.590   0.072 -14.358  1.00 98.88           C  \nATOM   1114  C   MET A 154      -3.372   1.024 -13.452  1.00 98.88           C  \nATOM   1115  CB  MET A 154      -1.643   0.835 -15.285  1.00 98.88           C  \nATOM   1116  O   MET A 154      -4.469   1.459 -13.802  1.00 98.88           O  \nATOM   1117  CG  MET A 154      -0.619  -0.083 -15.956  1.00 98.88           C  \nATOM   1118  SD  MET A 154       0.499   0.830 -17.047  1.00 98.88           S  \nATOM   1119  CE  MET A 154       1.669  -0.483 -17.474  1.00 98.88           C  \nATOM   1120  N   GLY A 155      -2.796   1.350 -12.297  1.00 98.94           N  \nATOM   1121  CA  GLY A 155      -3.346   2.288 -11.326  1.00 98.94           C  \nATOM   1122  C   GLY A 155      -2.293   3.252 -10.781  1.00 98.94           C  \nATOM   1123  O   GLY A 155      -1.086   3.022 -10.895  1.00 98.94           O  \nATOM   1124  N   VAL A 156      -2.758   4.347 -10.187  1.00 98.94           N  \nATOM   1125  CA  VAL A 156      -1.902   5.432  -9.686  1.00 98.94           C  \nATOM   1126  C   VAL A 156      -2.271   5.823  -8.259  1.00 98.94           C  \nATOM   1127  CB  VAL A 156      -1.911   6.651 -10.630  1.00 98.94           C  \nATOM   1128  O   VAL A 156      -3.442   5.918  -7.897  1.00 98.94           O  \nATOM   1129  CG1 VAL A 156      -1.173   6.325 -11.935  1.00 98.94           C  \nATOM   1130  CG2 VAL A 156      -3.313   7.174 -10.960  1.00 98.94           C  \nATOM   1131  N   MET A 157      -1.263   6.054  -7.426  1.00 98.94           N  \nATOM   1132  CA  MET A 157      -1.420   6.592  -6.075  1.00 98.94           C  \nATOM   1133  C   MET A 157      -0.302   7.581  -5.781  1.00 98.94           C  \nATOM   1134  CB  MET A 157      -1.380   5.485  -5.014  1.00 98.94           C  \nATOM   1135  O   MET A 157       0.771   7.547  -6.383  1.00 98.94           O  \nATOM   1136  CG  MET A 157      -2.468   4.434  -5.181  1.00 98.94           C  \nATOM   1137  SD  MET A 157      -2.392   3.080  -3.979  1.00 98.94           S  \nATOM   1138  CE  MET A 157      -2.729   3.948  -2.424  1.00 98.94           C  \nATOM   1139  N   GLY A 158      -0.507   8.431  -4.790  1.00 98.94           N  \nATOM   1140  CA  GLY A 158       0.559   9.299  -4.333  1.00 98.94           C  \nATOM   1141  C   GLY A 158       0.105  10.221  -3.230  1.00 98.94           C  \nATOM   1142  O   GLY A 158      -1.091  10.402  -3.009  1.00 98.94           O  \nATOM   1143  N   TRP A 159       1.080  10.782  -2.531  1.00 98.94           N  \nATOM   1144  CA  TRP A 159       0.856  11.643  -1.380  1.00 98.94           C  \nATOM   1145  C   TRP A 159       1.132  13.105  -1.716  1.00 98.94           C  \nATOM   1146  CB  TRP A 159       1.721  11.145  -0.222  1.00 98.94           C  \nATOM   1147  O   TRP A 159       2.143  13.384  -2.361  1.00 98.94           O  \nATOM   1148  CG  TRP A 159       1.678  12.001   1.001  1.00 98.94           C  \nATOM   1149  CD1 TRP A 159       0.569  12.524   1.572  1.00 98.94           C  \nATOM   1150  CD2 TRP A 159       2.794  12.480   1.806  1.00 98.94           C  \nATOM   1151  CE2 TRP A 159       2.278  13.310   2.844  1.00 98.94           C  \nATOM   1152  CE3 TRP A 159       4.190  12.282   1.775  1.00 98.94           C  \nATOM   1153  NE1 TRP A 159       0.919  13.296   2.655  1.00 98.94           N  \nATOM   1154  CH2 TRP A 159       4.489  13.727   3.727  1.00 98.94           C  \nATOM   1155  CZ2 TRP A 159       3.100  13.922   3.796  1.00 98.94           C  \nATOM   1156  CZ3 TRP A 159       5.026  12.894   2.729  1.00 98.94           C  \nATOM   1157  N   SER A 160       0.290  14.041  -1.261  1.00 98.81           N  \nATOM   1158  CA  SER A 160       0.548  15.482  -1.409  1.00 98.81           C  \nATOM   1159  C   SER A 160       0.695  15.872  -2.884  1.00 98.81           C  \nATOM   1160  CB  SER A 160       1.727  15.846  -0.501  1.00 98.81           C  \nATOM   1161  O   SER A 160      -0.154  15.505  -3.700  1.00 98.81           O  \nATOM   1162  OG  SER A 160       1.889  17.240  -0.400  1.00 98.81           O  \nATOM   1163  N   MET A 161       1.794  16.523  -3.274  1.00 98.81           N  \nATOM   1164  CA  MET A 161       2.133  16.766  -4.684  1.00 98.81           C  \nATOM   1165  C   MET A 161       2.199  15.485  -5.538  1.00 98.81           C  \nATOM   1166  CB  MET A 161       3.480  17.493  -4.781  1.00 98.81           C  \nATOM   1167  O   MET A 161       1.935  15.529  -6.736  1.00 98.81           O  \nATOM   1168  CG  MET A 161       3.407  18.950  -4.338  1.00 98.81           C  \nATOM   1169  SD  MET A 161       4.960  19.819  -4.645  1.00 98.81           S  \nATOM   1170  CE  MET A 161       4.517  21.481  -4.069  1.00 98.81           C  \nATOM   1171  N   GLY A 162       2.533  14.324  -4.964  1.00 98.88           N  \nATOM   1172  CA  GLY A 162       2.426  13.027  -5.647  1.00 98.88           C  \nATOM   1173  C   GLY A 162       0.974  12.572  -5.845  1.00 98.88           C  \nATOM   1174  O   GLY A 162       0.667  11.864  -6.804  1.00 98.88           O  \nATOM   1175  N   GLY A 163       0.065  13.012  -4.972  1.00 98.88           N  \nATOM   1176  CA  GLY A 163      -1.380  12.851  -5.125  1.00 98.88           C  \nATOM   1177  C   GLY A 163      -1.930  13.727  -6.253  1.00 98.88           C  \nATOM   1178  O   GLY A 163      -2.627  13.210  -7.122  1.00 98.88           O  \nATOM   1179  N   GLY A 164      -1.519  15.000  -6.314  1.00 98.88           N  \nATOM   1180  CA  GLY A 164      -1.769  15.876  -7.468  1.00 98.88           C  \nATOM   1181  C   GLY A 164      -1.229  15.279  -8.769  1.00 98.88           C  \nATOM   1182  O   GLY A 164      -1.957  15.146  -9.749  1.00 98.88           O  \nATOM   1183  N   GLY A 165       0.008  14.775  -8.745  1.00 98.88           N  \nATOM   1184  CA  GLY A 165       0.598  14.051  -9.872  1.00 98.88           C  \nATOM   1185  C   GLY A 165      -0.210  12.819 -10.300  1.00 98.88           C  \nATOM   1186  O   GLY A 165      -0.336  12.565 -11.492  1.00 98.88           O  \nATOM   1187  N   SER A 166      -0.823  12.094  -9.360  1.00 98.94           N  \nATOM   1188  CA  SER A 166      -1.697  10.951  -9.674  1.00 98.94           C  \nATOM   1189  C   SER A 166      -2.988  11.371 -10.369  1.00 98.94           C  \nATOM   1190  CB  SER A 166      -2.043  10.160  -8.413  1.00 98.94           C  \nATOM   1191  O   SER A 166      -3.441  10.693 -11.290  1.00 98.94           O  \nATOM   1192  OG  SER A 166      -0.861   9.701  -7.794  1.00 98.94           O  \nATOM   1193  N   LEU A 167      -3.560  12.505  -9.970  1.00 98.94           N  \nATOM   1194  CA  LEU A 167      -4.744  13.073 -10.609  1.00 98.94           C  \nATOM   1195  C   LEU A 167      -4.419  13.597 -12.016  1.00 98.94           C  \nATOM   1196  CB  LEU A 167      -5.302  14.178  -9.706  1.00 98.94           C  \nATOM   1197  O   LEU A 167      -5.176  13.339 -12.949  1.00 98.94           O  \nATOM   1198  CG  LEU A 167      -5.856  13.690  -8.361  1.00 98.94           C  \nATOM   1199  CD1 LEU A 167      -5.858  14.840  -7.363  1.00 98.94           C  \nATOM   1200  CD2 LEU A 167      -7.284  13.160  -8.517  1.00 98.94           C  \nATOM   1201  N   ILE A 168      -3.255  14.230 -12.207  1.00 98.94           N  \nATOM   1202  CA  ILE A 168      -2.753  14.620 -13.537  1.00 98.94           C  \nATOM   1203  C   ILE A 168      -2.534  13.376 -14.412  1.00 98.94           C  \nATOM   1204  CB  ILE A 168      -1.463  15.463 -13.410  1.00 98.94           C  \nATOM   1205  O   ILE A 168      -2.945  13.356 -15.572  1.00 98.94           O  \nATOM   1206  CG1 ILE A 168      -1.748  16.813 -12.713  1.00 98.94           C  \nATOM   1207  CG2 ILE A 168      -0.829  15.725 -14.791  1.00 98.94           C  \nATOM   1208  CD1 ILE A 168      -0.479  17.571 -12.306  1.00 98.94           C  \nATOM   1209  N   SER A 169      -1.948  12.308 -13.860  1.00 98.94           N  \nATOM   1210  CA  SER A 169      -1.773  11.037 -14.574  1.00 98.94           C  \nATOM   1211  C   SER A 169      -3.119  10.468 -15.026  1.00 98.94           C  \nATOM   1212  CB  SER A 169      -1.057  10.005 -13.698  1.00 98.94           C  \nATOM   1213  O   SER A 169      -3.242  10.068 -16.182  1.00 98.94           O  \nATOM   1214  OG  SER A 169       0.258  10.420 -13.391  1.00 98.94           O  \nATOM   1215  N   ALA A 170      -4.130  10.479 -14.152  1.00 98.88           N  \nATOM   1216  CA  ALA A 170      -5.485  10.024 -14.456  1.00 98.88           C  \nATOM   1217  C   ALA A 170      -6.186  10.887 -15.518  1.00 98.88           C  \nATOM   1218  CB  ALA A 170      -6.273   9.987 -13.143  1.00 98.88           C  \nATOM   1219  O   ALA A 170      -6.838  10.343 -16.407  1.00 98.88           O  \nATOM   1220  N   ALA A 171      -6.024  12.212 -15.461  1.00 98.81           N  \nATOM   1221  CA  ALA A 171      -6.569  13.138 -16.451  1.00 98.81           C  \nATOM   1222  C   ALA A 171      -5.957  12.918 -17.845  1.00 98.81           C  \nATOM   1223  CB  ALA A 171      -6.317  14.568 -15.959  1.00 98.81           C  \nATOM   1224  O   ALA A 171      -6.674  12.891 -18.842  1.00 98.81           O  \nATOM   1225  N   ASN A 172      -4.638  12.715 -17.910  1.00 98.75           N  \nATOM   1226  CA  ASN A 172      -3.905  12.574 -19.170  1.00 98.75           C  \nATOM   1227  C   ASN A 172      -3.938  11.148 -19.748  1.00 98.75           C  \nATOM   1228  CB  ASN A 172      -2.461  13.050 -18.935  1.00 98.75           C  \nATOM   1229  O   ASN A 172      -3.626  10.955 -20.920  1.00 98.75           O  \nATOM   1230  CG  ASN A 172      -2.355  14.543 -18.673  1.00 98.75           C  \nATOM   1231  ND2 ASN A 172      -1.245  14.995 -18.141  1.00 98.75           N  \nATOM   1232  OD1 ASN A 172      -3.241  15.331 -18.954  1.00 98.75           O  \nATOM   1233  N   ASN A 173      -4.282  10.139 -18.941  1.00 98.81           N  \nATOM   1234  CA  ASN A 173      -4.262   8.727 -19.331  1.00 98.81           C  \nATOM   1235  C   ASN A 173      -5.569   8.041 -18.894  1.00 98.81           C  \nATOM   1236  CB  ASN A 173      -3.013   8.024 -18.760  1.00 98.81           C  \nATOM   1237  O   ASN A 173      -5.561   7.238 -17.954  1.00 98.81           O  \nATOM   1238  CG  ASN A 173      -1.708   8.711 -19.104  1.00 98.81           C  \nATOM   1239  ND2 ASN A 173      -1.316   9.686 -18.319  1.00 98.81           N  \nATOM   1240  OD1 ASN A 173      -1.006   8.354 -20.035  1.00 98.81           O  \nATOM   1241  N   PRO A 174      -6.702   8.304 -19.574  1.00 98.62           N  \nATOM   1242  CA  PRO A 174      -8.017   7.794 -19.170  1.00 98.62           C  \nATOM   1243  C   PRO A 174      -8.160   6.262 -19.256  1.00 98.62           C  \nATOM   1244  CB  PRO A 174      -9.019   8.515 -20.082  1.00 98.62           C  \nATOM   1245  O   PRO A 174      -9.169   5.712 -18.824  1.00 98.62           O  \nATOM   1246  CG  PRO A 174      -8.201   8.839 -21.332  1.00 98.62           C  \nATOM   1247  CD  PRO A 174      -6.825   9.152 -20.754  1.00 98.62           C  \nATOM   1248  N   SER A 175      -7.163   5.556 -19.805  1.00 98.69           N  \nATOM   1249  CA  SER A 175      -7.086   4.090 -19.821  1.00 98.69           C  \nATOM   1250  C   SER A 175      -6.622   3.479 -18.489  1.00 98.69           C  \nATOM   1251  CB  SER A 175      -6.168   3.641 -20.961  1.00 98.69           C  \nATOM   1252  O   SER A 175      -6.802   2.268 -18.298  1.00 98.69           O  \nATOM   1253  OG  SER A 175      -4.876   4.197 -20.800  1.00 98.69           O  \nATOM   1254  N   LEU A 176      -6.064   4.287 -17.573  1.00 98.88           N  \nATOM   1255  CA  LEU A 176      -5.804   3.890 -16.187  1.00 98.88           C  \nATOM   1256  C   LEU A 176      -7.097   3.392 -15.533  1.00 98.88           C  \nATOM   1257  CB  LEU A 176      -5.217   5.067 -15.382  1.00 98.88           C  \nATOM   1258  O   LEU A 176      -8.183   3.916 -15.767  1.00 98.88           O  \nATOM   1259  CG  LEU A 176      -3.752   5.419 -15.700  1.00 98.88           C  \nATOM   1260  CD1 LEU A 176      -3.368   6.723 -15.007  1.00 98.88           C  \nATOM   1261  CD2 LEU A 176      -2.792   4.332 -15.215  1.00 98.88           C  \nATOM   1262  N   LYS A 177      -6.981   2.350 -14.712  1.00 98.88           N  \nATOM   1263  CA  LYS A 177      -8.136   1.641 -14.152  1.00 98.88           C  \nATOM   1264  C   LYS A 177      -8.564   2.160 -12.788  1.00 98.88           C  \nATOM   1265  CB  LYS A 177      -7.849   0.136 -14.108  1.00 98.88           C  \nATOM   1266  O   LYS A 177      -9.731   2.043 -12.447  1.00 98.88           O  \nATOM   1267  CG  LYS A 177      -7.472  -0.510 -15.447  1.00 98.88           C  \nATOM   1268  CD  LYS A 177      -8.529  -0.269 -16.527  1.00 98.88           C  \nATOM   1269  CE  LYS A 177      -8.091  -0.929 -17.832  1.00 98.88           C  \nATOM   1270  NZ  LYS A 177      -8.940  -0.469 -18.951  1.00 98.88           N  \nATOM   1271  N   ALA A 178      -7.642   2.726 -12.013  1.00 98.94           N  \nATOM   1272  CA  ALA A 178      -7.948   3.286 -10.703  1.00 98.94           C  \nATOM   1273  C   ALA A 178      -6.918   4.340 -10.285  1.00 98.94           C  \nATOM   1274  CB  ALA A 178      -7.987   2.146  -9.681  1.00 98.94           C  \nATOM   1275  O   ALA A 178      -5.727   4.221 -10.583  1.00 98.94           O  \nATOM   1276  N   ALA A 179      -7.378   5.334  -9.533  1.00 98.94           N  \nATOM   1277  CA  ALA A 179      -6.545   6.281  -8.814  1.00 98.94           C  \nATOM   1278  C   ALA A 179      -6.923   6.278  -7.326  1.00 98.94           C  \nATOM   1279  CB  ALA A 179      -6.678   7.663  -9.461  1.00 98.94           C  \nATOM   1280  O   ALA A 179      -8.109   6.246  -6.993  1.00 98.94           O  \nATOM   1281  N   ALA A 180      -5.934   6.315  -6.431  1.00 98.94           N  \nATOM   1282  CA  ALA A 180      -6.180   6.486  -4.997  1.00 98.94           C  \nATOM   1283  C   ALA A 180      -5.168   7.440  -4.322  1.00 98.94           C  \nATOM   1284  CB  ALA A 180      -6.325   5.125  -4.301  1.00 98.94           C  \nATOM   1285  O   ALA A 180      -4.325   6.988  -3.541  1.00 98.94           O  \nATOM   1286  N   PRO A 181      -5.158   8.742  -4.672  1.00 98.94           N  \nATOM   1287  CA  PRO A 181      -4.287   9.720  -4.024  1.00 98.94           C  \nATOM   1288  C   PRO A 181      -4.640   9.937  -2.541  1.00 98.94           C  \nATOM   1289  CB  PRO A 181      -4.456  11.004  -4.836  1.00 98.94           C  \nATOM   1290  O   PRO A 181      -5.787   9.779  -2.122  1.00 98.94           O  \nATOM   1291  CG  PRO A 181      -5.898  10.902  -5.325  1.00 98.94           C  \nATOM   1292  CD  PRO A 181      -6.011   9.415  -5.650  1.00 98.94           C  \nATOM   1293  N   GLN A 182      -3.647  10.355  -1.755  1.00 98.94           N  \nATOM   1294  CA  GLN A 182      -3.792  10.674  -0.335  1.00 98.94           C  \nATOM   1295  C   GLN A 182      -3.288  12.082  -0.018  1.00 98.94           C  \nATOM   1296  CB  GLN A 182      -3.119   9.595   0.530  1.00 98.94           C  \nATOM   1297  O   GLN A 182      -2.206  12.465  -0.465  1.00 98.94           O  \nATOM   1298  CG  GLN A 182      -1.594   9.504   0.349  1.00 98.94           C  \nATOM   1299  CD  GLN A 182      -0.976   8.364   1.135  1.00 98.94           C  \nATOM   1300  NE2 GLN A 182      -0.127   8.619   2.102  1.00 98.94           N  \nATOM   1301  OE1 GLN A 182      -1.230   7.198   0.890  1.00 98.94           O  \nATOM   1302  N   ALA A 183      -4.072  12.840   0.750  1.00 98.81           N  \nATOM   1303  CA  ALA A 183      -3.848  14.254   1.045  1.00 98.81           C  \nATOM   1304  C   ALA A 183      -3.401  15.044  -0.206  1.00 98.81           C  \nATOM   1305  CB  ALA A 183      -2.873  14.346   2.225  1.00 98.81           C  \nATOM   1306  O   ALA A 183      -2.327  15.642  -0.181  1.00 98.81           O  \nATOM   1307  N   PRO A 184      -4.114  14.936  -1.351  1.00 98.88           N  \nATOM   1308  CA  PRO A 184      -3.619  15.457  -2.619  1.00 98.88           C  \nATOM   1309  C   PRO A 184      -3.459  16.972  -2.578  1.00 98.88           C  \nATOM   1310  CB  PRO A 184      -4.648  15.050  -3.677  1.00 98.88           C  \nATOM   1311  O   PRO A 184      -4.267  17.673  -1.970  1.00 98.88           O  \nATOM   1312  CG  PRO A 184      -5.942  14.942  -2.872  1.00 98.88           C  \nATOM   1313  CD  PRO A 184      -5.462  14.407  -1.522  1.00 98.88           C  \nATOM   1314  N   TRP A 185      -2.448  17.460  -3.289  1.00 98.88           N  \nATOM   1315  CA  TRP A 185      -2.253  18.877  -3.566  1.00 98.88           C  \nATOM   1316  C   TRP A 185      -1.821  19.073  -5.014  1.00 98.88           C  \nATOM   1317  CB  TRP A 185      -1.222  19.457  -2.598  1.00 98.88           C  \nATOM   1318  O   TRP A 185      -0.919  18.383  -5.493  1.00 98.88           O  \nATOM   1319  CG  TRP A 185      -1.062  20.941  -2.692  1.00 98.88           C  \nATOM   1320  CD1 TRP A 185      -1.729  21.861  -1.960  1.00 98.88           C  \nATOM   1321  CD2 TRP A 185      -0.219  21.701  -3.609  1.00 98.88           C  \nATOM   1322  CE2 TRP A 185      -0.393  23.091  -3.343  1.00 98.88           C  \nATOM   1323  CE3 TRP A 185       0.649  21.352  -4.664  1.00 98.88           C  \nATOM   1324  NE1 TRP A 185      -1.329  23.130  -2.332  1.00 98.88           N  \nATOM   1325  CH2 TRP A 185       1.174  23.697  -5.085  1.00 98.88           C  \nATOM   1326  CZ2 TRP A 185       0.297  24.081  -4.057  1.00 98.88           C  \nATOM   1327  CZ3 TRP A 185       1.339  22.336  -5.397  1.00 98.88           C  \nATOM   1328  N   ASP A 186      -2.455  20.027  -5.682  1.00 98.81           N  \nATOM   1329  CA  ASP A 186      -2.043  20.589  -6.963  1.00 98.81           C  \nATOM   1330  C   ASP A 186      -2.388  22.086  -6.977  1.00 98.81           C  \nATOM   1331  CB  ASP A 186      -2.712  19.841  -8.126  1.00 98.81           C  \nATOM   1332  O   ASP A 186      -3.259  22.534  -6.232  1.00 98.81           O  \nATOM   1333  CG  ASP A 186      -2.133  20.329  -9.454  1.00 98.81           C  \nATOM   1334  OD1 ASP A 186      -0.972  19.956  -9.718  1.00 98.81           O  \nATOM   1335  OD2 ASP A 186      -2.756  21.177 -10.127  1.00 98.81           O  \nATOM   1336  N   SER A 187      -1.711  22.863  -7.823  1.00 98.19           N  \nATOM   1337  CA  SER A 187      -2.018  24.289  -8.007  1.00 98.19           C  \nATOM   1338  C   SER A 187      -3.426  24.552  -8.559  1.00 98.19           C  \nATOM   1339  CB  SER A 187      -0.979  24.917  -8.935  1.00 98.19           C  \nATOM   1340  O   SER A 187      -3.972  25.631  -8.343  1.00 98.19           O  \nATOM   1341  OG  SER A 187      -0.879  24.219 -10.167  1.00 98.19           O  \nATOM   1342  N   SER A 188      -4.015  23.582  -9.264  1.00 98.50           N  \nATOM   1343  CA  SER A 188      -5.404  23.599  -9.708  1.00 98.50           C  \nATOM   1344  C   SER A 188      -6.201  22.534  -8.970  1.00 98.50           C  \nATOM   1345  CB  SER A 188      -5.479  23.348 -11.213  1.00 98.50           C  \nATOM   1346  O   SER A 188      -5.770  21.394  -8.825  1.00 98.50           O  \nATOM   1347  OG  SER A 188      -6.809  23.512 -11.682  1.00 98.50           O  \nATOM   1348  N   THR A 189      -7.415  22.876  -8.560  1.00 98.75           N  \nATOM   1349  CA  THR A 189      -8.365  21.909  -7.999  1.00 98.75           C  \nATOM   1350  C   THR A 189      -9.353  21.387  -9.043  1.00 98.75           C  \nATOM   1351  CB  THR A 189      -9.091  22.504  -6.795  1.00 98.75           C  \nATOM   1352  O   THR A 189     -10.168  20.522  -8.740  1.00 98.75           O  \nATOM   1353  CG2 THR A 189      -8.122  22.857  -5.663  1.00 98.75           C  \nATOM   1354  OG1 THR A 189      -9.751  23.690  -7.171  1.00 98.75           O  \nATOM   1355  N   ASN A 190      -9.284  21.867 -10.291  1.00 98.75           N  \nATOM   1356  CA  ASN A 190     -10.184  21.435 -11.354  1.00 98.75           C  \nATOM   1357  C   ASN A 190      -9.664  20.172 -12.058  1.00 98.75           C  \nATOM   1358  CB  ASN A 190     -10.434  22.594 -12.330  1.00 98.75           C  \nATOM   1359  O   ASN A 190      -8.919  20.245 -13.034  1.00 98.75           O  \nATOM   1360  CG  ASN A 190     -11.413  22.219 -13.433  1.00 98.75           C  \nATOM   1361  ND2 ASN A 190     -11.572  23.069 -14.419  1.00 98.75           N  \nATOM   1362  OD1 ASN A 190     -12.053  21.173 -13.439  1.00 98.75           O  \nATOM   1363  N   PHE A 191     -10.145  19.017 -11.608  1.00 98.81           N  \nATOM   1364  CA  PHE A 191      -9.931  17.704 -12.218  1.00 98.81           C  \nATOM   1365  C   PHE A 191     -11.186  17.172 -12.925  1.00 98.81           C  \nATOM   1366  CB  PHE A 191      -9.350  16.747 -11.172  1.00 98.81           C  \nATOM   1367  O   PHE A 191     -11.359  15.965 -13.089  1.00 98.81           O  \nATOM   1368  CG  PHE A 191      -7.904  17.065 -10.867  1.00 98.81           C  \nATOM   1369  CD1 PHE A 191      -6.888  16.591 -11.717  1.00 98.81           C  \nATOM   1370  CD2 PHE A 191      -7.579  17.895  -9.782  1.00 98.81           C  \nATOM   1371  CE1 PHE A 191      -5.546  16.925 -11.459  1.00 98.81           C  \nATOM   1372  CE2 PHE A 191      -6.236  18.204  -9.509  1.00 98.81           C  \nATOM   1373  CZ  PHE A 191      -5.219  17.716 -10.345  1.00 98.81           C  \nATOM   1374  N   SER A 192     -12.048  18.063 -13.423  1.00 98.62           N  \nATOM   1375  CA  SER A 192     -13.251  17.689 -14.184  1.00 98.62           C  \nATOM   1376  C   SER A 192     -12.968  16.889 -15.459  1.00 98.62           C  \nATOM   1377  CB  SER A 192     -14.066  18.933 -14.543  1.00 98.62           C  \nATOM   1378  O   SER A 192     -13.887  16.297 -16.021  1.00 98.62           O  \nATOM   1379  OG  SER A 192     -13.304  19.784 -15.374  1.00 98.62           O  \nATOM   1380  N   SER A 193     -11.720  16.837 -15.938  1.00 98.56           N  \nATOM   1381  CA  SER A 193     -11.271  15.979 -17.043  1.00 98.56           C  \nATOM   1382  C   SER A 193     -11.193  14.487 -16.669  1.00 98.56           C  \nATOM   1383  CB  SER A 193      -9.927  16.468 -17.579  1.00 98.56           C  \nATOM   1384  O   SER A 193     -11.345  13.644 -17.555  1.00 98.56           O  \nATOM   1385  OG  SER A 193      -8.999  16.552 -16.524  1.00 98.56           O  \nATOM   1386  N   VAL A 194     -11.026  14.145 -15.387  1.00 98.81           N  \nATOM   1387  CA  VAL A 194     -10.785  12.772 -14.916  1.00 98.81           C  \nATOM   1388  C   VAL A 194     -12.020  11.882 -15.089  1.00 98.81           C  \nATOM   1389  CB  VAL A 194     -10.308  12.765 -13.445  1.00 98.81           C  \nATOM   1390  O   VAL A 194     -13.086  12.140 -14.534  1.00 98.81           O  \nATOM   1391  CG1 VAL A 194     -10.098  11.340 -12.910  1.00 98.81           C  \nATOM   1392  CG2 VAL A 194      -8.966  13.498 -13.297  1.00 98.81           C  \nATOM   1393  N   THR A 195     -11.849  10.777 -15.819  1.00 98.81           N  \nATOM   1394  CA  THR A 195     -12.846   9.696 -15.971  1.00 98.81           C  \nATOM   1395  C   THR A 195     -12.459   8.415 -15.224  1.00 98.81           C  \nATOM   1396  CB  THR A 195     -13.076   9.372 -17.451  1.00 98.81           C  \nATOM   1397  O   THR A 195     -13.262   7.493 -15.109  1.00 98.81           O  \nATOM   1398  CG2 THR A 195     -13.624  10.577 -18.217  1.00 98.81           C  \nATOM   1399  OG1 THR A 195     -11.857   8.996 -18.055  1.00 98.81           O  \nATOM   1400  N   VAL A 196     -11.232   8.342 -14.703  1.00 98.88           N  \nATOM   1401  CA  VAL A 196     -10.706   7.182 -13.969  1.00 98.88           C  \nATOM   1402  C   VAL A 196     -11.386   7.070 -12.596  1.00 98.88           C  \nATOM   1403  CB  VAL A 196      -9.180   7.301 -13.794  1.00 98.88           C  \nATOM   1404  O   VAL A 196     -11.495   8.090 -11.902  1.00 98.88           O  \nATOM   1405  CG1 VAL A 196      -8.576   6.108 -13.039  1.00 98.88           C  \nATOM   1406  CG2 VAL A 196      -8.470   7.420 -15.150  1.00 98.88           C  \nATOM   1407  N   PRO A 197     -11.793   5.862 -12.152  1.00 98.88           N  \nATOM   1408  CA  PRO A 197     -12.283   5.641 -10.795  1.00 98.88           C  \nATOM   1409  C   PRO A 197     -11.306   6.156  -9.730  1.00 98.88           C  \nATOM   1410  CB  PRO A 197     -12.527   4.136 -10.680  1.00 98.88           C  \nATOM   1411  O   PRO A 197     -10.211   5.609  -9.572  1.00 98.88           O  \nATOM   1412  CG  PRO A 197     -12.873   3.747 -12.114  1.00 98.88           C  \nATOM   1413  CD  PRO A 197     -11.950   4.644 -12.936  1.00 98.88           C  \nATOM   1414  N   THR A 198     -11.684   7.206  -8.996  1.00 98.94           N  \nATOM   1415  CA  THR A 198     -10.785   7.921  -8.074  1.00 98.94           C  \nATOM   1416  C   THR A 198     -11.265   7.919  -6.618  1.00 98.94           C  \nATOM   1417  CB  THR A 198     -10.544   9.347  -8.582  1.00 98.94           C  \nATOM   1418  O   THR A 198     -12.312   8.478  -6.299  1.00 98.94           O  \nATOM   1419  CG2 THR A 198      -9.544  10.109  -7.713  1.00 98.94           C  \nATOM   1420  OG1 THR A 198      -9.982   9.290  -9.875  1.00 98.94           O  \nATOM   1421  N   LEU A 199     -10.460   7.354  -5.714  1.00 98.94           N  \nATOM   1422  CA  LEU A 199     -10.613   7.467  -4.260  1.00 98.94           C  \nATOM   1423  C   LEU A 199      -9.649   8.521  -3.709  1.00 98.94           C  \nATOM   1424  CB  LEU A 199     -10.379   6.088  -3.620  1.00 98.94           C  \nATOM   1425  O   LEU A 199      -8.442   8.363  -3.810  1.00 98.94           O  \nATOM   1426  CG  LEU A 199     -10.175   6.076  -2.091  1.00 98.94           C  \nATOM   1427  CD1 LEU A 199     -11.355   6.681  -1.331  1.00 98.94           C  \nATOM   1428  CD2 LEU A 199     -10.020   4.626  -1.632  1.00 98.94           C  \nATOM   1429  N   ILE A 200     -10.154   9.566  -3.070  1.00 98.94           N  \nATOM   1430  CA  ILE A 200      -9.334  10.554  -2.370  1.00 98.94           C  \nATOM   1431  C   ILE A 200      -9.345  10.234  -0.879  1.00 98.94           C  \nATOM   1432  CB  ILE A 200      -9.822  11.980  -2.702  1.00 98.94           C  \nATOM   1433  O   ILE A 200     -10.387  10.314  -0.229  1.00 98.94           O  \nATOM   1434  CG1 ILE A 200      -9.509  12.276  -4.185  1.00 98.94           C  \nATOM   1435  CG2 ILE A 200      -9.141  13.022  -1.797  1.00 98.94           C  \nATOM   1436  CD1 ILE A 200     -10.234  13.494  -4.751  1.00 98.94           C  \nATOM   1437  N   PHE A 201      -8.176   9.895  -0.334  1.00 98.94           N  \nATOM   1438  CA  PHE A 201      -7.963   9.943   1.109  1.00 98.94           C  \nATOM   1439  C   PHE A 201      -7.632  11.380   1.508  1.00 98.94           C  \nATOM   1440  CB  PHE A 201      -6.851   8.981   1.538  1.00 98.94           C  \nATOM   1441  O   PHE A 201      -6.684  11.962   0.979  1.00 98.94           O  \nATOM   1442  CG  PHE A 201      -7.140   7.525   1.258  1.00 98.94           C  \nATOM   1443  CD1 PHE A 201      -8.030   6.824   2.092  1.00 98.94           C  \nATOM   1444  CD2 PHE A 201      -6.512   6.868   0.183  1.00 98.94           C  \nATOM   1445  CE1 PHE A 201      -8.274   5.460   1.868  1.00 98.94           C  \nATOM   1446  CE2 PHE A 201      -6.751   5.501  -0.037  1.00 98.94           C  \nATOM   1447  CZ  PHE A 201      -7.622   4.798   0.813  1.00 98.94           C  \nATOM   1448  N   ALA A 202      -8.385  11.939   2.443  1.00 98.88           N  \nATOM   1449  CA  ALA A 202      -8.147  13.251   3.029  1.00 98.88           C  \nATOM   1450  C   ALA A 202      -7.789  13.119   4.516  1.00 98.88           C  \nATOM   1451  CB  ALA A 202      -9.377  14.120   2.767  1.00 98.88           C  \nATOM   1452  O   ALA A 202      -8.005  12.071   5.130  1.00 98.88           O  \nATOM   1453  N   CYS A 203      -7.218  14.174   5.086  1.00 98.88           N  \nATOM   1454  CA  CYS A 203      -6.812  14.233   6.484  1.00 98.88           C  \nATOM   1455  C   CYS A 203      -7.447  15.485   7.102  1.00 98.88           C  \nATOM   1456  CB  CYS A 203      -5.275  14.274   6.533  1.00 98.88           C  \nATOM   1457  O   CYS A 203      -7.147  16.581   6.648  1.00 98.88           O  \nATOM   1458  SG  CYS A 203      -4.354  12.927   5.721  1.00 98.88           S  \nATOM   1459  N   GLU A 204      -8.336  15.335   8.086  1.00 98.81           N  \nATOM   1460  CA  GLU A 204      -9.189  16.431   8.591  1.00 98.81           C  \nATOM   1461  C   GLU A 204      -8.399  17.688   9.003  1.00 98.81           C  \nATOM   1462  CB  GLU A 204      -9.963  15.876   9.794  1.00 98.81           C  \nATOM   1463  O   GLU A 204      -8.715  18.801   8.591  1.00 98.81           O  \nATOM   1464  CG  GLU A 204     -10.964  16.865  10.414  1.00 98.81           C  \nATOM   1465  CD  GLU A 204     -11.621  16.283  11.671  1.00 98.81           C  \nATOM   1466  OE1 GLU A 204     -12.822  16.532  11.905  1.00 98.81           O  \nATOM   1467  OE2 GLU A 204     -10.894  15.586  12.418  1.00 98.81           O  \nATOM   1468  N   ASN A 205      -7.311  17.500   9.753  1.00 98.62           N  \nATOM   1469  CA  ASN A 205      -6.464  18.566  10.291  1.00 98.62           C  \nATOM   1470  C   ASN A 205      -5.214  18.802   9.426  1.00 98.62           C  \nATOM   1471  CB  ASN A 205      -6.150  18.256  11.765  1.00 98.62           C  \nATOM   1472  O   ASN A 205      -4.122  19.078   9.937  1.00 98.62           O  \nATOM   1473  CG  ASN A 205      -7.380  18.320  12.650  1.00 98.62           C  \nATOM   1474  ND2 ASN A 205      -7.406  17.570  13.726  1.00 98.62           N  \nATOM   1475  OD1 ASN A 205      -8.311  19.066  12.427  1.00 98.62           O  \nATOM   1476  N   ASP A 206      -5.337  18.643   8.108  1.00 98.75           N  \nATOM   1477  CA  ASP A 206      -4.255  18.920   7.168  1.00 98.75           C  \nATOM   1478  C   ASP A 206      -3.934  20.423   7.119  1.00 98.75           C  \nATOM   1479  CB  ASP A 206      -4.617  18.372   5.784  1.00 98.75           C  \nATOM   1480  O   ASP A 206      -4.728  21.232   6.646  1.00 98.75           O  \nATOM   1481  CG  ASP A 206      -3.406  18.240   4.878  1.00 98.75           C  \nATOM   1482  OD1 ASP A 206      -2.441  19.030   5.008  1.00 98.75           O  \nATOM   1483  OD2 ASP A 206      -3.369  17.256   4.107  1.00 98.75           O  \nATOM   1484  N   SER A 207      -2.750  20.806   7.599  1.00 98.31           N  \nATOM   1485  CA  SER A 207      -2.264  22.192   7.577  1.00 98.31           C  \nATOM   1486  C   SER A 207      -1.347  22.513   6.390  1.00 98.31           C  \nATOM   1487  CB  SER A 207      -1.568  22.514   8.900  1.00 98.31           C  \nATOM   1488  O   SER A 207      -0.899  23.650   6.262  1.00 98.31           O  \nATOM   1489  OG  SER A 207      -0.452  21.662   9.111  1.00 98.31           O  \nATOM   1490  N   ILE A 208      -1.051  21.531   5.530  1.00 98.62           N  \nATOM   1491  CA  ILE A 208      -0.144  21.678   4.378  1.00 98.62           C  \nATOM   1492  C   ILE A 208      -0.937  21.743   3.074  1.00 98.62           C  \nATOM   1493  CB  ILE A 208       0.867  20.512   4.344  1.00 98.62           C  \nATOM   1494  O   ILE A 208      -0.679  22.597   2.229  1.00 98.62           O  \nATOM   1495  CG1 ILE A 208       1.666  20.343   5.657  1.00 98.62           C  \nATOM   1496  CG2 ILE A 208       1.827  20.636   3.147  1.00 98.62           C  \nATOM   1497  CD1 ILE A 208       2.508  21.557   6.069  1.00 98.62           C  \nATOM   1498  N   ALA A 209      -1.924  20.861   2.926  1.00 98.50           N  \nATOM   1499  CA  ALA A 209      -2.878  20.836   1.827  1.00 98.50           C  \nATOM   1500  C   ALA A 209      -4.310  20.804   2.389  1.00 98.50           C  \nATOM   1501  CB  ALA A 209      -2.542  19.631   0.940  1.00 98.50           C  \nATOM   1502  O   ALA A 209      -4.979  19.772   2.275  1.00 98.50           O  \nATOM   1503  N   PRO A 210      -4.778  21.901   3.024  1.00 98.69           N  \nATOM   1504  CA  PRO A 210      -6.033  21.898   3.767  1.00 98.69           C  \nATOM   1505  C   PRO A 210      -7.197  21.382   2.932  1.00 98.69           C  \nATOM   1506  CB  PRO A 210      -6.256  23.333   4.253  1.00 98.69           C  \nATOM   1507  O   PRO A 210      -7.384  21.787   1.782  1.00 98.69           O  \nATOM   1508  CG  PRO A 210      -4.846  23.912   4.293  1.00 98.69           C  \nATOM   1509  CD  PRO A 210      -4.153  23.216   3.125  1.00 98.69           C  \nATOM   1510  N   VAL A 211      -7.990  20.481   3.513  1.00 98.81           N  \nATOM   1511  CA  VAL A 211      -9.031  19.741   2.787  1.00 98.81           C  \nATOM   1512  C   VAL A 211      -9.969  20.671   2.023  1.00 98.81           C  \nATOM   1513  CB  VAL A 211      -9.833  18.856   3.756  1.00 98.81           C  \nATOM   1514  O   VAL A 211     -10.213  20.453   0.836  1.00 98.81           O  \nATOM   1515  CG1 VAL A 211     -11.027  18.243   3.023  1.00 98.81           C  \nATOM   1516  CG2 VAL A 211      -8.924  17.756   4.308  1.00 98.81           C  \nATOM   1517  N   ASN A 212     -10.434  21.734   2.679  1.00 98.44           N  \nATOM   1518  CA  ASN A 212     -11.380  22.709   2.135  1.00 98.44           C  \nATOM   1519  C   ASN A 212     -10.853  23.524   0.937  1.00 98.44           C  \nATOM   1520  CB  ASN A 212     -11.824  23.635   3.283  1.00 98.44           C  \nATOM   1521  O   ASN A 212     -11.650  24.117   0.218  1.00 98.44           O  \nATOM   1522  CG  ASN A 212     -10.716  24.509   3.860  1.00 98.44           C  \nATOM   1523  ND2 ASN A 212     -11.042  25.345   4.813  1.00 98.44           N  \nATOM   1524  OD1 ASN A 212      -9.551  24.447   3.494  1.00 98.44           O  \nATOM   1525  N   SER A 213      -9.539  23.552   0.713  1.00 98.50           N  \nATOM   1526  CA  SER A 213      -8.890  24.282  -0.385  1.00 98.50           C  \nATOM   1527  C   SER A 213      -8.142  23.366  -1.353  1.00 98.50           C  \nATOM   1528  CB  SER A 213      -7.970  25.373   0.171  1.00 98.50           C  \nATOM   1529  O   SER A 213      -7.658  23.835  -2.380  1.00 98.50           O  \nATOM   1530  OG  SER A 213      -6.987  24.829   1.028  1.00 98.50           O  \nATOM   1531  N   SER A 214      -8.085  22.062  -1.063  1.00 98.81           N  \nATOM   1532  CA  SER A 214      -7.386  21.073  -1.880  1.00 98.81           C  \nATOM   1533  C   SER A 214      -8.241  19.826  -2.120  1.00 98.81           C  \nATOM   1534  CB  SER A 214      -6.012  20.768  -1.271  1.00 98.81           C  \nATOM   1535  O   SER A 214      -8.969  19.757  -3.111  1.00 98.81           O  \nATOM   1536  OG  SER A 214      -5.239  20.099  -2.240  1.00 98.81           O  \nATOM   1537  N   ALA A 215      -8.199  18.841  -1.217  1.00 98.81           N  \nATOM   1538  CA  ALA A 215      -8.774  17.517  -1.450  1.00 98.81           C  \nATOM   1539  C   ALA A 215     -10.284  17.536  -1.753  1.00 98.81           C  \nATOM   1540  CB  ALA A 215      -8.450  16.643  -0.233  1.00 98.81           C  \nATOM   1541  O   ALA A 215     -10.732  16.794  -2.631  1.00 98.81           O  \nATOM   1542  N   LEU A 216     -11.056  18.382  -1.064  1.00 98.81           N  \nATOM   1543  CA  LEU A 216     -12.499  18.471  -1.269  1.00 98.81           C  \nATOM   1544  C   LEU A 216     -12.859  19.181  -2.593  1.00 98.81           C  \nATOM   1545  CB  LEU A 216     -13.175  19.038  -0.007  1.00 98.81           C  \nATOM   1546  O   LEU A 216     -13.562  18.565  -3.395  1.00 98.81           O  \nATOM   1547  CG  LEU A 216     -14.681  19.279  -0.179  1.00 98.81           C  \nATOM   1548  CD1 LEU A 216     -15.421  17.971  -0.453  1.00 98.81           C  \nATOM   1549  CD2 LEU A 216     -15.257  19.922   1.083  1.00 98.81           C  \nATOM   1550  N   PRO A 217     -12.329  20.379  -2.920  1.00 98.81           N  \nATOM   1551  CA  PRO A 217     -12.521  20.974  -4.247  1.00 98.81           C  \nATOM   1552  C   PRO A 217     -12.095  20.062  -5.411  1.00 98.81           C  \nATOM   1553  CB  PRO A 217     -11.698  22.262  -4.217  1.00 98.81           C  \nATOM   1554  O   PRO A 217     -12.793  19.983  -6.423  1.00 98.81           O  \nATOM   1555  CG  PRO A 217     -11.745  22.681  -2.753  1.00 98.81           C  \nATOM   1556  CD  PRO A 217     -11.696  21.343  -2.023  1.00 98.81           C  \nATOM   1557  N   ILE A 218     -10.994  19.314  -5.255  1.00 98.94           N  \nATOM   1558  CA  ILE A 218     -10.569  18.288  -6.220  1.00 98.94           C  \nATOM   1559  C   ILE A 218     -11.663  17.228  -6.395  1.00 98.94           C  \nATOM   1560  CB  ILE A 218      -9.220  17.667  -5.792  1.00 98.94           C  \nATOM   1561  O   ILE A 218     -12.046  16.925  -7.528  1.00 98.94           O  \nATOM   1562  CG1 ILE A 218      -8.068  18.680  -5.986  1.00 98.94           C  \nATOM   1563  CG2 ILE A 218      -8.897  16.372  -6.560  1.00 98.94           C  \nATOM   1564  CD1 ILE A 218      -6.729  18.251  -5.371  1.00 98.94           C  \nATOM   1565  N   TYR A 219     -12.178  16.668  -5.295  1.00 98.88           N  \nATOM   1566  CA  TYR A 219     -13.262  15.684  -5.326  1.00 98.88           C  \nATOM   1567  C   TYR A 219     -14.520  16.223  -6.018  1.00 98.88           C  \nATOM   1568  CB  TYR A 219     -13.588  15.219  -3.901  1.00 98.88           C  \nATOM   1569  O   TYR A 219     -15.099  15.539  -6.871  1.00 98.88           O  \nATOM   1570  CG  TYR A 219     -14.831  14.355  -3.838  1.00 98.88           C  \nATOM   1571  CD1 TYR A 219     -16.065  14.937  -3.492  1.00 98.88           C  \nATOM   1572  CD2 TYR A 219     -14.764  12.989  -4.172  1.00 98.88           C  \nATOM   1573  CE1 TYR A 219     -17.235  14.159  -3.475  1.00 98.88           C  \nATOM   1574  CE2 TYR A 219     -15.928  12.199  -4.123  1.00 98.88           C  \nATOM   1575  OH  TYR A 219     -18.291  12.026  -3.774  1.00 98.88           O  \nATOM   1576  CZ  TYR A 219     -17.165  12.784  -3.780  1.00 98.88           C  \nATOM   1577  N   ASP A 220     -14.933  17.443  -5.676  1.00 98.81           N  \nATOM   1578  CA  ASP A 220     -16.153  18.061  -6.194  1.00 98.81           C  \nATOM   1579  C   ASP A 220     -16.057  18.349  -7.700  1.00 98.81           C  \nATOM   1580  CB  ASP A 220     -16.485  19.323  -5.377  1.00 98.81           C  \nATOM   1581  O   ASP A 220     -17.046  18.188  -8.416  1.00 98.81           O  \nATOM   1582  CG  ASP A 220     -17.025  19.007  -3.972  1.00 98.81           C  \nATOM   1583  OD1 ASP A 220     -17.607  17.909  -3.787  1.00 98.81           O  \nATOM   1584  OD2 ASP A 220     -16.891  19.878  -3.086  1.00 98.81           O  \nATOM   1585  N   SER A 221     -14.861  18.655  -8.215  1.00 98.81           N  \nATOM   1586  CA  SER A 221     -14.646  18.931  -9.643  1.00 98.81           C  \nATOM   1587  C   SER A 221     -14.895  17.731 -10.575  1.00 98.81           C  \nATOM   1588  CB  SER A 221     -13.228  19.466  -9.857  1.00 98.81           C  \nATOM   1589  O   SER A 221     -15.242  17.919 -11.743  1.00 98.81           O  \nATOM   1590  OG  SER A 221     -12.276  18.421  -9.803  1.00 98.81           O  \nATOM   1591  N   MET A 222     -14.726  16.492 -10.096  1.00 98.81           N  \nATOM   1592  CA  MET A 222     -14.836  15.284 -10.924  1.00 98.81           C  \nATOM   1593  C   MET A 222     -16.303  14.881 -11.131  1.00 98.81           C  \nATOM   1594  CB  MET A 222     -14.063  14.126 -10.285  1.00 98.81           C  \nATOM   1595  O   MET A 222     -16.980  14.490 -10.184  1.00 98.81           O  \nATOM   1596  CG  MET A 222     -12.549  14.339 -10.231  1.00 98.81           C  \nATOM   1597  SD  MET A 222     -11.651  12.917  -9.548  1.00 98.81           S  \nATOM   1598  CE  MET A 222     -12.059  13.117  -7.796  1.00 98.81           C  \nATOM   1599  N   SER A 223     -16.807  14.934 -12.366  1.00 98.44           N  \nATOM   1600  CA  SER A 223     -18.234  14.697 -12.663  1.00 98.44           C  \nATOM   1601  C   SER A 223     -18.526  13.464 -13.525  1.00 98.44           C  \nATOM   1602  CB  SER A 223     -18.820  15.956 -13.304  1.00 98.44           C  \nATOM   1603  O   SER A 223     -19.690  13.111 -13.700  1.00 98.44           O  \nATOM   1604  OG  SER A 223     -18.118  16.284 -14.493  1.00 98.44           O  \nATOM   1605  N   ARG A 224     -17.498  12.791 -14.064  1.00 98.25           N  \nATOM   1606  CA  ARG A 224     -17.641  11.709 -15.064  1.00 98.25           C  \nATOM   1607  C   ARG A 224     -17.061  10.359 -14.633  1.00 98.25           C  \nATOM   1608  CB  ARG A 224     -17.053  12.157 -16.415  1.00 98.25           C  \nATOM   1609  O   ARG A 224     -16.824   9.503 -15.481  1.00 98.25           O  \nATOM   1610  CG  ARG A 224     -17.762  13.375 -17.020  1.00 98.25           C  \nATOM   1611  CD  ARG A 224     -17.357  13.573 -18.491  1.00 98.25           C  \nATOM   1612  NE  ARG A 224     -15.906  13.778 -18.648  1.00 98.25           N  \nATOM   1613  NH1 ARG A 224     -15.848  16.024 -18.155  1.00 98.25           N  \nATOM   1614  NH2 ARG A 224     -13.965  14.881 -18.346  1.00 98.25           N  \nATOM   1615  CZ  ARG A 224     -15.254  14.892 -18.402  1.00 98.25           C  \nATOM   1616  N   ASN A 225     -16.788  10.164 -13.347  1.00 98.62           N  \nATOM   1617  CA  ASN A 225     -16.166   8.947 -12.829  1.00 98.62           C  \nATOM   1618  C   ASN A 225     -16.917   8.358 -11.630  1.00 98.62           C  \nATOM   1619  CB  ASN A 225     -14.684   9.225 -12.509  1.00 98.62           C  \nATOM   1620  O   ASN A 225     -17.730   9.028 -10.988  1.00 98.62           O  \nATOM   1621  CG  ASN A 225     -14.428  10.224 -11.392  1.00 98.62           C  \nATOM   1622  ND2 ASN A 225     -13.199  10.295 -10.944  1.00 98.62           N  \nATOM   1623  OD1 ASN A 225     -15.293  10.960 -10.947  1.00 98.62           O  \nATOM   1624  N   ALA A 226     -16.599   7.099 -11.319  1.00 98.81           N  \nATOM   1625  CA  ALA A 226     -16.764   6.563  -9.973  1.00 98.81           C  \nATOM   1626  C   ALA A 226     -15.790   7.296  -9.045  1.00 98.81           C  \nATOM   1627  CB  ALA A 226     -16.504   5.053  -9.993  1.00 98.81           C  \nATOM   1628  O   ALA A 226     -14.586   7.329  -9.314  1.00 98.81           O  \nATOM   1629  N   LYS A 227     -16.286   7.899  -7.968  1.00 98.88           N  \nATOM   1630  CA  LYS A 227     -15.438   8.661  -7.051  1.00 98.88           C  \nATOM   1631  C   LYS A 227     -15.830   8.470  -5.604  1.00 98.88           C  \nATOM   1632  CB  LYS A 227     -15.386  10.147  -7.444  1.00 98.88           C  \nATOM   1633  O   LYS A 227     -17.009   8.353  -5.288  1.00 98.88           O  \nATOM   1634  CG  LYS A 227     -16.744  10.865  -7.358  1.00 98.88           C  \nATOM   1635  CD  LYS A 227     -16.581  12.355  -7.677  1.00 98.88           C  \nATOM   1636  CE  LYS A 227     -17.863  13.140  -7.364  1.00 98.88           C  \nATOM   1637  NZ  LYS A 227     -17.651  14.596  -7.553  1.00 98.88           N  \nATOM   1638  N   GLN A 228     -14.840   8.510  -4.724  1.00 98.88           N  \nATOM   1639  CA  GLN A 228     -15.044   8.456  -3.284  1.00 98.88           C  \nATOM   1640  C   GLN A 228     -14.086   9.414  -2.577  1.00 98.88           C  \nATOM   1641  CB  GLN A 228     -14.917   6.998  -2.824  1.00 98.88           C  \nATOM   1642  O   GLN A 228     -12.947   9.591  -3.000  1.00 98.88           O  \nATOM   1643  CG  GLN A 228     -15.199   6.772  -1.330  1.00 98.88           C  \nATOM   1644  CD  GLN A 228     -15.283   5.281  -1.036  1.00 98.88           C  \nATOM   1645  NE2 GLN A 228     -14.186   4.567  -0.940  1.00 98.88           N  \nATOM   1646  OE1 GLN A 228     -16.335   4.680  -0.942  1.00 98.88           O  \nATOM   1647  N   PHE A 229     -14.561  10.022  -1.502  1.00 98.94           N  \nATOM   1648  CA  PHE A 229     -13.822  10.882  -0.597  1.00 98.94           C  \nATOM   1649  C   PHE A 229     -13.918  10.276   0.799  1.00 98.94           C  \nATOM   1650  CB  PHE A 229     -14.435  12.280  -0.641  1.00 98.94           C  \nATOM   1651  O   PHE A 229     -15.014   9.941   1.251  1.00 98.94           O  \nATOM   1652  CG  PHE A 229     -13.741  13.283   0.240  1.00 98.94           C  \nATOM   1653  CD1 PHE A 229     -14.238  13.538   1.531  1.00 98.94           C  \nATOM   1654  CD2 PHE A 229     -12.610  13.969  -0.238  1.00 98.94           C  \nATOM   1655  CE1 PHE A 229     -13.592  14.477   2.344  1.00 98.94           C  \nATOM   1656  CE2 PHE A 229     -11.961  14.901   0.586  1.00 98.94           C  \nATOM   1657  CZ  PHE A 229     -12.446  15.136   1.883  1.00 98.94           C  \nATOM   1658  N   LEU A 230     -12.776  10.084   1.451  1.00 98.88           N  \nATOM   1659  CA  LEU A 230     -12.695   9.531   2.798  1.00 98.88           C  \nATOM   1660  C   LEU A 230     -11.706  10.364   3.604  1.00 98.88           C  \nATOM   1661  CB  LEU A 230     -12.313   8.044   2.695  1.00 98.88           C  \nATOM   1662  O   LEU A 230     -10.494  10.260   3.408  1.00 98.88           O  \nATOM   1663  CG  LEU A 230     -12.406   7.273   4.027  1.00 98.88           C  \nATOM   1664  CD1 LEU A 230     -12.363   5.778   3.720  1.00 98.88           C  \nATOM   1665  CD2 LEU A 230     -11.264   7.555   5.005  1.00 98.88           C  \nATOM   1666  N   GLU A 231     -12.234  11.175   4.511  1.00 98.94           N  \nATOM   1667  CA  GLU A 231     -11.446  12.013   5.403  1.00 98.94           C  \nATOM   1668  C   GLU A 231     -11.182  11.307   6.729  1.00 98.94           C  \nATOM   1669  CB  GLU A 231     -12.162  13.341   5.608  1.00 98.94           C  \nATOM   1670  O   GLU A 231     -12.097  10.826   7.395  1.00 98.94           O  \nATOM   1671  CG  GLU A 231     -11.266  14.354   6.316  1.00 98.94           C  \nATOM   1672  CD  GLU A 231     -12.059  15.632   6.558  1.00 98.94           C  \nATOM   1673  OE1 GLU A 231     -12.841  15.643   7.530  1.00 98.94           O  \nATOM   1674  OE2 GLU A 231     -11.874  16.565   5.758  1.00 98.94           O  \nATOM   1675  N   ILE A 232      -9.909  11.218   7.105  1.00 98.88           N  \nATOM   1676  CA  ILE A 232      -9.484  10.616   8.364  1.00 98.88           C  \nATOM   1677  C   ILE A 232      -9.610  11.639   9.489  1.00 98.88           C  \nATOM   1678  CB  ILE A 232      -8.053  10.062   8.229  1.00 98.88           C  \nATOM   1679  O   ILE A 232      -8.893  12.645   9.498  1.00 98.88           O  \nATOM   1680  CG1 ILE A 232      -7.947   8.980   7.128  1.00 98.88           C  \nATOM   1681  CG2 ILE A 232      -7.526   9.515   9.567  1.00 98.88           C  \nATOM   1682  CD1 ILE A 232      -8.758   7.699   7.390  1.00 98.88           C  \nATOM   1683  N   ASN A 233     -10.458  11.304  10.461  1.00 98.69           N  \nATOM   1684  CA  ASN A 233     -10.694  12.079  11.676  1.00 98.69           C  \nATOM   1685  C   ASN A 233      -9.375  12.313  12.432  1.00 98.69           C  \nATOM   1686  CB  ASN A 233     -11.728  11.318  12.529  1.00 98.69           C  \nATOM   1687  O   ASN A 233      -8.580  11.384  12.632  1.00 98.69           O  \nATOM   1688  CG  ASN A 233     -12.113  12.035  13.812  1.00 98.69           C  \nATOM   1689  ND2 ASN A 233     -12.795  11.373  14.711  1.00 98.69           N  \nATOM   1690  OD1 ASN A 233     -11.794  13.170  14.069  1.00 98.69           O  \nATOM   1691  N   GLY A 234      -9.127  13.559  12.821  1.00 98.38           N  \nATOM   1692  CA  GLY A 234      -7.935  14.027  13.518  1.00 98.38           C  \nATOM   1693  C   GLY A 234      -6.641  13.955  12.701  1.00 98.38           C  \nATOM   1694  O   GLY A 234      -5.559  14.171  13.250  1.00 98.38           O  \nATOM   1695  N   GLY A 235      -6.705  13.605  11.413  1.00 98.50           N  \nATOM   1696  CA  GLY A 235      -5.528  13.319  10.600  1.00 98.50           C  \nATOM   1697  C   GLY A 235      -4.709  14.568  10.268  1.00 98.50           C  \nATOM   1698  O   GLY A 235      -5.240  15.511   9.699  1.00 98.50           O  \nATOM   1699  N   SER A 236      -3.399  14.549  10.533  1.00 98.56           N  \nATOM   1700  CA  SER A 236      -2.451  15.522   9.958  1.00 98.56           C  \nATOM   1701  C   SER A 236      -2.151  15.206   8.485  1.00 98.56           C  \nATOM   1702  CB  SER A 236      -1.155  15.526  10.777  1.00 98.56           C  \nATOM   1703  O   SER A 236      -2.516  14.135   8.005  1.00 98.56           O  \nATOM   1704  OG  SER A 236      -0.427  14.318  10.596  1.00 98.56           O  \nATOM   1705  N   HIS A 237      -1.399  16.058   7.772  1.00 98.75           N  \nATOM   1706  CA  HIS A 237      -1.038  15.841   6.355  1.00 98.75           C  \nATOM   1707  C   HIS A 237      -0.460  14.442   6.045  1.00 98.75           C  \nATOM   1708  CB  HIS A 237      -0.021  16.917   5.928  1.00 98.75           C  \nATOM   1709  O   HIS A 237      -0.572  13.934   4.932  1.00 98.75           O  \nATOM   1710  CG  HIS A 237       0.218  16.948   4.435  1.00 98.75           C  \nATOM   1711  CD2 HIS A 237       1.410  16.960   3.763  1.00 98.75           C  \nATOM   1712  ND1 HIS A 237      -0.758  17.057   3.481  1.00 98.75           N  \nATOM   1713  CE1 HIS A 237      -0.195  17.122   2.274  1.00 98.75           C  \nATOM   1714  NE2 HIS A 237       1.142  17.058   2.390  1.00 98.75           N  \nATOM   1715  N   SER A 238       0.160  13.780   7.030  1.00 98.69           N  \nATOM   1716  CA  SER A 238       0.731  12.435   6.909  1.00 98.69           C  \nATOM   1717  C   SER A 238      -0.212  11.299   7.340  1.00 98.69           C  \nATOM   1718  CB  SER A 238       2.066  12.383   7.660  1.00 98.69           C  \nATOM   1719  O   SER A 238       0.263  10.185   7.565  1.00 98.69           O  \nATOM   1720  OG  SER A 238       1.898  12.672   9.034  1.00 98.69           O  \nATOM   1721  N   CYS A 239      -1.524  11.527   7.479  1.00 98.62           N  \nATOM   1722  CA  CYS A 239      -2.459  10.568   8.089  1.00 98.62           C  \nATOM   1723  C   CYS A 239      -2.570   9.214   7.364  1.00 98.62           C  \nATOM   1724  CB  CYS A 239      -3.854  11.194   8.229  1.00 98.62           C  \nATOM   1725  O   CYS A 239      -3.002   8.233   7.966  1.00 98.62           O  \nATOM   1726  SG  CYS A 239      -4.873  11.244   6.728  1.00 98.62           S  \nATOM   1727  N   ALA A 240      -2.153   9.138   6.098  1.00 98.62           N  \nATOM   1728  CA  ALA A 240      -2.129   7.917   5.289  1.00 98.62           C  \nATOM   1729  C   ALA A 240      -0.702   7.437   4.942  1.00 98.62           C  \nATOM   1730  CB  ALA A 240      -2.997   8.170   4.053  1.00 98.62           C  \nATOM   1731  O   ALA A 240      -0.516   6.525   4.130  1.00 98.62           O  \nATOM   1732  N   ASN A 241       0.337   8.046   5.522  1.00 98.69           N  \nATOM   1733  CA  ASN A 241       1.730   7.663   5.283  1.00 98.69           C  \nATOM   1734  C   ASN A 241       2.113   6.381   6.035  1.00 98.69           C  \nATOM   1735  CB  ASN A 241       2.661   8.831   5.652  1.00 98.69           C  \nATOM   1736  O   ASN A 241       1.383   5.890   6.898  1.00 98.69           O  \nATOM   1737  CG  ASN A 241       2.624   9.969   4.649  1.00 98.69           C  \nATOM   1738  ND2 ASN A 241       3.445  10.968   4.858  1.00 98.69           N  \nATOM   1739  OD1 ASN A 241       1.899   9.965   3.670  1.00 98.69           O  \nATOM   1740  N   SER A 242       3.288   5.833   5.729  1.00 98.38           N  \nATOM   1741  CA  SER A 242       3.875   4.729   6.491  1.00 98.38           C  \nATOM   1742  C   SER A 242       3.936   5.073   7.979  1.00 98.38           C  \nATOM   1743  CB  SER A 242       5.295   4.435   6.013  1.00 98.38           C  \nATOM   1744  O   SER A 242       4.428   6.136   8.345  1.00 98.38           O  \nATOM   1745  OG  SER A 242       5.347   4.162   4.627  1.00 98.38           O  \nATOM   1746  N   GLY A 243       3.463   4.158   8.828  1.00 97.06           N  \nATOM   1747  CA  GLY A 243       3.408   4.366  10.279  1.00 97.06           C  \nATOM   1748  C   GLY A 243       2.111   4.999  10.788  1.00 97.06           C  \nATOM   1749  O   GLY A 243       1.992   5.219  11.987  1.00 97.06           O  \nATOM   1750  N   ASN A 244       1.123   5.259   9.922  1.00 98.12           N  \nATOM   1751  CA  ASN A 244      -0.181   5.739  10.376  1.00 98.12           C  \nATOM   1752  C   ASN A 244      -0.929   4.709  11.251  1.00 98.12           C  \nATOM   1753  CB  ASN A 244      -1.033   6.202   9.183  1.00 98.12           C  \nATOM   1754  O   ASN A 244      -0.759   3.496  11.097  1.00 98.12           O  \nATOM   1755  CG  ASN A 244      -1.651   5.067   8.384  1.00 98.12           C  \nATOM   1756  ND2 ASN A 244      -1.055   4.652   7.292  1.00 98.12           N  \nATOM   1757  OD1 ASN A 244      -2.672   4.516   8.759  1.00 98.12           O  \nATOM   1758  N   SER A 245      -1.814   5.207  12.117  1.00 97.81           N  \nATOM   1759  CA  SER A 245      -2.614   4.412  13.066  1.00 97.81           C  \nATOM   1760  C   SER A 245      -3.946   3.895  12.497  1.00 97.81           C  \nATOM   1761  CB  SER A 245      -2.879   5.234  14.335  1.00 97.81           C  \nATOM   1762  O   SER A 245      -4.746   3.313  13.230  1.00 97.81           O  \nATOM   1763  OG  SER A 245      -1.662   5.741  14.844  1.00 97.81           O  \nATOM   1764  N   ASN A 246      -4.212   4.112  11.205  1.00 98.31           N  \nATOM   1765  CA  ASN A 246      -5.463   3.767  10.521  1.00 98.31           C  \nATOM   1766  C   ASN A 246      -5.253   2.761   9.371  1.00 98.31           C  \nATOM   1767  CB  ASN A 246      -6.179   5.062  10.091  1.00 98.31           C  \nATOM   1768  O   ASN A 246      -6.108   2.631   8.496  1.00 98.31           O  \nATOM   1769  CG  ASN A 246      -6.778   5.805  11.270  1.00 98.31           C  \nATOM   1770  ND2 ASN A 246      -6.451   7.059  11.460  1.00 98.31           N  \nATOM   1771  OD1 ASN A 246      -7.537   5.249  12.048  1.00 98.31           O  \nATOM   1772  N   GLN A 247      -4.145   2.007   9.377  1.00 98.50           N  \nATOM   1773  CA  GLN A 247      -3.779   1.076   8.297  1.00 98.50           C  \nATOM   1774  C   GLN A 247      -4.830  -0.010   8.034  1.00 98.50           C  \nATOM   1775  CB  GLN A 247      -2.444   0.390   8.619  1.00 98.50           C  \nATOM   1776  O   GLN A 247      -4.942  -0.474   6.898  1.00 98.50           O  \nATOM   1777  CG  GLN A 247      -1.261   1.350   8.475  1.00 98.50           C  \nATOM   1778  CD  GLN A 247       0.053   0.692   8.870  1.00 98.50           C  \nATOM   1779  NE2 GLN A 247       0.767   1.259   9.818  1.00 98.50           N  \nATOM   1780  OE1 GLN A 247       0.448  -0.338   8.348  1.00 98.50           O  \nATOM   1781  N   ALA A 248      -5.604  -0.408   9.046  1.00 98.50           N  \nATOM   1782  CA  ALA A 248      -6.722  -1.332   8.871  1.00 98.50           C  \nATOM   1783  C   ALA A 248      -7.762  -0.746   7.902  1.00 98.50           C  \nATOM   1784  CB  ALA A 248      -7.314  -1.654  10.249  1.00 98.50           C  \nATOM   1785  O   ALA A 248      -7.992  -1.304   6.831  1.00 98.50           O  \nATOM   1786  N   LEU A 249      -8.302   0.436   8.222  1.00 98.69           N  \nATOM   1787  CA  LEU A 249      -9.323   1.119   7.427  1.00 98.69           C  \nATOM   1788  C   LEU A 249      -8.782   1.641   6.091  1.00 98.69           C  \nATOM   1789  CB  LEU A 249      -9.923   2.258   8.270  1.00 98.69           C  \nATOM   1790  O   LEU A 249      -9.340   1.328   5.039  1.00 98.69           O  \nATOM   1791  CG  LEU A 249     -10.927   3.137   7.502  1.00 98.69           C  \nATOM   1792  CD1 LEU A 249     -12.136   2.341   7.005  1.00 98.69           C  \nATOM   1793  CD2 LEU A 249     -11.417   4.258   8.406  1.00 98.69           C  \nATOM   1794  N   ILE A 250      -7.697   2.422   6.126  1.00 98.75           N  \nATOM   1795  CA  ILE A 250      -7.100   3.031   4.931  1.00 98.75           C  \nATOM   1796  C   ILE A 250      -6.685   1.930   3.961  1.00 98.75           C  \nATOM   1797  CB  ILE A 250      -5.910   3.943   5.313  1.00 98.75           C  \nATOM   1798  O   ILE A 250      -7.026   1.972   2.781  1.00 98.75           O  \nATOM   1799  CG1 ILE A 250      -6.421   5.168   6.097  1.00 98.75           C  \nATOM   1800  CG2 ILE A 250      -5.131   4.397   4.067  1.00 98.75           C  \nATOM   1801  CD1 ILE A 250      -5.304   6.067   6.639  1.00 98.75           C  \nATOM   1802  N   GLY A 251      -6.015   0.894   4.465  1.00 98.75           N  \nATOM   1803  CA  GLY A 251      -5.573  -0.204   3.625  1.00 98.75           C  \nATOM   1804  C   GLY A 251      -6.726  -1.069   3.109  1.00 98.75           C  \nATOM   1805  O   GLY A 251      -6.675  -1.455   1.944  1.00 98.75           O  \nATOM   1806  N   LYS A 252      -7.794  -1.300   3.897  1.00 98.81           N  \nATOM   1807  CA  LYS A 252      -9.020  -1.962   3.414  1.00 98.81           C  \nATOM   1808  C   LYS A 252      -9.582  -1.207   2.217  1.00 98.81           C  \nATOM   1809  CB  LYS A 252     -10.068  -2.094   4.545  1.00 98.81           C  \nATOM   1810  O   LYS A 252      -9.823  -1.806   1.173  1.00 98.81           O  \nATOM   1811  CG  LYS A 252     -11.456  -2.530   4.024  1.00 98.81           C  \nATOM   1812  CD  LYS A 252     -12.498  -2.820   5.112  1.00 98.81           C  \nATOM   1813  CE  LYS A 252     -12.967  -1.555   5.843  1.00 98.81           C  \nATOM   1814  NZ  LYS A 252     -13.984  -1.883   6.871  1.00 98.81           N  \nATOM   1815  N   LYS A 253      -9.746   0.111   2.349  1.00 98.88           N  \nATOM   1816  CA  LYS A 253     -10.307   0.964   1.295  1.00 98.88           C  \nATOM   1817  C   LYS A 253      -9.405   1.055   0.069  1.00 98.88           C  \nATOM   1818  CB  LYS A 253     -10.638   2.343   1.885  1.00 98.88           C  \nATOM   1819  O   LYS A 253      -9.906   0.925  -1.042  1.00 98.88           O  \nATOM   1820  CG  LYS A 253     -12.002   2.338   2.596  1.00 98.88           C  \nATOM   1821  CD  LYS A 253     -13.148   2.308   1.573  1.00 98.88           C  \nATOM   1822  CE  LYS A 253     -14.527   2.534   2.195  1.00 98.88           C  \nATOM   1823  NZ  LYS A 253     -15.595   2.309   1.192  1.00 98.88           N  \nATOM   1824  N   GLY A 254      -8.093   1.193   0.251  1.00 98.81           N  \nATOM   1825  CA  GLY A 254      -7.137   1.217  -0.858  1.00 98.81           C  \nATOM   1826  C   GLY A 254      -7.120  -0.102  -1.633  1.00 98.81           C  \nATOM   1827  O   GLY A 254      -7.247  -0.098  -2.855  1.00 98.81           O  \nATOM   1828  N   VAL A 255      -7.049  -1.239  -0.931  1.00 98.88           N  \nATOM   1829  CA  VAL A 255      -7.104  -2.572  -1.552  1.00 98.88           C  \nATOM   1830  C   VAL A 255      -8.439  -2.791  -2.259  1.00 98.88           C  \nATOM   1831  CB  VAL A 255      -6.853  -3.680  -0.509  1.00 98.88           C  \nATOM   1832  O   VAL A 255      -8.444  -3.210  -3.415  1.00 98.88           O  \nATOM   1833  CG1 VAL A 255      -7.147  -5.083  -1.055  1.00 98.88           C  \nATOM   1834  CG2 VAL A 255      -5.390  -3.676  -0.038  1.00 98.88           C  \nATOM   1835  N   ALA A 256      -9.565  -2.494  -1.601  1.00 98.88           N  \nATOM   1836  CA  ALA A 256     -10.888  -2.707  -2.175  1.00 98.88           C  \nATOM   1837  C   ALA A 256     -11.140  -1.842  -3.413  1.00 98.88           C  \nATOM   1838  CB  ALA A 256     -11.955  -2.470  -1.102  1.00 98.88           C  \nATOM   1839  O   ALA A 256     -11.674  -2.345  -4.398  1.00 98.88           O  \nATOM   1840  N   TRP A 257     -10.704  -0.579  -3.399  1.00 98.94           N  \nATOM   1841  CA  TRP A 257     -10.806   0.315  -4.552  1.00 98.94           C  \nATOM   1842  C   TRP A 257     -10.024  -0.220  -5.751  1.00 98.94           C  \nATOM   1843  CB  TRP A 257     -10.304   1.710  -4.165  1.00 98.94           C  \nATOM   1844  O   TRP A 257     -10.571  -0.377  -6.844  1.00 98.94           O  \nATOM   1845  CG  TRP A 257     -10.660   2.756  -5.166  1.00 98.94           C  \nATOM   1846  CD1 TRP A 257      -9.882   3.244  -6.160  1.00 98.94           C  \nATOM   1847  CD2 TRP A 257     -11.944   3.426  -5.292  1.00 98.94           C  \nATOM   1848  CE2 TRP A 257     -11.896   4.293  -6.422  1.00 98.94           C  \nATOM   1849  CE3 TRP A 257     -13.146   3.383  -4.550  1.00 98.94           C  \nATOM   1850  NE1 TRP A 257     -10.610   4.154  -6.909  1.00 98.94           N  \nATOM   1851  CH2 TRP A 257     -14.193   4.986  -6.057  1.00 98.94           C  \nATOM   1852  CZ2 TRP A 257     -13.015   5.041  -6.815  1.00 98.94           C  \nATOM   1853  CZ3 TRP A 257     -14.255   4.163  -4.921  1.00 98.94           C  \nATOM   1854  N   MET A 258      -8.755  -0.580  -5.534  1.00 98.94           N  \nATOM   1855  CA  MET A 258      -7.909  -1.131  -6.590  1.00 98.94           C  \nATOM   1856  C   MET A 258      -8.469  -2.452  -7.116  1.00 98.94           C  \nATOM   1857  CB  MET A 258      -6.468  -1.304  -6.088  1.00 98.94           C  \nATOM   1858  O   MET A 258      -8.556  -2.640  -8.323  1.00 98.94           O  \nATOM   1859  CG  MET A 258      -5.806   0.040  -5.774  1.00 98.94           C  \nATOM   1860  SD  MET A 258      -5.711   1.175  -7.181  1.00 98.94           S  \nATOM   1861  CE  MET A 258      -4.776   2.510  -6.421  1.00 98.94           C  \nATOM   1862  N   LYS A 259      -8.933  -3.346  -6.239  1.00 98.81           N  \nATOM   1863  CA  LYS A 259      -9.576  -4.604  -6.638  1.00 98.81           C  \nATOM   1864  C   LYS A 259     -10.823  -4.380  -7.478  1.00 98.81           C  \nATOM   1865  CB  LYS A 259      -9.928  -5.409  -5.380  1.00 98.81           C  \nATOM   1866  O   LYS A 259     -10.943  -4.943  -8.566  1.00 98.81           O  \nATOM   1867  CG  LYS A 259      -8.719  -6.180  -4.849  1.00 98.81           C  \nATOM   1868  CD  LYS A 259      -8.546  -7.435  -5.700  1.00 98.81           C  \nATOM   1869  CE  LYS A 259      -7.302  -8.191  -5.291  1.00 98.81           C  \nATOM   1870  NZ  LYS A 259      -6.964  -9.175  -6.335  1.00 98.81           N  \nATOM   1871  N   ARG A 260     -11.733  -3.527  -7.008  1.00 98.81           N  \nATOM   1872  CA  ARG A 260     -12.993  -3.276  -7.700  1.00 98.81           C  \nATOM   1873  C   ARG A 260     -12.759  -2.752  -9.112  1.00 98.81           C  \nATOM   1874  CB  ARG A 260     -13.869  -2.325  -6.865  1.00 98.81           C  \nATOM   1875  O   ARG A 260     -13.408  -3.243 -10.032  1.00 98.81           O  \nATOM   1876  CG  ARG A 260     -15.171  -1.913  -7.573  1.00 98.81           C  \nATOM   1877  CD  ARG A 260     -16.043  -3.115  -7.968  1.00 98.81           C  \nATOM   1878  NE  ARG A 260     -17.108  -2.717  -8.896  1.00 98.81           N  \nATOM   1879  NH1 ARG A 260     -16.046  -3.101 -10.912  1.00 98.81           N  \nATOM   1880  NH2 ARG A 260     -18.143  -2.399 -10.879  1.00 98.81           N  \nATOM   1881  CZ  ARG A 260     -17.087  -2.748 -10.215  1.00 98.81           C  \nATOM   1882  N   PHE A 261     -11.850  -1.797  -9.295  1.00 98.81           N  \nATOM   1883  CA  PHE A 261     -11.718  -1.088 -10.571  1.00 98.81           C  \nATOM   1884  C   PHE A 261     -10.566  -1.572 -11.462  1.00 98.81           C  \nATOM   1885  CB  PHE A 261     -11.653   0.414 -10.292  1.00 98.81           C  \nATOM   1886  O   PHE A 261     -10.699  -1.544 -12.681  1.00 98.81           O  \nATOM   1887  CG  PHE A 261     -12.911   0.973  -9.659  1.00 98.81           C  \nATOM   1888  CD1 PHE A 261     -14.103   1.030 -10.407  1.00 98.81           C  \nATOM   1889  CD2 PHE A 261     -12.900   1.435  -8.332  1.00 98.81           C  \nATOM   1890  CE1 PHE A 261     -15.277   1.545  -9.831  1.00 98.81           C  \nATOM   1891  CE2 PHE A 261     -14.082   1.920  -7.750  1.00 98.81           C  \nATOM   1892  CZ  PHE A 261     -15.267   1.985  -8.497  1.00 98.81           C  \nATOM   1893  N   MET A 262      -9.460  -2.078 -10.904  1.00 98.88           N  \nATOM   1894  CA  MET A 262      -8.365  -2.652 -11.703  1.00 98.88           C  \nATOM   1895  C   MET A 262      -8.600  -4.107 -12.095  1.00 98.88           C  \nATOM   1896  CB  MET A 262      -7.008  -2.566 -10.999  1.00 98.88           C  \nATOM   1897  O   MET A 262      -8.151  -4.513 -13.167  1.00 98.88           O  \nATOM   1898  CG  MET A 262      -6.604  -1.156 -10.564  1.00 98.88           C  \nATOM   1899  SD  MET A 262      -4.935  -1.040  -9.869  1.00 98.88           S  \nATOM   1900  CE  MET A 262      -3.928  -1.704 -11.218  1.00 98.88           C  \nATOM   1901  N   ASP A 263      -9.270  -4.890 -11.247  1.00 98.75           N  \nATOM   1902  CA  ASP A 263      -9.568  -6.302 -11.513  1.00 98.75           C  \nATOM   1903  C   ASP A 263     -11.031  -6.501 -11.956  1.00 98.75           C  \nATOM   1904  CB  ASP A 263      -9.207  -7.199 -10.302  1.00 98.75           C  \nATOM   1905  O   ASP A 263     -11.449  -7.630 -12.187  1.00 98.75           O  \nATOM   1906  CG  ASP A 263      -7.724  -7.253  -9.887  1.00 98.75           C  \nATOM   1907  OD1 ASP A 263      -6.855  -6.751 -10.638  1.00 98.75           O  \nATOM   1908  OD2 ASP A 263      -7.439  -7.848  -8.813  1.00 98.75           O  \nATOM   1909  N   ASN A 264     -11.840  -5.436 -12.063  1.00 98.50           N  \nATOM   1910  CA  ASN A 264     -13.294  -5.533 -12.273  1.00 98.50           C  \nATOM   1911  C   ASN A 264     -13.968  -6.517 -11.286  1.00 98.50           C  \nATOM   1912  CB  ASN A 264     -13.597  -5.828 -13.755  1.00 98.50           C  \nATOM   1913  O   ASN A 264     -14.927  -7.206 -11.629  1.00 98.50           O  \nATOM   1914  CG  ASN A 264     -13.181  -4.705 -14.684  1.00 98.50           C  \nATOM   1915  ND2 ASN A 264     -12.863  -5.018 -15.918  1.00 98.50           N  \nATOM   1916  OD1 ASN A 264     -13.174  -3.537 -14.337  1.00 98.50           O  \nATOM   1917  N   ASP A 265     -13.425  -6.640 -10.071  1.00 98.38           N  \nATOM   1918  CA  ASP A 265     -13.790  -7.694  -9.126  1.00 98.38           C  \nATOM   1919  C   ASP A 265     -14.952  -7.244  -8.234  1.00 98.38           C  \nATOM   1920  CB  ASP A 265     -12.542  -8.138  -8.348  1.00 98.38           C  \nATOM   1921  O   ASP A 265     -14.769  -6.567  -7.218  1.00 98.38           O  \nATOM   1922  CG  ASP A 265     -12.740  -9.452  -7.587  1.00 98.38           C  \nATOM   1923  OD1 ASP A 265     -13.891  -9.899  -7.388  1.00 98.38           O  \nATOM   1924  OD2 ASP A 265     -11.724 -10.062  -7.182  1.00 98.38           O  \nATOM   1925  N   THR A 266     -16.170  -7.620  -8.623  1.00 98.25           N  \nATOM   1926  CA  THR A 266     -17.415  -7.244  -7.933  1.00 98.25           C  \nATOM   1927  C   THR A 266     -17.541  -7.842  -6.533  1.00 98.25           C  \nATOM   1928  CB  THR A 266     -18.646  -7.636  -8.756  1.00 98.25           C  \nATOM   1929  O   THR A 266     -18.335  -7.354  -5.731  1.00 98.25           O  \nATOM   1930  CG2 THR A 266     -18.706  -6.881 -10.083  1.00 98.25           C  \nATOM   1931  OG1 THR A 266     -18.601  -9.014  -9.055  1.00 98.25           O  \nATOM   1932  N   ARG A 267     -16.692  -8.812  -6.159  1.00 98.12           N  \nATOM   1933  CA  ARG A 267     -16.580  -9.279  -4.765  1.00 98.12           C  \nATOM   1934  C   ARG A 267     -16.151  -8.160  -3.808  1.00 98.12           C  \nATOM   1935  CB  ARG A 267     -15.587 -10.444  -4.676  1.00 98.12           C  \nATOM   1936  O   ARG A 267     -16.313  -8.295  -2.600  1.00 98.12           O  \nATOM   1937  CG  ARG A 267     -15.950 -11.631  -5.582  1.00 98.12           C  \nATOM   1938  CD  ARG A 267     -14.916 -12.749  -5.439  1.00 98.12           C  \nATOM   1939  NE  ARG A 267     -13.597 -12.370  -5.979  1.00 98.12           N  \nATOM   1940  NH1 ARG A 267     -12.408 -14.124  -5.085  1.00 98.12           N  \nATOM   1941  NH2 ARG A 267     -11.338 -12.524  -6.183  1.00 98.12           N  \nATOM   1942  CZ  ARG A 267     -12.463 -13.006  -5.753  1.00 98.12           C  \nATOM   1943  N   TYR A 268     -15.604  -7.066  -4.343  1.00 98.62           N  \nATOM   1944  CA  TYR A 268     -15.153  -5.897  -3.595  1.00 98.62           C  \nATOM   1945  C   TYR A 268     -16.107  -4.699  -3.674  1.00 98.62           C  \nATOM   1946  CB  TYR A 268     -13.737  -5.520  -4.045  1.00 98.62           C  \nATOM   1947  O   TYR A 268     -15.815  -3.698  -3.027  1.00 98.62           O  \nATOM   1948  CG  TYR A 268     -12.689  -6.535  -3.646  1.00 98.62           C  \nATOM   1949  CD1 TYR A 268     -11.885  -6.298  -2.516  1.00 98.62           C  \nATOM   1950  CD2 TYR A 268     -12.504  -7.707  -4.403  1.00 98.62           C  \nATOM   1951  CE1 TYR A 268     -10.850  -7.191  -2.195  1.00 98.62           C  \nATOM   1952  CE2 TYR A 268     -11.517  -8.641  -4.038  1.00 98.62           C  \nATOM   1953  OH  TYR A 268      -9.683  -9.244  -2.606  1.00 98.62           O  \nATOM   1954  CZ  TYR A 268     -10.673  -8.374  -2.939  1.00 98.62           C  \nATOM   1955  N   SER A 269     -17.227  -4.770  -4.409  1.00 98.56           N  \nATOM   1956  CA  SER A 269     -18.144  -3.631  -4.597  1.00 98.56           C  \nATOM   1957  C   SER A 269     -18.590  -3.008  -3.271  1.00 98.56           C  \nATOM   1958  CB  SER A 269     -19.375  -4.052  -5.407  1.00 98.56           C  \nATOM   1959  O   SER A 269     -18.444  -1.801  -3.085  1.00 98.56           O  \nATOM   1960  OG  SER A 269     -19.021  -4.259  -6.759  1.00 98.56           O  \nATOM   1961  N   THR A 270     -19.032  -3.827  -2.308  1.00 98.44           N  \nATOM   1962  CA  THR A 270     -19.434  -3.343  -0.977  1.00 98.44           C  \nATOM   1963  C   THR A 270     -18.279  -2.633  -0.274  1.00 98.44           C  \nATOM   1964  CB  THR A 270     -19.972  -4.487  -0.105  1.00 98.44           C  \nATOM   1965  O   THR A 270     -18.389  -1.458   0.050  1.00 98.44           O  \nATOM   1966  CG2 THR A 270     -20.590  -3.995   1.199  1.00 98.44           C  \nATOM   1967  OG1 THR A 270     -20.980  -5.170  -0.813  1.00 98.44           O  \nATOM   1968  N   PHE A 271     -17.123  -3.283  -0.128  1.00 98.62           N  \nATOM   1969  CA  PHE A 271     -15.968  -2.699   0.566  1.00 98.62           C  \nATOM   1970  C   PHE A 271     -15.434  -1.427  -0.120  1.00 98.62           C  \nATOM   1971  CB  PHE A 271     -14.863  -3.760   0.656  1.00 98.62           C  \nATOM   1972  O   PHE A 271     -14.993  -0.484   0.546  1.00 98.62           O  \nATOM   1973  CG  PHE A 271     -15.282  -5.032   1.366  1.00 98.62           C  \nATOM   1974  CD1 PHE A 271     -15.584  -4.998   2.740  1.00 98.62           C  \nATOM   1975  CD2 PHE A 271     -15.401  -6.243   0.656  1.00 98.62           C  \nATOM   1976  CE1 PHE A 271     -15.988  -6.170   3.403  1.00 98.62           C  \nATOM   1977  CE2 PHE A 271     -15.821  -7.410   1.318  1.00 98.62           C  \nATOM   1978  CZ  PHE A 271     -16.107  -7.376   2.692  1.00 98.62           C  \nATOM   1979  N   ALA A 272     -15.482  -1.376  -1.452  1.00 98.69           N  \nATOM   1980  CA  ALA A 272     -15.048  -0.224  -2.230  1.00 98.69           C  \nATOM   1981  C   ALA A 272     -15.979   0.978  -2.020  1.00 98.69           C  \nATOM   1982  CB  ALA A 272     -14.943  -0.621  -3.707  1.00 98.69           C  \nATOM   1983  O   ALA A 272     -15.465   2.070  -1.772  1.00 98.69           O  \nATOM   1984  N   CYS A 273     -17.299   0.763  -2.036  1.00 98.69           N  \nATOM   1985  CA  CYS A 273     -18.314   1.822  -2.105  1.00 98.69           C  \nATOM   1986  C   CYS A 273     -19.017   2.155  -0.777  1.00 98.69           C  \nATOM   1987  CB  CYS A 273     -19.345   1.416  -3.169  1.00 98.69           C  \nATOM   1988  O   CYS A 273     -19.641   3.208  -0.680  1.00 98.69           O  \nATOM   1989  SG  CYS A 273     -18.663   1.130  -4.824  1.00 98.69           S  \nATOM   1990  N   GLU A 274     -18.940   1.293   0.242  1.00 98.31           N  \nATOM   1991  CA  GLU A 274     -19.640   1.499   1.519  1.00 98.31           C  \nATOM   1992  C   GLU A 274     -19.143   2.744   2.260  1.00 98.31           C  \nATOM   1993  CB  GLU A 274     -19.543   0.252   2.429  1.00 98.31           C  \nATOM   1994  O   GLU A 274     -17.938   3.002   2.305  1.00 98.31           O  \nATOM   1995  CG  GLU A 274     -18.119  -0.048   2.954  1.00 98.31           C  \nATOM   1996  CD  GLU A 274     -17.972  -1.370   3.734  1.00 98.31           C  \nATOM   1997  OE1 GLU A 274     -16.832  -1.625   4.208  1.00 98.31           O  \nATOM   1998  OE2 GLU A 274     -18.956  -2.134   3.835  1.00 98.31           O  \nATOM   1999  N   ASN A 275     -20.038   3.471   2.930  1.00 98.31           N  \nATOM   2000  CA  ASN A 275     -19.599   4.317   4.035  1.00 98.31           C  \nATOM   2001  C   ASN A 275     -19.115   3.389   5.165  1.00 98.31           C  \nATOM   2002  CB  ASN A 275     -20.730   5.253   4.491  1.00 98.31           C  \nATOM   2003  O   ASN A 275     -19.905   2.562   5.623  1.00 98.31           O  \nATOM   2004  CG  ASN A 275     -20.257   6.245   5.546  1.00 98.31           C  \nATOM   2005  ND2 ASN A 275     -20.962   7.334   5.728  1.00 98.31           N  \nATOM   2006  OD1 ASN A 275     -19.247   6.066   6.211  1.00 98.31           O  \nATOM   2007  N   PRO A 276     -17.856   3.495   5.638  1.00 97.62           N  \nATOM   2008  CA  PRO A 276     -17.367   2.658   6.728  1.00 97.62           C  \nATOM   2009  C   PRO A 276     -18.133   2.820   8.047  1.00 97.62           C  \nATOM   2010  CB  PRO A 276     -15.897   3.053   6.920  1.00 97.62           C  \nATOM   2011  O   PRO A 276     -17.925   2.001   8.939  1.00 97.62           O  \nATOM   2012  CG  PRO A 276     -15.499   3.654   5.576  1.00 97.62           C  \nATOM   2013  CD  PRO A 276     -16.783   4.332   5.119  1.00 97.62           C  \nATOM   2014  N   ASN A 277     -18.943   3.882   8.201  1.00 96.88           N  \nATOM   2015  CA  ASN A 277     -19.676   4.223   9.427  1.00 96.88           C  \nATOM   2016  C   ASN A 277     -18.769   4.164  10.671  1.00 96.88           C  \nATOM   2017  CB  ASN A 277     -20.949   3.365   9.528  1.00 96.88           C  \nATOM   2018  O   ASN A 277     -19.113   3.591  11.703  1.00 96.88           O  \nATOM   2019  CG  ASN A 277     -21.928   3.623   8.397  1.00 96.88           C  \nATOM   2020  ND2 ASN A 277     -22.610   2.604   7.930  1.00 96.88           N  \nATOM   2021  OD1 ASN A 277     -22.105   4.732   7.922  1.00 96.88           O  \nATOM   2022  N   SER A 278     -17.555   4.705  10.537  1.00 97.88           N  \nATOM   2023  CA  SER A 278     -16.480   4.581  11.519  1.00 97.88           C  \nATOM   2024  C   SER A 278     -16.188   5.922  12.174  1.00 97.88           C  \nATOM   2025  CB  SER A 278     -15.218   4.052  10.832  1.00 97.88           C  \nATOM   2026  O   SER A 278     -16.045   6.922  11.483  1.00 97.88           O  \nATOM   2027  OG  SER A 278     -14.178   3.858  11.776  1.00 97.88           O  \nATOM   2028  N   THR A 279     -15.956   5.927  13.489  1.00 98.25           N  \nATOM   2029  CA  THR A 279     -15.493   7.114  14.236  1.00 98.25           C  \nATOM   2030  C   THR A 279     -14.119   7.625  13.788  1.00 98.25           C  \nATOM   2031  CB  THR A 279     -15.419   6.810  15.738  1.00 98.25           C  \nATOM   2032  O   THR A 279     -13.720   8.729  14.147  1.00 98.25           O  \nATOM   2033  CG2 THR A 279     -16.788   6.475  16.331  1.00 98.25           C  \nATOM   2034  OG1 THR A 279     -14.588   5.690  15.972  1.00 98.25           O  \nATOM   2035  N   ARG A 280     -13.381   6.837  12.995  1.00 98.44           N  \nATOM   2036  CA  ARG A 280     -12.104   7.223  12.373  1.00 98.44           C  \nATOM   2037  C   ARG A 280     -12.288   8.063  11.104  1.00 98.44           C  \nATOM   2038  CB  ARG A 280     -11.283   5.958  12.076  1.00 98.44           C  \nATOM   2039  O   ARG A 280     -11.287   8.475  10.523  1.00 98.44           O  \nATOM   2040  CG  ARG A 280     -10.894   5.196  13.351  1.00 98.44           C  \nATOM   2041  CD  ARG A 280     -10.310   3.824  12.999  1.00 98.44           C  \nATOM   2042  NE  ARG A 280      -9.922   3.075  14.209  1.00 98.44           N  \nATOM   2043  NH1 ARG A 280      -7.742   3.800  14.410  1.00 98.44           N  \nATOM   2044  NH2 ARG A 280      -8.569   2.350  15.882  1.00 98.44           N  \nATOM   2045  CZ  ARG A 280      -8.751   3.086  14.823  1.00 98.44           C  \nATOM   2046  N   VAL A 281     -13.528   8.276  10.659  1.00 98.75           N  \nATOM   2047  CA  VAL A 281     -13.880   9.017   9.443  1.00 98.75           C  \nATOM   2048  C   VAL A 281     -14.719  10.231   9.828  1.00 98.75           C  \nATOM   2049  CB  VAL A 281     -14.635   8.116   8.445  1.00 98.75           C  \nATOM   2050  O   VAL A 281     -15.786  10.064  10.412  1.00 98.75           O  \nATOM   2051  CG1 VAL A 281     -14.963   8.843   7.140  1.00 98.75           C  \nATOM   2052  CG2 VAL A 281     -13.801   6.884   8.071  1.00 98.75           C  \nATOM   2053  N   SER A 282     -14.225  11.432   9.532  1.00 98.62           N  \nATOM   2054  CA  SER A 282     -14.893  12.711   9.836  1.00 98.62           C  \nATOM   2055  C   SER A 282     -15.825  13.166   8.706  1.00 98.62           C  \nATOM   2056  CB  SER A 282     -13.840  13.788  10.131  1.00 98.62           C  \nATOM   2057  O   SER A 282     -16.893  13.708   8.975  1.00 98.62           O  \nATOM   2058  OG  SER A 282     -12.794  13.693   9.186  1.00 98.62           O  \nATOM   2059  N   ASP A 283     -15.479  12.861   7.452  1.00 98.75           N  \nATOM   2060  CA  ASP A 283     -16.306  13.112   6.264  1.00 98.75           C  \nATOM   2061  C   ASP A 283     -16.167  11.944   5.267  1.00 98.75           C  \nATOM   2062  CB  ASP A 283     -15.947  14.487   5.653  1.00 98.75           C  \nATOM   2063  O   ASP A 283     -15.078  11.400   5.037  1.00 98.75           O  \nATOM   2064  CG  ASP A 283     -16.882  14.978   4.528  1.00 98.75           C  \nATOM   2065  OD1 ASP A 283     -17.708  14.195   4.006  1.00 98.75           O  \nATOM   2066  OD2 ASP A 283     -16.775  16.147   4.093  1.00 98.75           O  \nATOM   2067  N   PHE A 284     -17.290  11.530   4.680  1.00 98.81           N  \nATOM   2068  CA  PHE A 284     -17.362  10.482   3.670  1.00 98.81           C  \nATOM   2069  C   PHE A 284     -18.335  10.868   2.561  1.00 98.81           C  \nATOM   2070  CB  PHE A 284     -17.783   9.147   4.292  1.00 98.81           C  \nATOM   2071  O   PHE A 284     -19.502  11.171   2.813  1.00 98.81           O  \nATOM   2072  CG  PHE A 284     -17.917   8.037   3.265  1.00 98.81           C  \nATOM   2073  CD1 PHE A 284     -19.154   7.807   2.632  1.00 98.81           C  \nATOM   2074  CD2 PHE A 284     -16.792   7.279   2.896  1.00 98.81           C  \nATOM   2075  CE1 PHE A 284     -19.269   6.816   1.643  1.00 98.81           C  \nATOM   2076  CE2 PHE A 284     -16.911   6.277   1.916  1.00 98.81           C  \nATOM   2077  CZ  PHE A 284     -18.149   6.046   1.291  1.00 98.81           C  \nATOM   2078  N   ARG A 285     -17.883  10.757   1.311  1.00 98.75           N  \nATOM   2079  CA  ARG A 285     -18.705  11.020   0.122  1.00 98.75           C  \nATOM   2080  C   ARG A 285     -18.409   9.984  -0.947  1.00 98.75           C  \nATOM   2081  CB  ARG A 285     -18.478  12.452  -0.403  1.00 98.75           C  \nATOM   2082  O   ARG A 285     -17.266   9.569  -1.114  1.00 98.75           O  \nATOM   2083  CG  ARG A 285     -18.448  13.512   0.710  1.00 98.75           C  \nATOM   2084  CD  ARG A 285     -18.236  14.916   0.159  1.00 98.75           C  \nATOM   2085  NE  ARG A 285     -18.048  15.858   1.269  1.00 98.75           N  \nATOM   2086  NH1 ARG A 285     -18.736  17.782   0.208  1.00 98.75           N  \nATOM   2087  NH2 ARG A 285     -17.968  17.844   2.315  1.00 98.75           N  \nATOM   2088  CZ  ARG A 285     -18.259  17.157   1.254  1.00 98.75           C  \nATOM   2089  N   THR A 286     -19.410   9.575  -1.714  1.00 98.69           N  \nATOM   2090  CA  THR A 286     -19.214   8.662  -2.847  1.00 98.69           C  \nATOM   2091  C   THR A 286     -20.247   8.936  -3.935  1.00 98.69           C  \nATOM   2092  CB  THR A 286     -19.204   7.198  -2.380  1.00 98.69           C  \nATOM   2093  O   THR A 286     -21.371   9.331  -3.634  1.00 98.69           O  \nATOM   2094  CG2 THR A 286     -20.574   6.649  -1.991  1.00 98.69           C  \nATOM   2095  OG1 THR A 286     -18.681   6.366  -3.385  1.00 98.69           O  \nATOM   2096  N   ALA A 287     -19.862   8.772  -5.199  1.00 98.56           N  \nATOM   2097  CA  ALA A 287     -20.737   8.953  -6.353  1.00 98.56           C  \nATOM   2098  C   ALA A 287     -20.357   7.973  -7.470  1.00 98.56           C  \nATOM   2099  CB  ALA A 287     -20.669  10.409  -6.833  1.00 98.56           C  \nATOM   2100  O   ALA A 287     -19.178   7.663  -7.657  1.00 98.56           O  \nATOM   2101  N   ASN A 288     -21.359   7.494  -8.217  1.00 98.00           N  \nATOM   2102  CA  ASN A 288     -21.207   6.529  -9.316  1.00 98.00           C  \nATOM   2103  C   ASN A 288     -20.401   5.268  -8.934  1.00 98.00           C  \nATOM   2104  CB  ASN A 288     -20.641   7.240 -10.559  1.00 98.00           C  \nATOM   2105  O   ASN A 288     -19.641   4.745  -9.746  1.00 98.00           O  \nATOM   2106  CG  ASN A 288     -21.371   8.510 -10.934  1.00 98.00           C  \nATOM   2107  ND2 ASN A 288     -20.645   9.558 -11.245  1.00 98.00           N  \nATOM   2108  OD1 ASN A 288     -22.586   8.578 -10.927  1.00 98.00           O  \nATOM   2109  N   CYS A 289     -20.515   4.802  -7.685  1.00 97.31           N  \nATOM   2110  CA  CYS A 289     -19.765   3.653  -7.181  1.00 97.31           C  \nATOM   2111  C   CYS A 289     -20.651   2.402  -7.159  1.00 97.31           C  \nATOM   2112  CB  CYS A 289     -19.167   3.995  -5.813  1.00 97.31           C  \nATOM   2113  O   CYS A 289     -21.633   2.342  -6.421  1.00 97.31           O  \nATOM   2114  SG  CYS A 289     -17.803   2.917  -5.292  1.00 97.31           S  \nATOM   2115  N   SER A 290     -20.278   1.413  -7.967  1.00 90.31           N  \nATOM   2116  CA  SER A 290     -20.911   0.091  -8.096  1.00 90.31           C  \nATOM   2117  C   SER A 290     -19.838  -0.938  -8.369  1.00 90.31           C  \nATOM   2118  CB  SER A 290     -21.923   0.102  -9.244  1.00 90.31           C  \nATOM   2119  O   SER A 290     -18.926  -0.568  -9.136  1.00 90.31           O  \nATOM   2120  OG  SER A 290     -21.215   0.337 -10.445  1.00 90.31           O  \nATOM   2121  OXT SER A 290     -19.923  -2.097  -7.925  1.00 90.31           O  \nTER    2122      SER A 290                                                      \nEND                                                                             \n",
                    "binary": false
                  }
                ],
                "kwargs": {
                  "defaultRepresentation": true,
                  "ext": "pdb"
                }
              },
              {
                "component_index": 0,
                "target": "compList",
                "type": "call_method",
                "methodName": "removeAllRepresentations",
                "reconstruc_color_scheme": false,
                "args": [],
                "kwargs": {}
              },
              {
                "component_index": 1,
                "target": "compList",
                "type": "call_method",
                "methodName": "removeAllRepresentations",
                "reconstruc_color_scheme": false,
                "args": [],
                "kwargs": {}
              },
              {
                "component_index": 0,
                "target": "compList",
                "type": "call_method",
                "methodName": "addRepresentation",
                "reconstruc_color_scheme": false,
                "args": [
                  "cartoon"
                ],
                "kwargs": {
                  "sele": "all",
                  "colorScheme": "uniform",
                  "color": "red"
                }
              },
              {
                "component_index": 1,
                "target": "compList",
                "type": "call_method",
                "methodName": "addRepresentation",
                "reconstruc_color_scheme": false,
                "args": [
                  "cartoon"
                ],
                "kwargs": {
                  "sele": "all",
                  "colorScheme": "uniform",
                  "color": "blue",
                  "opacity": 0.7
                }
              }
            ],
            "_ngl_original_stage_parameters": {
              "impostor": true,
              "quality": "medium",
              "workerDefault": true,
              "sampleLevel": 0,
              "backgroundColor": "white",
              "rotateSpeed": 2,
              "zoomSpeed": 1.2,
              "panSpeed": 1,
              "clipNear": 0,
              "clipFar": 100,
              "clipDist": 10,
              "clipMode": "scene",
              "clipScale": "relative",
              "fogNear": 50,
              "fogFar": 100,
              "cameraFov": 40,
              "cameraEyeSep": 0.3,
              "cameraType": "perspective",
              "lightColor": 14540253,
              "lightIntensity": 1.2,
              "ambientColor": 14540253,
              "ambientIntensity": 0.3,
              "hoverTimeout": 0,
              "tooltip": true,
              "mousePreset": "default"
            },
            "_ngl_repr_dict": {
              "0": {
                "0": {
                  "type": "cartoon",
                  "params": {
                    "lazy": false,
                    "visible": true,
                    "quality": "medium",
                    "aspectRatio": 5,
                    "subdiv": 6,
                    "radialSegments": 10,
                    "tension": null,
                    "capped": true,
                    "smoothSheet": false,
                    "radiusType": "sstruc",
                    "radiusData": {},
                    "radiusSize": 1,
                    "radiusScale": 0.7,
                    "assembly": "default",
                    "defaultAssembly": "",
                    "clipNear": 0,
                    "clipRadius": 0,
                    "clipCenter": {
                      "x": 0,
                      "y": 0,
                      "z": 0
                    },
                    "flatShaded": false,
                    "opacity": 1,
                    "depthWrite": true,
                    "side": "double",
                    "wireframe": false,
                    "colorScheme": "uniform",
                    "colorScale": "RdYlBu",
                    "colorReverse": false,
                    "colorValue": 16711680,
                    "colorMode": "hcl",
                    "roughness": 0.4,
                    "metalness": 0,
                    "diffuse": 16777215,
                    "diffuseInterior": false,
                    "useInteriorColor": true,
                    "interiorColor": 2236962,
                    "interiorDarkening": 0,
                    "matrix": {
                      "elements": [
                        1,
                        0,
                        0,
                        0,
                        0,
                        1,
                        0,
                        0,
                        0,
                        0,
                        1,
                        0,
                        0,
                        0,
                        0,
                        1
                      ]
                    },
                    "disablePicking": false,
                    "sele": "all"
                  }
                },
                "1": {
                  "type": "cartoon",
                  "params": {
                    "lazy": false,
                    "visible": true,
                    "quality": "medium",
                    "aspectRatio": 5,
                    "subdiv": 6,
                    "radialSegments": 10,
                    "tension": null,
                    "capped": true,
                    "smoothSheet": false,
                    "radiusType": "sstruc",
                    "radiusData": {},
                    "radiusSize": 1,
                    "radiusScale": 0.7,
                    "assembly": "default",
                    "defaultAssembly": "",
                    "clipNear": 0,
                    "clipRadius": 0,
                    "clipCenter": {
                      "x": 0,
                      "y": 0,
                      "z": 0
                    },
                    "flatShaded": false,
                    "opacity": 1,
                    "depthWrite": true,
                    "side": "double",
                    "wireframe": false,
                    "colorScheme": "uniform",
                    "colorScale": "RdYlBu",
                    "colorReverse": false,
                    "colorValue": 16711680,
                    "colorMode": "hcl",
                    "roughness": 0.4,
                    "metalness": 0,
                    "diffuse": 16777215,
                    "diffuseInterior": false,
                    "useInteriorColor": true,
                    "interiorColor": 2236962,
                    "interiorDarkening": 0,
                    "matrix": {
                      "elements": [
                        1,
                        0,
                        0,
                        0,
                        0,
                        1,
                        0,
                        0,
                        0,
                        0,
                        1,
                        0,
                        0,
                        0,
                        0,
                        1
                      ]
                    },
                    "disablePicking": false,
                    "sele": "all"
                  }
                }
              },
              "1": {
                "0": {
                  "type": "cartoon",
                  "params": {
                    "lazy": false,
                    "visible": true,
                    "quality": "medium",
                    "aspectRatio": 5,
                    "subdiv": 6,
                    "radialSegments": 10,
                    "tension": null,
                    "capped": true,
                    "smoothSheet": false,
                    "radiusType": "sstruc",
                    "radiusData": {},
                    "radiusSize": 1,
                    "radiusScale": 0.7,
                    "assembly": "default",
                    "defaultAssembly": "",
                    "clipNear": 0,
                    "clipRadius": 0,
                    "clipCenter": {
                      "x": 0,
                      "y": 0,
                      "z": 0
                    },
                    "flatShaded": false,
                    "opacity": 0.7,
                    "depthWrite": true,
                    "side": "double",
                    "wireframe": false,
                    "colorScheme": "uniform",
                    "colorScale": "RdYlBu",
                    "colorReverse": false,
                    "colorValue": 255,
                    "colorMode": "hcl",
                    "roughness": 0.4,
                    "metalness": 0,
                    "diffuse": 16777215,
                    "diffuseInterior": false,
                    "useInteriorColor": true,
                    "interiorColor": 2236962,
                    "interiorDarkening": 0,
                    "matrix": {
                      "elements": [
                        1,
                        0,
                        0,
                        0,
                        0,
                        1,
                        0,
                        0,
                        0,
                        0,
                        1,
                        0,
                        0,
                        0,
                        0,
                        1
                      ]
                    },
                    "disablePicking": false,
                    "sele": "all"
                  }
                },
                "1": {
                  "type": "cartoon",
                  "params": {
                    "lazy": false,
                    "visible": true,
                    "quality": "medium",
                    "aspectRatio": 5,
                    "subdiv": 6,
                    "radialSegments": 10,
                    "tension": null,
                    "capped": true,
                    "smoothSheet": false,
                    "radiusType": "sstruc",
                    "radiusData": {},
                    "radiusSize": 1,
                    "radiusScale": 0.7,
                    "assembly": "default",
                    "defaultAssembly": "",
                    "clipNear": 0,
                    "clipRadius": 0,
                    "clipCenter": {
                      "x": 0,
                      "y": 0,
                      "z": 0
                    },
                    "flatShaded": false,
                    "opacity": 0.7,
                    "depthWrite": true,
                    "side": "double",
                    "wireframe": false,
                    "colorScheme": "uniform",
                    "colorScale": "RdYlBu",
                    "colorReverse": false,
                    "colorValue": 255,
                    "colorMode": "hcl",
                    "roughness": 0.4,
                    "metalness": 0,
                    "diffuse": 16777215,
                    "diffuseInterior": false,
                    "useInteriorColor": true,
                    "interiorColor": 2236962,
                    "interiorDarkening": 0,
                    "matrix": {
                      "elements": [
                        1,
                        0,
                        0,
                        0,
                        0,
                        1,
                        0,
                        0,
                        0,
                        0,
                        1,
                        0,
                        0,
                        0,
                        0,
                        1
                      ]
                    },
                    "disablePicking": false,
                    "sele": "all"
                  }
                }
              },
              "2": {
                "0": {
                  "type": "cartoon",
                  "params": {
                    "lazy": false,
                    "visible": true,
                    "quality": "high",
                    "aspectRatio": 5,
                    "subdiv": 6,
                    "radialSegments": 20,
                    "tension": null,
                    "capped": true,
                    "smoothSheet": false,
                    "radiusType": "sstruc",
                    "radiusData": {},
                    "radiusSize": 1,
                    "radiusScale": 0.7,
                    "assembly": "default",
                    "defaultAssembly": "",
                    "clipNear": 0,
                    "clipRadius": 0,
                    "clipCenter": {
                      "x": 0,
                      "y": 0,
                      "z": 0
                    },
                    "flatShaded": false,
                    "opacity": 1,
                    "depthWrite": true,
                    "side": "double",
                    "wireframe": false,
                    "colorScheme": "residueindex",
                    "colorScale": "Spectral",
                    "colorReverse": true,
                    "colorValue": 9474192,
                    "colorMode": "hcl",
                    "roughness": 0.4,
                    "metalness": 0,
                    "diffuse": 16777215,
                    "diffuseInterior": false,
                    "useInteriorColor": true,
                    "interiorColor": 2236962,
                    "interiorDarkening": 0,
                    "matrix": {
                      "elements": [
                        1,
                        0,
                        0,
                        0,
                        0,
                        1,
                        0,
                        0,
                        0,
                        0,
                        1,
                        0,
                        0,
                        0,
                        0,
                        1
                      ]
                    },
                    "disablePicking": false,
                    "sele": ""
                  }
                },
                "1": {
                  "type": "base",
                  "params": {
                    "lazy": false,
                    "visible": true,
                    "quality": "high",
                    "sphereDetail": 2,
                    "radialSegments": 20,
                    "openEnded": true,
                    "disableImpostor": false,
                    "aspectRatio": 1,
                    "lineOnly": false,
                    "cylinderOnly": false,
                    "bondScale": 0.4,
                    "linewidth": 2,
                    "radiusType": "size",
                    "radiusData": {},
                    "radiusSize": 0.3,
                    "radiusScale": 1,
                    "assembly": "default",
                    "defaultAssembly": "",
                    "clipNear": 0,
                    "clipRadius": 0,
                    "clipCenter": {
                      "x": 0,
                      "y": 0,
                      "z": 0
                    },
                    "flatShaded": false,
                    "opacity": 1,
                    "depthWrite": true,
                    "side": "double",
                    "wireframe": false,
                    "colorScheme": "residueindex",
                    "colorScale": "Spectral",
                    "colorReverse": true,
                    "colorValue": 9474192,
                    "colorMode": "hcl",
                    "roughness": 0.4,
                    "metalness": 0,
                    "diffuse": 16777215,
                    "diffuseInterior": false,
                    "useInteriorColor": true,
                    "interiorColor": 2236962,
                    "interiorDarkening": 0,
                    "matrix": {
                      "elements": [
                        1,
                        0,
                        0,
                        0,
                        0,
                        1,
                        0,
                        0,
                        0,
                        0,
                        1,
                        0,
                        0,
                        0,
                        0,
                        1
                      ]
                    },
                    "disablePicking": false,
                    "sele": ""
                  }
                },
                "2": {
                  "type": "ball+stick",
                  "params": {
                    "lazy": false,
                    "visible": true,
                    "quality": "high",
                    "sphereDetail": 2,
                    "radialSegments": 20,
                    "openEnded": true,
                    "disableImpostor": false,
                    "aspectRatio": 1.5,
                    "lineOnly": false,
                    "cylinderOnly": false,
                    "multipleBond": "off",
                    "bondScale": 0.3,
                    "bondSpacing": 0.75,
                    "linewidth": 2,
                    "radiusType": "size",
                    "radiusData": {},
                    "radiusSize": 0.15,
                    "radiusScale": 2,
                    "assembly": "default",
                    "defaultAssembly": "",
                    "clipNear": 0,
                    "clipRadius": 0,
                    "clipCenter": {
                      "x": 0,
                      "y": 0,
                      "z": 0
                    },
                    "flatShaded": false,
                    "opacity": 1,
                    "depthWrite": true,
                    "side": "double",
                    "wireframe": false,
                    "colorScheme": "element",
                    "colorScale": "",
                    "colorReverse": false,
                    "colorValue": 9474192,
                    "colorMode": "hcl",
                    "roughness": 0.4,
                    "metalness": 0,
                    "diffuse": 16777215,
                    "diffuseInterior": false,
                    "useInteriorColor": true,
                    "interiorColor": 2236962,
                    "interiorDarkening": 0,
                    "matrix": {
                      "elements": [
                        1,
                        0,
                        0,
                        0,
                        0,
                        1,
                        0,
                        0,
                        0,
                        0,
                        1,
                        0,
                        0,
                        0,
                        0,
                        1
                      ]
                    },
                    "disablePicking": false,
                    "sele": "ligand"
                  }
                }
              },
              "3": {
                "0": {
                  "type": "cartoon",
                  "params": {
                    "lazy": false,
                    "visible": true,
                    "quality": "high",
                    "aspectRatio": 5,
                    "subdiv": 6,
                    "radialSegments": 20,
                    "tension": null,
                    "capped": true,
                    "smoothSheet": false,
                    "radiusType": "sstruc",
                    "radiusData": {},
                    "radiusSize": 1,
                    "radiusScale": 0.7,
                    "assembly": "default",
                    "defaultAssembly": "",
                    "clipNear": 0,
                    "clipRadius": 0,
                    "clipCenter": {
                      "x": 0,
                      "y": 0,
                      "z": 0
                    },
                    "flatShaded": false,
                    "opacity": 1,
                    "depthWrite": true,
                    "side": "double",
                    "wireframe": false,
                    "colorScheme": "residueindex",
                    "colorScale": "Spectral",
                    "colorReverse": true,
                    "colorValue": 9474192,
                    "colorMode": "hcl",
                    "roughness": 0.4,
                    "metalness": 0,
                    "diffuse": 16777215,
                    "diffuseInterior": false,
                    "useInteriorColor": true,
                    "interiorColor": 2236962,
                    "interiorDarkening": 0,
                    "matrix": {
                      "elements": [
                        1,
                        0,
                        0,
                        0,
                        0,
                        1,
                        0,
                        0,
                        0,
                        0,
                        1,
                        0,
                        0,
                        0,
                        0,
                        1
                      ]
                    },
                    "disablePicking": false,
                    "sele": ""
                  }
                },
                "1": {
                  "type": "base",
                  "params": {
                    "lazy": false,
                    "visible": true,
                    "quality": "high",
                    "sphereDetail": 2,
                    "radialSegments": 20,
                    "openEnded": true,
                    "disableImpostor": false,
                    "aspectRatio": 1,
                    "lineOnly": false,
                    "cylinderOnly": false,
                    "bondScale": 0.4,
                    "linewidth": 2,
                    "radiusType": "size",
                    "radiusData": {},
                    "radiusSize": 0.3,
                    "radiusScale": 1,
                    "assembly": "default",
                    "defaultAssembly": "",
                    "clipNear": 0,
                    "clipRadius": 0,
                    "clipCenter": {
                      "x": 0,
                      "y": 0,
                      "z": 0
                    },
                    "flatShaded": false,
                    "opacity": 1,
                    "depthWrite": true,
                    "side": "double",
                    "wireframe": false,
                    "colorScheme": "residueindex",
                    "colorScale": "Spectral",
                    "colorReverse": true,
                    "colorValue": 9474192,
                    "colorMode": "hcl",
                    "roughness": 0.4,
                    "metalness": 0,
                    "diffuse": 16777215,
                    "diffuseInterior": false,
                    "useInteriorColor": true,
                    "interiorColor": 2236962,
                    "interiorDarkening": 0,
                    "matrix": {
                      "elements": [
                        1,
                        0,
                        0,
                        0,
                        0,
                        1,
                        0,
                        0,
                        0,
                        0,
                        1,
                        0,
                        0,
                        0,
                        0,
                        1
                      ]
                    },
                    "disablePicking": false,
                    "sele": ""
                  }
                },
                "2": {
                  "type": "ball+stick",
                  "params": {
                    "lazy": false,
                    "visible": true,
                    "quality": "high",
                    "sphereDetail": 2,
                    "radialSegments": 20,
                    "openEnded": true,
                    "disableImpostor": false,
                    "aspectRatio": 1.5,
                    "lineOnly": false,
                    "cylinderOnly": false,
                    "multipleBond": "off",
                    "bondScale": 0.3,
                    "bondSpacing": 0.75,
                    "linewidth": 2,
                    "radiusType": "size",
                    "radiusData": {},
                    "radiusSize": 0.15,
                    "radiusScale": 2,
                    "assembly": "default",
                    "defaultAssembly": "",
                    "clipNear": 0,
                    "clipRadius": 0,
                    "clipCenter": {
                      "x": 0,
                      "y": 0,
                      "z": 0
                    },
                    "flatShaded": false,
                    "opacity": 1,
                    "depthWrite": true,
                    "side": "double",
                    "wireframe": false,
                    "colorScheme": "element",
                    "colorScale": "",
                    "colorReverse": false,
                    "colorValue": 9474192,
                    "colorMode": "hcl",
                    "roughness": 0.4,
                    "metalness": 0,
                    "diffuse": 16777215,
                    "diffuseInterior": false,
                    "useInteriorColor": true,
                    "interiorColor": 2236962,
                    "interiorDarkening": 0,
                    "matrix": {
                      "elements": [
                        1,
                        0,
                        0,
                        0,
                        0,
                        1,
                        0,
                        0,
                        0,
                        0,
                        1,
                        0,
                        0,
                        0,
                        0,
                        1
                      ]
                    },
                    "disablePicking": false,
                    "sele": "ligand"
                  }
                }
              }
            },
            "_ngl_serialize": false,
            "_ngl_version": "2.4.0",
            "_ngl_view_id": [
              "18FBDFCD-8B74-4F2B-8431-97A86312685F"
            ],
            "_player_dict": {},
            "_scene_position": {},
            "_scene_rotation": {},
            "_synced_model_ids": [],
            "_synced_repr_model_ids": [],
            "_view_count": null,
            "_view_height": "",
            "_view_module": "nglview-js-widgets",
            "_view_module_version": "3.1.5",
            "_view_name": "NGLView",
            "_view_width": "",
            "background": "white",
            "frame": 0,
            "gui_style": null,
            "layout": "IPY_MODEL_f8ce6b8f649e4c35bca89fbb573e61da",
            "max_frame": 0,
            "n_components": 4,
            "picked": {},
            "tabbable": null,
            "tooltip": null
          }
        },
        "b7bef3eeb335450292f0b76a43a54dc0": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "2.0.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "2.0.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "2.0.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border_bottom": null,
            "border_left": null,
            "border_right": null,
            "border_top": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "c6370a6952144c73bf7fdc39cf3d8a3b": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "2.0.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "2.0.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "2.0.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "f3712a1a444d4cd293b7565cc305c0c1": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "2.0.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "2.0.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "2.0.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border_bottom": null,
            "border_left": null,
            "border_right": null,
            "border_top": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "3233db53b097403a95ade1daf78e1fbb": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "SliderStyleModel",
          "model_module_version": "2.0.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "2.0.0",
            "_model_name": "SliderStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "2.0.0",
            "_view_name": "StyleView",
            "description_width": "",
            "handle_color": null
          }
        },
        "f8ce6b8f649e4c35bca89fbb573e61da": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "2.0.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "2.0.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "2.0.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border_bottom": null,
            "border_left": null,
            "border_right": null,
            "border_top": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "1516b48783174c179c0072f2e152b2fe": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "ButtonModel",
          "model_module_version": "2.0.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "2.0.0",
            "_model_name": "ButtonModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "2.0.0",
            "_view_name": "ButtonView",
            "button_style": "",
            "description": "",
            "disabled": false,
            "icon": "compress",
            "layout": "IPY_MODEL_a317ad2cea59403383d87adbddafb780",
            "style": "IPY_MODEL_c8d25d4fac8541c1a98a4dde6d1b62af",
            "tabbable": null,
            "tooltip": null
          }
        },
        "d432e843235d47d0b4e4c7344a29ef37": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HBoxModel",
          "model_module_version": "2.0.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "2.0.0",
            "_model_name": "HBoxModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "2.0.0",
            "_view_name": "HBoxView",
            "box_style": "",
            "children": [
              "IPY_MODEL_2f5c908b47c34249a826f542f5adf054",
              "IPY_MODEL_c1b930a01abf44a086ff2fad756f0b94"
            ],
            "layout": "IPY_MODEL_da738cfd54bd4cdaad4fb0c906f29106",
            "tabbable": null,
            "tooltip": null
          }
        },
        "a317ad2cea59403383d87adbddafb780": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "2.0.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "2.0.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "2.0.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border_bottom": null,
            "border_left": null,
            "border_right": null,
            "border_top": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": "34px"
          }
        },
        "c8d25d4fac8541c1a98a4dde6d1b62af": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "ButtonStyleModel",
          "model_module_version": "2.0.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "2.0.0",
            "_model_name": "ButtonStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "2.0.0",
            "_view_name": "StyleView",
            "button_color": null,
            "font_family": null,
            "font_size": null,
            "font_style": null,
            "font_variant": null,
            "font_weight": null,
            "text_color": null,
            "text_decoration": null
          }
        },
        "da738cfd54bd4cdaad4fb0c906f29106": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "2.0.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "2.0.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "2.0.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border_bottom": null,
            "border_left": null,
            "border_right": null,
            "border_top": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        }
      }
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/bordin89/ML4NGP_tutorials/blob/main/ml4ngp_2_profam.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "##  Generating novel sequences and structures with ProFam-1\n",
        "\n",
        "This notebook presents an end-to-end workflow for **protein sequence generation and structural validation** using **ProFam-1**, a family-aware generative model trained on protein domain and family information. The goal of the workshop is to guide you from an input protein sequence, through controlled generative modelling, to quantitative and visual structural comparison against a reference structure.\n",
        "\n",
        "The notebook is designed to be fully reproducible in a Google Colab environment with GPU support and does not require Flash Attention. All steps are executed interactively, allowing participants to explore how generative parameters affect sequence quality and downstream structural fidelity.\n",
        "\n",
        "We begin by setting up the computational environment and installing the required dependencies. The ProFam repository is cloned, and a pretrained ProFam-1 checkpoint is located and prepared for inference. Care is taken to ensure compatibility with standard Colab GPU instances, avoiding optional components that are not universally available.\n",
        "\n",
        "Next, an input protein sequence is provided. By default, the notebook uses a known PETase sequence from UniProt as a single-sequence prompt, but participants may alternatively paste their own sequence directly into the notebook. Any extraneous whitespace or line breaks are removed automatically, and the sequence is written to FASTA format for downstream use.\n",
        "\n",
        "Using ProFam-1, one or more novel protein sequences are then generated, conditioned on the input sequence. The generation step exposes a small number of adjustable parameters, such as the number of sequences to generate and the sampling temperature, while keeping most model settings fixed to sensible defaults. Additional quality filters are applied to enforce reasonable sequence lengths and minimum similarity to the prompt, helping to avoid pathological or truncated outputs.\n",
        "\n",
        "Structure prediction is performed outside the notebook using a tool such as AlphaFold or ESMFold. Once predicted, the structures of both the generated sequence and the original input sequence are uploaded back into the notebook in either PDB or mmCIF format. These structures form the basis for structural validation.\n",
        "\n",
        "Structural comparison is carried out using **TM-align**, which provides quantitative measures of similarity including TM-score, RMSD, and aligned length. The TM-score is particularly informative: values above 0.5 generally indicate a shared fold, while values above 0.8 suggest strong structural agreement. These metrics allow an objective assessment of whether the generated sequence preserves the overall fold of the reference protein.\n",
        "\n",
        "Finally, the two structures are superposed and visualised together using **NGLView** in cartoon (ribbon/strand) representation. The generated structure is rigidly transformed onto the reference structure using the rotation matrix produced by TM-align, allowing direct visual inspection of conserved cores, flexible regions, and any structural deviations. This final step helps connect quantitative scores to intuitive, three-dimensional understanding.\n",
        "\n",
        "By the end of this notebook, you will have walked through a complete generative protein design loop, from sequence generation to structure-based validation, and gained practical insight into how modern protein language models can be evaluated using structural biology tools."
      ],
      "metadata": {
        "id": "jtC3R7aciTht"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "#@title 1) GPU sanity check\n",
        "\n",
        "import torch, subprocess\n",
        "\n",
        "print(\"Torch version:\", torch.__version__)\n",
        "print(\"CUDA available:\", torch.cuda.is_available())\n",
        "\n",
        "if torch.cuda.is_available():\n",
        "    print(\"GPU:\", torch.cuda.get_device_name(0))\n",
        "    print(\"CUDA version (torch):\", torch.version.cuda)\n",
        "\n",
        "# Optional: show nvidia-smi if available\n",
        "try:\n",
        "    out = subprocess.check_output([\"nvidia-smi\"], text=True)\n",
        "    print(\"\\n\" + out[:1500])\n",
        "except Exception:\n",
        "    print(\"nvidia-smi not available\")"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "QtenLPbjgB7d",
        "outputId": "14fca507-b1fa-4dd7-f81c-10176a2931fb"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Torch version: 2.9.0+cpu\n",
            "CUDA available: False\n",
            "nvidia-smi not available\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#@title 2) Clone ProFam\n",
        "\n",
        "import os, pathlib\n",
        "\n",
        "BASE_DIR = pathlib.Path(\"/content\").resolve()\n",
        "WORK_DIR = BASE_DIR / \"profam_workshop\"\n",
        "WORK_DIR.mkdir(exist_ok=True)\n",
        "\n",
        "os.chdir(WORK_DIR)\n",
        "print(\"Working directory:\", WORK_DIR)\n",
        "\n",
        "if not (WORK_DIR / \"profam\").exists():\n",
        "    !git clone https://github.com/alex-hh/profam.git\n",
        "else:\n",
        "    print(\"ProFam repository already present\")\n",
        "\n",
        "%cd {WORK_DIR / \"profam\"}\n",
        "!git rev-parse --short HEAD"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "ddgjQqXPgocQ",
        "outputId": "22a0c892-ab6a-4503-af28-023e197cf836"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Working directory: /content/profam_workshop\n",
            "ProFam repository already present\n",
            "/content/profam_workshop/profam\n",
            "4f89b22\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#@title 3) Install ProFam (GPU runtime, no flash-attn, Colab-friendly)\n",
        "\n",
        "import re, pathlib, os\n",
        "\n",
        "repo = pathlib.Path(\"/content/profam_workshop/profam\").resolve()\n",
        "os.chdir(repo)\n",
        "print(\"Repository:\", repo)\n",
        "\n",
        "# Ensure up-to-date build tools\n",
        "!pip -q install -U pip setuptools wheel\n",
        "\n",
        "req_in = repo / \"requirements.txt\"\n",
        "if not req_in.exists():\n",
        "    raise FileNotFoundError(f\"Missing {req_in}\")\n",
        "\n",
        "# Packages we deliberately do not override in Colab\n",
        "SKIP_PREFIXES = (\n",
        "    \"flash-attn\", \"flash_attn\",\n",
        "    \"numpy\", \"pandas\", \"requests\",\n",
        "    \"torch\", \"torchvision\", \"torchaudio\",\n",
        "    \"jax\", \"jaxlib\",\n",
        ")\n",
        "\n",
        "filtered = []\n",
        "removed = []\n",
        "\n",
        "for ln in req_in.read_text().splitlines():\n",
        "    s = ln.strip()\n",
        "    if not s or s.startswith(\"#\"):\n",
        "        filtered.append(ln)\n",
        "        continue\n",
        "\n",
        "    head = re.split(r\"[<>=!\\s\\[]\", s, maxsplit=1)[0].lower()\n",
        "\n",
        "    if any(head.startswith(p) for p in SKIP_PREFIXES):\n",
        "        removed.append(ln)\n",
        "        continue\n",
        "\n",
        "    filtered.append(ln)\n",
        "\n",
        "req_out = repo / \"requirements.colab_noflash.txt\"\n",
        "req_out.write_text(\"\\n\".join(filtered) + \"\\n\")\n",
        "\n",
        "print(f\"Wrote {req_out.name}\")\n",
        "\n",
        "if removed:\n",
        "    print(\"Skipped packages:\")\n",
        "    for r in removed:\n",
        "        print(\" \", r)\n",
        "\n",
        "# Install filtered dependencies\n",
        "!pip -q install -r requirements.colab_noflash.txt\n",
        "\n",
        "# Patch common Colab/runtime dependencies\n",
        "!pip -q install \\\n",
        "    \"pandas==2.2.2\" \\\n",
        "    \"requests==2.32.4\" \\\n",
        "    \"packaging>=24.2\" \\\n",
        "    \"typing-extensions>=4.12.0\" \\\n",
        "    \"xxhash>=3.5.0\" \\\n",
        "    \"jedi>=0.16\"\n",
        "\n",
        "# Install ProFam in editable mode\n",
        "!pip -q install -e .\n",
        "\n",
        "print(\"ProFam installation complete.\")"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "OgVyS5BKgteH",
        "outputId": "51b1d379-c682-493d-f3d2-081aea2242c6"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Repository: /content/profam_workshop/profam\n",
            "Wrote requirements.colab_noflash.txt\n",
            "Skipped packages:\n",
            "  numpy==1.26.4\n",
            "  pandas==2.3.3\n",
            "  requests==2.32.2\n",
            "  torch==2.6.0\n",
            "  torchvision==0.21.0\n",
            "  torchaudio==2.6.0\n",
            "  torchmetrics==1.8.2\n",
            "\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n",
            "virtualenv 20.36.0 requires filelock<4,>=3.20.1; python_version >= \"3.10\", but you have filelock 3.14.0 which is incompatible.\n",
            "langgraph 1.0.5 requires xxhash>=3.5.0, but you have xxhash 3.4.1 which is incompatible.\n",
            "torchvision 0.24.0+cpu requires torch==2.9.0, but you have torch 2.6.0 which is incompatible.\n",
            "db-dtypes 1.4.4 requires packaging>=24.2.0, but you have packaging 24.0 which is incompatible.\n",
            "pytensor 2.35.1 requires filelock>=3.15, but you have filelock 3.14.0 which is incompatible.\n",
            "grpcio 1.76.0 requires typing-extensions~=4.12, but you have typing-extensions 4.11.0 which is incompatible.\n",
            "diffusers 0.36.0 requires huggingface-hub<2.0,>=0.34.0, but you have huggingface-hub 0.24.6 which is incompatible.\n",
            "gradio 5.50.0 requires huggingface-hub<2.0,>=0.33.5, but you have huggingface-hub 0.24.6 which is incompatible.\n",
            "alembic 1.17.2 requires typing-extensions>=4.12, but you have typing-extensions 4.11.0 which is incompatible.\n",
            "typeguard 4.4.4 requires typing_extensions>=4.14.0, but you have typing-extensions 4.11.0 which is incompatible.\n",
            "peft 0.18.0 requires huggingface_hub>=0.25.0, but you have huggingface-hub 0.24.6 which is incompatible.\n",
            "xarray 2025.12.0 requires packaging>=24.1, but you have packaging 24.0 which is incompatible.\n",
            "torchaudio 2.9.0+cpu requires torch==2.9.0, but you have torch 2.6.0 which is incompatible.\n",
            "google-cloud-bigquery 3.38.0 requires packaging>=24.2.0, but you have packaging 24.0 which is incompatible.\n",
            "mcp 1.24.0 requires pydantic<3.0.0,>=2.11.0, but you have pydantic 2.9.2 which is incompatible.\n",
            "typing-inspection 0.4.2 requires typing-extensions>=4.12.0, but you have typing-extensions 4.11.0 which is incompatible.\u001b[0m\u001b[31m\n",
            "\u001b[0m\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n",
            "lightning 2.4.0 requires packaging<25.0,>=20.0, but you have packaging 25.0 which is incompatible.\n",
            "torchvision 0.24.0+cpu requires torch==2.9.0, but you have torch 2.6.0 which is incompatible.\n",
            "diffusers 0.36.0 requires huggingface-hub<2.0,>=0.34.0, but you have huggingface-hub 0.24.6 which is incompatible.\n",
            "gradio 5.50.0 requires huggingface-hub<2.0,>=0.33.5, but you have huggingface-hub 0.24.6 which is incompatible.\n",
            "peft 0.18.0 requires huggingface_hub>=0.25.0, but you have huggingface-hub 0.24.6 which is incompatible.\n",
            "torchaudio 2.9.0+cpu requires torch==2.9.0, but you have torch 2.6.0 which is incompatible.\n",
            "mcp 1.24.0 requires pydantic<3.0.0,>=2.11.0, but you have pydantic 2.9.2 which is incompatible.\u001b[0m\u001b[31m\n",
            "\u001b[0m  Installing build dependencies ... \u001b[?25l\u001b[?25hdone\n",
            "  Checking if build backend supports build_editable ... \u001b[?25l\u001b[?25hdone\n",
            "  Getting requirements to build editable ... \u001b[?25l\u001b[?25hdone\n",
            "  Preparing editable metadata (pyproject.toml) ... \u001b[?25l\u001b[?25hdone\n",
            "  Building editable for profam (pyproject.toml) ... \u001b[?25l\u001b[?25hdone\n",
            "\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n",
            "torchvision 0.24.0+cpu requires torch==2.9.0, but you have torch 2.6.0 which is incompatible.\n",
            "diffusers 0.36.0 requires huggingface-hub<2.0,>=0.34.0, but you have huggingface-hub 0.24.6 which is incompatible.\n",
            "gradio 5.50.0 requires huggingface-hub<2.0,>=0.33.5, but you have huggingface-hub 0.24.6 which is incompatible.\n",
            "peft 0.18.0 requires huggingface_hub>=0.25.0, but you have huggingface-hub 0.24.6 which is incompatible.\n",
            "torchaudio 2.9.0+cpu requires torch==2.9.0, but you have torch 2.6.0 which is incompatible.\u001b[0m\u001b[31m\n",
            "\u001b[0mProFam installation complete.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#@title 4) Download ProFam model checkpoint from Hugging Face\n",
        "\n",
        "import os, pathlib\n",
        "\n",
        "repo = pathlib.Path(\"/content/profam_workshop/profam\").resolve()\n",
        "os.chdir(repo)\n",
        "\n",
        "os.environ[\"HF_HOME\"] = \"/content/hf_cache\"\n",
        "pathlib.Path(os.environ[\"HF_HOME\"]).mkdir(exist_ok=True, parents=True)\n",
        "\n",
        "!python scripts/hf_download_checkpoint.py\n",
        "\n",
        "print(\"Checkpoint download complete.\")"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "_AE6Ezw9rIHX",
        "outputId": "253293d4-0ab8-44fb-9bb3-d073d7071c88"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\rFetching 5 files:   0% 0/5 [00:00<?, ?it/s]/usr/local/lib/python3.12/dist-packages/huggingface_hub/file_download.py:1212: UserWarning: `local_dir_use_symlinks` parameter is deprecated and will be ignored. The process to download files to a local folder has been updated and do not rely on symlinks anymore. You only need to pass a destination folder as`local_dir`.\n",
            "For more details, check out https://huggingface.co/docs/huggingface_hub/main/en/guides/download#download-files-to-local-folder.\n",
            "  warnings.warn(\n",
            "\rFetching 5 files: 100% 5/5 [00:00<00:00, 2384.48it/s]\n",
            "Downloaded to: /content/profam_workshop/profam/model_checkpoints/profam-1\n",
            "Checkpoint download complete.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#@title 5) Fix Colab torch/torchvision mismatch (remove torchvision)\n",
        "\n",
        "# Remove torchvision to prevent torchmetrics/lightning import issues\n",
        "!pip -q uninstall -y torchvision || true\n",
        "\n",
        "print(\"torchvision removed.\")"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "wRl_-dT7xpC4",
        "outputId": "613aafae-f3a4-4928-b1a9-cc3103b20dbb"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "torchvision removed.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#@title 6) Show ProFam CLI help (score + generate)\n",
        "\n",
        "import pathlib, os\n",
        "\n",
        "repo = pathlib.Path(\"/content/profam_workshop/profam\").resolve()\n",
        "os.chdir(repo)\n",
        "\n",
        "print(\"=== score_sequences.py ===\")\n",
        "!python scripts/score_sequences.py -h | head -n 120\n",
        "\n",
        "print(\"\\n=== generate_sequences.py ===\")\n",
        "!python scripts/generate_sequences.py -h | head -n 120"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "ZkCDYcH-u5Vt",
        "outputId": "155f8356-df54-4923-c6e8-b3376480c2cc"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "=== score_sequences.py ===\n",
            "/usr/local/lib/python3.12/dist-packages/lightning_utilities/core/imports.py:14: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
            "  import pkg_resources\n",
            "2026-01-08 17:28:21.525886: E external/local_xla/xla/stream_executor/cuda/cuda_fft.cc:467] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered\n",
            "WARNING: All log messages before absl::InitializeLog() is called are written to STDERR\n",
            "E0000 00:00:1767893301.942237    2638 cuda_dnn.cc:8579] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered\n",
            "E0000 00:00:1767893302.056363    2638 cuda_blas.cc:1407] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered\n",
            "W0000 00:00:1767893302.884201    2638 computation_placer.cc:177] computation placer already registered. Please check linkage and avoid linking the same target more than once.\n",
            "W0000 00:00:1767893302.884339    2638 computation_placer.cc:177] computation placer already registered. Please check linkage and avoid linking the same target more than once.\n",
            "W0000 00:00:1767893302.884348    2638 computation_placer.cc:177] computation placer already registered. Please check linkage and avoid linking the same target more than once.\n",
            "W0000 00:00:1767893302.884357    2638 computation_placer.cc:177] computation placer already registered. Please check linkage and avoid linking the same target more than once.\n",
            "2026-01-08 17:28:22.965781: I tensorflow/core/platform/cpu_feature_guard.cc:210] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.\n",
            "To enable the following instructions: AVX2 FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.\n",
            "usage: score_sequences.py [-h] [--checkpoint_dir CHECKPOINT_DIR]\n",
            "                          [--conditioning_fasta CONDITIONING_FASTA]\n",
            "                          [--candidates_file CANDIDATES_FILE]\n",
            "                          [--save_dir SAVE_DIR] [--device DEVICE]\n",
            "                          [--dtype {float32,float16,bfloat16}] [--seed SEED]\n",
            "                          [--max_tokens MAX_TOKENS]\n",
            "                          [--scoring_max_tokens SCORING_MAX_TOKENS]\n",
            "                          [--ensemble_number ENSEMBLE_NUMBER]\n",
            "                          [--use_diversity_weights | --no-use_diversity_weights]\n",
            "                          [--diversity_theta DIVERSITY_THETA]\n",
            "                          [--recompute_diversity_weights | --no-recompute_diversity_weights]\n",
            "                          [--attn_implementation {sdpa,flash_attention_2,eager}]\n",
            "\n",
            "Compute conditional likelihoods of candidate sequences given conditioning\n",
            "sequences\n",
            "\n",
            "options:\n",
            "  -h, --help            show this help message and exit\n",
            "  --checkpoint_dir CHECKPOINT_DIR\n",
            "                        Checkpoint run directory (contains\n",
            "                        checkpoints/last.ckpt)\n",
            "  --conditioning_fasta CONDITIONING_FASTA\n",
            "                        Path to conditioning FASTA/MSA file\n",
            "  --candidates_file CANDIDATES_FILE\n",
            "                        Path to candidate sequences FASTA file or csv file\n",
            "                        with columns: 'mutated_sequence', and optionally\n",
            "                        'DMS_score'\n",
            "  --save_dir SAVE_DIR   Directory to save output files\n",
            "  --device DEVICE\n",
            "  --dtype {float32,float16,bfloat16}\n",
            "  --seed SEED\n",
            "  --max_tokens MAX_TOKENS\n",
            "                        Token budget (prompt+completion) used for batch size\n",
            "                        heuristics\n",
            "  --scoring_max_tokens SCORING_MAX_TOKENS\n",
            "                        Token budget used ONLY to dynamically set the scoring\n",
            "                        batch size to stay within memory constraints. This is\n",
            "                        typically higher than --max_tokens.\n",
            "  --ensemble_number ENSEMBLE_NUMBER\n",
            "                        Number of prompts used to generate the ensemble score\n",
            "  --use_diversity_weights, --no-use_diversity_weights\n",
            "                        If set, sample conditioning sequences with homology-\n",
            "                        based diversity weights (1/#neighbors).\n",
            "  --diversity_theta DIVERSITY_THETA\n",
            "                        Theta used for homology neighbor definition when\n",
            "                        computing diversity weights.\n",
            "  --recompute_diversity_weights, --no-recompute_diversity_weights\n",
            "                        If set, ignore any on-disk cached weights and\n",
            "                        recompute.\n",
            "  --attn_implementation {sdpa,flash_attention_2,eager}\n",
            "                        Override attention implementation before model init\n",
            "                        (e.g. flash_attention_2)\n",
            "\n",
            "=== generate_sequences.py ===\n",
            "/usr/local/lib/python3.12/dist-packages/lightning_utilities/core/imports.py:14: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
            "  import pkg_resources\n",
            "2026-01-08 17:28:45.180279: E external/local_xla/xla/stream_executor/cuda/cuda_fft.cc:467] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered\n",
            "WARNING: All log messages before absl::InitializeLog() is called are written to STDERR\n",
            "E0000 00:00:1767893325.204518    2765 cuda_dnn.cc:8579] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered\n",
            "E0000 00:00:1767893325.211626    2765 cuda_blas.cc:1407] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered\n",
            "W0000 00:00:1767893325.230531    2765 computation_placer.cc:177] computation placer already registered. Please check linkage and avoid linking the same target more than once.\n",
            "W0000 00:00:1767893325.230590    2765 computation_placer.cc:177] computation placer already registered. Please check linkage and avoid linking the same target more than once.\n",
            "W0000 00:00:1767893325.230598    2765 computation_placer.cc:177] computation placer already registered. Please check linkage and avoid linking the same target more than once.\n",
            "W0000 00:00:1767893325.230605    2765 computation_placer.cc:177] computation placer already registered. Please check linkage and avoid linking the same target more than once.\n",
            "2026-01-08 17:28:45.236065: I tensorflow/core/platform/cpu_feature_guard.cc:210] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.\n",
            "To enable the following instructions: AVX2 FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.\n",
            "usage: generate_sequences.py [-h] [--checkpoint_dir CHECKPOINT_DIR]\n",
            "                             [--file_path FILE_PATH] [--save_dir SAVE_DIR]\n",
            "                             [--sampler {ensemble,single}]\n",
            "                             [--num_prompts_in_ensemble NUM_PROMPTS_IN_ENSEMBLE]\n",
            "                             [--num_samples NUM_SAMPLES]\n",
            "                             [--max_tokens MAX_TOKENS]\n",
            "                             [--max_generated_length MAX_GENERATED_LENGTH]\n",
            "                             [--temperature TEMPERATURE] [--top_p TOP_P]\n",
            "                             [--reduction {mean_probs,sum_log_probs}]\n",
            "                             [--device DEVICE]\n",
            "                             [--dtype {float32,float16,bfloat16}]\n",
            "                             [--continuous_sampling]\n",
            "                             [--max_sequence_length_multiplier MAX_SEQUENCE_LENGTH_MULTIPLIER]\n",
            "                             [--minimum_sequence_length_proportion MINIMUM_SEQUENCE_LENGTH_PROPORTION]\n",
            "                             [--minimum_sequence_identity MINIMUM_SEQUENCE_IDENTITY]\n",
            "                             [--maximum_retries MAXIMUM_RETRIES]\n",
            "                             [--task_index TASK_INDEX] [--num_tasks NUM_TASKS]\n",
            "                             [--disable_repeat_guard] [--seed SEED]\n",
            "                             [--attn_implementation {sdpa,flash_attention_2,eager}]\n",
            "\n",
            "Debug ensemble decoder sampling\n",
            "\n",
            "options:\n",
            "  -h, --help            show this help message and exit\n",
            "  --checkpoint_dir CHECKPOINT_DIR\n",
            "                        Checkpoint run directory (contains .hydra)\n",
            "  --file_path FILE_PATH\n",
            "                        Filepath for input FASTA/MSA files (e.g.\n",
            "                        'data/4_1_1_39_cluster.filtered.fasta' or\n",
            "                        '../data/val/*.a3m')\n",
            "  --save_dir SAVE_DIR   Directory to save generated FASTA files\n",
            "  --sampler {ensemble,single}\n",
            "                        Sampler type: ensemble or single\n",
            "  --num_prompts_in_ensemble NUM_PROMPTS_IN_ENSEMBLE\n",
            "  --num_samples NUM_SAMPLES\n",
            "  --max_tokens MAX_TOKENS\n",
            "  --max_generated_length MAX_GENERATED_LENGTH\n",
            "  --temperature TEMPERATURE\n",
            "  --top_p TOP_P         Nucleus sampling probability mass (0<p<=1)\n",
            "  --reduction {mean_probs,sum_log_probs}\n",
            "  --device DEVICE\n",
            "  --dtype {float32,float16,bfloat16}\n",
            "  --continuous_sampling\n",
            "                        Ignore [SEP] EOS and generate until token budget; drop\n",
            "                        final partial segment\n",
            "  --max_sequence_length_multiplier MAX_SEQUENCE_LENGTH_MULTIPLIER\n",
            "                        Limits the maximum generated length to be no longer\n",
            "                        than this factor longer than the longest sequence in\n",
            "                        the prompt\n",
            "  --minimum_sequence_length_proportion MINIMUM_SEQUENCE_LENGTH_PROPORTION\n",
            "                        Discard sequences that end with length < this\n",
            "                        proportion times the minimum sequence length in the\n",
            "                        prompt\n",
            "  --minimum_sequence_identity MINIMUM_SEQUENCE_IDENTITY\n",
            "                        Discard sequences that have an aligned identity\n",
            "                        fraction less than this quantity\n",
            "  --maximum_retries MAXIMUM_RETRIES\n",
            "                        If a sequence is aborted by filters, retry up to this\n",
            "                        many times before returning the last attempt\n",
            "  --task_index TASK_INDEX\n",
            "                        Task index\n",
            "  --num_tasks NUM_TASKS\n",
            "                        Number of tasks\n",
            "  --disable_repeat_guard\n",
            "                        Disable repeat guard: repeat guard aborts and retries\n",
            "                        the sequence generation if it detects too many repeats\n",
            "  --seed SEED           Random seed for reproducible sampling\n",
            "  --attn_implementation {sdpa,flash_attention_2,eager}\n",
            "                        Override attention implementation before model init\n",
            "                        (e.g. flash_attention_2)\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#@title 7) Create prompt FASTA (default PETase or user-provided sequence)\n",
        "\n",
        "import pathlib, textwrap, re, os\n",
        "\n",
        "repo = pathlib.Path(\"/content/profam_workshop/profam\").resolve()\n",
        "os.chdir(repo)\n",
        "\n",
        "# -----------------------------------------------------------------------------\n",
        "# DISCLAIMER\n",
        "# -----------------------------------------------------------------------------\n",
        "# ProFam can also take a multi-sequence FASTA or an aligned MSA (e.g. A3M).\n",
        "#\n",
        "# - A single sequence is sufficient and recommended for sequence generation.\n",
        "# - Multiple FASTA entries are treated as independent prompts.\n",
        "# - Providing an MSA can be used to condition generation on a protein family,\n",
        "#   but alignment is NOT required for generation.\n",
        "# - MSAs are primarily important for variant scoring, not for generation.\n",
        "#\n",
        "# For this workshop, we use a single-sequence prompt by default.\n",
        "# -----------------------------------------------------------------------------\n",
        "\n",
        "# =========================\n",
        "# USER CHOICES\n",
        "# =========================\n",
        "USE_DEFAULT = True  #@param {type:\"boolean\"}\n",
        "\n",
        "# If USE_DEFAULT = False, paste either:\n",
        "#  - FASTA (one or more records), OR\n",
        "#  - a raw amino-acid sequence\n",
        "USER_SEQUENCE_TEXT = \"MNKFLALALAVSLSASAAPVPSQAFGDLGKDTVAV GDSGVPVSPQTDPSATVGRRLTAAALDALDAGADVV VPGSAGTFSVTLGATNATVVGVDLQLAGADATVTLA AGATGNSGGYVVWGGHGTQATQVVAGLPQLAVAGAD VVIVDNNRAGADVVAVSGGTTSTTTW\"  #@param {type:\"string\"}\n",
        "\n",
        "USER_SEQ_NAME = \"TEST\"  #@param {type:\"string\"}\n",
        "\n",
        "out_fa = repo / \"prompts.fasta\"\n",
        "\n",
        "# =========================\n",
        "# DEFAULT: PETase (Ideonella sakaiensis)\n",
        "# UniProt: QRG82925\n",
        "# =========================\n",
        "DEFAULT_FASTA = textwrap.dedent(\"\"\"\\\n",
        ">PETase_QRG82925\n",
        "MNFPRASRLMQAAVLGGLMAVSAAATAQTNPYARGPNPTAASLEASAGPFTVRSFTVSRP\n",
        "SGYGAGTVYYPTNAGGTVGAIAIVPGYTARQSSIKWWGPRLASHGFVVITIDTNSTLDQP\n",
        "SSRSSQQMAALRQVASLNGTSSSPIYGKVDTARMGVMGWSMGGGGSLISAANNPSLKAAA\n",
        "PQAPWDSSTNFSSVTVPTLIFACENDSIAPVNSSALPIYDSMSRNAKQFLEINGGSHSCA\n",
        "NSGNSNQALIGKKGVAWMKRFMDNDTRYSTFACENPNSTRVSDFRTANCS\n",
        "\"\"\")\n",
        "\n",
        "AA_RE = re.compile(r\"^[ACDEFGHIKLMNPQRSTVWYBXZJUO\\-\\.\\s]+$\", re.IGNORECASE)\n",
        "\n",
        "def normalize_raw_sequence(s: str) -> str:\n",
        "    \"\"\"\n",
        "    Accept raw amino-acid sequences that may contain spaces or newlines.\n",
        "    Strip everything except letters AZ and uppercase.\n",
        "    \"\"\"\n",
        "    s = s.strip().upper()\n",
        "    s = re.sub(r\"\\s+\", \"\", s)\n",
        "    s = re.sub(r\"[^A-Z]\", \"\", s)\n",
        "    return s\n",
        "\n",
        "def to_fasta_from_raw(seq: str, name: str) -> str:\n",
        "    seq = normalize_raw_sequence(seq)\n",
        "    if not seq:\n",
        "        raise ValueError(\"Sequence is empty after cleaning.\")\n",
        "    wrapped = \"\\n\".join(seq[i:i+60] for i in range(0, len(seq), 60))\n",
        "    return f\">{name}\\n{wrapped}\\n\"\n",
        "\n",
        "def is_fasta(txt: str) -> bool:\n",
        "    return txt.lstrip().startswith(\">\")\n",
        "\n",
        "# =========================\n",
        "# WRITE FASTA\n",
        "# =========================\n",
        "if USE_DEFAULT:\n",
        "    out_fa.write_text(DEFAULT_FASTA)\n",
        "    print(\"Using default PETase sequence (UniProt QRG82925)\")\n",
        "else:\n",
        "    txt = (USER_SEQUENCE_TEXT or \"\").strip()\n",
        "    if not txt:\n",
        "        raise ValueError(\"USE_DEFAULT is False but USER_SEQUENCE_TEXT is empty.\")\n",
        "\n",
        "    if is_fasta(txt):\n",
        "        out_fa.write_text(txt.strip() + \"\\n\")\n",
        "        print(\"Using user-provided FASTA (single or multiple sequences)\")\n",
        "    else:\n",
        "        if not AA_RE.match(txt.replace(\"\\n\", \"\")):\n",
        "            raise ValueError(\"Input is neither FASTA nor a valid amino-acid sequence.\")\n",
        "        out_fa.write_text(to_fasta_from_raw(txt, USER_SEQ_NAME))\n",
        "        print(\"Using user-provided raw sequence\")\n",
        "\n",
        "print(\"\\nNote:\")\n",
        "print(\"This FASTA can contain a single sequence, multiple sequences, or an MSA.\")\n",
        "print(\"For generation, alignment is optional; for scoring, MSAs are recommended.\")\n",
        "\n",
        "print(\"\\n--- prompts.fasta ---\")\n",
        "print(out_fa.read_text())"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "wkZQIvrhzZox",
        "outputId": "b18495be-d009-4e65-8c3c-8b50c83c6057"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Using default PETase sequence (UniProt QRG82925)\n",
            "\n",
            "Note:\n",
            "This FASTA can contain a single sequence, multiple sequences, or an MSA.\n",
            "For generation, alignment is optional; for scoring, MSAs are recommended.\n",
            "\n",
            "--- prompts.fasta ---\n",
            ">PETase_QRG82925\n",
            "MNFPRASRLMQAAVLGGLMAVSAAATAQTNPYARGPNPTAASLEASAGPFTVRSFTVSRP\n",
            "SGYGAGTVYYPTNAGGTVGAIAIVPGYTARQSSIKWWGPRLASHGFVVITIDTNSTLDQP\n",
            "SSRSSQQMAALRQVASLNGTSSSPIYGKVDTARMGVMGWSMGGGGSLISAANNPSLKAAA\n",
            "PQAPWDSSTNFSSVTVPTLIFACENDSIAPVNSSALPIYDSMSRNAKQFLEINGGSHSCA\n",
            "NSGNSNQALIGKKGVAWMKRFMDNDTRYSTFACENPNSTRVSDFRTANCS\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#@title 8) Locate ProFam checkpoint_dir\n",
        "\n",
        "import pathlib, os\n",
        "\n",
        "repo = pathlib.Path(\"/content/profam_workshop/profam\").resolve()\n",
        "os.chdir(repo)\n",
        "\n",
        "candidates = []\n",
        "\n",
        "# Look inside the repository\n",
        "for p in pathlib.Path(\".\").rglob(\".hydra\"):\n",
        "    candidates.append(p.parent)\n",
        "\n",
        "# Look in Hugging Face cache\n",
        "hf_home = pathlib.Path(os.environ.get(\"HF_HOME\", \"/content/hf_cache\"))\n",
        "if hf_home.exists():\n",
        "    for p in hf_home.rglob(\".hydra\"):\n",
        "        candidates.append(p.parent)\n",
        "\n",
        "# De-duplicate\n",
        "uniq = []\n",
        "seen = set()\n",
        "for c in candidates:\n",
        "    c = c.resolve()\n",
        "    if c not in seen:\n",
        "        seen.add(c)\n",
        "        uniq.append(c)\n",
        "\n",
        "if not uniq:\n",
        "    raise RuntimeError(\n",
        "        \"Could not find a checkpoint directory containing '.hydra'. \"\n",
        "        \"Run the checkpoint download cell first.\"\n",
        "    )\n",
        "\n",
        "# Select the most recent checkpoint\n",
        "checkpoint_dir = sorted(uniq, key=lambda x: x.stat().st_mtime, reverse=True)[0]\n",
        "print(\"Using checkpoint_dir:\", checkpoint_dir)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "K01hNFNM0Ejb",
        "outputId": "ffe181d3-dd4b-456c-fa2e-8ccb1b24eee5"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Using checkpoint_dir: /content/profam_workshop/profam/model_checkpoints/profam-1\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "# Generate Sequences using ProFam-1\n",
        "\n",
        "In this section, we use **ProFam-1**, a protein familyaware generative model, to generate new protein sequences from a single input sequence.\n",
        "\n",
        "The model conditions on the input sequence and samples novel sequences that are consistent with the learned protein family constraints.\n",
        "\n",
        "**Notes for the workshop:**\n",
        "- The generation runs on **GPU** if available.\n",
        "- **Flash Attention is disabled** for compatibility with Google Colab.\n",
        "- Output sequences will be written as FASTA files in the `generated/` directory.\n",
        "\n"
      ],
      "metadata": {
        "id": "3nTvcavD0P9k"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "> **Runtime & Resource Notice**\n",
        ">\n",
        "> - Sequence generation may take **13 minutes** on a Google Colab GPU.\n",
        "> - The first run may be slightly slower due to **model weight loading**.\n",
        "> - If you see CUDA or cuDNN warnings, these are expected and can be ignored.\n",
        ">\n",
        "> If the notebook appears idle, please wait  the model is still running."
      ],
      "metadata": {
        "id": "K0-MjOQO01-o"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "## Generation Parameters\n",
        "\n",
        "The sequence generation step uses probabilistic sampling. The key parameters are:\n",
        "\n",
        "| Parameter | Meaning |\n",
        "|---------|--------|\n",
        "| `temperature` | Controls randomness. Lower = more conservative, higher = more diverse |\n",
        "| `top_p` | Nucleus sampling threshold (keeps top probability mass) |\n",
        "| `num_samples` | Number of sequences generated per prompt |\n",
        "| `max_tokens` | Maximum total tokens generated |\n",
        "| `sampler` | `ensemble` uses multiple internal prompts for robustness |\n",
        "| `device` | `cuda` uses GPU if available |\n",
        "| `dtype` | `float16` reduces memory usage on GPU |\n",
        "| `attn_implementation` | `sdpa` is used instead of Flash Attention for Colab compatibility |\n",
        "\n",
        "**Tip:**  \n",
        "For more diversity, increase `temperature` (e.g. `1.2`).  \n",
        "For safer, more conservative sequences, decrease it (e.g. `0.8`)."
      ],
      "metadata": {
        "id": "59V1M2PH04E2"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "#@title 9) Install MMseqs2 (required for --minimum_sequence_identity filtering)\n",
        "\n",
        "import os, pathlib\n",
        "\n",
        "WORK_DIR = pathlib.Path(\"/content/profam_workshop\").resolve()\n",
        "mmseqs_dir = WORK_DIR / \"mmseqs\"\n",
        "mmseqs_bin = mmseqs_dir / \"bin\" / \"mmseqs\"\n",
        "\n",
        "# Clean and install\n",
        "!rm -rf \"{mmseqs_dir}\"\n",
        "!mkdir -p \"{mmseqs_dir}\"\n",
        "!wget -q https://mmseqs.com/latest/mmseqs-linux-avx2.tar.gz -O /tmp/mmseqs.tar.gz\n",
        "!tar -xzf /tmp/mmseqs.tar.gz -C \"{mmseqs_dir}\" --strip-components=1\n",
        "!rm /tmp/mmseqs.tar.gz\n",
        "\n",
        "# Add to PATH\n",
        "os.environ[\"PATH\"] = f\"{mmseqs_dir}/bin:\" + os.environ[\"PATH\"]\n",
        "\n",
        "print(\"mmseqs binary:\", mmseqs_bin)\n",
        "!{mmseqs_bin} version"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "mgF_3pAUCdUQ",
        "outputId": "5b10514f-3468-466c-9130-ca12fe309386"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "mmseqs binary: /content/profam_workshop/mmseqs/bin/mmseqs\n",
            "bd01c2229f027d8d8e61947f44d11ef1a7669212\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#@title 10) Device & dtype selection (GPU with safe fallback)\n",
        "\n",
        "import torch\n",
        "\n",
        "# Preferred defaults (workshop intent)\n",
        "PREFERRED_DEVICE = \"cuda\"\n",
        "PREFERRED_DTYPE  = \"float16\"\n",
        "\n",
        "if torch.cuda.is_available():\n",
        "    device = PREFERRED_DEVICE\n",
        "    dtype  = PREFERRED_DTYPE\n",
        "    gpu_name = torch.cuda.get_device_name(0)\n",
        "    print(f\"Using GPU: {gpu_name}\")\n",
        "else:\n",
        "    device = \"cpu\"\n",
        "    dtype  = \"float32\"\n",
        "    print(\n",
        "        \"No NVIDIA GPU detected.\\n\"\n",
        "        \"Falling back to CPU (this will be slower, but functional).\"\n",
        "    )\n",
        "\n",
        "print(f\"device = {device}\")\n",
        "print(f\"dtype  = {dtype}\")"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "sPMylnhyqXsA",
        "outputId": "0d3f5daf-8f24-4067-95dc-d7cde36cc54c"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "No NVIDIA GPU detected.\n",
            "Falling back to CPU (this will be slower, but functional).\n",
            "device = cpu\n",
            "dtype  = float32\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#@title 11) Generate sequences with ProFam-1\n",
        "\n",
        "import pathlib, os, shutil\n",
        "import torch\n",
        "\n",
        "# ---------------- User parameters ----------------\n",
        "num_sequences = 10  #@param {type:\"integer\", min:1, max:50, step:1}\n",
        "temperature   = 0.8  #@param {type:\"number\", min:0.5, max:1.5, step:0.1}\n",
        "\n",
        "max_tokens = 2048  #@param {type:\"integer\", min:256, max:8192, step:256}\n",
        "max_generated_length = 512  #@param {type:\"integer\", min:64, max:4096, step:64}\n",
        "\n",
        "min_seq_len_prop = 0.8   #@param {type:\"number\", min:0.5, max:1.0, step:0.05}\n",
        "max_len_mult     = 1.2   #@param {type:\"number\", min:1.0, max:2.0, step:0.1}\n",
        "min_seq_identity = 0.8   #@param {type:\"number\", min:0.0, max:0.9, step:0.05}\n",
        "\n",
        "# ---------------- Fixed workshop defaults ----------------\n",
        "top_p      = 0.95\n",
        "sampler    = \"ensemble\"\n",
        "attn_impl  = \"sdpa\"   # no flash-attn\n",
        "seed       = 1\n",
        "\n",
        "# ---------------- Auto device / dtype ----------------\n",
        "if torch.cuda.is_available():\n",
        "    device = \"cuda\"\n",
        "    dtype  = \"float16\"\n",
        "    print(\"Using GPU:\", torch.cuda.get_device_name(0))\n",
        "else:\n",
        "    device = \"cpu\"\n",
        "    dtype  = \"float32\"\n",
        "    print(\"No GPU detected; using CPU (slower).\")\n",
        "\n",
        "# ---------------- Paths ----------------\n",
        "repo = pathlib.Path(\"/content/profam_workshop/profam\").resolve()\n",
        "os.chdir(repo)\n",
        "\n",
        "prompt_fa = repo / \"prompts.fasta\"\n",
        "save_dir  = repo / \"generated\"\n",
        "\n",
        "if not prompt_fa.exists():\n",
        "    raise FileNotFoundError(\"prompts.fasta not found. Run the prompt FASTA cell first.\")\n",
        "\n",
        "if \"checkpoint_dir\" not in globals():\n",
        "    raise RuntimeError(\"checkpoint_dir not set. Run the checkpoint discovery cell first.\")\n",
        "\n",
        "# Overwrite previous results\n",
        "if save_dir.exists():\n",
        "    shutil.rmtree(save_dir)\n",
        "save_dir.mkdir(exist_ok=True)\n",
        "\n",
        "# ---------------- Run generation ----------------\n",
        "cmd = (\n",
        "    f\"python scripts/generate_sequences.py \"\n",
        "    f\"--checkpoint_dir '{checkpoint_dir}' \"\n",
        "    f\"--file_path '{prompt_fa}' \"\n",
        "    f\"--save_dir '{save_dir}' \"\n",
        "    f\"--sampler {sampler} \"\n",
        "    f\"--num_prompts_in_ensemble 1 \"\n",
        "    f\"--num_samples {num_sequences} \"\n",
        "    f\"--max_tokens {max_tokens} \"\n",
        "    f\"--max_generated_length {max_generated_length} \"\n",
        "    f\"--temperature {temperature} \"\n",
        "    f\"--top_p {top_p} \"\n",
        "    f\"--device {device} \"\n",
        "    f\"--dtype {dtype} \"\n",
        "    f\"--attn_implementation {attn_impl} \"\n",
        "    f\"--minimum_sequence_length_proportion {min_seq_len_prop} \"\n",
        "    f\"--max_sequence_length_multiplier {max_len_mult} \"\n",
        "    f\"--seed {seed}\"\n",
        ")\n",
        "\n",
        "# Only enable identity filter when mmseqs exists\n",
        "mmseqs_ok = shutil.which(\"mmseqs\") is not None\n",
        "if mmseqs_ok:\n",
        "    cmd += f\" --minimum_sequence_identity {min_seq_identity}\"\n",
        "else:\n",
        "    print(\"mmseqs not found in PATH; skipping --minimum_sequence_identity filtering.\")\n",
        "\n",
        "print(\"\\nCommand:\\n\", cmd, \"\\n\")\n",
        "!bash -lc \"{cmd}\"\n",
        "\n",
        "print(\"\\nGenerated files:\")\n",
        "!ls -lh \"{save_dir}\" | head -n 50"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "VOI6swA61QnR",
        "outputId": "c538e93d-ecac-4917-eccb-55fb03bbb048"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "No GPU detected; using CPU (slower).\n",
            "\n",
            "Command:\n",
            " python scripts/generate_sequences.py --checkpoint_dir '/content/profam_workshop/profam/model_checkpoints/profam-1' --file_path '/content/profam_workshop/profam/prompts.fasta' --save_dir '/content/profam_workshop/profam/generated' --sampler ensemble --num_prompts_in_ensemble 1 --num_samples 10 --max_tokens 2048 --max_generated_length 512 --temperature 0.8 --top_p 0.95 --device cpu --dtype float32 --attn_implementation sdpa --minimum_sequence_length_proportion 0.8 --max_sequence_length_multiplier 1.2 --seed 1 --minimum_sequence_identity 0.8 \n",
            "\n",
            "/usr/local/lib/python3.12/dist-packages/lightning_utilities/core/imports.py:14: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
            "  import pkg_resources\n",
            "2026-01-08 17:32:55.649477: E external/local_xla/xla/stream_executor/cuda/cuda_fft.cc:467] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered\n",
            "WARNING: All log messages before absl::InitializeLog() is called are written to STDERR\n",
            "E0000 00:00:1767893575.675085    3823 cuda_dnn.cc:8579] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered\n",
            "E0000 00:00:1767893575.682350    3823 cuda_blas.cc:1407] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered\n",
            "W0000 00:00:1767893575.701023    3823 computation_placer.cc:177] computation placer already registered. Please check linkage and avoid linking the same target more than once.\n",
            "W0000 00:00:1767893575.701085    3823 computation_placer.cc:177] computation placer already registered. Please check linkage and avoid linking the same target more than once.\n",
            "W0000 00:00:1767893575.701090    3823 computation_placer.cc:177] computation placer already registered. Please check linkage and avoid linking the same target more than once.\n",
            "W0000 00:00:1767893575.701095    3823 computation_placer.cc:177] computation placer already registered. Please check linkage and avoid linking the same target more than once.\n",
            "2026-01-08 17:32:55.706688: I tensorflow/core/platform/cpu_feature_guard.cc:210] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.\n",
            "To enable the following instructions: AVX2 FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.\n",
            "Initialised Llama model, attention implementation:  sdpa\n",
            "/usr/local/lib/python3.12/dist-packages/lightning/pytorch/core/saving.py:195: Found keys that are not in the model state dict but in the checkpoint: ['model.token_embedder.weight']\n",
            "Wrote 10 sequences -> /content/profam_workshop/profam/generated/prompts_generated_ensemble.fasta\n",
            "\n",
            "Generated files:\n",
            "total 4.0K\n",
            "-rw-r--r-- 1 root root 3.3K Jan  8 17:45 prompts_generated_ensemble.fasta\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#@title 12) Preview generated FASTA (first file)\n",
        "\n",
        "import pathlib\n",
        "\n",
        "repo = pathlib.Path(\"/content/profam_workshop/profam\").resolve()\n",
        "save_dir = repo / \"generated\"\n",
        "\n",
        "fasta_files = sorted(save_dir.glob(\"*.fa*\"))\n",
        "if not fasta_files:\n",
        "    raise RuntimeError(\"No FASTA files found in generated/. Run the generation cell first.\")\n",
        "\n",
        "fp = fasta_files[0]\n",
        "print(\"Showing:\", fp, \"\\n\")\n",
        "print(\"\\n\".join(fp.read_text().splitlines()[:120]))"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "gEH2TpMt1sVx",
        "outputId": "952a791b-581d-4812-9c50-b219bcf077d4"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Showing: /content/profam_workshop/profam/generated/prompts_generated_ensemble.fasta \n",
            "\n",
            ">prompts_sample_0_log_likelihood_-0.342\n",
            "MTKYRGLVRLAALGAALVFAAPASAQLPPAGTRSATPVTLQASGGPFTVRTFTVARPAGYGAGTVYYPTNPGGTVGAIAIVPGYTARQSSIKWWGPRLASHGFVVITIDTNSTLDQPSSRSSQQMAALRQVASLNGTSSSPIYGKVDTTRMGVMGWSMGGGGSLISAANNPSLKAAAPQAPWDSSTNFSSVTVPTLIFACENDSIAPVNSSALPIYDSMSRNAKQFLEINGGSHSCANSGNSNQALIGKKGVAWMKRFMDNDTRYSTFACENPNSTRVSDFRTANCS\n",
            ">prompts_sample_1_log_likelihood_-0.377\n",
            "MFSGPLRISLLLVLLACAASLQAQTNPYARPTGPTVAALERASGPFSVTASPLSRPSGFGAGTVYYPSNAGGTVGAIAVVPGYTARQSSIKWWGPRLASHGFVVITIDTNSTLDQPSSRSSQQMAALRQVASLNGTSSSPIYGKVDTARMGVMGWSMGGGGSLISAANNPSLKAAAPQAPWDSSTNFSSVTVPTLIFACENDSIAPVNSSALPIYDSMSRNAKQFLEINGGSHSCANSGNSNQALIGKKGVAWMKRFMDNDTRYSTFACENPNSTRVSDFRTANCS\n",
            ">prompts_sample_2_log_likelihood_-0.900\n",
            "MFDRALIRFSVVGGALALSACGTTDTTGSGSNGTGGGGGGGGGGGSTSSPSGAVPQIVTLSNADAPTGTYTLPTGAVTLNAYLVRNSSGYGGGTIYYPTSGTSGTVGAIAIIPGYTARSSSVAWWGPRLASHGFVVVTVDTNNTFDNPSSRSAQQLAALRAVATLGGTSGSPIYGRVDTTRIGIMGWSMGGGGSLIAAATNPSLRAAAPQAPWDSSTNFTTVSVPTLIFACENDTIAPVNSSALPIYDSMSRNAKQFVEINGGSHSCANSGNSNQALIGAQGVAWMKRFMDNDTRYATFACENPNSTRVSDFRTANCS\n",
            ">prompts_sample_3_log_likelihood_-0.384\n",
            "MLLRALFPFAAAISALTLAGSPVAAASADAAGYARGTGPTAAALQASAGPFDVRAYTVTRPAGYGPGTVYYPTAAGGAVGAIAIVPGYTSRQSSIKWWGPRLASHGFVVITIDTNSTLDQPSSRSSQQMAALRQVASLNGTSSSPIYGKVDTARMGVMGWSMGGGGSLISAANNPSLKAAAPQAPWDSSTNFSSVTVPTLIFACENDSIAPVNSSALPIYDSMSRNAKQFLEINGGSHSCANSGNSNQALIGKKGVAWMKRFMDNDTRYSTFACENPNSTRVSDFRTANCS\n",
            ">prompts_sample_4_log_likelihood_-0.460\n",
            "MTLSNAVKTLLALALIAPALGGAQATEPTPAATSTPVLAPAGPYPVQTRTVSQPSGYGAGTIYFPTNPAGTVGAIAVVPGYTARQSSIKWWGPRLASHGFVVITIDTNSTLDQPASRSSQQMAALRQVAALNGTSSSPIYGKVDTARMGVMGWSMGGGGSLISAANNPSLKAAAPQAPWDSSTNFSSVTVPTLIFACENDSIAPVNSSALPIYDSMSRNAKQFLEINGGSHSCANSGNSNQALIGKKGVAWMKRFMDNDTRYSTFACENPNSTRVSDFRTANCS\n",
            ">prompts_sample_5_log_likelihood_-0.329\n",
            "MAAATVAAPVLAFAQSSPTPAAPAAAGTAPSATTVPAATASAPVTAPAASGPFTVRSFTVSRPSGYGAGTVYYPTNAGGTVGAIAIVPGYTARQSSIKWWGPRLASHGFVVITIDTNSTLDQPSSRSSQQMAALRQVASLNGTSSSPIYGKVDTARMGVMGWSMGGGGSLISAANNPSLKAAAPQAPWDSSTNFSSVTVPTLIFACENDSIAPVNSSALPIYDSMSRNAKQFLEINGGSHSCANSGNSNQALIGKKGVAWMKRFMDNDTRYSTFACENPNSTRVSDFRTANCS\n",
            ">prompts_sample_6_log_likelihood_-0.534\n",
            "MRLFLFTLFVTGSVALAAEAEKPRPGTEKEALTRGFKVRTYDVDRPSGYGGGTVYYPAKRTGTVGAIAIVPGYTARQSSITWWGPRLASHGFVVITIDTNSTLDQPSSRSTQQMEALRQVASLNGTSSAPIYGKVDTARMGVMGWSMGGGGSLISAANNPALKAAAPQAPWDSSTDFSKVTVPTLIFACENDSIAPVNSSALPIYDSMSRNAKQFLEINGGSHSCANSGNSNQALIGKKGVAWMKRFMDNDTRYSTFACENPNSTRVSDFRTANCS\n",
            ">prompts_sample_7_log_likelihood_-0.356\n",
            "MALIGFATGAVAQGLTAPPPLSAEGLDKAVGKLTVRSFTVSRPSGYGSGTVYYPTDASGKVGAIAIVPGYTARQSSIKWWGPRLAAHGFVVVTIDTNSTLDQPSSRSNQQMAALRQVASLNGTSSSPIYGKVDTTRMGVMGWSMGGGGSLISAANNPSLKAAAPQAPWDSSTNFSSVTVPTLIFACENDSIAPVNSSALPIYDSMSRNAKQFLEINGGSHSCANSGNSNQALIGKKGVAWMKRFMDNDTRYSTFACENPNSTRVSDFRTANCS\n",
            ">prompts_sample_8_log_likelihood_-0.214\n",
            "MNRSIFAAATLLSASAAFADQTNPYARGPNPAPATLTASAGPFTVRSFTVNRPSGYGAGTVYYPTNAGGTVGAIAIVPGYTARQSSIKWWGPRLASHGFVVITIDTNSTLDQPSSRSSQQMAALRQVASLNGTSSSPIYGKVDTARMGVMGWSMGGGGSLISAANNPSLKAAAPQAPWDSSTNFSSVTVPTLIFACENDSIAPVNSSALPIYDSMSRNAKQFLEINGGSHSCANSGNSNQALIGKKGVAWMKRFMDNDTRYSTFACENPNSTRVSDFRTANCS\n",
            ">prompts_sample_9_log_likelihood_-0.623\n",
            "MQRTPAVAALLTATAFASTPALAQGSFSGPPVPTPATLTATPGGFTARSFAVTRPAAYGPGTVYYPASGTGPAAAVVIIPGYSGRQSSIKWWGPRLASHGFVVITVDTDSSLDQPSSRSAQQMAALQRVTALNGTSAAPIYGKVDTARLGVMGWSMGGGGSLIAAANNPALKAAAPAAPWDSSANFSNVTVPTLIFACENDSIAPVNSSALPIYDSMSRNAKQFLEINGGSHSCANSGNSNQALIGKKGVAWMKRFMDNDTRYSTFACTNPNSTRVSDFRTANCS\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#@title 13) Rank generated sequences and pick the best candidate\n",
        "\n",
        "import pathlib, re\n",
        "import pandas as pd\n",
        "\n",
        "repo = pathlib.Path(\"/content/profam_workshop/profam\").resolve()\n",
        "prompt_fa = repo / \"prompts.fasta\"\n",
        "gen_dir = repo / \"generated\"\n",
        "out_best = repo / \"best_generated.fasta\"\n",
        "\n",
        "if not prompt_fa.exists():\n",
        "    raise FileNotFoundError(\"prompts.fasta not found.\")\n",
        "if not gen_dir.exists():\n",
        "    raise FileNotFoundError(\"generated/ directory not found. Run generation first.\")\n",
        "\n",
        "AA = set(\"ACDEFGHIKLMNPQRSTVWYBXZJUO\")\n",
        "\n",
        "def read_fasta(path: pathlib.Path):\n",
        "    records = []\n",
        "    name, seq = None, []\n",
        "    for line in path.read_text().splitlines():\n",
        "        line = line.strip()\n",
        "        if not line:\n",
        "            continue\n",
        "        if line.startswith(\">\"):\n",
        "            if name is not None:\n",
        "                records.append((name, \"\".join(seq)))\n",
        "            name = line[1:].strip()\n",
        "            seq = []\n",
        "        else:\n",
        "            seq.append(line)\n",
        "    if name is not None:\n",
        "        records.append((name, \"\".join(seq)))\n",
        "    return records\n",
        "\n",
        "def clean_seq(s: str) -> str:\n",
        "    s = re.sub(r\"\\s+\", \"\", s).upper()\n",
        "    return \"\".join(c for c in s if c.isalpha())\n",
        "\n",
        "def longest_run_fraction(s: str) -> float:\n",
        "    if not s:\n",
        "        return 0.0\n",
        "    best = 1\n",
        "    cur = 1\n",
        "    for i in range(1, len(s)):\n",
        "        if s[i] == s[i - 1]:\n",
        "            cur += 1\n",
        "            if cur > best:\n",
        "                best = cur\n",
        "        else:\n",
        "            cur = 1\n",
        "    return best / len(s)\n",
        "\n",
        "def kmer_low_complexity(s: str, k: int = 3) -> float:\n",
        "    if len(s) < k + 1:\n",
        "        return 0.0\n",
        "    kmers = [s[i:i + k] for i in range(len(s) - k + 1)]\n",
        "    return 1.0 - (len(set(kmers)) / len(kmers))\n",
        "\n",
        "def identity_to_prompt(g: str, p: str) -> float:\n",
        "    n = min(len(g), len(p))\n",
        "    if n == 0:\n",
        "        return 0.0\n",
        "    matches = sum(1 for i in range(n) if g[i] == p[i])\n",
        "    return matches / n\n",
        "\n",
        "prompt_records = read_fasta(prompt_fa)\n",
        "if not prompt_records:\n",
        "    raise RuntimeError(\"No FASTA records found in prompts.fasta.\")\n",
        "prompt_name, prompt_seq = prompt_records[0][0], clean_seq(prompt_records[0][1])\n",
        "\n",
        "rows = []\n",
        "for fp in sorted(gen_dir.glob(\"*.fa*\")):\n",
        "    for name, seq in read_fasta(fp):\n",
        "        seq = clean_seq(seq)\n",
        "        if not seq:\n",
        "            continue\n",
        "        if any(c not in AA for c in seq):\n",
        "            continue\n",
        "\n",
        "        rows.append(\n",
        "            {\n",
        "                \"file\": fp.name,\n",
        "                \"id\": name,\n",
        "                \"length\": len(seq),\n",
        "                \"len_ratio_to_prompt\": len(seq) / max(1, len(prompt_seq)),\n",
        "                \"prefix_identity_to_prompt\": identity_to_prompt(seq, prompt_seq),\n",
        "                \"longest_run_frac\": longest_run_fraction(seq),\n",
        "                \"kmer_low_complexity\": kmer_low_complexity(seq, k=3),\n",
        "                \"seq\": seq,\n",
        "            }\n",
        "        )\n",
        "\n",
        "if not rows:\n",
        "    raise RuntimeError(\"No generated sequences found/parsed in generated/.\")\n",
        "\n",
        "df = pd.DataFrame(rows)\n",
        "\n",
        "target_min_id = 0.20\n",
        "target_max_id = 0.60\n",
        "\n",
        "def rank_score(r):\n",
        "    len_pen = abs(r[\"len_ratio_to_prompt\"] - 1.0)\n",
        "\n",
        "    idv = r[\"prefix_identity_to_prompt\"]\n",
        "    if idv < target_min_id:\n",
        "        id_pen = (target_min_id - idv) * 2.0\n",
        "    elif idv > target_max_id:\n",
        "        id_pen = (idv - target_max_id) * 2.0\n",
        "    else:\n",
        "        id_pen = 0.0\n",
        "\n",
        "    rep_pen = r[\"kmer_low_complexity\"] * 1.5 + r[\"longest_run_frac\"] * 2.0\n",
        "\n",
        "    return len_pen + id_pen + rep_pen\n",
        "\n",
        "df[\"rank_score\"] = df.apply(rank_score, axis=1)\n",
        "df_sorted = df.sort_values(\"rank_score\", ascending=True).reset_index(drop=True)\n",
        "\n",
        "display_cols = [\n",
        "    \"file\",\n",
        "    \"id\",\n",
        "    \"length\",\n",
        "    \"len_ratio_to_prompt\",\n",
        "    \"prefix_identity_to_prompt\",\n",
        "    \"kmer_low_complexity\",\n",
        "    \"longest_run_frac\",\n",
        "    \"rank_score\",\n",
        "]\n",
        "display(df_sorted[display_cols].head(15))\n",
        "\n",
        "best = df_sorted.iloc[0]\n",
        "best_id = best[\"id\"]\n",
        "best_seq = best[\"seq\"]\n",
        "\n",
        "wrapped = \"\\n\".join(best_seq[i:i + 60] for i in range(0, len(best_seq), 60))\n",
        "out_best.write_text(f\">{best_id}\\n{wrapped}\\n\")\n",
        "\n",
        "print(\"Recommended sequence:\")\n",
        "print(\"  ID:\", best_id)\n",
        "print(\"  Length:\", len(best_seq))\n",
        "print(\"  Score:\", float(best[\"rank_score\"]))\n",
        "print(\"  Saved:\", out_best)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 464
        },
        "id": "m-pK0YQKBFk0",
        "outputId": "7ae076fd-9300-47b0-d519-8f35b33bb488"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "                               file                                      id  \\\n",
              "0  prompts_generated_ensemble.fasta  prompts_sample_3_log_likelihood_-0.384   \n",
              "1  prompts_generated_ensemble.fasta  prompts_sample_4_log_likelihood_-0.460   \n",
              "2  prompts_generated_ensemble.fasta  prompts_sample_8_log_likelihood_-0.214   \n",
              "3  prompts_generated_ensemble.fasta  prompts_sample_0_log_likelihood_-0.342   \n",
              "4  prompts_generated_ensemble.fasta  prompts_sample_7_log_likelihood_-0.356   \n",
              "5  prompts_generated_ensemble.fasta  prompts_sample_1_log_likelihood_-0.377   \n",
              "6  prompts_generated_ensemble.fasta  prompts_sample_6_log_likelihood_-0.534   \n",
              "7  prompts_generated_ensemble.fasta  prompts_sample_5_log_likelihood_-0.329   \n",
              "8  prompts_generated_ensemble.fasta  prompts_sample_9_log_likelihood_-0.623   \n",
              "9  prompts_generated_ensemble.fasta  prompts_sample_2_log_likelihood_-0.900   \n",
              "\n",
              "   length  len_ratio_to_prompt  prefix_identity_to_prompt  \\\n",
              "0     291             1.003448                   0.103448   \n",
              "1     284             0.979310                   0.102113   \n",
              "2     283             0.975862                   0.102473   \n",
              "3     287             0.989655                   0.083624   \n",
              "4     273             0.941379                   0.080586   \n",
              "5     286             0.986207                   0.055944   \n",
              "6     276             0.951724                   0.068841   \n",
              "7     293             1.010345                   0.079310   \n",
              "8     285             0.982759                   0.070175   \n",
              "9     318             1.096552                   0.086207   \n",
              "\n",
              "   kmer_low_complexity  longest_run_frac  rank_score  \n",
              "0             0.062284          0.013746    0.317469  \n",
              "1             0.053191          0.014085    0.324421  \n",
              "2             0.053381          0.014134    0.327531  \n",
              "3             0.052632          0.013937    0.349919  \n",
              "4             0.040590          0.014652    0.387638  \n",
              "5             0.045775          0.013986    0.398539  \n",
              "6             0.043796          0.014493    0.405274  \n",
              "7             0.092784          0.013652    0.418203  \n",
              "8             0.077739          0.014035    0.421568  \n",
              "9             0.110759          0.034591    0.559460  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-1db09dc9-74b8-4a73-9efe-b0a9803bfe42\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>file</th>\n",
              "      <th>id</th>\n",
              "      <th>length</th>\n",
              "      <th>len_ratio_to_prompt</th>\n",
              "      <th>prefix_identity_to_prompt</th>\n",
              "      <th>kmer_low_complexity</th>\n",
              "      <th>longest_run_frac</th>\n",
              "      <th>rank_score</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>prompts_generated_ensemble.fasta</td>\n",
              "      <td>prompts_sample_3_log_likelihood_-0.384</td>\n",
              "      <td>291</td>\n",
              "      <td>1.003448</td>\n",
              "      <td>0.103448</td>\n",
              "      <td>0.062284</td>\n",
              "      <td>0.013746</td>\n",
              "      <td>0.317469</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>prompts_generated_ensemble.fasta</td>\n",
              "      <td>prompts_sample_4_log_likelihood_-0.460</td>\n",
              "      <td>284</td>\n",
              "      <td>0.979310</td>\n",
              "      <td>0.102113</td>\n",
              "      <td>0.053191</td>\n",
              "      <td>0.014085</td>\n",
              "      <td>0.324421</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>prompts_generated_ensemble.fasta</td>\n",
              "      <td>prompts_sample_8_log_likelihood_-0.214</td>\n",
              "      <td>283</td>\n",
              "      <td>0.975862</td>\n",
              "      <td>0.102473</td>\n",
              "      <td>0.053381</td>\n",
              "      <td>0.014134</td>\n",
              "      <td>0.327531</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>prompts_generated_ensemble.fasta</td>\n",
              "      <td>prompts_sample_0_log_likelihood_-0.342</td>\n",
              "      <td>287</td>\n",
              "      <td>0.989655</td>\n",
              "      <td>0.083624</td>\n",
              "      <td>0.052632</td>\n",
              "      <td>0.013937</td>\n",
              "      <td>0.349919</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>prompts_generated_ensemble.fasta</td>\n",
              "      <td>prompts_sample_7_log_likelihood_-0.356</td>\n",
              "      <td>273</td>\n",
              "      <td>0.941379</td>\n",
              "      <td>0.080586</td>\n",
              "      <td>0.040590</td>\n",
              "      <td>0.014652</td>\n",
              "      <td>0.387638</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5</th>\n",
              "      <td>prompts_generated_ensemble.fasta</td>\n",
              "      <td>prompts_sample_1_log_likelihood_-0.377</td>\n",
              "      <td>286</td>\n",
              "      <td>0.986207</td>\n",
              "      <td>0.055944</td>\n",
              "      <td>0.045775</td>\n",
              "      <td>0.013986</td>\n",
              "      <td>0.398539</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6</th>\n",
              "      <td>prompts_generated_ensemble.fasta</td>\n",
              "      <td>prompts_sample_6_log_likelihood_-0.534</td>\n",
              "      <td>276</td>\n",
              "      <td>0.951724</td>\n",
              "      <td>0.068841</td>\n",
              "      <td>0.043796</td>\n",
              "      <td>0.014493</td>\n",
              "      <td>0.405274</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7</th>\n",
              "      <td>prompts_generated_ensemble.fasta</td>\n",
              "      <td>prompts_sample_5_log_likelihood_-0.329</td>\n",
              "      <td>293</td>\n",
              "      <td>1.010345</td>\n",
              "      <td>0.079310</td>\n",
              "      <td>0.092784</td>\n",
              "      <td>0.013652</td>\n",
              "      <td>0.418203</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>8</th>\n",
              "      <td>prompts_generated_ensemble.fasta</td>\n",
              "      <td>prompts_sample_9_log_likelihood_-0.623</td>\n",
              "      <td>285</td>\n",
              "      <td>0.982759</td>\n",
              "      <td>0.070175</td>\n",
              "      <td>0.077739</td>\n",
              "      <td>0.014035</td>\n",
              "      <td>0.421568</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>9</th>\n",
              "      <td>prompts_generated_ensemble.fasta</td>\n",
              "      <td>prompts_sample_2_log_likelihood_-0.900</td>\n",
              "      <td>318</td>\n",
              "      <td>1.096552</td>\n",
              "      <td>0.086207</td>\n",
              "      <td>0.110759</td>\n",
              "      <td>0.034591</td>\n",
              "      <td>0.559460</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-1db09dc9-74b8-4a73-9efe-b0a9803bfe42')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-1db09dc9-74b8-4a73-9efe-b0a9803bfe42 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-1db09dc9-74b8-4a73-9efe-b0a9803bfe42');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "    <div id=\"df-ed40bd2f-2eac-427d-a8ef-e35c77fd5169\">\n",
              "      <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-ed40bd2f-2eac-427d-a8ef-e35c77fd5169')\"\n",
              "                title=\"Suggest charts\"\n",
              "                style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "      </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "      <script>\n",
              "        async function quickchart(key) {\n",
              "          const quickchartButtonEl =\n",
              "            document.querySelector('#' + key + ' button');\n",
              "          quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "          quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "          try {\n",
              "            const charts = await google.colab.kernel.invokeFunction(\n",
              "                'suggestCharts', [key], {});\n",
              "          } catch (error) {\n",
              "            console.error('Error during call to suggestCharts:', error);\n",
              "          }\n",
              "          quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "          quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "        }\n",
              "        (() => {\n",
              "          let quickchartButtonEl =\n",
              "            document.querySelector('#df-ed40bd2f-2eac-427d-a8ef-e35c77fd5169 button');\n",
              "          quickchartButtonEl.style.display =\n",
              "            google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "        })();\n",
              "      </script>\n",
              "    </div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "summary": "{\n  \"name\": \"print(\\\"  Saved:\\\", out_best)\",\n  \"rows\": 10,\n  \"fields\": [\n    {\n      \"column\": \"file\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \"prompts_generated_ensemble.fasta\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"id\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 10,\n        \"samples\": [\n          \"prompts_sample_9_log_likelihood_-0.623\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"length\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 12,\n        \"min\": 273,\n        \"max\": 318,\n        \"num_unique_values\": 10,\n        \"samples\": [\n          285\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"len_ratio_to_prompt\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 0.0423325702678551,\n        \"min\": 0.9413793103448276,\n        \"max\": 1.096551724137931,\n        \"num_unique_values\": 10,\n        \"samples\": [\n          0.9827586206896551\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"prefix_identity_to_prompt\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 0.01594782895497514,\n        \"min\": 0.055944055944055944,\n        \"max\": 0.10344827586206896,\n        \"num_unique_values\": 10,\n        \"samples\": [\n          0.07017543859649122\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"kmer_low_complexity\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 0.023211242893168987,\n        \"min\": 0.040590405904059046,\n        \"max\": 0.11075949367088611,\n        \"num_unique_values\": 10,\n        \"samples\": [\n          0.07773851590106007\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"longest_run_frac\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 0.00649330411244062,\n        \"min\": 0.013651877133105802,\n        \"max\": 0.03459119496855346,\n        \"num_unique_values\": 10,\n        \"samples\": [\n          0.014035087719298246\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"rank_score\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 0.07139917902110088,\n        \"min\": 0.31746873860897096,\n        \"max\": 0.5594595614779188,\n        \"num_unique_values\": 10,\n        \"samples\": [\n          0.42156845140754906\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Recommended sequence:\n",
            "  ID: prompts_sample_3_log_likelihood_-0.384\n",
            "  Length: 291\n",
            "  Score: 0.31746873860897096\n",
            "  Saved: /content/profam_workshop/profam/best_generated.fasta\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#@title 14) Upload two structures for TM-score and visualisation\n",
        "\n",
        "from google.colab import files\n",
        "from pathlib import Path\n",
        "\n",
        "UPLOAD_DIR = Path(\"/content/tmalign_uploads\")\n",
        "UPLOAD_DIR.mkdir(exist_ok=True)\n",
        "\n",
        "def upload_one(label):\n",
        "    print(f\"\\nUpload {label} (PDB .pdb/.ent or mmCIF .cif/.mmcif)\")\n",
        "    uploaded = files.upload()\n",
        "\n",
        "    if len(uploaded) != 1:\n",
        "        raise RuntimeError(\n",
        "            f\"{label}: Please upload exactly one file (got {len(uploaded)}). \"\n",
        "            \"Re-run the cell.\"\n",
        "        )\n",
        "\n",
        "    name, data = next(iter(uploaded.items()))\n",
        "    dst = UPLOAD_DIR / name\n",
        "\n",
        "    with open(dst, \"wb\") as f:\n",
        "        f.write(data)\n",
        "\n",
        "    print(f\"{label} saved to: {dst}\")\n",
        "    return str(dst)\n",
        "\n",
        "# Upload structure A\n",
        "struct_a = upload_one(\"Structure A (e.g. ProFam-generated model)\")\n",
        "\n",
        "# Upload structure B\n",
        "struct_b = upload_one(\"Structure B (e.g. AFDB or reference structure)\")\n",
        "\n",
        "print(\"\\nStructures ready for TM-score:\")\n",
        "print(\"  Structure A:\", struct_a)\n",
        "print(\"  Structure B:\", struct_b)\n",
        "print(\"\\nRun the TM-score alignment cell next.\")"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 325
        },
        "id": "0tSjFaKYHzzy",
        "outputId": "30de26f6-f58a-406d-8744-80676ad2fc51"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "Upload Structure A (e.g. ProFam-generated model) (PDB .pdb/.ent or mmCIF .cif/.mmcif)\n"
          ]
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<IPython.core.display.HTML object>"
            ],
            "text/html": [
              "\n",
              "     <input type=\"file\" id=\"files-cf9f285c-a946-434d-a644-4d2bbd3481db\" name=\"files[]\" multiple disabled\n",
              "        style=\"border:none\" />\n",
              "     <output id=\"result-cf9f285c-a946-434d-a644-4d2bbd3481db\">\n",
              "      Upload widget is only available when the cell has been executed in the\n",
              "      current browser session. Please rerun this cell to enable.\n",
              "      </output>\n",
              "      <script>// Copyright 2017 Google LLC\n",
              "//\n",
              "// Licensed under the Apache License, Version 2.0 (the \"License\");\n",
              "// you may not use this file except in compliance with the License.\n",
              "// You may obtain a copy of the License at\n",
              "//\n",
              "//      http://www.apache.org/licenses/LICENSE-2.0\n",
              "//\n",
              "// Unless required by applicable law or agreed to in writing, software\n",
              "// distributed under the License is distributed on an \"AS IS\" BASIS,\n",
              "// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n",
              "// See the License for the specific language governing permissions and\n",
              "// limitations under the License.\n",
              "\n",
              "/**\n",
              " * @fileoverview Helpers for google.colab Python module.\n",
              " */\n",
              "(function(scope) {\n",
              "function span(text, styleAttributes = {}) {\n",
              "  const element = document.createElement('span');\n",
              "  element.textContent = text;\n",
              "  for (const key of Object.keys(styleAttributes)) {\n",
              "    element.style[key] = styleAttributes[key];\n",
              "  }\n",
              "  return element;\n",
              "}\n",
              "\n",
              "// Max number of bytes which will be uploaded at a time.\n",
              "const MAX_PAYLOAD_SIZE = 100 * 1024;\n",
              "\n",
              "function _uploadFiles(inputId, outputId) {\n",
              "  const steps = uploadFilesStep(inputId, outputId);\n",
              "  const outputElement = document.getElementById(outputId);\n",
              "  // Cache steps on the outputElement to make it available for the next call\n",
              "  // to uploadFilesContinue from Python.\n",
              "  outputElement.steps = steps;\n",
              "\n",
              "  return _uploadFilesContinue(outputId);\n",
              "}\n",
              "\n",
              "// This is roughly an async generator (not supported in the browser yet),\n",
              "// where there are multiple asynchronous steps and the Python side is going\n",
              "// to poll for completion of each step.\n",
              "// This uses a Promise to block the python side on completion of each step,\n",
              "// then passes the result of the previous step as the input to the next step.\n",
              "function _uploadFilesContinue(outputId) {\n",
              "  const outputElement = document.getElementById(outputId);\n",
              "  const steps = outputElement.steps;\n",
              "\n",
              "  const next = steps.next(outputElement.lastPromiseValue);\n",
              "  return Promise.resolve(next.value.promise).then((value) => {\n",
              "    // Cache the last promise value to make it available to the next\n",
              "    // step of the generator.\n",
              "    outputElement.lastPromiseValue = value;\n",
              "    return next.value.response;\n",
              "  });\n",
              "}\n",
              "\n",
              "/**\n",
              " * Generator function which is called between each async step of the upload\n",
              " * process.\n",
              " * @param {string} inputId Element ID of the input file picker element.\n",
              " * @param {string} outputId Element ID of the output display.\n",
              " * @return {!Iterable<!Object>} Iterable of next steps.\n",
              " */\n",
              "function* uploadFilesStep(inputId, outputId) {\n",
              "  const inputElement = document.getElementById(inputId);\n",
              "  inputElement.disabled = false;\n",
              "\n",
              "  const outputElement = document.getElementById(outputId);\n",
              "  outputElement.innerHTML = '';\n",
              "\n",
              "  const pickedPromise = new Promise((resolve) => {\n",
              "    inputElement.addEventListener('change', (e) => {\n",
              "      resolve(e.target.files);\n",
              "    });\n",
              "  });\n",
              "\n",
              "  const cancel = document.createElement('button');\n",
              "  inputElement.parentElement.appendChild(cancel);\n",
              "  cancel.textContent = 'Cancel upload';\n",
              "  const cancelPromise = new Promise((resolve) => {\n",
              "    cancel.onclick = () => {\n",
              "      resolve(null);\n",
              "    };\n",
              "  });\n",
              "\n",
              "  // Wait for the user to pick the files.\n",
              "  const files = yield {\n",
              "    promise: Promise.race([pickedPromise, cancelPromise]),\n",
              "    response: {\n",
              "      action: 'starting',\n",
              "    }\n",
              "  };\n",
              "\n",
              "  cancel.remove();\n",
              "\n",
              "  // Disable the input element since further picks are not allowed.\n",
              "  inputElement.disabled = true;\n",
              "\n",
              "  if (!files) {\n",
              "    return {\n",
              "      response: {\n",
              "        action: 'complete',\n",
              "      }\n",
              "    };\n",
              "  }\n",
              "\n",
              "  for (const file of files) {\n",
              "    const li = document.createElement('li');\n",
              "    li.append(span(file.name, {fontWeight: 'bold'}));\n",
              "    li.append(span(\n",
              "        `(${file.type || 'n/a'}) - ${file.size} bytes, ` +\n",
              "        `last modified: ${\n",
              "            file.lastModifiedDate ? file.lastModifiedDate.toLocaleDateString() :\n",
              "                                    'n/a'} - `));\n",
              "    const percent = span('0% done');\n",
              "    li.appendChild(percent);\n",
              "\n",
              "    outputElement.appendChild(li);\n",
              "\n",
              "    const fileDataPromise = new Promise((resolve) => {\n",
              "      const reader = new FileReader();\n",
              "      reader.onload = (e) => {\n",
              "        resolve(e.target.result);\n",
              "      };\n",
              "      reader.readAsArrayBuffer(file);\n",
              "    });\n",
              "    // Wait for the data to be ready.\n",
              "    let fileData = yield {\n",
              "      promise: fileDataPromise,\n",
              "      response: {\n",
              "        action: 'continue',\n",
              "      }\n",
              "    };\n",
              "\n",
              "    // Use a chunked sending to avoid message size limits. See b/62115660.\n",
              "    let position = 0;\n",
              "    do {\n",
              "      const length = Math.min(fileData.byteLength - position, MAX_PAYLOAD_SIZE);\n",
              "      const chunk = new Uint8Array(fileData, position, length);\n",
              "      position += length;\n",
              "\n",
              "      const base64 = btoa(String.fromCharCode.apply(null, chunk));\n",
              "      yield {\n",
              "        response: {\n",
              "          action: 'append',\n",
              "          file: file.name,\n",
              "          data: base64,\n",
              "        },\n",
              "      };\n",
              "\n",
              "      let percentDone = fileData.byteLength === 0 ?\n",
              "          100 :\n",
              "          Math.round((position / fileData.byteLength) * 100);\n",
              "      percent.textContent = `${percentDone}% done`;\n",
              "\n",
              "    } while (position < fileData.byteLength);\n",
              "  }\n",
              "\n",
              "  // All done.\n",
              "  yield {\n",
              "    response: {\n",
              "      action: 'complete',\n",
              "    }\n",
              "  };\n",
              "}\n",
              "\n",
              "scope.google = scope.google || {};\n",
              "scope.google.colab = scope.google.colab || {};\n",
              "scope.google.colab._files = {\n",
              "  _uploadFiles,\n",
              "  _uploadFilesContinue,\n",
              "};\n",
              "})(self);\n",
              "</script> "
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Saving 64f5bad8-de03-477c-8a34-a001dc4dd897.pdb to 64f5bad8-de03-477c-8a34-a001dc4dd897.pdb\n",
            "Structure A (e.g. ProFam-generated model) saved to: /content/tmalign_uploads/64f5bad8-de03-477c-8a34-a001dc4dd897.pdb\n",
            "\n",
            "Upload Structure B (e.g. AFDB or reference structure) (PDB .pdb/.ent or mmCIF .cif/.mmcif)\n"
          ]
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<IPython.core.display.HTML object>"
            ],
            "text/html": [
              "\n",
              "     <input type=\"file\" id=\"files-4dddf974-6a84-4575-aa52-d0213452f7d0\" name=\"files[]\" multiple disabled\n",
              "        style=\"border:none\" />\n",
              "     <output id=\"result-4dddf974-6a84-4575-aa52-d0213452f7d0\">\n",
              "      Upload widget is only available when the cell has been executed in the\n",
              "      current browser session. Please rerun this cell to enable.\n",
              "      </output>\n",
              "      <script>// Copyright 2017 Google LLC\n",
              "//\n",
              "// Licensed under the Apache License, Version 2.0 (the \"License\");\n",
              "// you may not use this file except in compliance with the License.\n",
              "// You may obtain a copy of the License at\n",
              "//\n",
              "//      http://www.apache.org/licenses/LICENSE-2.0\n",
              "//\n",
              "// Unless required by applicable law or agreed to in writing, software\n",
              "// distributed under the License is distributed on an \"AS IS\" BASIS,\n",
              "// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n",
              "// See the License for the specific language governing permissions and\n",
              "// limitations under the License.\n",
              "\n",
              "/**\n",
              " * @fileoverview Helpers for google.colab Python module.\n",
              " */\n",
              "(function(scope) {\n",
              "function span(text, styleAttributes = {}) {\n",
              "  const element = document.createElement('span');\n",
              "  element.textContent = text;\n",
              "  for (const key of Object.keys(styleAttributes)) {\n",
              "    element.style[key] = styleAttributes[key];\n",
              "  }\n",
              "  return element;\n",
              "}\n",
              "\n",
              "// Max number of bytes which will be uploaded at a time.\n",
              "const MAX_PAYLOAD_SIZE = 100 * 1024;\n",
              "\n",
              "function _uploadFiles(inputId, outputId) {\n",
              "  const steps = uploadFilesStep(inputId, outputId);\n",
              "  const outputElement = document.getElementById(outputId);\n",
              "  // Cache steps on the outputElement to make it available for the next call\n",
              "  // to uploadFilesContinue from Python.\n",
              "  outputElement.steps = steps;\n",
              "\n",
              "  return _uploadFilesContinue(outputId);\n",
              "}\n",
              "\n",
              "// This is roughly an async generator (not supported in the browser yet),\n",
              "// where there are multiple asynchronous steps and the Python side is going\n",
              "// to poll for completion of each step.\n",
              "// This uses a Promise to block the python side on completion of each step,\n",
              "// then passes the result of the previous step as the input to the next step.\n",
              "function _uploadFilesContinue(outputId) {\n",
              "  const outputElement = document.getElementById(outputId);\n",
              "  const steps = outputElement.steps;\n",
              "\n",
              "  const next = steps.next(outputElement.lastPromiseValue);\n",
              "  return Promise.resolve(next.value.promise).then((value) => {\n",
              "    // Cache the last promise value to make it available to the next\n",
              "    // step of the generator.\n",
              "    outputElement.lastPromiseValue = value;\n",
              "    return next.value.response;\n",
              "  });\n",
              "}\n",
              "\n",
              "/**\n",
              " * Generator function which is called between each async step of the upload\n",
              " * process.\n",
              " * @param {string} inputId Element ID of the input file picker element.\n",
              " * @param {string} outputId Element ID of the output display.\n",
              " * @return {!Iterable<!Object>} Iterable of next steps.\n",
              " */\n",
              "function* uploadFilesStep(inputId, outputId) {\n",
              "  const inputElement = document.getElementById(inputId);\n",
              "  inputElement.disabled = false;\n",
              "\n",
              "  const outputElement = document.getElementById(outputId);\n",
              "  outputElement.innerHTML = '';\n",
              "\n",
              "  const pickedPromise = new Promise((resolve) => {\n",
              "    inputElement.addEventListener('change', (e) => {\n",
              "      resolve(e.target.files);\n",
              "    });\n",
              "  });\n",
              "\n",
              "  const cancel = document.createElement('button');\n",
              "  inputElement.parentElement.appendChild(cancel);\n",
              "  cancel.textContent = 'Cancel upload';\n",
              "  const cancelPromise = new Promise((resolve) => {\n",
              "    cancel.onclick = () => {\n",
              "      resolve(null);\n",
              "    };\n",
              "  });\n",
              "\n",
              "  // Wait for the user to pick the files.\n",
              "  const files = yield {\n",
              "    promise: Promise.race([pickedPromise, cancelPromise]),\n",
              "    response: {\n",
              "      action: 'starting',\n",
              "    }\n",
              "  };\n",
              "\n",
              "  cancel.remove();\n",
              "\n",
              "  // Disable the input element since further picks are not allowed.\n",
              "  inputElement.disabled = true;\n",
              "\n",
              "  if (!files) {\n",
              "    return {\n",
              "      response: {\n",
              "        action: 'complete',\n",
              "      }\n",
              "    };\n",
              "  }\n",
              "\n",
              "  for (const file of files) {\n",
              "    const li = document.createElement('li');\n",
              "    li.append(span(file.name, {fontWeight: 'bold'}));\n",
              "    li.append(span(\n",
              "        `(${file.type || 'n/a'}) - ${file.size} bytes, ` +\n",
              "        `last modified: ${\n",
              "            file.lastModifiedDate ? file.lastModifiedDate.toLocaleDateString() :\n",
              "                                    'n/a'} - `));\n",
              "    const percent = span('0% done');\n",
              "    li.appendChild(percent);\n",
              "\n",
              "    outputElement.appendChild(li);\n",
              "\n",
              "    const fileDataPromise = new Promise((resolve) => {\n",
              "      const reader = new FileReader();\n",
              "      reader.onload = (e) => {\n",
              "        resolve(e.target.result);\n",
              "      };\n",
              "      reader.readAsArrayBuffer(file);\n",
              "    });\n",
              "    // Wait for the data to be ready.\n",
              "    let fileData = yield {\n",
              "      promise: fileDataPromise,\n",
              "      response: {\n",
              "        action: 'continue',\n",
              "      }\n",
              "    };\n",
              "\n",
              "    // Use a chunked sending to avoid message size limits. See b/62115660.\n",
              "    let position = 0;\n",
              "    do {\n",
              "      const length = Math.min(fileData.byteLength - position, MAX_PAYLOAD_SIZE);\n",
              "      const chunk = new Uint8Array(fileData, position, length);\n",
              "      position += length;\n",
              "\n",
              "      const base64 = btoa(String.fromCharCode.apply(null, chunk));\n",
              "      yield {\n",
              "        response: {\n",
              "          action: 'append',\n",
              "          file: file.name,\n",
              "          data: base64,\n",
              "        },\n",
              "      };\n",
              "\n",
              "      let percentDone = fileData.byteLength === 0 ?\n",
              "          100 :\n",
              "          Math.round((position / fileData.byteLength) * 100);\n",
              "      percent.textContent = `${percentDone}% done`;\n",
              "\n",
              "    } while (position < fileData.byteLength);\n",
              "  }\n",
              "\n",
              "  // All done.\n",
              "  yield {\n",
              "    response: {\n",
              "      action: 'complete',\n",
              "    }\n",
              "  };\n",
              "}\n",
              "\n",
              "scope.google = scope.google || {};\n",
              "scope.google.colab = scope.google.colab || {};\n",
              "scope.google.colab._files = {\n",
              "  _uploadFiles,\n",
              "  _uploadFilesContinue,\n",
              "};\n",
              "})(self);\n",
              "</script> "
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Saving AF-A0A0K8P6T7-F1-model_v6.pdb to AF-A0A0K8P6T7-F1-model_v6.pdb\n",
            "Structure B (e.g. AFDB or reference structure) saved to: /content/tmalign_uploads/AF-A0A0K8P6T7-F1-model_v6.pdb\n",
            "\n",
            "Structures ready for TM-score:\n",
            "  Structure A: /content/tmalign_uploads/64f5bad8-de03-477c-8a34-a001dc4dd897.pdb\n",
            "  Structure B: /content/tmalign_uploads/AF-A0A0K8P6T7-F1-model_v6.pdb\n",
            "\n",
            "Run the TM-score alignment cell next.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#@title TM-align: align A vs B, report TM-score, and visualise superposition (single output)\n",
        "\n",
        "import re, subprocess\n",
        "from pathlib import Path\n",
        "\n",
        "SHOW_3DMOL = False  #@param {type:\"boolean\"}  # If True, shows 3Dmol output before NGLView\n",
        "\n",
        "# --- Inputs from upload cells ------------------------------------------------\n",
        "if \"struct_a\" not in globals() or \"struct_b\" not in globals():\n",
        "    raise RuntimeError(\"struct_a / struct_b not set. Run the upload cell first.\")\n",
        "\n",
        "struct_a = str(struct_a)\n",
        "struct_b = str(struct_b)\n",
        "\n",
        "chain_a = \"\"  #@param {type:\"string\"}  # leave blank = first chain\n",
        "chain_b = \"\"  #@param {type:\"string\"}  # leave blank = first chain\n",
        "\n",
        "WORK_DIR = Path(\"/content/tmalign_work\").resolve()\n",
        "WORK_DIR.mkdir(exist_ok=True)\n",
        "\n",
        "# Install deps (quiet)\n",
        "!pip -q install gemmi nglview\n",
        "if SHOW_3DMOL:\n",
        "    !pip -q install py3Dmol\n",
        "\n",
        "# Enable widget manager for NGLView in Colab\n",
        "from google.colab import output\n",
        "output.enable_custom_widget_manager()\n",
        "\n",
        "import gemmi\n",
        "import nglview as nv\n",
        "\n",
        "def ensure_tmalign(bin_path: Path) -> Path:\n",
        "    if bin_path.exists():\n",
        "        return bin_path\n",
        "    bin_path.parent.mkdir(parents=True, exist_ok=True)\n",
        "    url = \"https://zhanggroup.org/TM-align/TMalign\"\n",
        "    subprocess.run([\"bash\", \"-lc\", f\"wget -q '{url}' -O '{bin_path}'\"], check=True)\n",
        "    subprocess.run([\"bash\", \"-lc\", f\"chmod +x '{bin_path}'\"], check=True)\n",
        "    return bin_path\n",
        "\n",
        "TMALIGN = ensure_tmalign(WORK_DIR / \"bin\" / \"TMalign\")\n",
        "\n",
        "def first_chain_id(path: str) -> str:\n",
        "    st = gemmi.read_structure(path)\n",
        "    for model in st:\n",
        "        for ch in model:\n",
        "            return ch.name\n",
        "    return \"\"\n",
        "\n",
        "def to_pdb(in_path: str, out_path: Path, chain_id: str = \"\") -> Path:\n",
        "    st = gemmi.read_structure(str(in_path))\n",
        "    if chain_id.strip():\n",
        "        keep = chain_id.strip()\n",
        "        for model in st:\n",
        "            for ch in list(model):\n",
        "                if ch.name != keep:\n",
        "                    model.remove_chain(ch.name)\n",
        "        st.remove_empty_chains()\n",
        "    st.write_pdb(str(out_path))\n",
        "    return out_path\n",
        "\n",
        "# Choose default chains if blank\n",
        "if chain_a.strip() == \"\":\n",
        "    chain_a = first_chain_id(struct_a)\n",
        "if chain_b.strip() == \"\":\n",
        "    chain_b = first_chain_id(struct_b)\n",
        "\n",
        "# Prepare inputs\n",
        "pdb_a = WORK_DIR / \"A.pdb\"\n",
        "pdb_b = WORK_DIR / \"B.pdb\"\n",
        "to_pdb(struct_a, pdb_a, chain_a)\n",
        "to_pdb(struct_b, pdb_b, chain_b)\n",
        "\n",
        "# --- Run TM-align with matrix output -----------------------------------------\n",
        "matrix_file = WORK_DIR / \"tmalign_matrix.txt\"\n",
        "if matrix_file.exists():\n",
        "    matrix_file.unlink()\n",
        "\n",
        "res = subprocess.run(\n",
        "    [str(TMALIGN), str(pdb_a), str(pdb_b), \"-m\", str(matrix_file)],\n",
        "    capture_output=True, text=True\n",
        ")\n",
        "out = (res.stdout or \"\") + \"\\n\" + (res.stderr or \"\")\n",
        "if res.returncode != 0:\n",
        "    print(out)\n",
        "    raise RuntimeError(f\"TM-align failed with exit code {res.returncode}\")\n",
        "\n",
        "# --- Parse summary ------------------------------------------------------------\n",
        "tm_scores = [float(x) for x in re.findall(r\"TM-score=\\s*([0-9]*\\.[0-9]+)\", out)]\n",
        "rmsd_m = re.search(r\"RMSD=\\s*([0-9]*\\.[0-9]+)\", out)\n",
        "aln_m  = re.search(r\"Aligned length=\\s*(\\d+)\", out)\n",
        "\n",
        "print(\"TM-align summary\")\n",
        "print(\"  A:\", Path(struct_a).name, f\"(chain {chain_a})\")\n",
        "print(\"  B:\", Path(struct_b).name, f\"(chain {chain_b})\")\n",
        "if aln_m:\n",
        "    print(f\"  Aligned length: {int(aln_m.group(1))}\")\n",
        "if rmsd_m:\n",
        "    print(f\"  RMSD: {float(rmsd_m.group(1)):.3f}\")\n",
        "if tm_scores:\n",
        "    if len(tm_scores) >= 2:\n",
        "        print(f\"  TM-score (norm by A length): {tm_scores[0]:.5f}\")\n",
        "        print(f\"  TM-score (norm by B length): {tm_scores[1]:.5f}\")\n",
        "        print(f\"  TM-score (max): {max(tm_scores[0], tm_scores[1]):.5f}\")\n",
        "    else:\n",
        "        print(f\"  TM-score: {tm_scores[0]:.5f}\")\n",
        "print()\n",
        "\n",
        "# --- Robust parse of 3x4 matrix from -m file ---------------------------------\n",
        "if not matrix_file.exists() or matrix_file.stat().st_size == 0:\n",
        "    raise RuntimeError(\"TM-align did not create a matrix file. Something went wrong with -m.\")\n",
        "\n",
        "lines = [ln.strip() for ln in matrix_file.read_text().splitlines() if ln.strip()]\n",
        "\n",
        "rows = {}\n",
        "for ln in lines:\n",
        "    parts = ln.split()\n",
        "    if len(parts) >= 5 and parts[0] in (\"1\", \"2\", \"3\"):\n",
        "        m = int(parts[0])\n",
        "        rows[m] = [float(parts[1]), float(parts[2]), float(parts[3]), float(parts[4])]\n",
        "\n",
        "if not all(k in rows for k in (1, 2, 3)):\n",
        "    raise RuntimeError(\n",
        "        \"Could not parse the 3 rotation-matrix rows from TM-align -m output.\\n\"\n",
        "        \"First 30 non-empty lines:\\n\" + \"\\n\".join(lines[:30])\n",
        "    )\n",
        "\n",
        "t = [rows[1][0], rows[2][0], rows[3][0]]\n",
        "U = [\n",
        "    [rows[1][1], rows[1][2], rows[1][3]],\n",
        "    [rows[2][1], rows[2][2], rows[2][3]],\n",
        "    [rows[3][1], rows[3][2], rows[3][3]],\n",
        "]\n",
        "\n",
        "# TM-align: X2 = t + U * x1  (Chain_1 -> Chain_2)\n",
        "# We want to move B onto A => apply inverse transform to B:\n",
        "UT = [\n",
        "    [U[0][0], U[1][0], U[2][0]],\n",
        "    [U[0][1], U[1][1], U[2][1]],\n",
        "    [U[0][2], U[1][2], U[2][2]],\n",
        "]\n",
        "t_inv = [\n",
        "    -(UT[0][0]*t[0] + UT[0][1]*t[1] + UT[0][2]*t[2]),\n",
        "    -(UT[1][0]*t[0] + UT[1][1]*t[1] + UT[1][2]*t[2]),\n",
        "    -(UT[2][0]*t[0] + UT[2][1]*t[1] + UT[2][2]*t[2]),\n",
        "]\n",
        "\n",
        "M_inv = [\n",
        "    UT[0][0], UT[0][1], UT[0][2], t_inv[0],\n",
        "    UT[1][0], UT[1][1], UT[1][2], t_inv[1],\n",
        "    UT[2][0], UT[2][1], UT[2][2], t_inv[2],\n",
        "    0,        0,        0,        1\n",
        "]\n",
        "\n",
        "# --- Write B superposed onto A (for NGLView) ---------------------------------\n",
        "B_SUP = WORK_DIR / \"B_superposed.pdb\"\n",
        "\n",
        "def apply_transform_to_structure(in_pdb: Path, out_pdb: Path, M):\n",
        "    st = gemmi.read_structure(str(in_pdb))\n",
        "    r11,r12,r13,t1, r21,r22,r23,t2, r31,r32,r33,t3, _,_,_,_ = M\n",
        "    for model in st:\n",
        "        for chain in model:\n",
        "            for res in chain:\n",
        "                for atom in res:\n",
        "                    x,y,z = atom.pos.x, atom.pos.y, atom.pos.z\n",
        "                    X = t1 + r11*x + r12*y + r13*z\n",
        "                    Y = t2 + r21*x + r22*y + r23*z\n",
        "                    Z = t3 + r31*x + r32*y + r33*z\n",
        "                    atom.pos = gemmi.Position(X, Y, Z)\n",
        "    st.write_pdb(str(out_pdb))\n",
        "\n",
        "apply_transform_to_structure(pdb_b, B_SUP, M_inv)\n",
        "\n",
        "# --- Optional: 3Dmol (only if requested) -------------------------------------\n",
        "if SHOW_3DMOL:\n",
        "    import py3Dmol\n",
        "    v3 = py3Dmol.view(width=900, height=520)\n",
        "    v3.addModel(pdb_a.read_text(), \"pdb\")\n",
        "    v3.addModel(pdb_b.read_text(), \"pdb\")\n",
        "    v3.setTransform({\"model\": 1}, M_inv)\n",
        "    v3.setStyle({\"model\": 0}, {\"cartoon\": {\"color\": \"dodgerblue\"}})\n",
        "    v3.setStyle({\"model\": 1}, {\"cartoon\": {\"color\": \"tomato\", \"opacity\": 0.70}})\n",
        "    v3.zoomTo()\n",
        "    v3  # render as output (no .show())\n",
        "\n",
        "# Superposition viewer (NGLView, force uniform colors; defeat rainbow defaults)\n",
        "\n",
        "from google.colab import output\n",
        "output.enable_custom_widget_manager()\n",
        "\n",
        "!pip -q install nglview gemmi\n",
        "\n",
        "import nglview as nv\n",
        "from pathlib import Path\n",
        "\n",
        "WORK_DIR = Path(\"/content/tmalign_work\").resolve()\n",
        "pdb_a = WORK_DIR / \"A.pdb\"\n",
        "b_sup = WORK_DIR / \"B_superposed.pdb\"\n",
        "\n",
        "if not pdb_a.exists():\n",
        "    raise FileNotFoundError(pdb_a)\n",
        "if not b_sup.exists():\n",
        "    raise FileNotFoundError(b_sup)\n",
        "\n",
        "view = nv.NGLWidget()\n",
        "\n",
        "# Add components\n",
        "compA = view.add_component(str(pdb_a))\n",
        "compB = view.add_component(str(b_sup))\n",
        "\n",
        "compA.clear_representations()\n",
        "compB.clear_representations()\n",
        "\n",
        "\n",
        "compA.add_cartoon(colorScheme=\"uniform\", color=\"red\")\n",
        "compB.add_cartoon(colorScheme=\"uniform\", color=\"blue\", opacity=0.7)\n",
        "\n",
        "view.center()\n",
        "view"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 490,
          "referenced_widgets": [
            "2f5c908b47c34249a826f542f5adf054",
            "c1b930a01abf44a086ff2fad756f0b94",
            "4dcfe5e1f00a469d8696518d42d95c0e",
            "b7bef3eeb335450292f0b76a43a54dc0",
            "c6370a6952144c73bf7fdc39cf3d8a3b",
            "f3712a1a444d4cd293b7565cc305c0c1",
            "3233db53b097403a95ade1daf78e1fbb",
            "f8ce6b8f649e4c35bca89fbb573e61da",
            "1516b48783174c179c0072f2e152b2fe",
            "d432e843235d47d0b4e4c7344a29ef37",
            "a317ad2cea59403383d87adbddafb780",
            "c8d25d4fac8541c1a98a4dde6d1b62af",
            "da738cfd54bd4cdaad4fb0c906f29106"
          ]
        },
        "id": "hgBnxxixT-3T",
        "outputId": "b515fa7e-a771-4aac-c84b-3371e6b7ce3d"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "TM-align summary\n",
            "  A: 64f5bad8-de03-477c-8a34-a001dc4dd897.pdb (chain A)\n",
            "  B: AF-A0A0K8P6T7-F1-model_v6.pdb (chain A)\n",
            "  Aligned length: 250\n",
            "  RMSD: 1.380\n",
            "  TM-score (norm by A length): 0.84977\n",
            "  TM-score (norm by B length): 0.84686\n",
            "  TM-score (max): 0.84977\n",
            "\n"
          ]
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "NGLWidget()"
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "4dcfe5e1f00a469d8696518d42d95c0e"
            }
          },
          "metadata": {
            "application/vnd.jupyter.widget-view+json": {
              "colab": {
                "custom_widget_manager": {
                  "url": "https://ssl.gstatic.com/colaboratory-static/widgets/colab-cdn-widget-manager/2b70e893a8ba7c0f/manager.min.js"
                }
              }
            }
          }
        }
      ]
    }
  ]
}